{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "colab_type": "text",
        "id": "view-in-github"
      },
      "source": [
        "<a href=\"https://polly.elucidata.io/manage/workspaces?action=open_polly_notebook&amp;source=github&amp;path=ElucidataInc%2Fpolly-python%2Fblob%2Fmain%2FDiscover%2Ffind_datasets_of_interest.ipynb&amp;kernel=elucidata%2FPython+3&amp;machine=small\" target=\"_parent\"><img alt=\"Open in Polly\" src=\"https://elucidatainc.github.io/PublicAssets/open_polly.svg\"/></a>\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "kernel": "SoS"
      },
      "source": [
        "## Install polly-python"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {
        "kernel": "Bash"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\u001b[33mWARNING: The directory '/home/polly/.cache/pip/http' or its parent directory is not owned by the current user and the cache has been disabled. Please check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.\u001b[0m\n",
            "\u001b[33mWARNING: The directory '/home/polly/.cache/pip' or its parent directory is not owned by the current user and caching wheels has been disabled. check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.\u001b[0m\n",
            "Looking in indexes: https://pypi.org/simple, http://54.245.179.143:80/\n",
            "Collecting polly-python\n",
            "  Downloading https://files.pythonhosted.org/packages/9a/4b/fc1433a5f8f214d2a6ef54d593c0302db7b85b9c5fbc3f316418c5fb8669/polly_python-0.0.6-py3-none-any.whl\n",
            "Collecting python-magic==0.4.24 (from polly-python)\n",
            "  Downloading https://files.pythonhosted.org/packages/d3/99/c89223c6547df268596899334ee77b3051f606077317023617b1c43162fb/python_magic-0.4.24-py2.py3-none-any.whl\n",
            "Collecting chardet==4.0.0 (from polly-python)\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/19/c7/fa589626997dd07bd87d9269342ccb74b1720384a4d739a1872bd84fbe68/chardet-4.0.0-py2.py3-none-any.whl (178kB)\n",
            "\u001b[K     |\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 184kB 7.7MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting six==1.16.0 (from polly-python)\n",
            "  Downloading https://files.pythonhosted.org/packages/d9/5a/e7c31adbe875f2abbb91bd84cf2dc52d792b5a01506781dbcf25c91daf11/six-1.16.0-py2.py3-none-any.whl\n",
            "Collecting requests==2.25.1 (from polly-python)\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/29/c1/24814557f1d22c56d50280771a17307e6bf87b70727d975fd6b2ce6b014a/requests-2.25.1-py2.py3-none-any.whl (61kB)\n",
            "\u001b[K     |\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 61kB 57.1MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting pytz==2021.1 (from polly-python)\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/70/94/784178ca5dd892a98f113cdd923372024dc04b8d40abe77ca76b5fb90ca6/pytz-2021.1-py2.py3-none-any.whl (510kB)\n",
            "\u001b[K     |\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 512kB 86.2MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting postpy2==0.0.6 (from polly-python)\n",
            "  Downloading https://files.pythonhosted.org/packages/7e/25/d98f5ea87d937bf44963be78b504dbb3ddea1c6ddfcc0985b02d062b41dd/postpy2-0.0.6-py3-none-any.whl\n",
            "Requirement already satisfied: pandas in /usr/local/lib/python3.6/dist-packages (from polly-python) (1.1.4)\n",
            "Collecting urllib3==1.26.6 (from polly-python)\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/5f/64/43575537846896abac0b15c3e5ac678d787a4021e906703f1766bfb8ea11/urllib3-1.26.6-py2.py3-none-any.whl (138kB)\n",
            "\u001b[K     |\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 143kB 89.1MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting idna==2.10 (from polly-python)\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/a2/38/928ddce2273eaa564f6f50de919327bf3a00f091b5baba8dfa9460f3a8a8/idna-2.10-py2.py3-none-any.whl (58kB)\n",
            "\u001b[K     |\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 61kB 57.8MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting python-dateutil==2.8.1 (from polly-python)\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/d4/70/d60450c3dd48ef87586924207ae8907090de0b306af2bce5d134d78615cb/python_dateutil-2.8.1-py2.py3-none-any.whl (227kB)\n",
            "\u001b[K     |\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 235kB 95.6MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting certifi==2021.5.30 (from polly-python)\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/05/1b/0a0dece0e8aa492a6ec9e4ad2fe366b511558cdc73fd3abc82ba7348e875/certifi-2021.5.30-py2.py3-none-any.whl (145kB)\n",
            "\u001b[K     |\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 153kB 93.8MB/s eta 0:00:01\n",
            "\u001b[?25hRequirement already satisfied: numpy>=1.15.4 in /usr/local/lib/python3.6/dist-packages (from pandas->polly-python) (1.19.4)\n",
            "Installing collected packages: python-magic, chardet, six, idna, certifi, urllib3, requests, pytz, postpy2, python-dateutil, polly-python\n",
            "  Found existing installation: chardet 3.0.4\n",
            "    Uninstalling chardet-3.0.4:\n",
            "      Successfully uninstalled chardet-3.0.4\n",
            "  Found existing installation: six 1.14.0\n",
            "    Uninstalling six-1.14.0:\n",
            "      Successfully uninstalled six-1.14.0\n",
            "  Found existing installation: idna 2.8\n",
            "    Uninstalling idna-2.8:\n",
            "      Successfully uninstalled idna-2.8\n",
            "  Found existing installation: certifi 2019.11.28\n",
            "    Uninstalling certifi-2019.11.28:\n",
            "      Successfully uninstalled certifi-2019.11.28\n",
            "  Found existing installation: urllib3 1.26.2\n",
            "    Uninstalling urllib3-1.26.2:\n",
            "      Successfully uninstalled urllib3-1.26.2\n",
            "  Found existing installation: requests 2.25.0\n",
            "    Uninstalling requests-2.25.0:\n",
            "      Successfully uninstalled requests-2.25.0\n",
            "  Found existing installation: pytz 2019.3\n",
            "    Uninstalling pytz-2019.3:\n",
            "      Successfully uninstalled pytz-2019.3\n",
            "  Found existing installation: python-dateutil 2.8.0\n",
            "    Uninstalling python-dateutil-2.8.0:\n",
            "      Successfully uninstalled python-dateutil-2.8.0\n",
            "Successfully installed certifi-2021.5.30 chardet-4.0.0 idna-2.10 polly-python-0.0.6 postpy2-0.0.6 python-dateutil-2.8.1 python-magic-0.4.24 pytz-2021.1 requests-2.25.1 six-1.16.0 urllib3-1.26.6\n",
            "\u001b[33mWARNING: You are using pip version 19.2.3, however version 21.3.1 is available.\n",
            "You should consider upgrading via the 'pip install --upgrade pip' command.\u001b[0m\n"
          ]
        }
      ],
      "source": [
        "!sudo pip3 install polly-python "
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "kernel": "Python3"
      },
      "source": [
        "## Import libraries"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 2,
      "metadata": {
        "kernel": "Python3"
      },
      "outputs": [],
      "source": [
        "from polly.omixatlas import OmixAtlas\n",
        "import pandas as pd\n",
        "from multiprocessing import Pool"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "kernel": "Python3"
      },
      "source": [
        "## Authentication with polly"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 3,
      "metadata": {
        "kernel": "Python3"
      },
      "outputs": [],
      "source": [
        "omix_atlas = OmixAtlas(os.environ['POLLY_REFRESH_TOKEN'])"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## Define helper functions"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 4,
      "metadata": {},
      "outputs": [],
      "source": [
        "def get_matching_datasets_count(match_params):\n",
        "    \"\"\"\n",
        "    Returns the count of matching datasets with given parameteres\n",
        "    \"\"\"\n",
        "    keys = list(match_params)\n",
        "    q = f\"SELECT COUNT(*) FROM geo_files WHERE MATCH_QUERY({keys[0]}, '{params[keys[0]]}') \"\n",
        "    for key in keys:\n",
        "        q = q + f\"AND MATCH_QUERY({key},'{params[key]}') \"\n",
        "    counts = omix_atlas.query_metadata(q).get('aggregations').get('COUNT(*)').get('value')\n",
        "\n",
        "    return counts\n",
        "\n",
        "def construct_query(match_params, offset, limit):\n",
        "    \"\"\"\n",
        "    Constructs a query string using the key-value pairs from the input dictionary\n",
        "    \"\"\"\n",
        "    keys = list(match_params)\n",
        "    q = f\"SELECT * FROM geo_files WHERE MATCH_QUERY({keys[0]}, '{params[keys[0]]}') \"\n",
        "    for key in keys:\n",
        "        q = q + f\"AND MATCH_QUERY({key},'{params[key]}') \"\n",
        "    q = q + f\" LIMIT {offset}, {limit}\"\n",
        "    return q\n",
        "\n",
        "def query_small_batch(args):\n",
        "    \"\"\"\n",
        "    Returns a data frame containing dataset level metadata for a small batch of datasets in GEO \n",
        "    based on the input search parameters\n",
        "    \"\"\"\n",
        "    q  = construct_query(*args)\n",
        "    df = omix_atlas.query_metadata(q)\n",
        "    return df\n",
        "\n",
        "def _construct_parameters_for_map(*args):\n",
        "    \"\"\"\n",
        "    Returns a list to be used while passing arguments to pool map function\n",
        "    \"\"\"\n",
        "    params = args[0]\n",
        "    counts = args[1]\n",
        "    limit  = args[2]\n",
        "    total_queries = counts//limit + 1\n",
        "    params_list = [params]*total_queries\n",
        "    offset_list = list(range(0, counts, limit))\n",
        "    limit_list  = [limit]*total_queries\n",
        "    \n",
        "    return list(zip(params_list, offset_list, limit_list))\n",
        "\n",
        "def search_geo_datasets(match_params):\n",
        "    \"\"\"\n",
        "    Returns a data frame containing dataset level metadata for all datasets in GEO based on the input search parameters\n",
        "    \"\"\"\n",
        "    counts = get_matching_datasets_count(match_params)\n",
        "    \n",
        "    pool = Pool()\n",
        "    limit = 2000\n",
        "    query_results = pool.map(query_small_batch, _construct_parameters_for_map(match_params, counts, limit))\n",
        "    df = pd.concat(query_results)\n",
        "    \n",
        "    return df"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## Define parameters for search"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 8,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/plain": [
              "{'disease': 'neuroendocrine', 'organism': 'homo sapiens', 'platform': 'RNASeq'}"
            ]
          },
          "execution_count": 8,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "params = {\n",
        "    'disease':'neuroendocrine',\n",
        "    'organism':'homo sapiens',\n",
        "    'platform': 'RNASeq'\n",
        "}\n",
        "params"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## Search datasets"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 9,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "{'query_used': \"SELECT COUNT(*) FROM geo_files WHERE MATCH_QUERY(disease, 'neuroendocrine') AND MATCH_QUERY(disease,'neuroendocrine') AND MATCH_QUERY(organism,'homo sapiens') AND MATCH_QUERY(platform,'RNASeq')  LIMIT 100\"}\n",
            "{'query_used': \"SELECT * FROM geo_files WHERE MATCH_QUERY(disease, 'neuroendocrine') AND MATCH_QUERY(disease,'neuroendocrine') AND MATCH_QUERY(organism,'homo sapiens') AND MATCH_QUERY(platform,'RNASeq')  LIMIT 0, 2000\", 'message': 'Showing 1 - 9 of 9 matching results'}\n"
          ]
        }
      ],
      "source": [
        "df = search_geo_datasets(params)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 10,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "(9, 36)\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>publication_name</th>\n",
              "      <th>tissue</th>\n",
              "      <th>dataset_source</th>\n",
              "      <th>description</th>\n",
              "      <th>organism</th>\n",
              "      <th>year</th>\n",
              "      <th>disease</th>\n",
              "      <th>operation</th>\n",
              "      <th>platform</th>\n",
              "      <th>dataset_id</th>\n",
              "      <th>...</th>\n",
              "      <th>kw_filetype</th>\n",
              "      <th>kw_region</th>\n",
              "      <th>kw_location</th>\n",
              "      <th>kw_timestamp</th>\n",
              "      <th>author</th>\n",
              "      <th>abstract</th>\n",
              "      <th>overall_design</th>\n",
              "      <th>summary</th>\n",
              "      <th>type</th>\n",
              "      <th>experimental_design</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>28864682</td>\n",
              "      <td>[None]</td>\n",
              "      <td>GEO</td>\n",
              "      <td>mTOR kinase inhibition effectively decreases p...</td>\n",
              "      <td>[Homo sapiens]</td>\n",
              "      <td>2018</td>\n",
              "      <td>[Gastro-enteropancreatic neuroendocrine tumor,...</td>\n",
              "      <td>{'is_normalized': 'true', 'batch_corrected_var...</td>\n",
              "      <td>RNASeq</td>\n",
              "      <td>GSE102246_GPL15433</td>\n",
              "      <td>...</td>\n",
              "      <td>gct</td>\n",
              "      <td>us-west-2</td>\n",
              "      <td>https://discover-prod-datalake-v1.s3-us-west-2...</td>\n",
              "      <td>1642014810453</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>30287662</td>\n",
              "      <td>[lung, prostate gland]</td>\n",
              "      <td>GEO</td>\n",
              "      <td>PARCB Project: Reprogramming normal human epit...</td>\n",
              "      <td>[Homo sapiens]</td>\n",
              "      <td>Mar 27 2019</td>\n",
              "      <td>[Neuroendocrine Tumors]</td>\n",
              "      <td>{'is_normalized': 'true', 'batch_corrected_var...</td>\n",
              "      <td>RNASeq</td>\n",
              "      <td>GSE118207_GPL21290</td>\n",
              "      <td>...</td>\n",
              "      <td>gct</td>\n",
              "      <td>us-west-2</td>\n",
              "      <td>https://discover-prod-datalake-v1.s3-us-west-2...</td>\n",
              "      <td>1642015405173</td>\n",
              "      <td>Owen,N,Witte</td>\n",
              "      <td>This SuperSeries is composed of the SubSeries ...</td>\n",
              "      <td>Refer to individual Series</td>\n",
              "      <td>This SuperSeries is composed of the SubSeries ...</td>\n",
              "      <td>Genome binding/occupancy profiling by high thr...</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>29915428</td>\n",
              "      <td>[small intestine, liver, lymph node, rectum, p...</td>\n",
              "      <td>GEO</td>\n",
              "      <td>Expression profile of Gastro-Entero-Pancreatic...</td>\n",
              "      <td>[Homo sapiens]</td>\n",
              "      <td>2018</td>\n",
              "      <td>[Gastro-enteropancreatic neuroendocrine tumor,...</td>\n",
              "      <td>{'is_normalized': 'true', 'batch_corrected_var...</td>\n",
              "      <td>RNASeq</td>\n",
              "      <td>GSE98894_GPL16791</td>\n",
              "      <td>...</td>\n",
              "      <td>gct</td>\n",
              "      <td>us-west-2</td>\n",
              "      <td>https://discover-prod-datalake-v1.s3-us-west-2...</td>\n",
              "      <td>1642017949425</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>30315258</td>\n",
              "      <td>[lymph node, pancreatic islet, liver]</td>\n",
              "      <td>GEO</td>\n",
              "      <td>Transcriptome of human non-functional pancreat...</td>\n",
              "      <td>[Homo sapiens]</td>\n",
              "      <td>2018</td>\n",
              "      <td>[Gastro-enteropancreatic neuroendocrine tumor,...</td>\n",
              "      <td>{'is_normalized': 'true', 'batch_corrected_var...</td>\n",
              "      <td>RNASeq</td>\n",
              "      <td>GSE118014_GPL11154</td>\n",
              "      <td>...</td>\n",
              "      <td>gct</td>\n",
              "      <td>us-west-2</td>\n",
              "      <td>https://discover-prod-datalake-v1.s3-us-west-2...</td>\n",
              "      <td>1642015434539</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>26460041</td>\n",
              "      <td>[prostate gland]</td>\n",
              "      <td>GEO</td>\n",
              "      <td>A Basal Stem Cell Signature Identifies Aggress...</td>\n",
              "      <td>[Homo sapiens]</td>\n",
              "      <td>2018</td>\n",
              "      <td>[Prostatic Neoplasms, Carcinoma, Neuroendocrine]</td>\n",
              "      <td>{'is_normalized': 'true', 'batch_corrected_var...</td>\n",
              "      <td>RNASeq</td>\n",
              "      <td>GSE82071_GPL11154</td>\n",
              "      <td>...</td>\n",
              "      <td>gct</td>\n",
              "      <td>us-west-2</td>\n",
              "      <td>https://discover-prod-datalake-v1.s3-us-west-2...</td>\n",
              "      <td>1640216370001</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>{'categorical_variables': {'cell_population': ...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>5 rows \u00d7 36 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "  publication_name                                             tissue  \\\n",
              "0         28864682                                             [None]   \n",
              "1         30287662                             [lung, prostate gland]   \n",
              "2         29915428  [small intestine, liver, lymph node, rectum, p...   \n",
              "3         30315258              [lymph node, pancreatic islet, liver]   \n",
              "4         26460041                                   [prostate gland]   \n",
              "\n",
              "  dataset_source                                        description  \\\n",
              "0            GEO  mTOR kinase inhibition effectively decreases p...   \n",
              "1            GEO  PARCB Project: Reprogramming normal human epit...   \n",
              "2            GEO  Expression profile of Gastro-Entero-Pancreatic...   \n",
              "3            GEO  Transcriptome of human non-functional pancreat...   \n",
              "4            GEO  A Basal Stem Cell Signature Identifies Aggress...   \n",
              "\n",
              "         organism         year  \\\n",
              "0  [Homo sapiens]         2018   \n",
              "1  [Homo sapiens]  Mar 27 2019   \n",
              "2  [Homo sapiens]         2018   \n",
              "3  [Homo sapiens]         2018   \n",
              "4  [Homo sapiens]         2018   \n",
              "\n",
              "                                             disease  \\\n",
              "0  [Gastro-enteropancreatic neuroendocrine tumor,...   \n",
              "1                            [Neuroendocrine Tumors]   \n",
              "2  [Gastro-enteropancreatic neuroendocrine tumor,...   \n",
              "3  [Gastro-enteropancreatic neuroendocrine tumor,...   \n",
              "4   [Prostatic Neoplasms, Carcinoma, Neuroendocrine]   \n",
              "\n",
              "                                           operation platform  \\\n",
              "0  {'is_normalized': 'true', 'batch_corrected_var...   RNASeq   \n",
              "1  {'is_normalized': 'true', 'batch_corrected_var...   RNASeq   \n",
              "2  {'is_normalized': 'true', 'batch_corrected_var...   RNASeq   \n",
              "3  {'is_normalized': 'true', 'batch_corrected_var...   RNASeq   \n",
              "4  {'is_normalized': 'true', 'batch_corrected_var...   RNASeq   \n",
              "\n",
              "           dataset_id  ... kw_filetype  kw_region  \\\n",
              "0  GSE102246_GPL15433  ...         gct  us-west-2   \n",
              "1  GSE118207_GPL21290  ...         gct  us-west-2   \n",
              "2   GSE98894_GPL16791  ...         gct  us-west-2   \n",
              "3  GSE118014_GPL11154  ...         gct  us-west-2   \n",
              "4   GSE82071_GPL11154  ...         gct  us-west-2   \n",
              "\n",
              "                                         kw_location   kw_timestamp  \\\n",
              "0  https://discover-prod-datalake-v1.s3-us-west-2...  1642014810453   \n",
              "1  https://discover-prod-datalake-v1.s3-us-west-2...  1642015405173   \n",
              "2  https://discover-prod-datalake-v1.s3-us-west-2...  1642017949425   \n",
              "3  https://discover-prod-datalake-v1.s3-us-west-2...  1642015434539   \n",
              "4  https://discover-prod-datalake-v1.s3-us-west-2...  1640216370001   \n",
              "\n",
              "         author                                           abstract  \\\n",
              "0           NaN                                                NaN   \n",
              "1  Owen,N,Witte  This SuperSeries is composed of the SubSeries ...   \n",
              "2           NaN                                                NaN   \n",
              "3           NaN                                                NaN   \n",
              "4           NaN                                                NaN   \n",
              "\n",
              "               overall_design  \\\n",
              "0                         NaN   \n",
              "1  Refer to individual Series   \n",
              "2                         NaN   \n",
              "3                         NaN   \n",
              "4                         NaN   \n",
              "\n",
              "                                             summary  \\\n",
              "0                                                NaN   \n",
              "1  This SuperSeries is composed of the SubSeries ...   \n",
              "2                                                NaN   \n",
              "3                                                NaN   \n",
              "4                                                NaN   \n",
              "\n",
              "                                                type  \\\n",
              "0                                                NaN   \n",
              "1  Genome binding/occupancy profiling by high thr...   \n",
              "2                                                NaN   \n",
              "3                                                NaN   \n",
              "4                                                NaN   \n",
              "\n",
              "                                 experimental_design  \n",
              "0                                                NaN  \n",
              "1                                                NaN  \n",
              "2                                                NaN  \n",
              "3                                                NaN  \n",
              "4  {'categorical_variables': {'cell_population': ...  \n",
              "\n",
              "[5 rows x 36 columns]"
            ]
          },
          "execution_count": 10,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "print(df.shape)\n",
        "df.head()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 63,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>dataset_id</th>\n",
              "      <th>description</th>\n",
              "      <th>disease</th>\n",
              "      <th>tissue</th>\n",
              "      <th>total_num_samples</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>GSE100215_GPL16791</td>\n",
              "      <td>Functional co-operativity of long-noncoding RN...</td>\n",
              "      <td>[Neuroblastoma, Nijmegen Breakage Syndrome]</td>\n",
              "      <td>[None]</td>\n",
              "      <td>11</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>GSE107707_GPL18573</td>\n",
              "      <td>Transcriptional changes associated with resist...</td>\n",
              "      <td>[Neuroblastoma]</td>\n",
              "      <td>[None]</td>\n",
              "      <td>24</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>GSE139442_GPL18573</td>\n",
              "      <td>miRNA analysis in depolarized neuroblastoma cells</td>\n",
              "      <td>[Neuroblastoma]</td>\n",
              "      <td>[None]</td>\n",
              "      <td>8</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>GSE136135_GPL20795</td>\n",
              "      <td>Transcriptomic analysis of neuroblastoma cells...</td>\n",
              "      <td>[Neuroblastoma]</td>\n",
              "      <td>[None]</td>\n",
              "      <td>2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>GSE36350_GPL9115</td>\n",
              "      <td>DNMT3B7, an aberrant DNMT3B isoform, suppresse...</td>\n",
              "      <td>[Neuroblastoma]</td>\n",
              "      <td>[None]</td>\n",
              "      <td>3</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>78</th>\n",
              "      <td>GSE53695_GPL16791</td>\n",
              "      <td>nELAVL HITS-CLIP in Alzheimer's Disease patients</td>\n",
              "      <td>[Neuroblastoma, Alzheimer Disease]</td>\n",
              "      <td>[prefrontal cortex]</td>\n",
              "      <td>10</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>79</th>\n",
              "      <td>GSE22785_GPL9128</td>\n",
              "      <td>Integrated genome, transcriptome and translato...</td>\n",
              "      <td>[Neuroblastoma]</td>\n",
              "      <td>[adrenal gland, bone marrow, brain]</td>\n",
              "      <td>14</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>80</th>\n",
              "      <td>GSE90683_GPL16791</td>\n",
              "      <td>Heterogeneity of neuroblastoma cell identity r...</td>\n",
              "      <td>[Neuroblastoma]</td>\n",
              "      <td>[neural crest]</td>\n",
              "      <td>36</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>81</th>\n",
              "      <td>GSE91377_GPL16791</td>\n",
              "      <td>RNA-Seq comparative analysis of human neurobla...</td>\n",
              "      <td>[Neuroblastoma]</td>\n",
              "      <td>[None]</td>\n",
              "      <td>4</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>82</th>\n",
              "      <td>GSE90711_GPL11154</td>\n",
              "      <td>Proteomics and transcriptomics of peripheral n...</td>\n",
              "      <td>[Neuroblastoma, Peripheral Nerve Injuries]</td>\n",
              "      <td>[peripheral nerve, bone, nerve]</td>\n",
              "      <td>27</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>83 rows \u00d7 5 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "            dataset_id                                        description  \\\n",
              "0   GSE100215_GPL16791  Functional co-operativity of long-noncoding RN...   \n",
              "1   GSE107707_GPL18573  Transcriptional changes associated with resist...   \n",
              "2   GSE139442_GPL18573  miRNA analysis in depolarized neuroblastoma cells   \n",
              "3   GSE136135_GPL20795  Transcriptomic analysis of neuroblastoma cells...   \n",
              "4     GSE36350_GPL9115  DNMT3B7, an aberrant DNMT3B isoform, suppresse...   \n",
              "..                 ...                                                ...   \n",
              "78   GSE53695_GPL16791   nELAVL HITS-CLIP in Alzheimer's Disease patients   \n",
              "79    GSE22785_GPL9128  Integrated genome, transcriptome and translato...   \n",
              "80   GSE90683_GPL16791  Heterogeneity of neuroblastoma cell identity r...   \n",
              "81   GSE91377_GPL16791  RNA-Seq comparative analysis of human neurobla...   \n",
              "82   GSE90711_GPL11154  Proteomics and transcriptomics of peripheral n...   \n",
              "\n",
              "                                        disease  \\\n",
              "0   [Neuroblastoma, Nijmegen Breakage Syndrome]   \n",
              "1                               [Neuroblastoma]   \n",
              "2                               [Neuroblastoma]   \n",
              "3                               [Neuroblastoma]   \n",
              "4                               [Neuroblastoma]   \n",
              "..                                          ...   \n",
              "78           [Neuroblastoma, Alzheimer Disease]   \n",
              "79                              [Neuroblastoma]   \n",
              "80                              [Neuroblastoma]   \n",
              "81                              [Neuroblastoma]   \n",
              "82   [Neuroblastoma, Peripheral Nerve Injuries]   \n",
              "\n",
              "                                 tissue total_num_samples  \n",
              "0                                [None]                11  \n",
              "1                                [None]                24  \n",
              "2                                [None]                 8  \n",
              "3                                [None]                 2  \n",
              "4                                [None]                 3  \n",
              "..                                  ...               ...  \n",
              "78                  [prefrontal cortex]                10  \n",
              "79  [adrenal gland, bone marrow, brain]                14  \n",
              "80                       [neural crest]                36  \n",
              "81                               [None]                 4  \n",
              "82      [peripheral nerve, bone, nerve]                27  \n",
              "\n",
              "[83 rows x 5 columns]"
            ]
          },
          "execution_count": 63,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "df[['dataset_id','description','disease','tissue','total_num_samples']]"
      ]
    }
  ],
  "metadata": {
    "kernelspec": {
      "display_name": "Python 3",
      "language": "python",
      "name": "python3"
    },
    "language_info": {
      "codemirror_mode": {
        "name": "ipython",
        "version": 3
      },
      "file_extension": ".py",
      "mimetype": "text/x-python",
      "name": "python",
      "nbconvert_exporter": "python",
      "pygments_lexer": "ipython3",
      "version": "3.6.9"
    },
    "sos": {
      "kernels": [
        [
          "Bash",
          "bash",
          "Bash",
          "#E6EEFF",
          ""
        ],
        [
          "Python3",
          "python3",
          "Python3",
          "#FFD91A",
          {
            "name": "ipython",
            "version": 3
          }
        ]
      ],
      "panel": {
        "displayed": true,
        "height": 0
      },
      "version": "1.0.0"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 2
}