{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "colab_type": "text",
        "id": "view-in-github"
      },
      "source": [
        "<a href=\"https://polly.elucidata.io/manage/workspaces?action=open_polly_notebook&amp;source=github&amp;path=ElucidataInc%2Fpolly-python%2Fblob%2Fmain%2FDiscover%2Fontology_recommendation_disease_tissue.ipynb&amp;kernel=elucidata%2FPython+3&amp;machine=medium\" target=\"_parent\"><img alt=\"Open in Polly\" src=\"https://elucidatainc.github.io/PublicAssets/open_polly.svg\"/></a>\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "# Ontology recommendations for disease and tissue using polly-python"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "Ontology recommendation functionality for disease and tissue are added in Polly-Python. In the existing SQL query itslef, the users would now be able to call a function - 'recommend' on disease and tissue column of metadata to get recommendations. \n",
        "\n",
        "Usage of 'recommend' function - \n",
        "\n",
        "recommend(field_name, search_term, key - ['match' | 'related'])\n",
        "\n",
        "field_name -> It can take value: disease, tissue, curated_disease, curated_tissue based on V1 or V2 APIs.\n",
        "\n",
        "search_term -> Disease or tissue name for which recommendations are required.\n",
        "\n",
        "key -> Can be \"match\" or \"related\"\n",
        "\n",
        "    match - Only the terms that have an exact match of the keyword in them will be returned as an output.\n",
        "        \n",
        "    related - The list of expanded terms would contain the matched terms, the synonyms, and hypernyms of the keyword as per MeSH ontology. "
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## For users querying V2 infrastructure"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "For 'match' query in disease - \n",
        "\n",
        "query = \"\"\"SELECT * FROM geo.datasets WHERE CONTAINS(curated_disease, recommend('curated_disease', 'breast neoplasms', 'match'))\"\"\"\n",
        "\n",
        "For 'related' query in tissue - \n",
        "\n",
        "query = \"\"\"SELECT * FROM geo.datasets WHERE CONTAINS(curated_tissue, recommend( 'curated_tissue', 'liver', 'related'))\"\"\"\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 3,
      "metadata": {},
      "outputs": [],
      "source": [
        "# please do not modify\n",
        "from IPython.display import display_html\n",
        "def restartkernel() :\n",
        "    display_html(\"<script>Jupyter.notebook.kernel.restart()</script>\",raw=True)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "# Import polly-python"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 4,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Collecting polly-python\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/15/bc/3bd95540fad7f863898cd0f095720996931e5c49e40e2fe8d5894f29c684/polly_python-0.1.3-py3-none-any.whl (41kB)\n",
            "\u001b[K     |################################| 51kB 14.5MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting rst2txt (from polly-python)\n",
            "  Downloading https://files.pythonhosted.org/packages/9d/d4/cee0341774dbdd4c128816ebc31fbbf39a11c79220a96a4f1152aa511caa/rst2txt-1.1.0-py2.py3-none-any.whl\n",
            "Collecting requests==2.25.1 (from polly-python)\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/29/c1/24814557f1d22c56d50280771a17307e6bf87b70727d975fd6b2ce6b014a/requests-2.25.1-py2.py3-none-any.whl (61kB)\n",
            "\u001b[K     |################################| 61kB 7.6MB/s eta 0:00:011\n",
            "\u001b[?25hCollecting Deprecated (from polly-python)\n",
            "  Downloading https://files.pythonhosted.org/packages/51/6a/c3a0408646408f7283b7bc550c30a32cc791181ec4618592eec13e066ce3/Deprecated-1.2.13-py2.py3-none-any.whl\n",
            "Collecting chardet==4.0.0 (from polly-python)\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/19/c7/fa589626997dd07bd87d9269342ccb74b1720384a4d739a1872bd84fbe68/chardet-4.0.0-py2.py3-none-any.whl (178kB)\n",
            "\u001b[K     |################################| 184kB 44.5MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting postpy2==0.0.6 (from polly-python)\n",
            "  Downloading https://files.pythonhosted.org/packages/7e/25/d98f5ea87d937bf44963be78b504dbb3ddea1c6ddfcc0985b02d062b41dd/postpy2-0.0.6-py3-none-any.whl\n",
            "Collecting retrying==1.3.3 (from polly-python)\n",
            "  Downloading https://files.pythonhosted.org/packages/44/ef/beae4b4ef80902f22e3af073397f079c96969c69b2c7d52a57ea9ae61c9d/retrying-1.3.3.tar.gz\n",
            "Requirement already satisfied: python-dateutil==2.8.1 in /usr/local/lib/python3.6/dist-packages (from polly-python) (2.8.1)\n",
            "Collecting urllib3==1.26.6 (from polly-python)\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/5f/64/43575537846896abac0b15c3e5ac678d787a4021e906703f1766bfb8ea11/urllib3-1.26.6-py2.py3-none-any.whl (138kB)\n",
            "\u001b[K     |################################| 143kB 88.7MB/s eta 0:00:01\n",
            "\u001b[?25hRequirement already satisfied: Cerberus==1.3.2 in /usr/local/lib/python3.6/dist-packages/Cerberus-1.3.2-py3.6.egg (from polly-python) (1.3.2)\n",
            "Collecting sqlparse (from polly-python)\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/05/40/d836d55fb3f467243ee839ab7b814822fda522cd395fa41e282684e71ee5/sqlparse-0.4.2-py3-none-any.whl (42kB)\n",
            "\u001b[K     |################################| 51kB 46.6MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting certifi==2021.10.8 (from polly-python)\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/37/45/946c02767aabb873146011e665728b680884cd8fe70dde973c640e45b775/certifi-2021.10.8-py2.py3-none-any.whl (149kB)\n",
            "\u001b[K     |################################| 153kB 93.9MB/s eta 0:00:01\n",
            "\u001b[?25hRequirement already satisfied: pandas in /usr/local/lib/python3.6/dist-packages (from polly-python) (1.0.1)\n",
            "Requirement already satisfied: botocore>=1.20.73 in /usr/local/lib/python3.6/dist-packages (from polly-python) (1.20.112)\n",
            "Collecting black (from polly-python)\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/2e/ef/a38a2189959246543e60859fb65bd3143129f6d18dfc7bcdd79217f81ca2/black-22.3.0-py3-none-any.whl (153kB)\n",
            "\u001b[K     |################################| 163kB 101.2MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting joblib (from polly-python)\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/3e/d5/0163eb0cfa0b673aa4fe1cd3ea9d8a81ea0f32e50807b0c295871e4aab2e/joblib-1.1.0-py2.py3-none-any.whl (306kB)\n",
            "\u001b[K     |################################| 307kB 92.9MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting elucidatacloudpathlib==0.6.6 (from polly-python)\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/b1/d6/5162b68817dd93130dc9421f23a299ee6b243c0ff8a806370727ff10090f/elucidatacloudpathlib-0.6.6-py3-none-any.whl (50kB)\n",
            "\u001b[K     |################################| 51kB 43.6MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting pytz==2021.1 (from polly-python)\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/70/94/784178ca5dd892a98f113cdd923372024dc04b8d40abe77ca76b5fb90ca6/pytz-2021.1-py2.py3-none-any.whl (510kB)\n",
            "\u001b[K     |################################| 512kB 90.5MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting boto3>=1.17.73 (from polly-python)\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/75/ca/d917b244919f1ebf96f7bbd5a00e4641f7e9191b0d070258f5dc10f5eaad/boto3-1.23.10-py3-none-any.whl (132kB)\n",
            "\u001b[K     |################################| 133kB 93.3MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting idna==2.10 (from polly-python)\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/a2/38/928ddce2273eaa564f6f50de919327bf3a00f091b5baba8dfa9460f3a8a8/idna-2.10-py2.py3-none-any.whl (58kB)\n",
            "\u001b[K     |################################| 61kB 47.2MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting six==1.16.0 (from polly-python)\n",
            "  Downloading https://files.pythonhosted.org/packages/d9/5a/e7c31adbe875f2abbb91bd84cf2dc52d792b5a01506781dbcf25c91daf11/six-1.16.0-py2.py3-none-any.whl\n",
            "Collecting python-magic==0.4.24 (from polly-python)\n",
            "  Downloading https://files.pythonhosted.org/packages/d3/99/c89223c6547df268596899334ee77b3051f606077317023617b1c43162fb/python_magic-0.4.24-py2.py3-none-any.whl\n",
            "Collecting tqdm (from polly-python)\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/8a/c4/d15f1e627fff25443ded77ea70a7b5532d6371498f9285d44d62587e209c/tqdm-4.64.0-py2.py3-none-any.whl (78kB)\n",
            "\u001b[K     |################################| 81kB 60.6MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting cmapPy (from polly-python)\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/e8/e1/9e90b0dc2051f4a3a2b9553b254889e93a2a3c7a3b925433c6298d99d596/cmapPy-4.0.1-py2.py3-none-any.whl (150kB)\n",
            "\u001b[K     |################################| 153kB 104.6MB/s eta 0:00:01\n",
            "\u001b[?25hRequirement already satisfied: docutils>=0.11 in /usr/local/lib/python3.6/dist-packages (from rst2txt->polly-python) (0.15.2)\n",
            "Requirement already satisfied: pygments in /usr/local/lib/python3.6/dist-packages (from rst2txt->polly-python) (2.5.2)\n",
            "Collecting wrapt<2,>=1.10 (from Deprecated->polly-python)\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/e0/80/af9da7379ee6df583875d0aeb80f9d5f0bd5f081dd1ee5ce06587d8bfec7/wrapt-1.14.1-cp36-cp36m-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (74kB)\n",
            "\u001b[K     |################################| 81kB 56.5MB/s eta 0:00:01\n",
            "\u001b[?25hRequirement already satisfied: setuptools in /usr/local/lib/python3.6/dist-packages (from Cerberus==1.3.2->polly-python) (45.2.0)\n",
            "Requirement already satisfied: numpy>=1.13.3 in /usr/local/lib/python3.6/dist-packages (from pandas->polly-python) (1.18.1)\n",
            "Requirement already satisfied: jmespath<1.0.0,>=0.7.1 in /usr/local/lib/python3.6/dist-packages (from botocore>=1.20.73->polly-python) (0.9.5)\n",
            "Collecting dataclasses>=0.6; python_version < \"3.7\" (from black->polly-python)\n",
            "  Downloading https://files.pythonhosted.org/packages/fe/ca/75fac5856ab5cfa51bbbcefa250182e50441074fdc3f803f6e76451fab43/dataclasses-0.8-py3-none-any.whl\n",
            "Collecting typing-extensions>=3.10.0.0; python_version < \"3.10\" (from black->polly-python)\n",
            "  Downloading https://files.pythonhosted.org/packages/45/6b/44f7f8f1e110027cf88956b59f2fad776cca7e1704396d043f89effd3a0e/typing_extensions-4.1.1-py3-none-any.whl\n",
            "Collecting mypy-extensions>=0.4.3 (from black->polly-python)\n",
            "  Downloading https://files.pythonhosted.org/packages/5c/eb/975c7c080f3223a5cdaff09612f3a5221e4ba534f7039db34c35d95fa6a5/mypy_extensions-0.4.3-py2.py3-none-any.whl\n",
            "Collecting click>=8.0.0 (from black->polly-python)\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/4a/a8/0b2ced25639fb20cc1c9784de90a8c25f9504a7f18cd8b5397bd61696d7d/click-8.0.4-py3-none-any.whl (97kB)\n",
            "\u001b[K     |################################| 102kB 61.4MB/s ta 0:00:01\n",
            "\u001b[?25hCollecting typed-ast>=1.4.2; python_version < \"3.8\" and implementation_name == \"cpython\" (from black->polly-python)\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/e3/7c/7407838e9c540031439f2948bce2763cdd6882ebb72cc0a25b763c10529e/typed_ast-1.5.4-cp36-cp36m-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (831kB)\n",
            "\u001b[K     |################################| 839kB 97.8MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting platformdirs>=2 (from black->polly-python)\n"
          ]
        },
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "  Downloading https://files.pythonhosted.org/packages/b1/78/dcfd84d3aabd46a9c77260fb47ea5d244806e4daef83aa6fe5d83adb182c/platformdirs-2.4.0-py3-none-any.whl\n",
            "Collecting pathspec>=0.9.0 (from black->polly-python)\n",
            "  Downloading https://files.pythonhosted.org/packages/42/ba/a9d64c7bcbc7e3e8e5f93a52721b377e994c22d16196e2b0f1236774353a/pathspec-0.9.0-py2.py3-none-any.whl\n",
            "Collecting tomli>=1.1.0; python_version < \"3.11\" (from black->polly-python)\n",
            "  Downloading https://files.pythonhosted.org/packages/05/e4/74f9440db36734d7ba83c574c1e7024009ce849208a41f90e94a134dc6d1/tomli-1.2.3-py3-none-any.whl\n",
            "Requirement already satisfied: importlib-metadata; python_version < \"3.8\" in /usr/local/lib/python3.6/dist-packages (from elucidatacloudpathlib==0.6.6->polly-python) (1.5.0)\n",
            "Collecting s3transfer<0.6.0,>=0.5.0 (from boto3>=1.17.73->polly-python)\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/7b/9c/f51775ebe7df5a7aa4e7c79ed671bde94e154bd968aca8d65bb24aba0c8c/s3transfer-0.5.2-py3-none-any.whl (79kB)\n",
            "\u001b[K     |################################| 81kB 57.5MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting importlib-resources; python_version < \"3.7\" (from tqdm->polly-python)\n",
            "  Downloading https://files.pythonhosted.org/packages/24/1b/33e489669a94da3ef4562938cd306e8fa915e13939d7b8277cb5569cb405/importlib_resources-5.4.0-py3-none-any.whl\n",
            "Collecting h5py>=2.6.0 (from cmapPy->polly-python)\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/70/7a/e53e500335afb6b1aade11227cdf107fca54106a1dca5c9d13242a043f3b/h5py-3.1.0-cp36-cp36m-manylinux1_x86_64.whl (4.0MB)\n",
            "\u001b[K     |################################| 4.0MB 95.9MB/s eta 0:00:01\n",
            "\u001b[?25hRequirement already satisfied: zipp>=0.5 in /usr/local/lib/python3.6/dist-packages (from importlib-metadata; python_version < \"3.8\"->elucidatacloudpathlib==0.6.6->polly-python) (3.1.0)\n",
            "Collecting cached-property; python_version < \"3.8\" (from h5py>=2.6.0->cmapPy->polly-python)\n",
            "  Downloading https://files.pythonhosted.org/packages/48/19/f2090f7dad41e225c7f2326e4cfe6fff49e57dedb5b53636c9551f86b069/cached_property-1.5.2-py2.py3-none-any.whl\n",
            "Building wheels for collected packages: retrying\n",
            "  Building wheel for retrying (setup.py) ... \u001b[?25ldone\n",
            "\u001b[?25h  Created wheel for retrying: filename=retrying-1.3.3-cp36-none-any.whl size=11431 sha256=442149bfc6f71df4a9bd2dcd28d703e12d3260472c1df0ea0ca2200ecb5763f3\n",
            "  Stored in directory: /home/polly/.cache/pip/wheels/d7/a9/33/acc7b709e2a35caa7d4cae442f6fe6fbf2c43f80823d46460c\n",
            "Successfully built retrying\n",
            "\u001b[31mERROR: awscli 1.18.178 has requirement botocore==1.19.18, but you'll have botocore 1.20.112 which is incompatible.\u001b[0m\n",
            "\u001b[31mERROR: awscli 1.18.178 has requirement s3transfer<0.4.0,>=0.3.0, but you'll have s3transfer 0.5.2 which is incompatible.\u001b[0m\n",
            "\u001b[31mERROR: boto3 1.23.10 has requirement botocore<1.27.0,>=1.26.10, but you'll have botocore 1.20.112 which is incompatible.\u001b[0m\n",
            "Installing collected packages: rst2txt, certifi, idna, urllib3, chardet, requests, wrapt, Deprecated, python-magic, postpy2, six, retrying, sqlparse, dataclasses, typing-extensions, mypy-extensions, click, typed-ast, platformdirs, pathspec, tomli, black, joblib, elucidatacloudpathlib, pytz, s3transfer, boto3, importlib-resources, tqdm, cached-property, h5py, cmapPy, polly-python\n",
            "  Found existing installation: certifi 2019.11.28\n",
            "    Uninstalling certifi-2019.11.28:\n",
            "      Successfully uninstalled certifi-2019.11.28\n",
            "  Found existing installation: idna 2.8\n",
            "    Uninstalling idna-2.8:\n",
            "      Successfully uninstalled idna-2.8\n",
            "  Found existing installation: urllib3 1.26.2\n",
            "    Uninstalling urllib3-1.26.2:\n",
            "      Successfully uninstalled urllib3-1.26.2\n",
            "  Found existing installation: chardet 3.0.4\n",
            "    Uninstalling chardet-3.0.4:\n",
            "      Successfully uninstalled chardet-3.0.4\n",
            "  Found existing installation: requests 2.25.0\n",
            "    Uninstalling requests-2.25.0:\n",
            "      Successfully uninstalled requests-2.25.0\n",
            "  Found existing installation: six 1.14.0\n",
            "    Uninstalling six-1.14.0:\n",
            "      Successfully uninstalled six-1.14.0\n",
            "  Found existing installation: pytz 2019.3\n",
            "    Uninstalling pytz-2019.3:\n",
            "      Successfully uninstalled pytz-2019.3\n",
            "  Found existing installation: s3transfer 0.3.3\n",
            "    Uninstalling s3transfer-0.3.3:\n",
            "      Successfully uninstalled s3transfer-0.3.3\n",
            "  Found existing installation: boto3 1.17.4\n",
            "    Uninstalling boto3-1.17.4:\n",
            "      Successfully uninstalled boto3-1.17.4\n",
            "Successfully installed Deprecated-1.2.13 black-22.3.0 boto3-1.23.10 cached-property-1.5.2 certifi-2021.10.8 chardet-4.0.0 click-8.0.4 cmapPy-4.0.1 dataclasses-0.8 elucidatacloudpathlib-0.6.6 h5py-3.1.0 idna-2.10 importlib-resources-5.4.0 joblib-1.1.0 mypy-extensions-0.4.3 pathspec-0.9.0 platformdirs-2.4.0 polly-python-0.1.3 postpy2-0.0.6 python-magic-0.4.24 pytz-2021.1 requests-2.25.1 retrying-1.3.3 rst2txt-1.1.0 s3transfer-0.5.2 six-1.16.0 sqlparse-0.4.2 tomli-1.2.3 tqdm-4.64.0 typed-ast-1.5.4 typing-extensions-4.1.1 urllib3-1.26.6 wrapt-1.14.1\n",
            "\u001b[33mWARNING: You are using pip version 19.2.3, however version 21.3.1 is available.\n",
            "You should consider upgrading via the 'pip install --upgrade pip' command.\u001b[0m\n"
          ]
        }
      ],
      "source": [
        "!sudo pip3 install polly-python  #Restart kernel after the cell executes."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/html": [
              "<script>Jupyter.notebook.kernel.restart()</script>"
            ]
          },
          "metadata": {},
          "output_type": "display_data"
        }
      ],
      "source": [
        "restartkernel() #Pause for a few seconds before the kernel is refreshed"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "# Import Dependencies"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {},
      "outputs": [],
      "source": [
        "import os\n",
        "from polly.auth import Polly\n",
        "from polly.omixatlas import OmixAtlas"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "# Auth With Token on Polly"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 3,
      "metadata": {},
      "outputs": [],
      "source": [
        "POLLY_REFRESH_TOKEN = os.environ['POLLY_REFRESH_TOKEN']\n",
        "omixatlas = OmixAtlas(POLLY_REFRESH_TOKEN)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "# SQL Queries for V2 storage infrastructure"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## Previous query on V2 infrastructure\n",
        "Before implementation of this feature, users query for a given tissue and disease as shown below. \n",
        "\n",
        "For this query, user is able to fetch 1388 datasets for the given disease and tissue combination."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 4,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Query execution succeeded (time taken: 1.96 seconds, data scanned: 0.908 MB)\n",
            "Fetched 1388 rows\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>dataset_id</th>\n",
              "      <th>curated_disease</th>\n",
              "      <th>curated_tissue</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>GSE9691_GPL3921</td>\n",
              "      <td>[Breast Neoplasms]</td>\n",
              "      <td>[breast]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>GSE97221_GPL10558</td>\n",
              "      <td>[Breast Neoplasms]</td>\n",
              "      <td>[breast]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>GSE97317_GPL11154</td>\n",
              "      <td>[Breast Neoplasms]</td>\n",
              "      <td>[breast]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>GSE9734_GPL4742</td>\n",
              "      <td>[Breast Neoplasms,  Carcinoma,  Pancreatic Duc...</td>\n",
              "      <td>[pancreas,  breast,  kidney,  colon]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>GSE97482_GPL10332</td>\n",
              "      <td>[Breast Neoplasms]</td>\n",
              "      <td>[breast]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1383</th>\n",
              "      <td>GSE9483_GPL6071</td>\n",
              "      <td>[Neoplasms,  Basal Cell,  Breast Cancer,  Fami...</td>\n",
              "      <td>[breast]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1384</th>\n",
              "      <td>GSE95035_GPL10558</td>\n",
              "      <td>[Breast Neoplasms]</td>\n",
              "      <td>[breast]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1385</th>\n",
              "      <td>GSE95087_GPL16956</td>\n",
              "      <td>[Breast Neoplasms]</td>\n",
              "      <td>[breast]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1386</th>\n",
              "      <td>GSE95304_GPL11154</td>\n",
              "      <td>[Breast Neoplasms]</td>\n",
              "      <td>[breast]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1387</th>\n",
              "      <td>GSE95554_GPL17117</td>\n",
              "      <td>[Breast Neoplasms]</td>\n",
              "      <td>[breast,  oil secretion]</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>1388 rows \u00d7 3 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "             dataset_id                                    curated_disease  \\\n",
              "0       GSE9691_GPL3921                                 [Breast Neoplasms]   \n",
              "1     GSE97221_GPL10558                                 [Breast Neoplasms]   \n",
              "2     GSE97317_GPL11154                                 [Breast Neoplasms]   \n",
              "3       GSE9734_GPL4742  [Breast Neoplasms,  Carcinoma,  Pancreatic Duc...   \n",
              "4     GSE97482_GPL10332                                 [Breast Neoplasms]   \n",
              "...                 ...                                                ...   \n",
              "1383    GSE9483_GPL6071  [Neoplasms,  Basal Cell,  Breast Cancer,  Fami...   \n",
              "1384  GSE95035_GPL10558                                 [Breast Neoplasms]   \n",
              "1385  GSE95087_GPL16956                                 [Breast Neoplasms]   \n",
              "1386  GSE95304_GPL11154                                 [Breast Neoplasms]   \n",
              "1387  GSE95554_GPL17117                                 [Breast Neoplasms]   \n",
              "\n",
              "                            curated_tissue  \n",
              "0                                 [breast]  \n",
              "1                                 [breast]  \n",
              "2                                 [breast]  \n",
              "3     [pancreas,  breast,  kidney,  colon]  \n",
              "4                                 [breast]  \n",
              "...                                    ...  \n",
              "1383                              [breast]  \n",
              "1384                              [breast]  \n",
              "1385                              [breast]  \n",
              "1386                              [breast]  \n",
              "1387              [breast,  oil secretion]  \n",
              "\n",
              "[1388 rows x 3 columns]"
            ]
          },
          "execution_count": 4,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "sql_query = \"\"\"SELECT dataset_id, curated_disease, curated_tissue FROM geo.datasets WHERE \n",
        "        CONTAINS(curated_disease,'Breast Neoplasms') AND \n",
        "        CONTAINS(curated_tissue,'breast')\"\"\" \n",
        "result = omixatlas.query_metadata(sql_query)\n",
        "result"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## New queries after implementation of ontology recommendations\n",
        "Now the users can query as shown below. \n",
        "\n",
        "For query with ontology recommendations, the user is able to fetch 2223 datasets for the given disease and tissue combination. This is ~60% higher than previous ones."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 6,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Query execution succeeded (time taken: 2.04 seconds, data scanned: 0.908 MB)\n",
            "Fetched 2223 rows\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>dataset_id</th>\n",
              "      <th>curated_disease</th>\n",
              "      <th>curated_tissue</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>GSE96058_GPL11154</td>\n",
              "      <td>[Triple Negative Breast Neoplasms,  Brittle co...</td>\n",
              "      <td>[breast]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>GSE96058_GPL18573</td>\n",
              "      <td>[Triple Negative Breast Neoplasms,  Brittle co...</td>\n",
              "      <td>[breast]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>GSE96085_GPL15084</td>\n",
              "      <td>[Carcinoma,  Neoplasms,  Second Primary,  Brea...</td>\n",
              "      <td>[mammary gland]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>GSE96520_GPL4135</td>\n",
              "      <td>[Mammary Neoplasms,  Animal,  Breast Neoplasms]</td>\n",
              "      <td>[mammary gland]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>GSE96567_GPL15084</td>\n",
              "      <td>[Breast Neoplasms,  Carcinoma,  Neoplasms,  Se...</td>\n",
              "      <td>[mammary gland]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2218</th>\n",
              "      <td>GSE38912_GPL11154</td>\n",
              "      <td>[Breast Neoplasms]</td>\n",
              "      <td>[colon,  breast]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2219</th>\n",
              "      <td>GSE38912_GPL15433</td>\n",
              "      <td>[Breast Neoplasms]</td>\n",
              "      <td>[colon,  breast]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2220</th>\n",
              "      <td>GSE38912_GPL3921</td>\n",
              "      <td>[Breast Neoplasms]</td>\n",
              "      <td>[colon,  breast]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2221</th>\n",
              "      <td>GSE38912_GPL9052</td>\n",
              "      <td>[Breast Neoplasms]</td>\n",
              "      <td>[breast,  colon]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2222</th>\n",
              "      <td>GSE38959_GPL4133</td>\n",
              "      <td>[Triple Negative Breast Neoplasms]</td>\n",
              "      <td>[heart,  kidney,  mammary gland,  lung,  liver]</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>2223 rows \u00d7 3 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "             dataset_id                                    curated_disease  \\\n",
              "0     GSE96058_GPL11154  [Triple Negative Breast Neoplasms,  Brittle co...   \n",
              "1     GSE96058_GPL18573  [Triple Negative Breast Neoplasms,  Brittle co...   \n",
              "2     GSE96085_GPL15084  [Carcinoma,  Neoplasms,  Second Primary,  Brea...   \n",
              "3      GSE96520_GPL4135    [Mammary Neoplasms,  Animal,  Breast Neoplasms]   \n",
              "4     GSE96567_GPL15084  [Breast Neoplasms,  Carcinoma,  Neoplasms,  Se...   \n",
              "...                 ...                                                ...   \n",
              "2218  GSE38912_GPL11154                                 [Breast Neoplasms]   \n",
              "2219  GSE38912_GPL15433                                 [Breast Neoplasms]   \n",
              "2220   GSE38912_GPL3921                                 [Breast Neoplasms]   \n",
              "2221   GSE38912_GPL9052                                 [Breast Neoplasms]   \n",
              "2222   GSE38959_GPL4133                 [Triple Negative Breast Neoplasms]   \n",
              "\n",
              "                                       curated_tissue  \n",
              "0                                            [breast]  \n",
              "1                                            [breast]  \n",
              "2                                     [mammary gland]  \n",
              "3                                     [mammary gland]  \n",
              "4                                     [mammary gland]  \n",
              "...                                               ...  \n",
              "2218                                 [colon,  breast]  \n",
              "2219                                 [colon,  breast]  \n",
              "2220                                 [colon,  breast]  \n",
              "2221                                 [breast,  colon]  \n",
              "2222  [heart,  kidney,  mammary gland,  lung,  liver]  \n",
              "\n",
              "[2223 rows x 3 columns]"
            ]
          },
          "execution_count": 6,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "sql_query = \"\"\"SELECT dataset_id, curated_disease, curated_tissue FROM geo.datasets WHERE \n",
        "        CONTAINS(curated_disease, recommend('curated_disease', 'breast neoplasms', 'related')) AND \n",
        "        CONTAINS(curated_tissue, recommend('curated_tissue', 'breast', 'related'))\"\"\" \n",
        "result = omixatlas.query_metadata(sql_query)\n",
        "result"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## Other query examples on V2 infrastructure"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 7,
      "metadata": {
        "scrolled": true
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Query execution succeeded (time taken: 2.07 seconds, data scanned: 38.845 MB)\n",
            "Fetched 554 rows\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>data_matrix_available</th>\n",
              "      <th>curated_organism</th>\n",
              "      <th>src_uri</th>\n",
              "      <th>total_num_samples</th>\n",
              "      <th>year</th>\n",
              "      <th>description</th>\n",
              "      <th>curated_cell_line</th>\n",
              "      <th>data_table_name</th>\n",
              "      <th>data_table_version</th>\n",
              "      <th>platform</th>\n",
              "      <th>...</th>\n",
              "      <th>abstract</th>\n",
              "      <th>version</th>\n",
              "      <th>curated_strain</th>\n",
              "      <th>bucket</th>\n",
              "      <th>curated_tissue</th>\n",
              "      <th>dataset_source</th>\n",
              "      <th>data_type</th>\n",
              "      <th>overall_design</th>\n",
              "      <th>is_current</th>\n",
              "      <th>region</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>NaN</td>\n",
              "      <td>[Homo sapiens]</td>\n",
              "      <td>polly:data://GEO_data_lake/data/Microarray/GSE...</td>\n",
              "      <td>34.0</td>\n",
              "      <td>2018.0</td>\n",
              "      <td>Role of Humoral Immunity against Hepatitis B V...</td>\n",
              "      <td>[None]</td>\n",
              "      <td>geo__gse96851_gpl570</td>\n",
              "      <td>0.0</td>\n",
              "      <td>Microarray</td>\n",
              "      <td>...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0</td>\n",
              "      <td>[None]</td>\n",
              "      <td>discover-prod-datalake-v1</td>\n",
              "      <td>[liver]</td>\n",
              "      <td>GEO</td>\n",
              "      <td>Transcriptomics</td>\n",
              "      <td>Liver samples were obtained from 4 patients wi...</td>\n",
              "      <td>True</td>\n",
              "      <td>us-west-2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>False</td>\n",
              "      <td>[Homo sapiens]</td>\n",
              "      <td>polly:data://GEO_data_lake/data/GEO_metadata/G...</td>\n",
              "      <td>54.0</td>\n",
              "      <td>2017.0</td>\n",
              "      <td>A Pharmacogenomic Landscape in Human Liver Can...</td>\n",
              "      <td>[SK-HEP-1,  CLC33,  CLC26,  CLC17,  CLC30,  HL...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>RNAseq</td>\n",
              "      <td>...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0</td>\n",
              "      <td>[None]</td>\n",
              "      <td>discover-prod-datalake-v1</td>\n",
              "      <td>[None]</td>\n",
              "      <td>GEO</td>\n",
              "      <td>Transcriptomics</td>\n",
              "      <td>RNAseq for 81 liver cancer cell models was per...</td>\n",
              "      <td>True</td>\n",
              "      <td>us-west-2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>False</td>\n",
              "      <td>[Homo sapiens]</td>\n",
              "      <td>polly:data://GEO_data_lake/data/GEO_metadata/G...</td>\n",
              "      <td>3.0</td>\n",
              "      <td>2017.0</td>\n",
              "      <td>A Pharmacogenomic Landscape in Human Liver Can...</td>\n",
              "      <td>[SK-HEP-1,  CLC49,  CLC26,  SNU-354,  Mahlavu,...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>RNAseq</td>\n",
              "      <td>...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0</td>\n",
              "      <td>[None]</td>\n",
              "      <td>discover-prod-datalake-v1</td>\n",
              "      <td>[None]</td>\n",
              "      <td>GEO</td>\n",
              "      <td>Transcriptomics</td>\n",
              "      <td>RNAseq for 81 liver cancer cell models was per...</td>\n",
              "      <td>True</td>\n",
              "      <td>us-west-2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>False</td>\n",
              "      <td>[Homo sapiens]</td>\n",
              "      <td>polly:data://GEO_data_lake/data/GEO_metadata/G...</td>\n",
              "      <td>16.0</td>\n",
              "      <td>2018.0</td>\n",
              "      <td>A Pharmacogenomic Landscape in Human Liver Can...</td>\n",
              "      <td>[SK-HEP-1,  SNU-398,  JHH-4,  SNU-886,  CLC43,...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>RNAseq</td>\n",
              "      <td>...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0</td>\n",
              "      <td>[None]</td>\n",
              "      <td>discover-prod-datalake-v1</td>\n",
              "      <td>[None]</td>\n",
              "      <td>GEO</td>\n",
              "      <td>Transcriptomics</td>\n",
              "      <td>RNAseq for 81 liver cancer cell models was per...</td>\n",
              "      <td>True</td>\n",
              "      <td>us-west-2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>NaN</td>\n",
              "      <td>[Mus musculus]</td>\n",
              "      <td>polly:data://GEO_data_lake/data/RNASeq/GSE9723...</td>\n",
              "      <td>10.0</td>\n",
              "      <td>2018.0</td>\n",
              "      <td>Pyroptosis by Caspase11/4-Gasdermin-D Pathway ...</td>\n",
              "      <td>[None]</td>\n",
              "      <td>geo__gse97234_gpl13112</td>\n",
              "      <td>0.0</td>\n",
              "      <td>RNASeq</td>\n",
              "      <td>...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0</td>\n",
              "      <td>[C57BL/6]</td>\n",
              "      <td>discover-prod-datalake-v1</td>\n",
              "      <td>[liver]</td>\n",
              "      <td>GEO</td>\n",
              "      <td>Transcriptomics</td>\n",
              "      <td>9 total samples = 3 AH liver, 3 ASH liver, 3 c...</td>\n",
              "      <td>True</td>\n",
              "      <td>us-west-2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>549</th>\n",
              "      <td>NaN</td>\n",
              "      <td>[Mus musculus]</td>\n",
              "      <td>polly:data://GEO_data_lake/data/Microarray/GSE...</td>\n",
              "      <td>20.0</td>\n",
              "      <td>2019.0</td>\n",
              "      <td>Analysis of differentially expressed genes in ...</td>\n",
              "      <td>[None]</td>\n",
              "      <td>geo__gse138916_gpl21163</td>\n",
              "      <td>0.0</td>\n",
              "      <td>Microarray</td>\n",
              "      <td>...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0</td>\n",
              "      <td>[None]</td>\n",
              "      <td>discover-prod-datalake-v1</td>\n",
              "      <td>[liver]</td>\n",
              "      <td>GEO</td>\n",
              "      <td>Transcriptomics</td>\n",
              "      <td>snap-frozen liver samples were obtained from g...</td>\n",
              "      <td>True</td>\n",
              "      <td>us-west-2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>550</th>\n",
              "      <td>NaN</td>\n",
              "      <td>[Mus musculus]</td>\n",
              "      <td>polly:data://GEO_data_lake/data/RNASeq/GSE1389...</td>\n",
              "      <td>10.0</td>\n",
              "      <td>2019.0</td>\n",
              "      <td>MBOAT7's role in the progression of Non-alcoho...</td>\n",
              "      <td>[None]</td>\n",
              "      <td>geo__gse138945_gpl13112</td>\n",
              "      <td>0.0</td>\n",
              "      <td>RNASeq</td>\n",
              "      <td>...</td>\n",
              "      <td>Recent studies have identified a genetic varia...</td>\n",
              "      <td>0</td>\n",
              "      <td>[None]</td>\n",
              "      <td>discover-prod-datalake-v1</td>\n",
              "      <td>[liver]</td>\n",
              "      <td>GEO</td>\n",
              "      <td>Transcriptomics</td>\n",
              "      <td>RNAseq of liver homogenates from high fat diet...</td>\n",
              "      <td>True</td>\n",
              "      <td>us-west-2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>551</th>\n",
              "      <td>NaN</td>\n",
              "      <td>[Mus musculus]</td>\n",
              "      <td>polly:data://GEO_data_lake/data/RNASeq/GSE1389...</td>\n",
              "      <td>10.0</td>\n",
              "      <td>2019.0</td>\n",
              "      <td>LPI's role in the progression of Non-alcoholic...</td>\n",
              "      <td>[None]</td>\n",
              "      <td>geo__gse138946_gpl13112</td>\n",
              "      <td>0.0</td>\n",
              "      <td>RNASeq</td>\n",
              "      <td>...</td>\n",
              "      <td>Recent studies have identified a genetic varia...</td>\n",
              "      <td>0</td>\n",
              "      <td>[None]</td>\n",
              "      <td>discover-prod-datalake-v1</td>\n",
              "      <td>[liver]</td>\n",
              "      <td>GEO</td>\n",
              "      <td>Transcriptomics</td>\n",
              "      <td>RNAseq of Liver homogenate +/- 18:0 Lysophosph...</td>\n",
              "      <td>True</td>\n",
              "      <td>us-west-2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>552</th>\n",
              "      <td>False</td>\n",
              "      <td>[Homo sapiens]</td>\n",
              "      <td>polly:data://GEO_data_lake/data/GEO_metadata/G...</td>\n",
              "      <td>3.0</td>\n",
              "      <td>2019.0</td>\n",
              "      <td>Rimonabant suppresses RNA transcription of hep...</td>\n",
              "      <td>[Hep-G2]</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>RNAseq</td>\n",
              "      <td>...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0</td>\n",
              "      <td>[None]</td>\n",
              "      <td>discover-prod-datalake-v1</td>\n",
              "      <td>[None]</td>\n",
              "      <td>GEO</td>\n",
              "      <td>Transcriptomics</td>\n",
              "      <td>Transcriptome analysis of PHH treated with DMS...</td>\n",
              "      <td>True</td>\n",
              "      <td>us-west-2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>553</th>\n",
              "      <td>False</td>\n",
              "      <td>[Mus musculus]</td>\n",
              "      <td>polly:data://GEO_data_lake/data/GEO_metadata/G...</td>\n",
              "      <td>21.0</td>\n",
              "      <td>2019.0</td>\n",
              "      <td>Loss of hepatic Mboat7 leads to liver fibrosis.</td>\n",
              "      <td>[None]</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0</td>\n",
              "      <td>[None]</td>\n",
              "      <td>discover-prod-datalake-v1</td>\n",
              "      <td>[adipose tissue,  liver]</td>\n",
              "      <td>GEO</td>\n",
              "      <td>Transcriptomics</td>\n",
              "      <td>Identification of deregulated gene expression ...</td>\n",
              "      <td>True</td>\n",
              "      <td>us-west-2</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>554 rows \u00d7 35 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "    data_matrix_available curated_organism  \\\n",
              "0                     NaN   [Homo sapiens]   \n",
              "1                   False   [Homo sapiens]   \n",
              "2                   False   [Homo sapiens]   \n",
              "3                   False   [Homo sapiens]   \n",
              "4                     NaN   [Mus musculus]   \n",
              "..                    ...              ...   \n",
              "549                   NaN   [Mus musculus]   \n",
              "550                   NaN   [Mus musculus]   \n",
              "551                   NaN   [Mus musculus]   \n",
              "552                 False   [Homo sapiens]   \n",
              "553                 False   [Mus musculus]   \n",
              "\n",
              "                                               src_uri  total_num_samples  \\\n",
              "0    polly:data://GEO_data_lake/data/Microarray/GSE...               34.0   \n",
              "1    polly:data://GEO_data_lake/data/GEO_metadata/G...               54.0   \n",
              "2    polly:data://GEO_data_lake/data/GEO_metadata/G...                3.0   \n",
              "3    polly:data://GEO_data_lake/data/GEO_metadata/G...               16.0   \n",
              "4    polly:data://GEO_data_lake/data/RNASeq/GSE9723...               10.0   \n",
              "..                                                 ...                ...   \n",
              "549  polly:data://GEO_data_lake/data/Microarray/GSE...               20.0   \n",
              "550  polly:data://GEO_data_lake/data/RNASeq/GSE1389...               10.0   \n",
              "551  polly:data://GEO_data_lake/data/RNASeq/GSE1389...               10.0   \n",
              "552  polly:data://GEO_data_lake/data/GEO_metadata/G...                3.0   \n",
              "553  polly:data://GEO_data_lake/data/GEO_metadata/G...               21.0   \n",
              "\n",
              "       year                                        description  \\\n",
              "0    2018.0  Role of Humoral Immunity against Hepatitis B V...   \n",
              "1    2017.0  A Pharmacogenomic Landscape in Human Liver Can...   \n",
              "2    2017.0  A Pharmacogenomic Landscape in Human Liver Can...   \n",
              "3    2018.0  A Pharmacogenomic Landscape in Human Liver Can...   \n",
              "4    2018.0  Pyroptosis by Caspase11/4-Gasdermin-D Pathway ...   \n",
              "..      ...                                                ...   \n",
              "549  2019.0  Analysis of differentially expressed genes in ...   \n",
              "550  2019.0  MBOAT7's role in the progression of Non-alcoho...   \n",
              "551  2019.0  LPI's role in the progression of Non-alcoholic...   \n",
              "552  2019.0  Rimonabant suppresses RNA transcription of hep...   \n",
              "553  2019.0    Loss of hepatic Mboat7 leads to liver fibrosis.   \n",
              "\n",
              "                                     curated_cell_line  \\\n",
              "0                                               [None]   \n",
              "1    [SK-HEP-1,  CLC33,  CLC26,  CLC17,  CLC30,  HL...   \n",
              "2    [SK-HEP-1,  CLC49,  CLC26,  SNU-354,  Mahlavu,...   \n",
              "3    [SK-HEP-1,  SNU-398,  JHH-4,  SNU-886,  CLC43,...   \n",
              "4                                               [None]   \n",
              "..                                                 ...   \n",
              "549                                             [None]   \n",
              "550                                             [None]   \n",
              "551                                             [None]   \n",
              "552                                           [Hep-G2]   \n",
              "553                                             [None]   \n",
              "\n",
              "             data_table_name  data_table_version    platform  ...  \\\n",
              "0       geo__gse96851_gpl570                 0.0  Microarray  ...   \n",
              "1                        NaN                 NaN      RNAseq  ...   \n",
              "2                        NaN                 NaN      RNAseq  ...   \n",
              "3                        NaN                 NaN      RNAseq  ...   \n",
              "4     geo__gse97234_gpl13112                 0.0      RNASeq  ...   \n",
              "..                       ...                 ...         ...  ...   \n",
              "549  geo__gse138916_gpl21163                 0.0  Microarray  ...   \n",
              "550  geo__gse138945_gpl13112                 0.0      RNASeq  ...   \n",
              "551  geo__gse138946_gpl13112                 0.0      RNASeq  ...   \n",
              "552                      NaN                 NaN      RNAseq  ...   \n",
              "553                      NaN                 NaN         NaN  ...   \n",
              "\n",
              "                                              abstract version curated_strain  \\\n",
              "0                                                  NaN       0         [None]   \n",
              "1                                                  NaN       0         [None]   \n",
              "2                                                  NaN       0         [None]   \n",
              "3                                                  NaN       0         [None]   \n",
              "4                                                  NaN       0      [C57BL/6]   \n",
              "..                                                 ...     ...            ...   \n",
              "549                                                NaN       0         [None]   \n",
              "550  Recent studies have identified a genetic varia...       0         [None]   \n",
              "551  Recent studies have identified a genetic varia...       0         [None]   \n",
              "552                                                NaN       0         [None]   \n",
              "553                                                NaN       0         [None]   \n",
              "\n",
              "                        bucket            curated_tissue dataset_source  \\\n",
              "0    discover-prod-datalake-v1                   [liver]            GEO   \n",
              "1    discover-prod-datalake-v1                    [None]            GEO   \n",
              "2    discover-prod-datalake-v1                    [None]            GEO   \n",
              "3    discover-prod-datalake-v1                    [None]            GEO   \n",
              "4    discover-prod-datalake-v1                   [liver]            GEO   \n",
              "..                         ...                       ...            ...   \n",
              "549  discover-prod-datalake-v1                   [liver]            GEO   \n",
              "550  discover-prod-datalake-v1                   [liver]            GEO   \n",
              "551  discover-prod-datalake-v1                   [liver]            GEO   \n",
              "552  discover-prod-datalake-v1                    [None]            GEO   \n",
              "553  discover-prod-datalake-v1  [adipose tissue,  liver]            GEO   \n",
              "\n",
              "           data_type                                     overall_design  \\\n",
              "0    Transcriptomics  Liver samples were obtained from 4 patients wi...   \n",
              "1    Transcriptomics  RNAseq for 81 liver cancer cell models was per...   \n",
              "2    Transcriptomics  RNAseq for 81 liver cancer cell models was per...   \n",
              "3    Transcriptomics  RNAseq for 81 liver cancer cell models was per...   \n",
              "4    Transcriptomics  9 total samples = 3 AH liver, 3 ASH liver, 3 c...   \n",
              "..               ...                                                ...   \n",
              "549  Transcriptomics  snap-frozen liver samples were obtained from g...   \n",
              "550  Transcriptomics  RNAseq of liver homogenates from high fat diet...   \n",
              "551  Transcriptomics  RNAseq of Liver homogenate +/- 18:0 Lysophosph...   \n",
              "552  Transcriptomics  Transcriptome analysis of PHH treated with DMS...   \n",
              "553  Transcriptomics  Identification of deregulated gene expression ...   \n",
              "\n",
              "    is_current     region  \n",
              "0         True  us-west-2  \n",
              "1         True  us-west-2  \n",
              "2         True  us-west-2  \n",
              "3         True  us-west-2  \n",
              "4         True  us-west-2  \n",
              "..         ...        ...  \n",
              "549       True  us-west-2  \n",
              "550       True  us-west-2  \n",
              "551       True  us-west-2  \n",
              "552       True  us-west-2  \n",
              "553       True  us-west-2  \n",
              "\n",
              "[554 rows x 35 columns]"
            ]
          },
          "execution_count": 7,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "sql_query = \"\"\"SELECT * FROM geo.datasets WHERE \n",
        "                CONTAINS(curated_disease, recommend('curated_disease', 'hepatitis', 'match'))\"\"\"\n",
        "result = omixatlas.query_metadata(sql_query)\n",
        "result"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 19,
      "metadata": {
        "scrolled": true
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Query execution succeeded (time taken: 3.07 seconds, data scanned: 0.574 MB)\n",
            "Fetched 7216 rows\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>dataset_id</th>\n",
              "      <th>curated_tissue</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>GSE9581_GPL6119</td>\n",
              "      <td>[brain,  liver,  testis,  heart]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>GSE9581_GPL6120</td>\n",
              "      <td>[brain,  liver,  testis,  heart]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>GSE9588_GPL4372</td>\n",
              "      <td>[liver]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>GSE96059_GPL17021</td>\n",
              "      <td>[liver]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>GSE96093_GPL17021</td>\n",
              "      <td>[liver]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7211</th>\n",
              "      <td>GSE75277_GPL1261</td>\n",
              "      <td>[liver]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7212</th>\n",
              "      <td>GSE75285_GPL16298</td>\n",
              "      <td>[liver,  blood,  blastema]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7213</th>\n",
              "      <td>GSE75285_GPL570</td>\n",
              "      <td>[blood,  liver,  blastema]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7214</th>\n",
              "      <td>GSE75285_GPL6801</td>\n",
              "      <td>[blood,  liver,  blastema]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7215</th>\n",
              "      <td>GSE75445_GPL1261</td>\n",
              "      <td>[liver]</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>7216 rows \u00d7 2 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "             dataset_id                    curated_tissue\n",
              "0       GSE9581_GPL6119  [brain,  liver,  testis,  heart]\n",
              "1       GSE9581_GPL6120  [brain,  liver,  testis,  heart]\n",
              "2       GSE9588_GPL4372                           [liver]\n",
              "3     GSE96059_GPL17021                           [liver]\n",
              "4     GSE96093_GPL17021                           [liver]\n",
              "...                 ...                               ...\n",
              "7211   GSE75277_GPL1261                           [liver]\n",
              "7212  GSE75285_GPL16298        [liver,  blood,  blastema]\n",
              "7213    GSE75285_GPL570        [blood,  liver,  blastema]\n",
              "7214   GSE75285_GPL6801        [blood,  liver,  blastema]\n",
              "7215   GSE75445_GPL1261                           [liver]\n",
              "\n",
              "[7216 rows x 2 columns]"
            ]
          },
          "execution_count": 19,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "sql_query = \"\"\"SELECT dataset_id, curated_tissue FROM geo.datasets WHERE \n",
        "            CONTAINS(curated_tissue, recommend('curated_tissue', 'liver', 'related'))\"\"\"\n",
        "result = omixatlas.query_metadata(sql_query)\n",
        "result"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 9,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Query execution succeeded (time taken: 2.15 seconds, data scanned: 0.908 MB)\n",
            "Fetched 2565 rows\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>dataset_id</th>\n",
              "      <th>curated_disease</th>\n",
              "      <th>curated_tissue</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>GSE96058_GPL11154</td>\n",
              "      <td>[Triple Negative Breast Neoplasms,  Brittle co...</td>\n",
              "      <td>[breast]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>GSE96058_GPL18573</td>\n",
              "      <td>[Triple Negative Breast Neoplasms,  Brittle co...</td>\n",
              "      <td>[breast]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>GSE96085_GPL15084</td>\n",
              "      <td>[Carcinoma,  Neoplasms,  Second Primary,  Brea...</td>\n",
              "      <td>[mammary gland]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>GSE96520_GPL4135</td>\n",
              "      <td>[Mammary Neoplasms,  Animal,  Breast Neoplasms]</td>\n",
              "      <td>[mammary gland]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>GSE96567_GPL15084</td>\n",
              "      <td>[Breast Neoplasms,  Carcinoma,  Neoplasms,  Se...</td>\n",
              "      <td>[mammary gland]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2560</th>\n",
              "      <td>GSE95304_GPL11154</td>\n",
              "      <td>[Breast Neoplasms]</td>\n",
              "      <td>[breast]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2561</th>\n",
              "      <td>GSE95472_GPL6244</td>\n",
              "      <td>[Triple Negative Breast Neoplasms]</td>\n",
              "      <td>[breast]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2562</th>\n",
              "      <td>GSE95554_GPL17117</td>\n",
              "      <td>[Breast Neoplasms]</td>\n",
              "      <td>[breast,  oil secretion]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2563</th>\n",
              "      <td>GSE95700_GPL570</td>\n",
              "      <td>[Triple Negative Breast Neoplasms]</td>\n",
              "      <td>[breast]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2564</th>\n",
              "      <td>GSE9574_GPL96</td>\n",
              "      <td>[Breast Neoplasms,  Carcinoma in Situ]</td>\n",
              "      <td>[breast epithelium]</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>2565 rows \u00d7 3 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "             dataset_id                                    curated_disease  \\\n",
              "0     GSE96058_GPL11154  [Triple Negative Breast Neoplasms,  Brittle co...   \n",
              "1     GSE96058_GPL18573  [Triple Negative Breast Neoplasms,  Brittle co...   \n",
              "2     GSE96085_GPL15084  [Carcinoma,  Neoplasms,  Second Primary,  Brea...   \n",
              "3      GSE96520_GPL4135    [Mammary Neoplasms,  Animal,  Breast Neoplasms]   \n",
              "4     GSE96567_GPL15084  [Breast Neoplasms,  Carcinoma,  Neoplasms,  Se...   \n",
              "...                 ...                                                ...   \n",
              "2560  GSE95304_GPL11154                                 [Breast Neoplasms]   \n",
              "2561   GSE95472_GPL6244                 [Triple Negative Breast Neoplasms]   \n",
              "2562  GSE95554_GPL17117                                 [Breast Neoplasms]   \n",
              "2563    GSE95700_GPL570                 [Triple Negative Breast Neoplasms]   \n",
              "2564      GSE9574_GPL96             [Breast Neoplasms,  Carcinoma in Situ]   \n",
              "\n",
              "                curated_tissue  \n",
              "0                     [breast]  \n",
              "1                     [breast]  \n",
              "2              [mammary gland]  \n",
              "3              [mammary gland]  \n",
              "4              [mammary gland]  \n",
              "...                        ...  \n",
              "2560                  [breast]  \n",
              "2561                  [breast]  \n",
              "2562  [breast,  oil secretion]  \n",
              "2563                  [breast]  \n",
              "2564       [breast epithelium]  \n",
              "\n",
              "[2565 rows x 3 columns]"
            ]
          },
          "execution_count": 9,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "sql_query = \"\"\"SELECT dataset_id, curated_disease, curated_tissue FROM geo.datasets \n",
        "WHERE (CONTAINS(curated_disease, recommend('curated_disease', 'breast neoplasms', 'related')) OR\n",
        "CONTAINS(curated_disease, recommend('curated_disease', 'pancreatic neoplasms', 'related')))AND \n",
        "(CONTAINS(curated_tissue, recommend('curated_tissue', 'breast', 'related')) OR \n",
        "CONTAINS(curated_tissue, recommend('curated_tissue', 'pancreas', 'related')))\"\"\"\n",
        "result = omixatlas.query_metadata(sql_query)\n",
        "result"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": []
    }
  ],
  "metadata": {
    "kernelspec": {
      "display_name": "Python 3",
      "language": "python",
      "name": "python3"
    },
    "language_info": {
      "codemirror_mode": {
        "name": "ipython",
        "version": 3
      },
      "file_extension": ".py",
      "mimetype": "text/x-python",
      "name": "python",
      "nbconvert_exporter": "python",
      "pygments_lexer": "ipython3",
      "version": "3.6.9"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 2
}