{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "colab_type": "text",
        "id": "view-in-github"
      },
      "source": [
        "<a href=\"https://polly.elucidata.io/manage/workspaces?action=open_polly_notebook&amp;source=github&amp;path=ElucidataInc%2Fpolly-python%2Fblob%2Fmain%2FDiscover%2Fpolly_python_query_examples.ipynb&amp;kernel=elucidata%2FPython%203.10;machine=medium\" target=\"_parent\"><img alt=\"Open in Polly\" src=\"https://elucidatainc.github.io/PublicAssets/open_polly.svg\"/></a>\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "# Welcome to polly-python Notebook consisting of sample queries for after data infrastructure upgrade\n",
        "\n",
        "\n",
        "For more details on how to use Notebooks on Polly, please visit [Polly Notebooks](https://docs.elucidata.io/Scaling%20compute/Polly%20Notebooks.html).\n",
        "\n",
        "For more details on API access to your OmixAtlas, please visit [Accessing OmixAtlas using polly-python through Polly Notebooks](https://docs.elucidata.io/OmixAtlas/Polly%20Python.html)."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {},
      "outputs": [],
      "source": [
        "# please do not modify\n",
        "from IPython.display import display_html\n",
        "def restartkernel() :\n",
        "    display_html(\"<script>Jupyter.notebook.kernel.restart()</script>\",raw=True)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## Install polly-python"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 3,
      "metadata": {
        "scrolled": true
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Requirement already satisfied: polly-python in /usr/local/lib/python3.6/dist-packages (0.1.0)\n",
            "Requirement already satisfied: python-dateutil==2.8.1 in /usr/local/lib/python3.6/dist-packages (from polly-python) (2.8.1)\n",
            "Requirement already satisfied: chardet==4.0.0 in /usr/local/lib/python3.6/dist-packages (from polly-python) (4.0.0)\n",
            "Requirement already satisfied: elucidatacloudpathlib==0.6.6 in /usr/local/lib/python3.6/dist-packages (from polly-python) (0.6.6)\n",
            "Requirement already satisfied: rst2txt in /usr/local/lib/python3.6/dist-packages (from polly-python) (1.1.0)\n",
            "Requirement already satisfied: boto3>=1.17.73 in /usr/local/lib/python3.6/dist-packages (from polly-python) (1.23.5)\n",
            "Requirement already satisfied: cmapPy in /usr/local/lib/python3.6/dist-packages (from polly-python) (4.0.1)\n",
            "Requirement already satisfied: postpy2==0.0.6 in /usr/local/lib/python3.6/dist-packages (from polly-python) (0.0.6)\n",
            "Requirement already satisfied: requests==2.25.1 in /usr/local/lib/python3.6/dist-packages (from polly-python) (2.25.1)\n",
            "Requirement already satisfied: six==1.16.0 in /usr/local/lib/python3.6/dist-packages (from polly-python) (1.16.0)\n",
            "Requirement already satisfied: retrying==1.3.3 in /usr/local/lib/python3.6/dist-packages (from polly-python) (1.3.3)\n",
            "Requirement already satisfied: black in /usr/local/lib/python3.6/dist-packages (from polly-python) (22.3.0)\n",
            "Requirement already satisfied: python-magic==0.4.24 in /usr/local/lib/python3.6/dist-packages (from polly-python) (0.4.24)\n",
            "Requirement already satisfied: idna==2.10 in /usr/local/lib/python3.6/dist-packages (from polly-python) (2.10)\n",
            "Requirement already satisfied: pandas in /usr/local/lib/python3.6/dist-packages (from polly-python) (1.0.1)\n",
            "Requirement already satisfied: botocore>=1.20.73 in /usr/local/lib/python3.6/dist-packages (from polly-python) (1.20.112)\n",
            "Requirement already satisfied: Cerberus==1.3.2 in /usr/local/lib/python3.6/dist-packages/Cerberus-1.3.2-py3.6.egg (from polly-python) (1.3.2)\n",
            "Requirement already satisfied: pytz==2021.1 in /usr/local/lib/python3.6/dist-packages (from polly-python) (2021.1)\n",
            "Requirement already satisfied: certifi==2021.10.8 in /usr/local/lib/python3.6/dist-packages (from polly-python) (2021.10.8)\n",
            "Requirement already satisfied: joblib in /usr/local/lib/python3.6/dist-packages (from polly-python) (1.1.0)\n",
            "Requirement already satisfied: urllib3==1.26.6 in /usr/local/lib/python3.6/dist-packages (from polly-python) (1.26.6)\n",
            "Requirement already satisfied: Deprecated in /usr/local/lib/python3.6/dist-packages (from polly-python) (1.2.13)\n",
            "Requirement already satisfied: importlib-metadata; python_version < \"3.8\" in /usr/local/lib/python3.6/dist-packages (from elucidatacloudpathlib==0.6.6->polly-python) (1.5.0)\n",
            "Requirement already satisfied: docutils>=0.11 in /usr/local/lib/python3.6/dist-packages (from rst2txt->polly-python) (0.15.2)\n",
            "Requirement already satisfied: pygments in /usr/local/lib/python3.6/dist-packages (from rst2txt->polly-python) (2.5.2)\n",
            "Requirement already satisfied: jmespath<2.0.0,>=0.7.1 in /usr/local/lib/python3.6/dist-packages (from boto3>=1.17.73->polly-python) (0.9.5)\n",
            "Requirement already satisfied: s3transfer<0.6.0,>=0.5.0 in /usr/local/lib/python3.6/dist-packages (from boto3>=1.17.73->polly-python) (0.5.2)\n",
            "Requirement already satisfied: numpy>=1.11.2 in /usr/local/lib/python3.6/dist-packages (from cmapPy->polly-python) (1.18.1)\n",
            "Requirement already satisfied: h5py>=2.6.0 in /usr/local/lib/python3.6/dist-packages (from cmapPy->polly-python) (3.1.0)\n",
            "Requirement already satisfied: pathspec>=0.9.0 in /usr/local/lib/python3.6/dist-packages (from black->polly-python) (0.9.0)\n",
            "Requirement already satisfied: typed-ast>=1.4.2; python_version < \"3.8\" and implementation_name == \"cpython\" in /usr/local/lib/python3.6/dist-packages (from black->polly-python) (1.5.4)\n",
            "Requirement already satisfied: mypy-extensions>=0.4.3 in /usr/local/lib/python3.6/dist-packages (from black->polly-python) (0.4.3)\n",
            "Requirement already satisfied: platformdirs>=2 in /usr/local/lib/python3.6/dist-packages (from black->polly-python) (2.4.0)\n",
            "Requirement already satisfied: dataclasses>=0.6; python_version < \"3.7\" in /usr/local/lib/python3.6/dist-packages (from black->polly-python) (0.8)\n",
            "Requirement already satisfied: typing-extensions>=3.10.0.0; python_version < \"3.10\" in /usr/local/lib/python3.6/dist-packages (from black->polly-python) (4.1.1)\n",
            "Requirement already satisfied: tomli>=1.1.0; python_version < \"3.11\" in /usr/local/lib/python3.6/dist-packages (from black->polly-python) (1.2.3)\n",
            "Requirement already satisfied: click>=8.0.0 in /usr/local/lib/python3.6/dist-packages (from black->polly-python) (8.0.4)\n",
            "Requirement already satisfied: setuptools in /usr/local/lib/python3.6/dist-packages (from Cerberus==1.3.2->polly-python) (45.2.0)\n",
            "Requirement already satisfied: wrapt<2,>=1.10 in /usr/local/lib/python3.6/dist-packages (from Deprecated->polly-python) (1.14.1)\n",
            "Requirement already satisfied: zipp>=0.5 in /usr/local/lib/python3.6/dist-packages (from importlib-metadata; python_version < \"3.8\"->elucidatacloudpathlib==0.6.6->polly-python) (3.1.0)\n",
            "Requirement already satisfied: cached-property; python_version < \"3.8\" in /usr/local/lib/python3.6/dist-packages (from h5py>=2.6.0->cmapPy->polly-python) (1.5.2)\n",
            "\u001b[33mWARNING: You are using pip version 19.2.3, however version 21.3.1 is available.\n",
            "You should consider upgrading via the 'pip install --upgrade pip' command.\u001b[0m\n"
          ]
        }
      ],
      "source": [
        "!sudo pip3 install polly-python"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/html": [
              "<script>Jupyter.notebook.kernel.restart()</script>"
            ]
          },
          "metadata": {},
          "output_type": "display_data"
        }
      ],
      "source": [
        "restartkernel() #Pause for a few seconds before the kernel is refreshed"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/html": [
              "<script type=\"text/javascript\"> Jupyter.notebook.kernel.execute(\"url = '\" + window.location + \"'\", {}, {}); </script>"
            ],
            "text/plain": [
              "<IPython.core.display.HTML object>"
            ]
          },
          "execution_count": 1,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "# please do not modify\n",
        "from IPython.display import HTML\n",
        "HTML('''<script type=\"text/javascript\"> Jupyter.notebook.kernel.execute(\"url = '\" + window.location + \"'\", {}, {}); </script>''')"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## Import required dependencies"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 4,
      "metadata": {},
      "outputs": [],
      "source": [
        "import os\n",
        "from polly.auth import Polly\n",
        "from polly.omixatlas import OmixAtlas\n",
        "from polly.workspaces import Workspaces"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## Auth with token on Polly"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 5,
      "metadata": {},
      "outputs": [],
      "source": [
        "AUTH_TOKEN=(os.environ['POLLY_REFRESH_TOKEN']) # Obtain authentication tokens\n",
        "Polly.auth(AUTH_TOKEN)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 6,
      "metadata": {},
      "outputs": [],
      "source": [
        "pd.set_option('display.max_columns', 500)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 7,
      "metadata": {},
      "outputs": [],
      "source": [
        "#Defining omixatlas object\n",
        "omixatlas = OmixAtlas()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## Get basic details of all the omixatlas users have access to"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 5,
      "metadata": {},
      "outputs": [],
      "source": [
        "all_omix_atlas = pd.DataFrame.from_dict(omixatlas.get_all_omixatlas()['data'])"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 7,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>repo_name</th>\n",
              "      <th>repo_id</th>\n",
              "      <th>indexes</th>\n",
              "      <th>diseases</th>\n",
              "      <th>organisms</th>\n",
              "      <th>sources</th>\n",
              "      <th>datatypes</th>\n",
              "      <th>dataset_count</th>\n",
              "      <th>disease_count</th>\n",
              "      <th>tissue_count</th>\n",
              "      <th>organism_count</th>\n",
              "      <th>cell_line_count</th>\n",
              "      <th>cell_type_count</th>\n",
              "      <th>drug_count</th>\n",
              "      <th>data_type_count</th>\n",
              "      <th>data_source_count</th>\n",
              "      <th>sample_count</th>\n",
              "      <th>normal_sample_count</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>hpa</td>\n",
              "      <td>1619514006627</td>\n",
              "      <td>{'gct_metadata': 'hpa_gct_metadata', 'h5ad_met...</td>\n",
              "      <td>[normal]</td>\n",
              "      <td>[homo sapiens, sus (pig), mus musculus]</td>\n",
              "      <td>[hpa]</td>\n",
              "      <td>[transcriptomics, gene expression reliability]</td>\n",
              "      <td>1755</td>\n",
              "      <td>1</td>\n",
              "      <td>78</td>\n",
              "      <td>3</td>\n",
              "      <td>70</td>\n",
              "      <td>150</td>\n",
              "      <td>1</td>\n",
              "      <td>2</td>\n",
              "      <td>1</td>\n",
              "      <td>678</td>\n",
              "      <td>678</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>gdc</td>\n",
              "      <td>1623221686703</td>\n",
              "      <td>{'gct_metadata': 'gdc_gct_metadata', 'h5ad_met...</td>\n",
              "      <td>[leukemia, myeloid, acute, precursor t-cell ly...</td>\n",
              "      <td>[homo sapiens]</td>\n",
              "      <td>[gdc]</td>\n",
              "      <td>[mirna expression, transcriptomics, copy numbe...</td>\n",
              "      <td>9869</td>\n",
              "      <td>75</td>\n",
              "      <td>41</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>3</td>\n",
              "      <td>1</td>\n",
              "      <td>5353</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>geo</td>\n",
              "      <td>9</td>\n",
              "      <td>{'gct_metadata': 'geo_gct_metadata', 'h5ad_met...</td>\n",
              "      <td>[normal, neoplasms, breast neoplasms, obesity,...</td>\n",
              "      <td>[homo sapiens, mus musculus, rattus norvegicus...</td>\n",
              "      <td>[geo]</td>\n",
              "      <td>[transcriptomics, raw counts transcriptomics]</td>\n",
              "      <td>10000</td>\n",
              "      <td>3030</td>\n",
              "      <td>985</td>\n",
              "      <td>68</td>\n",
              "      <td>4970</td>\n",
              "      <td>780</td>\n",
              "      <td>1379</td>\n",
              "      <td>2</td>\n",
              "      <td>1</td>\n",
              "      <td>1280558</td>\n",
              "      <td>840747</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>enterprise_atlas</td>\n",
              "      <td>1638441282192</td>\n",
              "      <td>{'gct_metadata': 'enterprise_atlas_gct_metadat...</td>\n",
              "      <td>[breast neoplasms, endometrial neoplasms, endo...</td>\n",
              "      <td>[homo sapiens, mus musculus]</td>\n",
              "      <td>[geo, proprietary]</td>\n",
              "      <td>[transcriptomics, single cell, mutation, mirna]</td>\n",
              "      <td>176</td>\n",
              "      <td>35</td>\n",
              "      <td>25</td>\n",
              "      <td>2</td>\n",
              "      <td>96</td>\n",
              "      <td>40</td>\n",
              "      <td>22</td>\n",
              "      <td>4</td>\n",
              "      <td>2</td>\n",
              "      <td>8952</td>\n",
              "      <td>3937</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>cptac</td>\n",
              "      <td>1609924165364</td>\n",
              "      <td>{'gct_metadata': 'cptac_gct_metadata', 'h5ad_m...</td>\n",
              "      <td>[carcinoma, hepatocellular, glucose-galactose ...</td>\n",
              "      <td>[homo sapiens]</td>\n",
              "      <td>[cptac]</td>\n",
              "      <td>[proteomics, phosphoproteomics, transcriptomic...</td>\n",
              "      <td>6554</td>\n",
              "      <td>13</td>\n",
              "      <td>33</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>10</td>\n",
              "      <td>1</td>\n",
              "      <td>2440</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>liveromix_atlas</td>\n",
              "      <td>1615965444377</td>\n",
              "      <td>{'gct_metadata': 'liveromix_atlas_gct_metadata...</td>\n",
              "      <td>[normal, carcinoma, hepatocellular, obesity, n...</td>\n",
              "      <td>[homo sapiens, mus musculus, rattus norvegicus...</td>\n",
              "      <td>[geo, lincs, tcga, metabolomics workbench, met...</td>\n",
              "      <td>[transcriptomics, mutation, metabolomics, sing...</td>\n",
              "      <td>6761</td>\n",
              "      <td>752</td>\n",
              "      <td>60</td>\n",
              "      <td>22</td>\n",
              "      <td>347</td>\n",
              "      <td>310</td>\n",
              "      <td>4704</td>\n",
              "      <td>12</td>\n",
              "      <td>10</td>\n",
              "      <td>143535</td>\n",
              "      <td>110202</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>cbioportal</td>\n",
              "      <td>1623986995264</td>\n",
              "      <td>{'gct_metadata': 'cbioportal_gct_metadata', 'h...</td>\n",
              "      <td>[prostatic neoplasms, carcinoma, ductal, breas...</td>\n",
              "      <td>[homo sapiens]</td>\n",
              "      <td>[cbioportal]</td>\n",
              "      <td>[mutation, copy number variation, fusion, tran...</td>\n",
              "      <td>10000</td>\n",
              "      <td>330</td>\n",
              "      <td>262</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>5</td>\n",
              "      <td>1</td>\n",
              "      <td>58994</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>pcd</td>\n",
              "      <td>1622113130397</td>\n",
              "      <td>{'gct_metadata': 'pcd_gct_metadata', 'h5ad_met...</td>\n",
              "      <td>[none, adenocarcinoma, carcinoma, squamous cel...</td>\n",
              "      <td>[homo sapiens]</td>\n",
              "      <td>[pharmacodb]</td>\n",
              "      <td>[drug response]</td>\n",
              "      <td>10000</td>\n",
              "      <td>133</td>\n",
              "      <td>34</td>\n",
              "      <td>1</td>\n",
              "      <td>1400</td>\n",
              "      <td>1</td>\n",
              "      <td>749</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>653500</td>\n",
              "      <td>262</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8</th>\n",
              "      <td>lincs</td>\n",
              "      <td>32</td>\n",
              "      <td>{'gct_metadata': 'lincs_gct_metadata', 'h5ad_m...</td>\n",
              "      <td>[prostatic neoplasms, normal, colorectal neopl...</td>\n",
              "      <td>[homo sapiens]</td>\n",
              "      <td>[geo, geo, lincs]</td>\n",
              "      <td>[transcriptomics]</td>\n",
              "      <td>10000</td>\n",
              "      <td>34</td>\n",
              "      <td>21</td>\n",
              "      <td>1</td>\n",
              "      <td>84</td>\n",
              "      <td>1</td>\n",
              "      <td>20347</td>\n",
              "      <td>1</td>\n",
              "      <td>2</td>\n",
              "      <td>1371945</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>metabolomics</td>\n",
              "      <td>23</td>\n",
              "      <td>{'gct_metadata': 'metabolomics_gct_metadata', ...</td>\n",
              "      <td>[normal, neoplasms, diabetes mellitus, obesity...</td>\n",
              "      <td>[homo sapiens, mus musculus, rattus norvegicus...</td>\n",
              "      <td>[metabolomics workbench, metabolights, publica...</td>\n",
              "      <td>[metabolomics, lipidomics, single cell]</td>\n",
              "      <td>1777</td>\n",
              "      <td>214</td>\n",
              "      <td>134</td>\n",
              "      <td>184</td>\n",
              "      <td>24</td>\n",
              "      <td>23</td>\n",
              "      <td>67</td>\n",
              "      <td>3</td>\n",
              "      <td>4</td>\n",
              "      <td>90677</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10</th>\n",
              "      <td>immport</td>\n",
              "      <td>1621422280385</td>\n",
              "      <td>{'gct_metadata': 'immport_gct_metadata', 'h5ad...</td>\n",
              "      <td>[normal, diabetes mellitus, type 1, kidney dis...</td>\n",
              "      <td>[homo sapiens, mus musculus]</td>\n",
              "      <td>[immport]</td>\n",
              "      <td>[lab measurement, proteomics, titer, pcr, cyto...</td>\n",
              "      <td>10000</td>\n",
              "      <td>86</td>\n",
              "      <td>10</td>\n",
              "      <td>2</td>\n",
              "      <td>1</td>\n",
              "      <td>2</td>\n",
              "      <td>47</td>\n",
              "      <td>5</td>\n",
              "      <td>1</td>\n",
              "      <td>51044</td>\n",
              "      <td>25886</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11</th>\n",
              "      <td>depmap_ccle</td>\n",
              "      <td>20</td>\n",
              "      <td>{'gct_metadata': 'depmap_ccle_gct_metadata', '...</td>\n",
              "      <td>[adrenal cancer, bile duct cancer, bladder can...</td>\n",
              "      <td>[homo sapiens]</td>\n",
              "      <td>[ccle]</td>\n",
              "      <td>[]</td>\n",
              "      <td>33</td>\n",
              "      <td>32</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "      <td>1177</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>12</th>\n",
              "      <td>sc_data_lake</td>\n",
              "      <td>17</td>\n",
              "      <td>{'gct_metadata': 'sc_data_lake_gct_metadata', ...</td>\n",
              "      <td>[normal, neoplasms, breast neoplasms, melanoma...</td>\n",
              "      <td>[mus musculus, homo sapiens, macaca fascicular...</td>\n",
              "      <td>[geo, humancellatlas, gene expression omnibus ...</td>\n",
              "      <td>[single cell]</td>\n",
              "      <td>3175</td>\n",
              "      <td>582</td>\n",
              "      <td>435</td>\n",
              "      <td>5</td>\n",
              "      <td>74</td>\n",
              "      <td>552</td>\n",
              "      <td>86</td>\n",
              "      <td>1</td>\n",
              "      <td>6</td>\n",
              "      <td>215639</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13</th>\n",
              "      <td>tcga</td>\n",
              "      <td>15</td>\n",
              "      <td>{'gct_metadata': 'tcga_gct_metadata', 'h5ad_me...</td>\n",
              "      <td>[breast neoplasms, clear-cell metastatic renal...</td>\n",
              "      <td>[homo sapiens]</td>\n",
              "      <td>[tcga]</td>\n",
              "      <td>[transcriptomics, mirna, copy number variation...</td>\n",
              "      <td>10000</td>\n",
              "      <td>41</td>\n",
              "      <td>62</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>606</td>\n",
              "      <td>6</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>14</th>\n",
              "      <td>gtex</td>\n",
              "      <td>14</td>\n",
              "      <td>{'gct_metadata': 'gtex_gct_metadata', 'h5ad_me...</td>\n",
              "      <td>[normal]</td>\n",
              "      <td>[homo sapiens]</td>\n",
              "      <td>[gtex]</td>\n",
              "      <td>[transcriptomics]</td>\n",
              "      <td>30</td>\n",
              "      <td>1</td>\n",
              "      <td>30</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>17381</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>15</th>\n",
              "      <td>gnomad</td>\n",
              "      <td>1628836648493</td>\n",
              "      <td>{'gct_metadata': 'gnomad_gct_metadata', 'h5ad_...</td>\n",
              "      <td>[none]</td>\n",
              "      <td>[homo sapiens]</td>\n",
              "      <td>[]</td>\n",
              "      <td>[gwas]</td>\n",
              "      <td>10000</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "           repo_name        repo_id  \\\n",
              "0                hpa  1619514006627   \n",
              "1                gdc  1623221686703   \n",
              "2                geo              9   \n",
              "3   enterprise_atlas  1638441282192   \n",
              "4              cptac  1609924165364   \n",
              "5    liveromix_atlas  1615965444377   \n",
              "6         cbioportal  1623986995264   \n",
              "7                pcd  1622113130397   \n",
              "8              lincs             32   \n",
              "9      metabolomics             23   \n",
              "10           immport  1621422280385   \n",
              "11       depmap_ccle             20   \n",
              "12      sc_data_lake             17   \n",
              "13              tcga             15   \n",
              "14              gtex             14   \n",
              "15            gnomad  1628836648493   \n",
              "\n",
              "                                              indexes  \\\n",
              "0   {'gct_metadata': 'hpa_gct_metadata', 'h5ad_met...   \n",
              "1   {'gct_metadata': 'gdc_gct_metadata', 'h5ad_met...   \n",
              "2   {'gct_metadata': 'geo_gct_metadata', 'h5ad_met...   \n",
              "3   {'gct_metadata': 'enterprise_atlas_gct_metadat...   \n",
              "4   {'gct_metadata': 'cptac_gct_metadata', 'h5ad_m...   \n",
              "5   {'gct_metadata': 'liveromix_atlas_gct_metadata...   \n",
              "6   {'gct_metadata': 'cbioportal_gct_metadata', 'h...   \n",
              "7   {'gct_metadata': 'pcd_gct_metadata', 'h5ad_met...   \n",
              "8   {'gct_metadata': 'lincs_gct_metadata', 'h5ad_m...   \n",
              "9  {'gct_metadata': 'metabolomics_gct_metadata', ...   \n",
              "10  {'gct_metadata': 'immport_gct_metadata', 'h5ad...   \n",
              "11  {'gct_metadata': 'depmap_ccle_gct_metadata', '...   \n",
              "12  {'gct_metadata': 'sc_data_lake_gct_metadata', ...   \n",
              "13  {'gct_metadata': 'tcga_gct_metadata', 'h5ad_me...   \n",
              "14  {'gct_metadata': 'gtex_gct_metadata', 'h5ad_me...   \n",
              "15  {'gct_metadata': 'gnomad_gct_metadata', 'h5ad_...   \n",
              "\n",
              "                                             diseases  \\\n",
              "0                                            [normal]   \n",
              "1   [leukemia, myeloid, acute, precursor t-cell ly...   \n",
              "2   [normal, neoplasms, breast neoplasms, obesity,...   \n",
              "3   [breast neoplasms, endometrial neoplasms, endo...   \n",
              "4   [carcinoma, hepatocellular, glucose-galactose ...   \n",
              "5   [normal, carcinoma, hepatocellular, obesity, n...   \n",
              "6   [prostatic neoplasms, carcinoma, ductal, breas...   \n",
              "7   [none, adenocarcinoma, carcinoma, squamous cel...   \n",
              "8   [prostatic neoplasms, normal, colorectal neopl...   \n",
              "9  [normal, neoplasms, diabetes mellitus, obesity...   \n",
              "10  [normal, diabetes mellitus, type 1, kidney dis...   \n",
              "11  [adrenal cancer, bile duct cancer, bladder can...   \n",
              "12  [normal, neoplasms, breast neoplasms, melanoma...   \n",
              "13  [breast neoplasms, clear-cell metastatic renal...   \n",
              "14                                           [normal]   \n",
              "15                                             [none]   \n",
              "\n",
              "                                            organisms  \\\n",
              "0             [homo sapiens, sus (pig), mus musculus]   \n",
              "1                                      [homo sapiens]   \n",
              "2   [homo sapiens, mus musculus, rattus norvegicus...   \n",
              "3                        [homo sapiens, mus musculus]   \n",
              "4                                      [homo sapiens]   \n",
              "5   [homo sapiens, mus musculus, rattus norvegicus...   \n",
              "6                                      [homo sapiens]   \n",
              "7                                      [homo sapiens]   \n",
              "8                                      [homo sapiens]   \n",
              "9  [homo sapiens, mus musculus, rattus norvegicus...   \n",
              "10                       [homo sapiens, mus musculus]   \n",
              "11                                     [homo sapiens]   \n",
              "12  [mus musculus, homo sapiens, macaca fascicular...   \n",
              "13                                     [homo sapiens]   \n",
              "14                                     [homo sapiens]   \n",
              "15                                     [homo sapiens]   \n",
              "\n",
              "                                              sources  \\\n",
              "0                                               [hpa]   \n",
              "1                                               [gdc]   \n",
              "2                                               [geo]   \n",
              "3                                  [geo, proprietary]   \n",
              "4                                             [cptac]   \n",
              "5   [geo, lincs, tcga, metabolomics workbench, met...   \n",
              "6                                        [cbioportal]   \n",
              "7                                        [pharmacodb]   \n",
              "8                                   [geo, geo, lincs]   \n",
              "9  [metabolomics workbench, metabolights, publica...   \n",
              "10                                          [immport]   \n",
              "11                                             [ccle]   \n",
              "12  [geo, humancellatlas, gene expression omnibus ...   \n",
              "13                                             [tcga]   \n",
              "14                                             [gtex]   \n",
              "15                                                 []   \n",
              "\n",
              "                                            datatypes  dataset_count  \\\n",
              "0      [transcriptomics, gene expression reliability]           1755   \n",
              "1   [mirna expression, transcriptomics, copy numbe...           9869   \n",
              "2       [transcriptomics, raw counts transcriptomics]          10000   \n",
              "3     [transcriptomics, single cell, mutation, mirna]            176   \n",
              "4   [proteomics, phosphoproteomics, transcriptomic...           6554   \n",
              "5   [transcriptomics, mutation, metabolomics, sing...           6761   \n",
              "6   [mutation, copy number variation, fusion, tran...          10000   \n",
              "7                                     [drug response]          10000   \n",
              "8                                   [transcriptomics]          10000   \n",
              "9            [metabolomics, lipidomics, single cell]           1777   \n",
              "10  [lab measurement, proteomics, titer, pcr, cyto...          10000   \n",
              "11                                                 []             33   \n",
              "12                                      [single cell]           3175   \n",
              "13  [transcriptomics, mirna, copy number variation...          10000   \n",
              "14                                  [transcriptomics]             30   \n",
              "15                                             [gwas]          10000   \n",
              "\n",
              "    disease_count  tissue_count  organism_count  cell_line_count  \\\n",
              "0               1            78               3               70   \n",
              "1              75            41               1                1   \n",
              "2            3030           985              68             4970   \n",
              "3              35            25               2               96   \n",
              "4              13            33               1                1   \n",
              "5             752            60              22              347   \n",
              "6             330           262               1                1   \n",
              "7             133            34               1             1400   \n",
              "8              34            21               1               84   \n",
              "9            214           134             184               24   \n",
              "10             86            10               2                1   \n",
              "11             32             0               1                0   \n",
              "12            582           435               5               74   \n",
              "13             41            62               1                1   \n",
              "14              1            30               1                1   \n",
              "15              1             1               1                1   \n",
              "\n",
              "    cell_type_count  drug_count  data_type_count  data_source_count  \\\n",
              "0               150           1                2                  1   \n",
              "1                 1           1                3                  1   \n",
              "2               780        1379                2                  1   \n",
              "3                40          22                4                  2   \n",
              "4                 1           1               10                  1   \n",
              "5               310        4704               12                 10   \n",
              "6                 1           1                5                  1   \n",
              "7                 1         749                1                  1   \n",
              "8                 1       20347                1                  2   \n",
              "9               23          67                3                  4   \n",
              "10                2          47                5                  1   \n",
              "11                0           0                0                  1   \n",
              "12              552          86                1                  6   \n",
              "13                1         606                6                  1   \n",
              "14                1           1                1                  1   \n",
              "15                0           1                1                  0   \n",
              "\n",
              "    sample_count  normal_sample_count  \n",
              "0            678                  678  \n",
              "1           5353                    0  \n",
              "2        1280558               840747  \n",
              "3           8952                 3937  \n",
              "4           2440                    0  \n",
              "5         143535               110202  \n",
              "6          58994                    0  \n",
              "7         653500                  262  \n",
              "8        1371945                    0  \n",
              "9         90677                    0  \n",
              "10         51044                25886  \n",
              "11          1177                    0  \n",
              "12        215639                    0  \n",
              "13             0                    0  \n",
              "14         17381                    0  \n",
              "15             0                    0  "
            ]
          },
          "execution_count": 7,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "all_omix_atlas"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## Get basic details of a given omixatlas"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 47,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/plain": [
              "cell_line_count                                                     5683\n",
              "cell_type_count                                                      872\n",
              "data_source_count                                                      1\n",
              "data_type_count                                                        2\n",
              "dataset_count                                                      97417\n",
              "datatypes              [{'transcriptomics': 97738}, {'raw counts tran...\n",
              "disease_count                                                       3899\n",
              "drug_count                                                          1891\n",
              "indexes                {'gct_metadata': 'geo_gct_metadata', 'h5ad_met...\n",
              "normal_sample_count                                              1604049\n",
              "organism_count                                                       299\n",
              "repo_id                                                                9\n",
              "repo_name                                                            geo\n",
              "sample_count                                                     2263489\n",
              "sources                                                 [{'geo': 97777}]\n",
              "tissue_count                                                        1106\n",
              "v2_indexes             {'csv': 'geo_csv', 'files': 'geo_files', 'gct_...\n",
              "Name: data, dtype: object"
            ]
          },
          "execution_count": 47,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "omix_atlas_summary = pd.DataFrame.from_dict(omixatlas.omixatlas_summary('geo'))['data']\n",
        "omix_atlas_summary"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### Checking for schema of a repository"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 10,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Query execution succeeded (time taken: 0.59 seconds, data scanned: 0.000 MB)\n",
            "Fetched 3 rows\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>table_name</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>geo.datasets</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>geo.samples</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>geo.features</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "     table_name\n",
              "0  geo.datasets\n",
              "1   geo.samples\n",
              "2  geo.features"
            ]
          },
          "execution_count": 10,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "query = \"SHOW TABLES IN geo\"\n",
        "results=omixatlas.query_metadata(query)\n",
        "results"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### Querying for schema of dataset level metadata of GEO"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 11,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Query execution succeeded (time taken: 1.48 seconds, data scanned: 0.000 MB)\n",
            "Fetched 35 rows\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>column_name</th>\n",
              "      <th>column_type</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>data_matrix_available</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>curated_organism</td>\n",
              "      <td>array&lt;string&gt;</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>src_uri</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>total_num_samples</td>\n",
              "      <td>bigint</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>year</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>description</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>curated_cell_line</td>\n",
              "      <td>array&lt;string&gt;</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>data_table_name</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8</th>\n",
              "      <td>data_table_version</td>\n",
              "      <td>bigint</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>platform</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10</th>\n",
              "      <td>timestamp_</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11</th>\n",
              "      <td>file_type</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>12</th>\n",
              "      <td>publication</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13</th>\n",
              "      <td>curated_cell_type</td>\n",
              "      <td>array&lt;string&gt;</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>14</th>\n",
              "      <td>key</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>15</th>\n",
              "      <td>summary</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>16</th>\n",
              "      <td>src_repo</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>17</th>\n",
              "      <td>drug_smiles</td>\n",
              "      <td>array&lt;string&gt;</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>18</th>\n",
              "      <td>package</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>19</th>\n",
              "      <td>file_location</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>20</th>\n",
              "      <td>author</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>21</th>\n",
              "      <td>dataset_id</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>22</th>\n",
              "      <td>curated_disease</td>\n",
              "      <td>array&lt;string&gt;</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>23</th>\n",
              "      <td>curated_drug</td>\n",
              "      <td>array&lt;string&gt;</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>24</th>\n",
              "      <td>curated_gene</td>\n",
              "      <td>array&lt;string&gt;</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>25</th>\n",
              "      <td>abstract</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>26</th>\n",
              "      <td>version</td>\n",
              "      <td>bigint</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>27</th>\n",
              "      <td>curated_strain</td>\n",
              "      <td>array&lt;string&gt;</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>28</th>\n",
              "      <td>bucket</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>29</th>\n",
              "      <td>curated_tissue</td>\n",
              "      <td>array&lt;string&gt;</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>30</th>\n",
              "      <td>dataset_source</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>31</th>\n",
              "      <td>data_type</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>32</th>\n",
              "      <td>overall_design</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>33</th>\n",
              "      <td>is_current</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>34</th>\n",
              "      <td>region</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "              column_name    column_type\n",
              "0   data_matrix_available         string\n",
              "1        curated_organism  array<string>\n",
              "2                 src_uri         string\n",
              "3       total_num_samples         bigint\n",
              "4                    year         string\n",
              "5             description         string\n",
              "6       curated_cell_line  array<string>\n",
              "7         data_table_name         string\n",
              "8      data_table_version         bigint\n",
              "9                platform         string\n",
              "10             timestamp_         string\n",
              "11              file_type         string\n",
              "12            publication         string\n",
              "13      curated_cell_type  array<string>\n",
              "14                    key         string\n",
              "15                summary         string\n",
              "16               src_repo         string\n",
              "17            drug_smiles  array<string>\n",
              "18                package         string\n",
              "19          file_location         string\n",
              "20                 author         string\n",
              "21             dataset_id         string\n",
              "22        curated_disease  array<string>\n",
              "23           curated_drug  array<string>\n",
              "24           curated_gene  array<string>\n",
              "25               abstract         string\n",
              "26                version         bigint\n",
              "27         curated_strain  array<string>\n",
              "28                 bucket         string\n",
              "29         curated_tissue  array<string>\n",
              "30         dataset_source         string\n",
              "31              data_type         string\n",
              "32         overall_design         string\n",
              "33             is_current         string\n",
              "34                 region         string"
            ]
          },
          "execution_count": 11,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "query = \"DESCRIBE geo.datasets\"\n",
        "results=omixatlas.query_metadata(query)\n",
        "results"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 16,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Query execution succeeded (time taken: 1.56 seconds, data scanned: 0.000 MB)\n",
            "Fetched 53 rows\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>column_name</th>\n",
              "      <th>column_type</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>growth_protocol_ch1</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>src_uri</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>sample_id</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>curated_gene_modified</td>\n",
              "      <td>array&lt;string&gt;</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>dose_ch1</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>curated_cohort_name</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>curated_control</td>\n",
              "      <td>bigint</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>src_dataset_id</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8</th>\n",
              "      <td>extract_protocol_ch1</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>characteristics_ch2</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10</th>\n",
              "      <td>characteristics_ch1</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11</th>\n",
              "      <td>extract_protocol_ch2</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>12</th>\n",
              "      <td>age_ch1</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13</th>\n",
              "      <td>curated_cell_type</td>\n",
              "      <td>array&lt;string&gt;</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>14</th>\n",
              "      <td>sex_ch1</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>15</th>\n",
              "      <td>treatment_ch1</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>16</th>\n",
              "      <td>dataset_id</td>\n",
              "      <td>array&lt;string&gt;</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>17</th>\n",
              "      <td>taxid_ch1</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>18</th>\n",
              "      <td>curated_gene</td>\n",
              "      <td>array&lt;string&gt;</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>19</th>\n",
              "      <td>curated_drug</td>\n",
              "      <td>array&lt;string&gt;</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>20</th>\n",
              "      <td>version</td>\n",
              "      <td>bigint</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>21</th>\n",
              "      <td>label_protocol_ch2</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>22</th>\n",
              "      <td>label_protocol_ch1</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>23</th>\n",
              "      <td>treatment_protocol_ch2</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>24</th>\n",
              "      <td>characteristics_ch2_1</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>25</th>\n",
              "      <td>treatment_protocol_ch1</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>26</th>\n",
              "      <td>instrument_model</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>27</th>\n",
              "      <td>name</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>28</th>\n",
              "      <td>curated_genetic_modification_type</td>\n",
              "      <td>array&lt;string&gt;</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>29</th>\n",
              "      <td>bmi_ch1</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>30</th>\n",
              "      <td>data_processing</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>31</th>\n",
              "      <td>curated_cell_line</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>32</th>\n",
              "      <td>description</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>33</th>\n",
              "      <td>title</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>34</th>\n",
              "      <td>timestamp_</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>35</th>\n",
              "      <td>curated_cohort_id</td>\n",
              "      <td>bigint</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>36</th>\n",
              "      <td>data_id</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>37</th>\n",
              "      <td>src_repo</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>38</th>\n",
              "      <td>id_key</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>39</th>\n",
              "      <td>gender_ch1</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>40</th>\n",
              "      <td>smoking_status_ch1</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>41</th>\n",
              "      <td>label_ch2</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>42</th>\n",
              "      <td>curated_disease</td>\n",
              "      <td>array&lt;string&gt;</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>43</th>\n",
              "      <td>label_ch1</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>44</th>\n",
              "      <td>time_point_ch1</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>45</th>\n",
              "      <td>characteristics_ch1_3</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>46</th>\n",
              "      <td>characteristics_ch1_2</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>47</th>\n",
              "      <td>curated_tissue</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>48</th>\n",
              "      <td>curated_drug_smiles_code</td>\n",
              "      <td>array&lt;string&gt;</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>49</th>\n",
              "      <td>hyb_protocol</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>50</th>\n",
              "      <td>platform_id</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>51</th>\n",
              "      <td>is_current</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>52</th>\n",
              "      <td>characteristics_ch1_1</td>\n",
              "      <td>string</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "                          column_name    column_type\n",
              "0                 growth_protocol_ch1         string\n",
              "1                             src_uri         string\n",
              "2                           sample_id         string\n",
              "3               curated_gene_modified  array<string>\n",
              "4                            dose_ch1         string\n",
              "5                 curated_cohort_name         string\n",
              "6                     curated_control         bigint\n",
              "7                      src_dataset_id         string\n",
              "8                extract_protocol_ch1         string\n",
              "9                 characteristics_ch2         string\n",
              "10                characteristics_ch1         string\n",
              "11               extract_protocol_ch2         string\n",
              "12                            age_ch1         string\n",
              "13                  curated_cell_type  array<string>\n",
              "14                            sex_ch1         string\n",
              "15                      treatment_ch1         string\n",
              "16                         dataset_id  array<string>\n",
              "17                          taxid_ch1         string\n",
              "18                       curated_gene  array<string>\n",
              "19                       curated_drug  array<string>\n",
              "20                            version         bigint\n",
              "21                 label_protocol_ch2         string\n",
              "22                 label_protocol_ch1         string\n",
              "23             treatment_protocol_ch2         string\n",
              "24              characteristics_ch2_1         string\n",
              "25             treatment_protocol_ch1         string\n",
              "26                   instrument_model         string\n",
              "27                               name         string\n",
              "28  curated_genetic_modification_type  array<string>\n",
              "29                            bmi_ch1         string\n",
              "30                    data_processing         string\n",
              "31                  curated_cell_line         string\n",
              "32                        description         string\n",
              "33                              title         string\n",
              "34                         timestamp_         string\n",
              "35                  curated_cohort_id         bigint\n",
              "36                            data_id         string\n",
              "37                           src_repo         string\n",
              "38                             id_key         string\n",
              "39                         gender_ch1         string\n",
              "40                 smoking_status_ch1         string\n",
              "41                          label_ch2         string\n",
              "42                    curated_disease  array<string>\n",
              "43                          label_ch1         string\n",
              "44                     time_point_ch1         string\n",
              "45              characteristics_ch1_3         string\n",
              "46              characteristics_ch1_2         string\n",
              "47                     curated_tissue         string\n",
              "48           curated_drug_smiles_code  array<string>\n",
              "49                       hyb_protocol         string\n",
              "50                        platform_id         string\n",
              "51                         is_current         string\n",
              "52              characteristics_ch1_1         string"
            ]
          },
          "execution_count": 16,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "query = \"DESCRIBE geo.samples\"\n",
        "results=omixatlas.query_metadata(query)\n",
        "results"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### Querying for dataset-id, description, number of samples etc. from GEO where platform is 'RNASeq', number of samples >20."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 18,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Query execution succeeded (time taken: 2.48 seconds, data scanned: 3.973 MB)\n",
            "Fetched 3160 rows\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>dataset_id</th>\n",
              "      <th>description</th>\n",
              "      <th>total_num_samples</th>\n",
              "      <th>curated_disease</th>\n",
              "      <th>curated_tissue</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>GSE89497_GPL20301</td>\n",
              "      <td>Single-cell RNA-seq reveals the diversity of t...</td>\n",
              "      <td>622</td>\n",
              "      <td>[Normal]</td>\n",
              "      <td>[None]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>GSE77980_GPL17021</td>\n",
              "      <td>Use of the Fluidigm C1 platform for RNA sequen...</td>\n",
              "      <td>622</td>\n",
              "      <td>[Normal]</td>\n",
              "      <td>[pancreatic islet]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>GSE74155_GPL13112</td>\n",
              "      <td>Single-cell analysis of allelic gene expressio...</td>\n",
              "      <td>617</td>\n",
              "      <td>[Normal]</td>\n",
              "      <td>[motoneuron]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>GSE98571_GPL17021</td>\n",
              "      <td>Analysis of induced cardiomyocytes during dire...</td>\n",
              "      <td>585</td>\n",
              "      <td>[Normal]</td>\n",
              "      <td>[None]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>GSE87069_GPL13112</td>\n",
              "      <td>Single-cell RNA Sequencing Uncovers A Close Re...</td>\n",
              "      <td>574</td>\n",
              "      <td>[Parkinson Disease]</td>\n",
              "      <td>[ventral midbrain, tegmentum]</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "          dataset_id                                        description  \\\n",
              "0  GSE89497_GPL20301  Single-cell RNA-seq reveals the diversity of t...   \n",
              "1  GSE77980_GPL17021  Use of the Fluidigm C1 platform for RNA sequen...   \n",
              "2  GSE74155_GPL13112  Single-cell analysis of allelic gene expressio...   \n",
              "3  GSE98571_GPL17021  Analysis of induced cardiomyocytes during dire...   \n",
              "4  GSE87069_GPL13112  Single-cell RNA Sequencing Uncovers A Close Re...   \n",
              "\n",
              "   total_num_samples      curated_disease                 curated_tissue  \n",
              "0                622             [Normal]                         [None]  \n",
              "1                622             [Normal]             [pancreatic islet]  \n",
              "2                617             [Normal]                   [motoneuron]  \n",
              "3                585             [Normal]                         [None]  \n",
              "4                574  [Parkinson Disease]  [ventral midbrain, tegmentum]  "
            ]
          },
          "execution_count": 18,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "query = \"\"\"SELECT dataset_id, description, total_num_samples, curated_disease, curated_tissue FROM geo.datasets \n",
        "            WHERE platform = 'RNASeq' AND total_num_samples > 20 ORDER BY total_num_samples desc\"\"\"\n",
        "results=omixatlas.query_metadata(query)\n",
        "results.head()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### Querying for datasets from GEO where disease array has any item consisting of keyword 'neoplasm'"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 20,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Query execution succeeded (time taken: 2.06 seconds, data scanned: 0.718 MB)\n",
            "Fetched 20253 rows\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>dataset_id</th>\n",
              "      <th>curated_disease</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>GSE97782_GPL6480</td>\n",
              "      <td>[Urinary Bladder Neoplasms]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>GSE9782_GPL97</td>\n",
              "      <td>[Neoplasms, Multiple Myeloma]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>GSE97831_GPL11154</td>\n",
              "      <td>[Prostatic Neoplasms, Castration-Resistant]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>GSE97854_GPL13497</td>\n",
              "      <td>[Neoplasms]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>GSE97856_GPL13497</td>\n",
              "      <td>[Liver Neoplasms]</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "          dataset_id                              curated_disease\n",
              "0   GSE97782_GPL6480                  [Urinary Bladder Neoplasms]\n",
              "1      GSE9782_GPL97                [Neoplasms, Multiple Myeloma]\n",
              "2  GSE97831_GPL11154  [Prostatic Neoplasms, Castration-Resistant]\n",
              "3  GSE97854_GPL13497                                  [Neoplasms]\n",
              "4  GSE97856_GPL13497                            [Liver Neoplasms]"
            ]
          },
          "execution_count": 20,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "#Querying for datasets from GEO where disease array has any item consisting of keyword 'neoplasm'\n",
        "query = \"SELECT dataset_id, curated_disease FROM geo.datasets WHERE CARDINALITY(FILTER(curated_disease, d -> LOWER(d) like '%neoplasm%')) > 0\"\n",
        "results=omixatlas.query_metadata(query)\n",
        "results.head()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### Querying for datasets and number of samples in a dataset from GEO where disease array has an exact term 'Obesity'"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 22,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Query execution succeeded (time taken: 2.10 seconds, data scanned: 0.810 MB)\n",
            "Fetched 1746 rows\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>dataset_id</th>\n",
              "      <th>total_num_samples</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>GSE97977_GPL10558</td>\n",
              "      <td>6.0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>GSE98153_GPL7202</td>\n",
              "      <td>17.0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>GSE98226_GPL21163</td>\n",
              "      <td>9.0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>GSE98287_GPL17021</td>\n",
              "      <td>16.0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>GSE98321_GPL17400</td>\n",
              "      <td>8.0</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "          dataset_id  total_num_samples\n",
              "0  GSE97977_GPL10558                6.0\n",
              "1   GSE98153_GPL7202               17.0\n",
              "2  GSE98226_GPL21163                9.0\n",
              "3  GSE98287_GPL17021               16.0\n",
              "4  GSE98321_GPL17400                8.0"
            ]
          },
          "execution_count": 22,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "query = \"SELECT dataset_id, total_num_samples FROM geo.datasets WHERE CONTAINS(curated_disease, 'Obesity')\"\n",
        "results=omixatlas.query_metadata(query)\n",
        "results.head()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### Get the abundance of different tissues across all datasets in GEO"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 24,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Query execution succeeded (time taken: 2.06 seconds, data scanned: 0.177 MB)\n",
            "Fetched 1108 rows\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>tissue_name</th>\n",
              "      <th>tissue_count</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>pancreas</td>\n",
              "      <td>834</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>lymph node</td>\n",
              "      <td>1518</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>white adipose tissue</td>\n",
              "      <td>494</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>trophectoderm</td>\n",
              "      <td>46</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>extraembryonic tissue</td>\n",
              "      <td>4</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "             tissue_name  tissue_count\n",
              "0               pancreas           834\n",
              "1             lymph node          1518\n",
              "2   white adipose tissue           494\n",
              "3          trophectoderm            46\n",
              "4  extraembryonic tissue             4"
            ]
          },
          "execution_count": 24,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "query = \"\"\"SELECT tissue_name, COUNT((tissue_name)) AS tissue_count FROM geo.datasets \n",
        "            CROSS JOIN UNNEST(curated_tissue) AS t(tissue_name) GROUP BY tissue_name\"\"\"\n",
        "results=omixatlas.query_metadata(query)\n",
        "results.head()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### Fetch the list of all the dataset IDs where expression value of gene 'brd4' is reported"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 25,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Query execution succeeded (time taken: 4.22 seconds, data scanned: 139.192 MB)\n",
            "Fetched 723 rows\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>src_dataset_id</th>\n",
              "      <th>name</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>GSE937_GPL81</td>\n",
              "      <td>Brd4</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>GSE67981_GPL10558</td>\n",
              "      <td>BRD4</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>GSE49376_GPL10904</td>\n",
              "      <td>BRD4</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>GSE15070_GPL2995</td>\n",
              "      <td>Brd4</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>GSE59841_GPL11154</td>\n",
              "      <td>BRD4</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "      src_dataset_id  name\n",
              "0       GSE937_GPL81  Brd4\n",
              "1  GSE67981_GPL10558  BRD4\n",
              "2  GSE49376_GPL10904  BRD4\n",
              "3   GSE15070_GPL2995  Brd4\n",
              "4  GSE59841_GPL11154  BRD4"
            ]
          },
          "execution_count": 25,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "query = \"SELECT src_dataset_id, name FROM geo.features WHERE LOWER(name) = 'brd4'\"\n",
        "results=omixatlas.query_metadata(query)\n",
        "results.head()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### Get the name of samples, dataset ID and extract_protocol_ch1 where spectroscopy is mentioned in the extract_protocol_ch1"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 26,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Query execution succeeded (time taken: 15.86 seconds, data scanned: 26.753 MB)\n",
            "Fetched 1562 rows\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>name</th>\n",
              "      <th>src_dataset_id</th>\n",
              "      <th>extract_protocol_ch1</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>GSM170868</td>\n",
              "      <td>GSE7103_GPL96</td>\n",
              "      <td>Total RNA was isolated using Trizol (INVITROGE...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>GSM170869</td>\n",
              "      <td>GSE7103_GPL96</td>\n",
              "      <td>Total RNA was isolated using Trizol (INVITROGE...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>GSM170870</td>\n",
              "      <td>GSE7103_GPL96</td>\n",
              "      <td>Total RNA was isolated using Trizol (INVITROGE...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>GSM170871</td>\n",
              "      <td>GSE7103_GPL96</td>\n",
              "      <td>Total RNA was isolated using Trizol (INVITROGE...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>GSM170872</td>\n",
              "      <td>GSE7103_GPL96</td>\n",
              "      <td>Total RNA was isolated using Trizol (INVITROGE...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "        name src_dataset_id                               extract_protocol_ch1\n",
              "0  GSM170868  GSE7103_GPL96  Total RNA was isolated using Trizol (INVITROGE...\n",
              "1  GSM170869  GSE7103_GPL96  Total RNA was isolated using Trizol (INVITROGE...\n",
              "2  GSM170870  GSE7103_GPL96  Total RNA was isolated using Trizol (INVITROGE...\n",
              "3  GSM170871  GSE7103_GPL96  Total RNA was isolated using Trizol (INVITROGE...\n",
              "4  GSM170872  GSE7103_GPL96  Total RNA was isolated using Trizol (INVITROGE..."
            ]
          },
          "execution_count": 26,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "query = \"\"\"SELECT name, src_dataset_id, extract_protocol_ch1 FROM geo.samples\n",
        "         WHERE LOWER(extract_protocol_ch1) LIKE '%spectroscopy%'\"\"\"\n",
        "results=omixatlas.query_metadata(query)\n",
        "results.head()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### Get the name of gene and number of samples where information about particular gene is curated "
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 28,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Query execution succeeded (time taken: 1.76 seconds, data scanned: 0.157 MB)\n",
            "Fetched 4234 rows\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>curated_gene</th>\n",
              "      <th>count</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>[]</td>\n",
              "      <td>1331997</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>[none]</td>\n",
              "      <td>1179456</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>[CD4]</td>\n",
              "      <td>7245</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>[TNF]</td>\n",
              "      <td>3160</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>[CD14]</td>\n",
              "      <td>2346</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "  curated_gene    count\n",
              "0           []  1331997\n",
              "1       [none]  1179456\n",
              "2        [CD4]     7245\n",
              "3        [TNF]     3160\n",
              "4       [CD14]     2346"
            ]
          },
          "execution_count": 28,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "query = \"\"\"SELECT curated_gene, COUNT(*) AS count FROM geo.samples GROUP BY curated_gene ORDER BY count DESC \"\"\"\n",
        "results=omixatlas.query_metadata(query)\n",
        "results.head()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### Get number of datasets with all combinations of disease and tissues in GEO"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 29,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Query execution succeeded (time taken: 2.62 seconds, data scanned: 0.498 MB)\n",
            "Fetched 40909 rows\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>tissue_name</th>\n",
              "      <th>disease_name</th>\n",
              "      <th>COUNT_COL</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>None</td>\n",
              "      <td>Normal</td>\n",
              "      <td>12833</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>liver</td>\n",
              "      <td>Normal</td>\n",
              "      <td>2738</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>None</td>\n",
              "      <td>Neoplasms</td>\n",
              "      <td>2221</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>bone marrow</td>\n",
              "      <td>Normal</td>\n",
              "      <td>1907</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>None</td>\n",
              "      <td>Breast Neoplasms</td>\n",
              "      <td>1897</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "   tissue_name      disease_name  COUNT_COL\n",
              "0         None            Normal      12833\n",
              "1        liver            Normal       2738\n",
              "2         None         Neoplasms       2221\n",
              "3  bone marrow            Normal       1907\n",
              "4         None  Breast Neoplasms       1897"
            ]
          },
          "execution_count": 29,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "query = \"\"\"SELECT tissue_name, disease_name, COUNT(*) AS COUNT_COL FROM geo.datasets \n",
        "    CROSS JOIN UNNEST(curated_tissue) AS t(tissue_name) \n",
        "    CROSS JOIN UNNEST (curated_disease) AS t(disease_name)\n",
        "    GROUP BY tissue_name, disease_name\n",
        "    ORDER BY COUNT_COL desc\"\"\"\n",
        "results=omixatlas.query_metadata(query)\n",
        "results.head()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### Fetch data matrix for a dataset_ID of interest"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 31,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Query execution succeeded (time taken: 1.20 seconds, data scanned: 0.922 MB)\n",
            "Fetched 20736 rows\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>rid</th>\n",
              "      <th>gsm96563</th>\n",
              "      <th>gsm96564</th>\n",
              "      <th>gsm96565</th>\n",
              "      <th>gsm96566</th>\n",
              "      <th>gsm96567</th>\n",
              "      <th>gsm96568</th>\n",
              "      <th>gsm96569</th>\n",
              "      <th>gsm96570</th>\n",
              "      <th>row_id</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>Hbb-y</td>\n",
              "      <td>12.7775</td>\n",
              "      <td>12.6816</td>\n",
              "      <td>12.8115</td>\n",
              "      <td>12.7138</td>\n",
              "      <td>4.6548</td>\n",
              "      <td>4.5429</td>\n",
              "      <td>4.4265</td>\n",
              "      <td>4.3696</td>\n",
              "      <td>hbb_y</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>Bbip1</td>\n",
              "      <td>12.6214</td>\n",
              "      <td>12.5121</td>\n",
              "      <td>12.3345</td>\n",
              "      <td>12.6402</td>\n",
              "      <td>9.5737</td>\n",
              "      <td>9.6951</td>\n",
              "      <td>10.6927</td>\n",
              "      <td>10.3277</td>\n",
              "      <td>bbip1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>Actb</td>\n",
              "      <td>13.9787</td>\n",
              "      <td>13.8807</td>\n",
              "      <td>13.9282</td>\n",
              "      <td>14.0340</td>\n",
              "      <td>12.8278</td>\n",
              "      <td>13.0275</td>\n",
              "      <td>13.6720</td>\n",
              "      <td>12.5043</td>\n",
              "      <td>actb</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>Fth1</td>\n",
              "      <td>11.6626</td>\n",
              "      <td>11.2601</td>\n",
              "      <td>11.9009</td>\n",
              "      <td>11.6864</td>\n",
              "      <td>12.9458</td>\n",
              "      <td>13.1211</td>\n",
              "      <td>12.8372</td>\n",
              "      <td>13.0097</td>\n",
              "      <td>fth1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>Rpgrip1</td>\n",
              "      <td>12.7654</td>\n",
              "      <td>12.7127</td>\n",
              "      <td>12.2721</td>\n",
              "      <td>12.6098</td>\n",
              "      <td>10.0882</td>\n",
              "      <td>10.1332</td>\n",
              "      <td>10.6733</td>\n",
              "      <td>10.4763</td>\n",
              "      <td>rpgrip1</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "       rid  gsm96563  gsm96564  gsm96565  gsm96566  gsm96567  gsm96568  \\\n",
              "0    Hbb-y   12.7775   12.6816   12.8115   12.7138    4.6548    4.5429   \n",
              "1    Bbip1   12.6214   12.5121   12.3345   12.6402    9.5737    9.6951   \n",
              "2     Actb   13.9787   13.8807   13.9282   14.0340   12.8278   13.0275   \n",
              "3     Fth1   11.6626   11.2601   11.9009   11.6864   12.9458   13.1211   \n",
              "4  Rpgrip1   12.7654   12.7127   12.2721   12.6098   10.0882   10.1332   \n",
              "\n",
              "   gsm96569  gsm96570   row_id  \n",
              "0    4.4265    4.3696    hbb_y  \n",
              "1   10.6927   10.3277    bbip1  \n",
              "2   13.6720   12.5043     actb  \n",
              "3   12.8372   13.0097     fth1  \n",
              "4   10.6733   10.4763  rpgrip1  "
            ]
          },
          "execution_count": 31,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "query = \"SELECT * FROM data_matrices.geo__GSE4230_GPL1261\"\n",
        "results=omixatlas.query_metadata(query)\n",
        "results.head()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### Fetch data matrix for selected genes for a dataset_ID of interest"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 32,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Query execution succeeded (time taken: 0.92 seconds, data scanned: 0.922 MB)\n",
            "Fetched 4 rows\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>rid</th>\n",
              "      <th>gsm96563</th>\n",
              "      <th>gsm96564</th>\n",
              "      <th>gsm96565</th>\n",
              "      <th>gsm96566</th>\n",
              "      <th>gsm96567</th>\n",
              "      <th>gsm96568</th>\n",
              "      <th>gsm96569</th>\n",
              "      <th>gsm96570</th>\n",
              "      <th>row_id</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>Hbb-y</td>\n",
              "      <td>12.7775</td>\n",
              "      <td>12.6816</td>\n",
              "      <td>12.8115</td>\n",
              "      <td>12.7138</td>\n",
              "      <td>4.6548</td>\n",
              "      <td>4.5429</td>\n",
              "      <td>4.4265</td>\n",
              "      <td>4.3696</td>\n",
              "      <td>hbb_y</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>Bbip1</td>\n",
              "      <td>12.6214</td>\n",
              "      <td>12.5121</td>\n",
              "      <td>12.3345</td>\n",
              "      <td>12.6402</td>\n",
              "      <td>9.5737</td>\n",
              "      <td>9.6951</td>\n",
              "      <td>10.6927</td>\n",
              "      <td>10.3277</td>\n",
              "      <td>bbip1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>Actb</td>\n",
              "      <td>13.9787</td>\n",
              "      <td>13.8807</td>\n",
              "      <td>13.9282</td>\n",
              "      <td>14.0340</td>\n",
              "      <td>12.8278</td>\n",
              "      <td>13.0275</td>\n",
              "      <td>13.6720</td>\n",
              "      <td>12.5043</td>\n",
              "      <td>actb</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>Fth1</td>\n",
              "      <td>11.6626</td>\n",
              "      <td>11.2601</td>\n",
              "      <td>11.9009</td>\n",
              "      <td>11.6864</td>\n",
              "      <td>12.9458</td>\n",
              "      <td>13.1211</td>\n",
              "      <td>12.8372</td>\n",
              "      <td>13.0097</td>\n",
              "      <td>fth1</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "     rid  gsm96563  gsm96564  gsm96565  gsm96566  gsm96567  gsm96568  \\\n",
              "0  Hbb-y   12.7775   12.6816   12.8115   12.7138    4.6548    4.5429   \n",
              "1  Bbip1   12.6214   12.5121   12.3345   12.6402    9.5737    9.6951   \n",
              "2   Actb   13.9787   13.8807   13.9282   14.0340   12.8278   13.0275   \n",
              "3   Fth1   11.6626   11.2601   11.9009   11.6864   12.9458   13.1211   \n",
              "\n",
              "   gsm96569  gsm96570 row_id  \n",
              "0    4.4265    4.3696  hbb_y  \n",
              "1   10.6927   10.3277  bbip1  \n",
              "2   13.6720   12.5043   actb  \n",
              "3   12.8372   13.0097   fth1  "
            ]
          },
          "execution_count": 32,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "gene = ('hbb-y', 'fth1', 'bbip1', 'actb')\n",
        "query = f\"SELECT * FROM data_matrices.geo__GSE4230_GPL1261 WHERE LOWER(rid) IN {gene}\"\n",
        "results=omixatlas.query_metadata(query)\n",
        "results.head()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### Get dataset_IDs where disease is obesity and organism can be either human or rat"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 33,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Query execution succeeded (time taken: 1.68 seconds, data scanned: 0.753 MB)\n",
            "Fetched 580 rows\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>dataset_id</th>\n",
              "      <th>organism_name</th>\n",
              "      <th>disease_name</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>GSE97977_GPL10558</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Obesity</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>GSE98460_GPL16686</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Obesity</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>GSE98680_GPL16699</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Obesity</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>GSE98682_GPL16699</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Obesity</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>GSE98682_GPL19730</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Obesity</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "          dataset_id organism_name disease_name\n",
              "0  GSE97977_GPL10558  Homo sapiens      Obesity\n",
              "1  GSE98460_GPL16686  Homo sapiens      Obesity\n",
              "2  GSE98680_GPL16699  Homo sapiens      Obesity\n",
              "3  GSE98682_GPL16699  Homo sapiens      Obesity\n",
              "4  GSE98682_GPL19730  Homo sapiens      Obesity"
            ]
          },
          "execution_count": 33,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "query = \"\"\"SELECT dataset_id, organism_name, disease_name FROM geo.datasets \n",
        "    CROSS JOIN UNNEST(curated_organism) AS t(organism_name) \n",
        "    CROSS JOIN UNNEST (curated_disease) AS t(disease_name) \n",
        "    WHERE LOWER (disease_name) LIKE 'obesity' AND (LOWER(organism_name) LIKE 'homo sapiens' OR LOWER (organism_name) LIKE 'rattus norvegicus')\"\"\"\n",
        "results=omixatlas.query_metadata(query)\n",
        "results.head()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## JOIN Queries"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### Get microarray datasets and their samples where the \u2018HeLa\u2019 cell line was used on a microarray platform"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 46,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Query execution succeeded (time taken: 3.71 seconds, data scanned: 7.240 MB)\n",
            "Fetched 2390 rows\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>dataset_id</th>\n",
              "      <th>sample_name</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>GSE101098_GPL16686</td>\n",
              "      <td>GSM2699152</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>GSE101098_GPL16686</td>\n",
              "      <td>GSM2699151</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>GSE101098_GPL16686</td>\n",
              "      <td>GSM2699150</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>GSE101098_GPL16686</td>\n",
              "      <td>GSM2699149</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>GSE101098_GPL16686</td>\n",
              "      <td>GSM2699148</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "           dataset_id sample_name\n",
              "0  GSE101098_GPL16686  GSM2699152\n",
              "1  GSE101098_GPL16686  GSM2699151\n",
              "2  GSE101098_GPL16686  GSM2699150\n",
              "3  GSE101098_GPL16686  GSM2699149\n",
              "4  GSE101098_GPL16686  GSM2699148"
            ]
          },
          "execution_count": 46,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "query = \"\"\"  SELECT datasets.dataset_id AS dataset_id, sample_name AS sample_name FROM geo.datasets AS datasets\n",
        "    JOIN (\n",
        "        SELECT src_dataset_id AS dataset_id, name AS sample_name FROM geo.samples \n",
        "        WHERE JSON_FORMAT(CAST(curated_cell_line as JSON)) LIKE '%HeLa%'\n",
        "         ) AS hela_samples\n",
        "    ON datasets.dataset_id = hela_samples.dataset_id WHERE LOWER(platform) = 'microarray'\n",
        "    ORDER BY datasets.dataset_id\"\"\"\n",
        "results=omixatlas.query_metadata(query)\n",
        "results.head()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### Queries using a sub-query"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 44,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Query execution succeeded (time taken: 4.09 seconds, data scanned: 7.240 MB)\n",
            "Fetched 2390 rows\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>dataset_id</th>\n",
              "      <th>sample_name</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>GSE47790_GPL17272</td>\n",
              "      <td>GSM1159682</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>GSE47790_GPL17272</td>\n",
              "      <td>GSM1159683</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>GSE47790_GPL17272</td>\n",
              "      <td>GSM1159684</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>GSE47790_GPL17272</td>\n",
              "      <td>GSM1159685</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>GSE47790_GPL17272</td>\n",
              "      <td>GSM1159686</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "          dataset_id sample_name\n",
              "0  GSE47790_GPL17272  GSM1159682\n",
              "1  GSE47790_GPL17272  GSM1159683\n",
              "2  GSE47790_GPL17272  GSM1159684\n",
              "3  GSE47790_GPL17272  GSM1159685\n",
              "4  GSE47790_GPL17272  GSM1159686"
            ]
          },
          "execution_count": 44,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "query = \"\"\"SELECT src_dataset_id AS dataset_id, name AS sample_name FROM geo.samples\n",
        "    WHERE JSON_FORMAT(CAST(curated_cell_line as JSON)) LIKE '%HeLa%' AND src_dataset_id \n",
        "    IN (SELECT dataset_id FROM geo.datasets WHERE LOWER (platform) = 'microarray')\"\"\"\n",
        "results=omixatlas.query_metadata(query)\n",
        "results.head()"
      ]
    }
  ],
  "metadata": {
    "kernelspec": {
      "display_name": "Python 3",
      "language": "python",
      "name": "python3"
    },
    "language_info": {
      "codemirror_mode": {
        "name": "ipython",
        "version": 3
      },
      "file_extension": ".py",
      "mimetype": "text/x-python",
      "name": "python",
      "nbconvert_exporter": "python",
      "pygments_lexer": "ipython3",
      "version": "3.6.9"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 2
}
