{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "colab_type": "text",
        "id": "view-in-github"
      },
      "source": [
        "<a href=\"https://polly.elucidata.io/manage/workspaces?action=open_polly_notebook&source=github&path=path_place_holder&kernel=elucidata/Python 3.10&machine=medium\" target=\"_parent\"><img src=\"https://elucidatainc.github.io/PublicAssets/open_polly.svg\" alt=\"Open in Polly\"/></a>\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "# Ingestion Case 1\n",
        "\n",
        "## Task: Upload selected datasets from an existing OmixAtlas (source) to a new OmixAtlas (destination)\n",
        "\n",
        "### Assumptions: \n",
        "1. No custom curation required\n",
        "2. Dataset level schema of destination OA to be different from the source OA\n",
        "3. Sample level schema of destination OA to be same as source OA\n",
        "4. Indexing to be done only for dataset and sample level metadata\n",
        "\n",
        "### Steps:\n",
        "#### Create OA\n",
        "1. Create OA on Test Polly\n",
        "\n",
        "#### Schema management\n",
        "2. Create schema in csv file\n",
        "3. Prepare schema payload for both dataset and sample only.\n",
        "4. Insert schema for dataset and sample level metadata\n",
        "5. Verify if schema is properly inserted\n",
        "\n",
        "#### Preparing files\n",
        "6. Fetch the template which dataset level metadata should follow\n",
        "7. Prepare dataset level metadata\n",
        "8. Verify if dataset level metadata is as per the template \n",
        "9. Prepare gct to be uploaded\n",
        "\n",
        "#### Ingestion\n",
        "10. Ingestion\n",
        "11. Verify indexing"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 2,
      "metadata": {
        "scrolled": true
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "/bin/bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)\n",
            "Collecting polly-python\n",
            "  Downloading polly_python-0.1.5-py3-none-any.whl (53 kB)\n",
            "\u001b[K     |████████████████████████████████| 53 kB 2.6 MB/s eta 0:00:011\n",
            "\u001b[?25hCollecting joblib\n",
            "  Downloading joblib-1.1.0-py2.py3-none-any.whl (306 kB)\n",
            "\u001b[K     |████████████████████████████████| 306 kB 9.0 MB/s eta 0:00:01\n",
            "\u001b[?25hRequirement already satisfied: tqdm in /usr/local/lib/python3.10/site-packages (from polly-python) (4.64.0)\n",
            "Collecting certifi==2021.10.8\n",
            "  Downloading certifi-2021.10.8-py2.py3-none-any.whl (149 kB)\n",
            "\u001b[K     |████████████████████████████████| 149 kB 147.3 MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting pytz==2021.1\n",
            "  Downloading pytz-2021.1-py2.py3-none-any.whl (510 kB)\n",
            "\u001b[K     |████████████████████████████████| 510 kB 129.4 MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting Cerberus==1.3.2\n",
            "  Downloading Cerberus-1.3.2.tar.gz (52 kB)\n",
            "\u001b[K     |████████████████████████████████| 52 kB 2.9 MB/s  eta 0:00:01\n",
            "\u001b[?25hCollecting requests==2.25.1\n",
            "  Downloading requests-2.25.1-py2.py3-none-any.whl (61 kB)\n",
            "\u001b[K     |████████████████████████████████| 61 kB 21.6 MB/s  eta 0:00:01\n",
            "\u001b[?25hRequirement already satisfied: urllib3==1.26.6 in /usr/local/lib/python3.10/site-packages (from polly-python) (1.26.6)\n",
            "Requirement already satisfied: boto3>=1.17.73 in /usr/local/lib/python3.10/site-packages (from polly-python) (1.21.19)\n",
            "Requirement already satisfied: python-dateutil==2.8.1 in /usr/local/lib/python3.10/site-packages (from polly-python) (2.8.1)\n",
            "Requirement already satisfied: idna==2.10 in /usr/local/lib/python3.10/site-packages (from polly-python) (2.10)\n",
            "Requirement already satisfied: botocore>=1.20.73 in /usr/local/lib/python3.10/site-packages (from polly-python) (1.24.19)\n",
            "Collecting sqlparse\n",
            "  Downloading sqlparse-0.4.2-py3-none-any.whl (42 kB)\n",
            "\u001b[K     |████████████████████████████████| 42 kB 3.4 MB/s s eta 0:00:01\n",
            "\u001b[?25hCollecting python-magic==0.4.24\n",
            "  Downloading python_magic-0.4.24-py2.py3-none-any.whl (12 kB)\n",
            "Collecting cmapPy\n",
            "  Downloading cmapPy-4.0.1-py2.py3-none-any.whl (150 kB)\n",
            "\u001b[K     |████████████████████████████████| 150 kB 79.1 MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting Deprecated\n",
            "  Downloading Deprecated-1.2.13-py2.py3-none-any.whl (9.6 kB)\n",
            "Requirement already satisfied: six==1.16.0 in /usr/local/lib/python3.10/site-packages (from polly-python) (1.16.0)\n",
            "Collecting black\n",
            "  Downloading black-22.6.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.5 MB)\n",
            "\u001b[K     |████████████████████████████████| 1.5 MB 127.7 MB/s eta 0:00:01\n",
            "\u001b[?25hRequirement already satisfied: chardet==4.0.0 in /usr/local/lib/python3.10/site-packages (from polly-python) (4.0.0)\n",
            "Collecting retrying==1.3.3\n",
            "  Downloading retrying-1.3.3.tar.gz (10 kB)\n",
            "Collecting pytest\n",
            "  Downloading pytest-7.1.2-py3-none-any.whl (297 kB)\n",
            "\u001b[K     |████████████████████████████████| 297 kB 122.0 MB/s eta 0:00:01\n",
            "\u001b[?25hRequirement already satisfied: pandas in /usr/local/lib/python3.10/site-packages (from polly-python) (1.4.1)\n",
            "Collecting postpy2==0.0.6\n",
            "  Downloading postpy2-0.0.6-py3-none-any.whl (17 kB)\n",
            "Collecting elucidatacloudpathlib==0.6.6\n",
            "  Downloading elucidatacloudpathlib-0.6.6-py3-none-any.whl (50 kB)\n",
            "\u001b[K     |████████████████████████████████| 50 kB 17.7 MB/s  eta 0:00:01\n",
            "\u001b[?25hCollecting rst2txt\n",
            "  Downloading rst2txt-1.1.0-py2.py3-none-any.whl (12 kB)\n",
            "Requirement already satisfied: setuptools in /usr/local/lib/python3.10/site-packages (from Cerberus==1.3.2->polly-python) (58.1.0)\n",
            "Requirement already satisfied: jmespath<1.0.0,>=0.7.1 in /usr/local/lib/python3.10/site-packages (from boto3>=1.17.73->polly-python) (0.10.0)\n",
            "Requirement already satisfied: s3transfer<0.6.0,>=0.5.0 in /usr/local/lib/python3.10/site-packages (from boto3>=1.17.73->polly-python) (0.5.2)\n",
            "Collecting platformdirs>=2\n",
            "  Downloading platformdirs-2.5.2-py3-none-any.whl (14 kB)\n",
            "Collecting pathspec>=0.9.0\n",
            "  Downloading pathspec-0.9.0-py2.py3-none-any.whl (31 kB)\n",
            "Collecting click>=8.0.0\n",
            "  Downloading click-8.1.3-py3-none-any.whl (96 kB)\n",
            "\u001b[K     |████████████████████████████████| 96 kB 11.8 MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting mypy-extensions>=0.4.3\n",
            "  Downloading mypy_extensions-0.4.3-py2.py3-none-any.whl (4.5 kB)\n",
            "Collecting tomli>=1.1.0\n",
            "  Downloading tomli-2.0.1-py3-none-any.whl (12 kB)\n",
            "Requirement already satisfied: numpy>=1.11.2 in /usr/local/lib/python3.10/site-packages (from cmapPy->polly-python) (1.22.3)\n",
            "Collecting h5py>=2.6.0\n",
            "  Downloading h5py-3.7.0-cp310-cp310-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (4.5 MB)\n",
            "\u001b[K     |████████████████████████████████| 4.5 MB 43.2 MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting wrapt<2,>=1.10\n",
            "  Downloading wrapt-1.14.1-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (77 kB)\n",
            "\u001b[K     |████████████████████████████████| 77 kB 13.7 MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting pluggy<2.0,>=0.12\n",
            "  Downloading pluggy-1.0.0-py2.py3-none-any.whl (13 kB)\n",
            "Collecting py>=1.8.2\n",
            "  Downloading py-1.11.0-py2.py3-none-any.whl (98 kB)\n",
            "\u001b[K     |████████████████████████████████| 98 kB 21.2 MB/s  eta 0:00:01\n",
            "\u001b[?25hRequirement already satisfied: attrs>=19.2.0 in /usr/local/lib/python3.10/site-packages (from pytest->polly-python) (21.2.0)\n",
            "Collecting iniconfig\n",
            "  Downloading iniconfig-1.1.1-py2.py3-none-any.whl (5.0 kB)\n",
            "Requirement already satisfied: packaging in /usr/local/lib/python3.10/site-packages (from pytest->polly-python) (21.3)\n",
            "Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in /usr/local/lib/python3.10/site-packages (from packaging->pytest->polly-python) (3.0.6)\n",
            "Requirement already satisfied: docutils>=0.11 in /usr/local/lib/python3.10/site-packages (from rst2txt->polly-python) (0.15.2)\n",
            "Requirement already satisfied: pygments in /usr/local/lib/python3.10/site-packages (from rst2txt->polly-python) (2.10.0)\n",
            "Using legacy 'setup.py install' for Cerberus, since package 'wheel' is not installed.\n",
            "Using legacy 'setup.py install' for retrying, since package 'wheel' is not installed.\n",
            "Installing collected packages: pytz, certifi, wrapt, tomli, requests, python-magic, py, pluggy, platformdirs, pathspec, mypy-extensions, iniconfig, h5py, click, sqlparse, rst2txt, retrying, pytest, postpy2, joblib, elucidatacloudpathlib, Deprecated, cmapPy, Cerberus, black, polly-python\n",
            "  Attempting uninstall: pytz\n",
            "    Found existing installation: pytz 2021.3\n",
            "    Uninstalling pytz-2021.3:\n",
            "      Successfully uninstalled pytz-2021.3\n",
            "  Attempting uninstall: certifi\n",
            "    Found existing installation: certifi 2021.5.30\n",
            "    Uninstalling certifi-2021.5.30:\n",
            "      Successfully uninstalled certifi-2021.5.30\n",
            "  Attempting uninstall: requests\n",
            "    Found existing installation: requests 2.27.1\n",
            "    Uninstalling requests-2.27.1:\n",
            "      Successfully uninstalled requests-2.27.1\n",
            "    Running setup.py install for retrying ... \u001b[?25ldone\n",
            "\u001b[?25h  Attempting uninstall: Cerberus\n",
            "    Found existing installation: Cerberus 1.3.4\n",
            "    Uninstalling Cerberus-1.3.4:\n",
            "      Successfully uninstalled Cerberus-1.3.4\n",
            "    Running setup.py install for Cerberus ... \u001b[?25ldone\n",
            "\u001b[?25hSuccessfully installed Cerberus-1.3.2 Deprecated-1.2.13 black-22.6.0 certifi-2021.10.8 click-8.1.3 cmapPy-4.0.1 elucidatacloudpathlib-0.6.6 h5py-3.7.0 iniconfig-1.1.1 joblib-1.1.0 mypy-extensions-0.4.3 pathspec-0.9.0 platformdirs-2.5.2 pluggy-1.0.0 polly-python-0.1.5 postpy2-0.0.6 py-1.11.0 pytest-7.1.2 python-magic-0.4.24 pytz-2021.1 requests-2.25.1 retrying-1.3.3 rst2txt-1.1.0 sqlparse-0.4.2 tomli-2.0.1 wrapt-1.14.1\n",
            "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\n",
            "\u001b[33mWARNING: You are using pip version 21.2.4; however, version 22.1.2 is available.\n",
            "You should consider upgrading via the '/usr/local/bin/python3.10 -m pip install --upgrade pip' command.\u001b[0m\n"
          ]
        }
      ],
      "source": [
        "!sudo pip3 install polly-python"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "#### Importing csv files containing schema for dataset level metadata"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 6,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "/bin/bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)\n",
            "\u001b[32m\u001b[1mA new version of Polly CLI is available. To update, execute the command npm update -g @elucidatainc/pollycli\u001b[22m\u001b[39m\n",
            "polly://dataset_schema.csv\n",
            "./\n",
            "\u001b7\u001b[?7l\u001b[1Gprogress [░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░] 0% | NA | ETA: 0s | time elapsed: 0s\u001b[0K\u001b[1Gprogress [████████████████████████████████████████] 100% | 1.198 KB/1.198 KB | ETA: 0s | time elapsed: 0s\u001b[0K\u001b[?7h\u001b8\n",
            "\u001b[32m\u001b[1mSuccess: Download complete\u001b[22m\u001b[39m\n"
          ]
        }
      ],
      "source": [
        "! polly files copy -y -s polly://dataset_schema.csv -d ./ "
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### Importing dependecies and authenticating into Polly"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 3,
      "metadata": {},
      "outputs": [],
      "source": [
        "import os\n",
        "from polly.omixatlas import OmixAtlas\n",
        "from polly.workspaces import Workspaces\n",
        "\n",
        "from polly.auth import Polly\n",
        "#AUTH_KEY = 'NWQ3N2xjcGFvZTo6QTBPQ3ZBeHpNSjRxZmpJc3hCbkpXMTdVZjRvT0VTbng5SjRIclVXcg==' # prod polly\n",
        "AUTH_KEY= \"dHg4MHBjczByMTo6MFY5cUllREFDbmFuaWhTT3hxTEcyMkJhTk1udFY2NWgyZ2RXeXQ4eg==\" # test polly\n",
        "Polly.auth(AUTH_KEY, env = \"testpolly\")\n",
        "\n",
        "omixatlas = OmixAtlas()\n",
        "workspaces = Workspaces()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## Creating a new OmixAtlas"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 8,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            " OmixAtlas 1657777478388 Created  \n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Repository Id</th>\n",
              "      <th>Repository Name</th>\n",
              "      <th>Display Name</th>\n",
              "      <th>Description</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>1657777478388</td>\n",
              "      <td>ingestion_testing_oa</td>\n",
              "      <td>ingestion testing OA</td>\n",
              "      <td>This OA is for testing the performance of inge...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "   Repository Id       Repository Name          Display Name  \\\n",
              "0  1657777478388  ingestion_testing_oa  ingestion testing OA   \n",
              "\n",
              "                                         Description  \n",
              "0  This OA is for testing the performance of inge...  "
            ]
          },
          "execution_count": 8,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "omixatlas.create(\"ingestion testing OA\", \"This OA is for testing the performance of ingestion functions\")"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "After creating the OmixAtlas, please ask the admin to map the resource with the organization"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### generating schema payload"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "**This block of code will be used to generate the payload of schema when the schema is prepared in the csv file**\n",
        "\n",
        "**Input:** csv file containing the schema \n",
        "\n",
        "**Output:** payload to be ingested to the OmixAtlas"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 9,
      "metadata": {},
      "outputs": [],
      "source": [
        "def get_fields(schema_file):\n",
        "    schema_df = pd.read_csv(schema_file)\n",
        "    col_types = {\n",
        "        \"field_name\": str,\n",
        "        \"original_name\": str,\n",
        "        \"type\": str,\n",
        "        \"is_keyword\": bool,\n",
        "        \"is_array\": bool,\n",
        "        \"is_filter\": bool,\n",
        "        \"is_column\": bool,\n",
        "        \"is_keyword\": bool,\n",
        "        \"filter_size\": int,\n",
        "        \"display_name\": str,\n",
        "        \"description\": str,\n",
        "    }\n",
        "    schema_df = schema_df.astype(col_types)\n",
        "    schema_df[\"field_name\"] = schema_df[\"field_name\"].apply(lambda x: x.lower().strip().replace(\" \", \"_\"))\n",
        "    schema_df[\"type\"] = schema_df[\"type\"].str.lower().str.strip()\n",
        "    schema_df[\"description\"] = schema_df[\"description\"].str.replace(\"nan\", \"NA\")\n",
        "    \n",
        "    object_rows = schema_df[schema_df[\"type\"] == \"object\"]\n",
        "    object_rows_array = object_rows[object_rows[\"is_array\"] == 1]\n",
        "    if len(object_rows_array.index) > 0:\n",
        "        raise ValueError(\"Fields with type 'object' cannot be arrays\")\n",
        "    object_rows_keyword = object_rows[object_rows[\"is_keyword\"] == 1]\n",
        "    if len(object_rows_keyword.index) > 0:\n",
        "        raise ValueError(\"Fields with type 'object' cannot be keywords\")\n",
        "    \n",
        "    field_list = schema_df.to_dict(orient=\"records\")\n",
        "    fields = {}\n",
        "    for field in field_list:\n",
        "        field_name = field.pop(\"field_name\")\n",
        "        fields[field_name] = field\n",
        "    \n",
        "    return fields\n",
        "\n",
        "def generate_schema_payload(schema_type, schema_file):\n",
        "    return {\n",
        "        \"data\": {\n",
        "            \"type\": \"schemas\",\n",
        "            \"id\": repo_id,\n",
        "            \"attributes\": {\n",
        "                \"repo_id\": repo_id,\n",
        "                \"schema_type\": schema_type,\n",
        "                \"schema\": {\n",
        "                    \"all\": {\n",
        "                        \"all\": get_fields(schema_file)\n",
        "                    }\n",
        "                }\n",
        "            }\n",
        "        }\n",
        "    }"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### generating schema for dataset level metadata (files)\n",
        "\n",
        "Calling the functions above to generate the schema payload. \n",
        "\n",
        "Here, the repo_id must be added as a string. The file is the input csv file which contains the schema in excel sheet."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 10,
      "metadata": {
        "scrolled": true
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Creating dataset-level metadata schema...\n"
          ]
        },
        {
          "data": {
            "text/plain": [
              "{'data': {'type': 'schemas',\n",
              "  'id': '1657777478388',\n",
              "  'attributes': {'repo_id': '1657777478388',\n",
              "   'schema_type': 'files',\n",
              "   'schema': {'all': {'all': {'dataset_source': {'original_name': 'dataset_source',\n",
              "       'type': 'text',\n",
              "       'is_keyword': True,\n",
              "       'is_array': False,\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Source',\n",
              "       'description': 'Source from where the data was fetched'},\n",
              "      'dataset_id': {'original_name': 'dataset_id',\n",
              "       'type': 'text',\n",
              "       'is_keyword': True,\n",
              "       'is_array': False,\n",
              "       'is_filter': False,\n",
              "       'is_column': True,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Dataset ID',\n",
              "       'description': 'Unique ID assocaited with every dataset'},\n",
              "      'description': {'original_name': 'description',\n",
              "       'type': 'text',\n",
              "       'is_keyword': False,\n",
              "       'is_array': False,\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Title',\n",
              "       'description': 'Description of the dataset'},\n",
              "      'curated_disease': {'original_name': 'curated_disease',\n",
              "       'type': 'text',\n",
              "       'is_keyword': True,\n",
              "       'is_array': True,\n",
              "       'is_filter': True,\n",
              "       'is_column': True,\n",
              "       'filter_size': 3000,\n",
              "       'display_name': 'Disease',\n",
              "       'description': 'Disease associated with the dataset'},\n",
              "      'curated_cell_line': {'original_name': 'curated_cell_line',\n",
              "       'type': 'text',\n",
              "       'is_keyword': True,\n",
              "       'is_array': True,\n",
              "       'is_filter': True,\n",
              "       'is_column': True,\n",
              "       'filter_size': 1000,\n",
              "       'display_name': 'Cell line',\n",
              "       'description': 'Cell lines from which the samples were derived for this dataset'},\n",
              "      'curated_cell_type': {'original_name': 'curated_cell_type',\n",
              "       'type': 'text',\n",
              "       'is_keyword': True,\n",
              "       'is_array': True,\n",
              "       'is_filter': False,\n",
              "       'is_column': True,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Cell type',\n",
              "       'description': 'Types of cell present in the dataset'},\n",
              "      'data_type': {'original_name': 'data_type',\n",
              "       'type': 'text',\n",
              "       'is_keyword': True,\n",
              "       'is_array': False,\n",
              "       'is_filter': True,\n",
              "       'is_column': True,\n",
              "       'filter_size': 10,\n",
              "       'display_name': 'Data type',\n",
              "       'description': 'The type of biomolecular data captured (eg - Exome Sequencing, Transcriptomics)'},\n",
              "      'curated_organism': {'original_name': 'curated_organism',\n",
              "       'type': 'text',\n",
              "       'is_keyword': True,\n",
              "       'is_array': True,\n",
              "       'is_filter': True,\n",
              "       'is_column': True,\n",
              "       'filter_size': 1000,\n",
              "       'display_name': 'Organism',\n",
              "       'description': 'Orgnism from which the samples were derived'},\n",
              "      'curated_tissue': {'original_name': 'curated_tissue',\n",
              "       'type': 'text',\n",
              "       'is_keyword': True,\n",
              "       'is_array': True,\n",
              "       'is_filter': True,\n",
              "       'is_column': True,\n",
              "       'filter_size': 1000,\n",
              "       'display_name': 'Tissue',\n",
              "       'description': 'Tissue from which the samples were derivved'},\n",
              "      'curated_drug': {'original_name': 'curated_drug',\n",
              "       'type': 'text',\n",
              "       'is_keyword': True,\n",
              "       'is_array': True,\n",
              "       'is_filter': True,\n",
              "       'is_column': True,\n",
              "       'filter_size': 3000,\n",
              "       'display_name': 'Drug',\n",
              "       'description': 'Drugs administered in the samples belonging to the dataset'},\n",
              "      'total_num_samples': {'original_name': 'total_num_samples',\n",
              "       'type': 'integer',\n",
              "       'is_keyword': True,\n",
              "       'is_array': False,\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Number of Samples',\n",
              "       'description': 'Total number of samples in a dataset'}}}}}}}"
            ]
          },
          "execution_count": 10,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "print(\"Creating dataset-level metadata schema...\")\n",
        "repo_id = \"1657777478388\"\n",
        "file = \"dataset_schema.csv\"\n",
        "payload_d = generate_schema_payload(\"files\", file)\n",
        "payload_d"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### adding dataset level schema to destination OA\n",
        "\n",
        "The payload above can be used with polly python function insert_schema to add the schema in a new OmixAtlas"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 11,
      "metadata": {
        "scrolled": true
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "'{\"data\": {\"type\": \"schemas\", \"id\": \"1657777478388\", \"attributes\": {\"repo_id\": \"1657777478388\", \"schema_type\": \"files\", \"schema\": {\"all\": {\"all\": {\"dataset_source\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"dataset_source\", \"display_name\": \"Source\", \"description\": \"Source from where the data was fetched\"}, \"dataset_id\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"dataset_id\", \"display_name\": \"Dataset ID\", \"description\": \"Unique ID assocaited with every dataset\"}, \"description\": {\"type\": \"text\", \"is_keyword\": false, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"description\", \"display_name\": \"Title\", \"description\": \"Description of the dataset\"}, \"curated_disease\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": true, \"is_filter\": true, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 3000, \"original_name\": \"curated_disease\", \"display_name\": \"Disease\", \"description\": \"Disease associated with the dataset\"}, \"curated_cell_line\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": true, \"is_filter\": true, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1000, \"original_name\": \"curated_cell_line\", \"display_name\": \"Cell line\", \"description\": \"Cell lines from which the samples were derived for this dataset\"}, \"curated_cell_type\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": true, \"is_filter\": false, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"curated_cell_type\", \"display_name\": \"Cell type\", \"description\": \"Types of cell present in the dataset\"}, \"data_type\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": true, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 10, \"original_name\": \"data_type\", \"display_name\": \"Data type\", \"description\": \"The type of biomolecular data captured (eg - Exome Sequencing, Transcriptomics)\"}, \"curated_organism\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": true, \"is_filter\": true, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1000, \"original_name\": \"curated_organism\", \"display_name\": \"Organism\", \"description\": \"Orgnism from which the samples were derived\"}, \"curated_tissue\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": true, \"is_filter\": true, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1000, \"original_name\": \"curated_tissue\", \"display_name\": \"Tissue\", \"description\": \"Tissue from which the samples were derivved\"}, \"curated_drug\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": true, \"is_filter\": true, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 3000, \"original_name\": \"curated_drug\", \"display_name\": \"Drug\", \"description\": \"Drugs administered in the samples belonging to the dataset\"}, \"total_num_samples\": {\"type\": \"integer\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"total_num_samples\", \"display_name\": \"Number of Samples\", \"description\": \"Total number of samples in a dataset\"}, \"version\": {\"type\": \"integer\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 2000, \"original_name\": \"version\", \"display_name\": \"Version\", \"description\": \"Version of data file\"}, \"is_current\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 2000, \"original_name\": \"is_current\", \"display_name\": \"Is current\", \"description\": \"Whether this is the current version of the data file or not\"}, \"src_repo\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 2000, \"original_name\": \"kw_src_repo\", \"display_name\": \"Source repository\", \"description\": \"Name of the repository this data file originated from\"}, \"package\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 2000, \"original_name\": \"kw_package\", \"display_name\": \"Package\", \"description\": \"Name of the package this data file was part of\"}, \"region\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 2000, \"original_name\": \"kw_region\", \"display_name\": \"Region\", \"description\": \"Region (AWS) in which this data is located\"}, \"bucket\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 2000, \"original_name\": \"kw_bucket\", \"display_name\": \"Bucket\", \"description\": \"S3 bucket in which the data file resides\"}, \"key\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 2000, \"original_name\": \"kw_key\", \"display_name\": \"Key\", \"description\": \"Key to the data file in source package\"}, \"file_type\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 2000, \"original_name\": \"kw_file_type\", \"display_name\": \"File type\", \"description\": \"Data file\\'s type\"}, \"file_location\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 2000, \"original_name\": \"kw_file_location\", \"display_name\": \"File location\", \"description\": \"Unique S3 location for this data file\"}, \"src_uri\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 2000, \"original_name\": \"kw_src_uri\", \"display_name\": \"Source URI\", \"description\": \"Unique URI derived from data file\\'s S3 location\"}, \"timestamp_\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 2000, \"original_name\": \"kw_timestamp_\", \"display_name\": \"Timestamp\", \"description\": \"Unix timestamp denoting time of creation for this entry\"}, \"data_table_name\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 2000, \"original_name\": \"data_table_name\", \"display_name\": \"Data-matrix table\", \"description\": \"Name of the data table associated with data file\"}, \"data_table_version\": {\"type\": \"integer\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 2000, \"original_name\": \"data_table_version\", \"display_name\": \"Data-matrix version\", \"description\": \"Current version of the data table associated with data file\"}}}}, \"created_time\": 1657777512553, \"last_updated_time\": 1657777512553, \"enabled\": true}, \"links\": {\"self\": \"/repositories/1657777478388/schema/files\"}}}'"
            ]
          },
          "execution_count": 11,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "omixatlas.insert_schema(1657777478388, payload_d)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### generating schema for sample level metadata (gct_metadata)\n",
        "\n",
        "Just like the dataset level metadata, the sample level metadata should also be added to the OmixAtlas.\n",
        "\n",
        "**In this case, because the sample level schema of destination OA is assumed to be same as source OA, we'll get schema dictionary from the source OA and then inserting it in the destination OA.**"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 18,
      "metadata": {},
      "outputs": [],
      "source": [
        "#source OA = 1657110718820\n",
        "schema = omixatlas.get_schema(\"1657110718820\", [\"sample\"], return_type = \"dict\")"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 19,
      "metadata": {
        "scrolled": true
      },
      "outputs": [],
      "source": [
        "payload_s = schema.sample"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 20,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/plain": [
              "{'data': {'type': 'schemas',\n",
              "  'id': '1657110718820',\n",
              "  'attributes': {'repo_id': '1657110718820',\n",
              "   'schema_type': 'gct_metadata',\n",
              "   'schema': {'all': {'all': {'growth_protocol_ch1': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': False,\n",
              "       'original_name': 'growth_protocol_ch1',\n",
              "       'description': 'NA',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Growth protocol ch1'},\n",
              "      'src_uri': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': True,\n",
              "       'original_name': 'kw_src_uri',\n",
              "       'description': \"Unique URI derived from source data file's S3 location\",\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 2000,\n",
              "       'display_name': 'Source URI'},\n",
              "      'sample_id': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': True,\n",
              "       'original_name': 'kw_column',\n",
              "       'description': 'Unique ID associated with every sample',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': True,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Sample ID'},\n",
              "      'curated_gene_modified': {'is_array': True,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': True,\n",
              "       'original_name': 'kw_curated_modified_gene',\n",
              "       'description': 'Gene modified through genetic modification',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': True,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Gene modified'},\n",
              "      'dose_ch1': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': False,\n",
              "       'original_name': 'dose_ch1',\n",
              "       'description': 'NA',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Dose ch1'},\n",
              "      'curated_cohort_name': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': True,\n",
              "       'original_name': 'curated_cohort_name',\n",
              "       'description': 'Name of the cohort to which the sample belongs',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': True,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Cohort name'},\n",
              "      'curated_control': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': True,\n",
              "       'original_name': 'curated_is_control',\n",
              "       'description': 'Signifies whether the given sample is a control or not, 1 signifies control and 0 signifies perturbed sample',\n",
              "       'type': 'integer',\n",
              "       'is_filter': False,\n",
              "       'is_column': True,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Control'},\n",
              "      'src_dataset_id': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': True,\n",
              "       'original_name': 'kw_src_dataset_id',\n",
              "       'description': 'Dataset ID of the file this data entity originated from',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 2000,\n",
              "       'display_name': 'Source dataset ID'},\n",
              "      'extract_protocol_ch1': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': False,\n",
              "       'original_name': 'extract_protocol_ch1',\n",
              "       'description': 'NA',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Extract protocol ch1'},\n",
              "      'characteristics_ch2': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': False,\n",
              "       'original_name': 'characteristics_ch2',\n",
              "       'description': 'NA',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Characteristics ch2'},\n",
              "      'perturbation_type': {'is_array': True,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': True,\n",
              "       'original_name': 'perturbation_type',\n",
              "       'description': 'Whether a chemical or molecular perturbation was done',\n",
              "       'type': 'text',\n",
              "       'is_filter': True,\n",
              "       'is_column': True,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1000,\n",
              "       'display_name': 'Perturbation Type'},\n",
              "      'characteristics_ch1': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': False,\n",
              "       'original_name': 'characteristics_ch1',\n",
              "       'description': 'NA',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Characteristics ch1'},\n",
              "      'extract_protocol_ch2': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': False,\n",
              "       'original_name': 'extract_protocol_ch2',\n",
              "       'description': 'NA',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Extract protocol ch2'},\n",
              "      'age_ch1': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': False,\n",
              "       'original_name': 'age_ch1',\n",
              "       'description': 'NA',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Age ch1'},\n",
              "      'curated_cell_type': {'is_array': True,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': True,\n",
              "       'original_name': 'kw_curated_cell_type',\n",
              "       'description': 'Types of cell present in the sample',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': True,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Cell type'},\n",
              "      'sex_ch1': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': False,\n",
              "       'original_name': 'sex_ch1',\n",
              "       'description': 'NA',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Sex ch1'},\n",
              "      'treatment_ch1': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': False,\n",
              "       'original_name': 'treatment_ch1',\n",
              "       'description': 'NA',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Treatment ch1'},\n",
              "      'dataset_id': {'is_array': True,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': True,\n",
              "       'original_name': 'kw_doc_id',\n",
              "       'description': 'Unique ID of the dataset to which the sample belongs',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': True,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Dataset ID'},\n",
              "      'taxid_ch1': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': False,\n",
              "       'original_name': 'taxid_ch1',\n",
              "       'description': 'NA',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Taxonomy ID ch1'},\n",
              "      'curated_gene': {'is_array': True,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': True,\n",
              "       'original_name': 'kw_curated_gene',\n",
              "       'description': 'Gene of interest in the sample',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': True,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Gene'},\n",
              "      'curated_drug': {'is_array': True,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': True,\n",
              "       'original_name': 'kw_curated_drug',\n",
              "       'description': 'Drug admistered in the sample',\n",
              "       'type': 'text',\n",
              "       'is_filter': True,\n",
              "       'is_column': True,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 3000,\n",
              "       'display_name': 'Drug'},\n",
              "      'version': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': True,\n",
              "       'original_name': 'version',\n",
              "       'description': 'Version of data entity',\n",
              "       'type': 'integer',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 2000,\n",
              "       'display_name': 'Version'},\n",
              "      'label_protocol_ch2': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': False,\n",
              "       'original_name': 'label_protocol_ch2',\n",
              "       'description': 'NA',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Label protocol ch2'},\n",
              "      'label_protocol_ch1': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': False,\n",
              "       'original_name': 'label_protocol_ch1',\n",
              "       'description': 'NA',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Label protocol ch1'},\n",
              "      'treatment_protocol_ch2': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': False,\n",
              "       'original_name': 'treatment_protocol_ch2',\n",
              "       'description': 'NA',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Treatment protocol ch2'},\n",
              "      'characteristics_ch2_1': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': False,\n",
              "       'original_name': 'characteristics_ch2_1',\n",
              "       'description': 'NA',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Characteristics ch2_1'},\n",
              "      'treatment_protocol_ch1': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': False,\n",
              "       'original_name': 'treatment_protocol_ch1',\n",
              "       'description': 'NA',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Treatment protocol ch1'},\n",
              "      'instrument_model': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': False,\n",
              "       'original_name': 'instrument_model',\n",
              "       'description': 'NA',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Instrument model'},\n",
              "      'name': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': True,\n",
              "       'original_name': 'kw_name',\n",
              "       'description': 'Name of the data entity (local to its source file)',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 2000,\n",
              "       'display_name': 'Name'},\n",
              "      'curated_genetic_modification_type': {'is_array': True,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': True,\n",
              "       'original_name': 'kw_curated_genetic_mod_type',\n",
              "       'description': 'Signifies the kind of genetic modification performed on the sample like Knockout or Mutation',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': True,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Genetic modification'},\n",
              "      'bmi_ch1': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': False,\n",
              "       'original_name': 'bmi_ch1',\n",
              "       'description': 'NA',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'BMI ch1'},\n",
              "      'data_processing': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': False,\n",
              "       'original_name': 'data_processing',\n",
              "       'description': 'NA',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Data processing'},\n",
              "      'dosage': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': True,\n",
              "       'original_name': 'dosage',\n",
              "       'description': 'Dosage of the treatment',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': True,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Dosage'},\n",
              "      'curated_cell_line': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': True,\n",
              "       'original_name': 'kw_curated_cell_line',\n",
              "       'description': 'Cell line from which the sample was derived',\n",
              "       'type': 'text',\n",
              "       'is_filter': True,\n",
              "       'is_column': True,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1000,\n",
              "       'display_name': 'Cell line'},\n",
              "      'description': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': True,\n",
              "       'original_name': 'description',\n",
              "       'description': 'Description of the sample',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Description'},\n",
              "      'title': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': True,\n",
              "       'original_name': 'title',\n",
              "       'description': 'NA',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Title'},\n",
              "      'exposure_time': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': True,\n",
              "       'original_name': 'exposure_time',\n",
              "       'description': 'Exposure time of treatment',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': True,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Exposure Time'},\n",
              "      'timestamp_': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': True,\n",
              "       'original_name': 'kw_timestamp_',\n",
              "       'description': 'Unix timestamp denoting time of creation for this entry',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 2000,\n",
              "       'display_name': 'Timestamp'},\n",
              "      'curated_cohort_id': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': True,\n",
              "       'original_name': 'curated_cohort_id',\n",
              "       'description': 'Numeric ID of the cohort to which the sample belongs',\n",
              "       'type': 'integer',\n",
              "       'is_filter': False,\n",
              "       'is_column': True,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Cohort ID'},\n",
              "      'data_id': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': True,\n",
              "       'original_name': 'kw_data_id',\n",
              "       'description': 'Unique ID for this data entity on Polly',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 2000,\n",
              "       'display_name': 'Data ID'},\n",
              "      'perturbation_modality': {'is_array': True,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': True,\n",
              "       'original_name': 'perturbation_modality',\n",
              "       'description': 'The modality of the perturbation performed on the sample',\n",
              "       'type': 'text',\n",
              "       'is_filter': True,\n",
              "       'is_column': True,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1000,\n",
              "       'display_name': 'Perturbation Modality'},\n",
              "      'src_repo': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': True,\n",
              "       'original_name': 'kw_src_repo',\n",
              "       'description': 'Name of the repository this data entity originated from',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 2000,\n",
              "       'display_name': 'Source repository'},\n",
              "      'id_key': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': True,\n",
              "       'original_name': 'kw_id_key',\n",
              "       'description': 'Name of the key that was used for creation of unique identity',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 2000,\n",
              "       'display_name': 'ID key'},\n",
              "      'gender_ch1': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': False,\n",
              "       'original_name': 'gender_ch1',\n",
              "       'description': 'NA',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Gender ch1'},\n",
              "      'smoking_status_ch1': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': False,\n",
              "       'original_name': 'smoking_status_ch1',\n",
              "       'description': 'NA',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Smoking status ch1'},\n",
              "      'label_ch2': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': False,\n",
              "       'original_name': 'label_ch2',\n",
              "       'description': 'NA',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Label ch2'},\n",
              "      'curated_disease': {'is_array': True,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': True,\n",
              "       'original_name': 'kw_curated_disease',\n",
              "       'description': 'Disease associated with the sample',\n",
              "       'type': 'text',\n",
              "       'is_filter': True,\n",
              "       'is_column': True,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 3000,\n",
              "       'display_name': 'Disease'},\n",
              "      'label_ch1': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': False,\n",
              "       'original_name': 'label_ch1',\n",
              "       'description': 'NA',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Label ch1'},\n",
              "      'time_point_ch1': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': False,\n",
              "       'original_name': 'time_point_ch1',\n",
              "       'description': 'NA',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Time point ch1'},\n",
              "      'pubchem_id': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': True,\n",
              "       'original_name': 'drug_common_name',\n",
              "       'description': 'PubChem ID of the drug',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': True,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'PubChem ID'},\n",
              "      'characteristics_ch1_3': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': False,\n",
              "       'original_name': 'characteristics_ch1_3',\n",
              "       'description': 'NA',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Characteristics ch1_2'},\n",
              "      'characteristics_ch1_2': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': False,\n",
              "       'original_name': 'characteristics_ch1_2',\n",
              "       'description': 'NA',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Characteristics ch1_2'},\n",
              "      'curated_tissue': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': True,\n",
              "       'original_name': 'kw_curated_tissue',\n",
              "       'description': 'Tissue to which the sample belongs',\n",
              "       'type': 'text',\n",
              "       'is_filter': True,\n",
              "       'is_column': True,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1000,\n",
              "       'display_name': 'Tissue'},\n",
              "      'curated_drug_smiles_code': {'is_array': True,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': True,\n",
              "       'original_name': 'kw_curated_smiles',\n",
              "       'description': 'SMILES code of the drug admistered in the sample',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': True,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Drug (SMILES code)'},\n",
              "      'hyb_protocol': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': False,\n",
              "       'original_name': 'hyb_protocol',\n",
              "       'description': 'NA',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Hybridization protocol'},\n",
              "      'platform_id': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': False,\n",
              "       'original_name': 'platform_id',\n",
              "       'description': 'NA',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Platform ID'},\n",
              "      'is_current': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': True,\n",
              "       'original_name': 'is_current',\n",
              "       'description': 'Whether this is the current version of the data entity or not',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 2000,\n",
              "       'display_name': 'Is current'},\n",
              "      'characteristics_ch1_1': {'is_array': False,\n",
              "       'is_ontology': False,\n",
              "       'is_keyword': False,\n",
              "       'original_name': 'characteristics_ch1_1',\n",
              "       'description': 'NA',\n",
              "       'type': 'text',\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'is_curated': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Characteristics ch1_1'}}}},\n",
              "   'created_time': 1657173913956,\n",
              "   'last_updated_time': 1657699208640,\n",
              "   'enabled': True}}}"
            ]
          },
          "execution_count": 20,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "payload_s"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### adding sample level schema to OA\n",
        "\n",
        "The payload above can be used with polly python function insert_schema to add the schema in a new OmixAtlas.\n",
        "\n",
        "Note: Even though the payload we fetched contained the \"id\" of source OA, when insert_schema is initiated with that payload, the \"id\" will get replpaced with destination OA."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 21,
      "metadata": {
        "scrolled": true
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "'{\"data\": {\"type\": \"schemas\", \"id\": \"1657777478388\", \"attributes\": {\"repo_id\": \"1657777478388\", \"schema_type\": \"gct_metadata\", \"schema\": {\"all\": {\"all\": {\"growth_protocol_ch1\": {\"type\": \"text\", \"is_keyword\": false, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"growth_protocol_ch1\", \"display_name\": \"Growth protocol ch1\", \"description\": \"NA\"}, \"src_uri\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 2000, \"original_name\": \"kw_src_uri\", \"display_name\": \"Source URI\", \"description\": \"Unique URI derived from source data file\\'s S3 location\"}, \"sample_id\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"kw_column\", \"display_name\": \"Sample ID\", \"description\": \"Unique ID associated with every sample\"}, \"curated_gene_modified\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": true, \"is_filter\": false, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"kw_curated_modified_gene\", \"display_name\": \"Gene modified\", \"description\": \"Gene modified through genetic modification\"}, \"dose_ch1\": {\"type\": \"text\", \"is_keyword\": false, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"dose_ch1\", \"display_name\": \"Dose ch1\", \"description\": \"NA\"}, \"curated_cohort_name\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"curated_cohort_name\", \"display_name\": \"Cohort name\", \"description\": \"Name of the cohort to which the sample belongs\"}, \"curated_control\": {\"type\": \"integer\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"curated_is_control\", \"display_name\": \"Control\", \"description\": \"Signifies whether the given sample is a control or not, 1 signifies control and 0 signifies perturbed sample\"}, \"src_dataset_id\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 2000, \"original_name\": \"kw_src_dataset_id\", \"display_name\": \"Source dataset ID\", \"description\": \"Dataset ID of the file this data entity originated from\"}, \"extract_protocol_ch1\": {\"type\": \"text\", \"is_keyword\": false, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"extract_protocol_ch1\", \"display_name\": \"Extract protocol ch1\", \"description\": \"NA\"}, \"characteristics_ch2\": {\"type\": \"text\", \"is_keyword\": false, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"characteristics_ch2\", \"display_name\": \"Characteristics ch2\", \"description\": \"NA\"}, \"perturbation_type\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": true, \"is_filter\": true, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1000, \"original_name\": \"perturbation_type\", \"display_name\": \"Perturbation Type\", \"description\": \"Whether a chemical or molecular perturbation was done\"}, \"characteristics_ch1\": {\"type\": \"text\", \"is_keyword\": false, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"characteristics_ch1\", \"display_name\": \"Characteristics ch1\", \"description\": \"NA\"}, \"extract_protocol_ch2\": {\"type\": \"text\", \"is_keyword\": false, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"extract_protocol_ch2\", \"display_name\": \"Extract protocol ch2\", \"description\": \"NA\"}, \"age_ch1\": {\"type\": \"text\", \"is_keyword\": false, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"age_ch1\", \"display_name\": \"Age ch1\", \"description\": \"NA\"}, \"curated_cell_type\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": true, \"is_filter\": false, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"kw_curated_cell_type\", \"display_name\": \"Cell type\", \"description\": \"Types of cell present in the sample\"}, \"sex_ch1\": {\"type\": \"text\", \"is_keyword\": false, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"sex_ch1\", \"display_name\": \"Sex ch1\", \"description\": \"NA\"}, \"treatment_ch1\": {\"type\": \"text\", \"is_keyword\": false, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"treatment_ch1\", \"display_name\": \"Treatment ch1\", \"description\": \"NA\"}, \"dataset_id\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": true, \"is_filter\": false, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"kw_doc_id\", \"display_name\": \"Dataset ID\", \"description\": \"Unique ID of the dataset to which the sample belongs\"}, \"taxid_ch1\": {\"type\": \"text\", \"is_keyword\": false, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"taxid_ch1\", \"display_name\": \"Taxonomy ID ch1\", \"description\": \"NA\"}, \"curated_gene\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": true, \"is_filter\": false, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"kw_curated_gene\", \"display_name\": \"Gene\", \"description\": \"Gene of interest in the sample\"}, \"curated_drug\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": true, \"is_filter\": true, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 3000, \"original_name\": \"kw_curated_drug\", \"display_name\": \"Drug\", \"description\": \"Drug admistered in the sample\"}, \"version\": {\"type\": \"integer\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 2000, \"original_name\": \"version\", \"display_name\": \"Version\", \"description\": \"Version of data entity\"}, \"label_protocol_ch2\": {\"type\": \"text\", \"is_keyword\": false, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"label_protocol_ch2\", \"display_name\": \"Label protocol ch2\", \"description\": \"NA\"}, \"label_protocol_ch1\": {\"type\": \"text\", \"is_keyword\": false, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"label_protocol_ch1\", \"display_name\": \"Label protocol ch1\", \"description\": \"NA\"}, \"treatment_protocol_ch2\": {\"type\": \"text\", \"is_keyword\": false, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"treatment_protocol_ch2\", \"display_name\": \"Treatment protocol ch2\", \"description\": \"NA\"}, \"characteristics_ch2_1\": {\"type\": \"text\", \"is_keyword\": false, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"characteristics_ch2_1\", \"display_name\": \"Characteristics ch2_1\", \"description\": \"NA\"}, \"treatment_protocol_ch1\": {\"type\": \"text\", \"is_keyword\": false, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"treatment_protocol_ch1\", \"display_name\": \"Treatment protocol ch1\", \"description\": \"NA\"}, \"instrument_model\": {\"type\": \"text\", \"is_keyword\": false, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"instrument_model\", \"display_name\": \"Instrument model\", \"description\": \"NA\"}, \"name\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 2000, \"original_name\": \"kw_name\", \"display_name\": \"Name\", \"description\": \"Name of the data entity (local to its source file)\"}, \"curated_genetic_modification_type\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": true, \"is_filter\": false, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"kw_curated_genetic_mod_type\", \"display_name\": \"Genetic modification\", \"description\": \"Signifies the kind of genetic modification performed on the sample like Knockout or Mutation\"}, \"bmi_ch1\": {\"type\": \"text\", \"is_keyword\": false, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"bmi_ch1\", \"display_name\": \"BMI ch1\", \"description\": \"NA\"}, \"data_processing\": {\"type\": \"text\", \"is_keyword\": false, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"data_processing\", \"display_name\": \"Data processing\", \"description\": \"NA\"}, \"dosage\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"dosage\", \"display_name\": \"Dosage\", \"description\": \"Dosage of the treatment\"}, \"curated_cell_line\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": true, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1000, \"original_name\": \"kw_curated_cell_line\", \"display_name\": \"Cell line\", \"description\": \"Cell line from which the sample was derived\"}, \"description\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"description\", \"display_name\": \"Description\", \"description\": \"Description of the sample\"}, \"title\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"title\", \"display_name\": \"Title\", \"description\": \"NA\"}, \"exposure_time\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"exposure_time\", \"display_name\": \"Exposure Time\", \"description\": \"Exposure time of treatment\"}, \"timestamp_\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 2000, \"original_name\": \"kw_timestamp_\", \"display_name\": \"Timestamp\", \"description\": \"Unix timestamp denoting time of creation for this entry\"}, \"curated_cohort_id\": {\"type\": \"integer\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"curated_cohort_id\", \"display_name\": \"Cohort ID\", \"description\": \"Numeric ID of the cohort to which the sample belongs\"}, \"data_id\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 2000, \"original_name\": \"kw_data_id\", \"display_name\": \"Data ID\", \"description\": \"Unique ID for this data entity on Polly\"}, \"perturbation_modality\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": true, \"is_filter\": true, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1000, \"original_name\": \"perturbation_modality\", \"display_name\": \"Perturbation Modality\", \"description\": \"The modality of the perturbation performed on the sample\"}, \"src_repo\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 2000, \"original_name\": \"kw_src_repo\", \"display_name\": \"Source repository\", \"description\": \"Name of the repository this data entity originated from\"}, \"id_key\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 2000, \"original_name\": \"kw_id_key\", \"display_name\": \"ID key\", \"description\": \"Name of the key that was used for creation of unique identity\"}, \"gender_ch1\": {\"type\": \"text\", \"is_keyword\": false, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"gender_ch1\", \"display_name\": \"Gender ch1\", \"description\": \"NA\"}, \"smoking_status_ch1\": {\"type\": \"text\", \"is_keyword\": false, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"smoking_status_ch1\", \"display_name\": \"Smoking status ch1\", \"description\": \"NA\"}, \"label_ch2\": {\"type\": \"text\", \"is_keyword\": false, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"label_ch2\", \"display_name\": \"Label ch2\", \"description\": \"NA\"}, \"curated_disease\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": true, \"is_filter\": true, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 3000, \"original_name\": \"kw_curated_disease\", \"display_name\": \"Disease\", \"description\": \"Disease associated with the sample\"}, \"label_ch1\": {\"type\": \"text\", \"is_keyword\": false, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"label_ch1\", \"display_name\": \"Label ch1\", \"description\": \"NA\"}, \"time_point_ch1\": {\"type\": \"text\", \"is_keyword\": false, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"time_point_ch1\", \"display_name\": \"Time point ch1\", \"description\": \"NA\"}, \"pubchem_id\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"drug_common_name\", \"display_name\": \"PubChem ID\", \"description\": \"PubChem ID of the drug\"}, \"characteristics_ch1_3\": {\"type\": \"text\", \"is_keyword\": false, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"characteristics_ch1_3\", \"display_name\": \"Characteristics ch1_2\", \"description\": \"NA\"}, \"characteristics_ch1_2\": {\"type\": \"text\", \"is_keyword\": false, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"characteristics_ch1_2\", \"display_name\": \"Characteristics ch1_2\", \"description\": \"NA\"}, \"curated_tissue\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": true, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1000, \"original_name\": \"kw_curated_tissue\", \"display_name\": \"Tissue\", \"description\": \"Tissue to which the sample belongs\"}, \"curated_drug_smiles_code\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": true, \"is_filter\": false, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"kw_curated_smiles\", \"display_name\": \"Drug (SMILES code)\", \"description\": \"SMILES code of the drug admistered in the sample\"}, \"hyb_protocol\": {\"type\": \"text\", \"is_keyword\": false, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"hyb_protocol\", \"display_name\": \"Hybridization protocol\", \"description\": \"NA\"}, \"platform_id\": {\"type\": \"text\", \"is_keyword\": false, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"platform_id\", \"display_name\": \"Platform ID\", \"description\": \"NA\"}, \"is_current\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 2000, \"original_name\": \"is_current\", \"display_name\": \"Is current\", \"description\": \"Whether this is the current version of the data entity or not\"}, \"characteristics_ch1_1\": {\"type\": \"text\", \"is_keyword\": false, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"characteristics_ch1_1\", \"display_name\": \"Characteristics ch1_1\", \"description\": \"NA\"}}}}, \"created_time\": 1657777814046, \"last_updated_time\": 1657777814046, \"enabled\": true}, \"links\": {\"self\": \"/repositories/1657777478388/schema/gct_metadata\"}}}'"
            ]
          },
          "execution_count": 21,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "#destination OA: 1657777478388\n",
        "omixatlas.insert_schema(1657777478388, payload_s)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### validating if the schema is appropriately added\n",
        "\n",
        "Note:- There are additional feilds in the schema (than ones in csv file) which gets auto-added upon inserting schema in a new OA"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 22,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Source</th>\n",
              "      <th>Datatype</th>\n",
              "      <th>Field Name</th>\n",
              "      <th>Field Description</th>\n",
              "      <th>Field Type</th>\n",
              "      <th>Is Curated</th>\n",
              "      <th>Is Array</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>curated_organism</td>\n",
              "      <td>Orgnism from which the samples were derived</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>src_uri</td>\n",
              "      <td>Unique URI derived from data file's S3 location</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>total_num_samples</td>\n",
              "      <td>Total number of samples in a dataset</td>\n",
              "      <td>integer</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>description</td>\n",
              "      <td>Description of the dataset</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>curated_cell_line</td>\n",
              "      <td>Cell lines from which the samples were derived...</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>data_table_name</td>\n",
              "      <td>Name of the data table associated with data file</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>data_table_version</td>\n",
              "      <td>Current version of the data table associated w...</td>\n",
              "      <td>integer</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>timestamp_</td>\n",
              "      <td>Unix timestamp denoting time of creation for t...</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>file_type</td>\n",
              "      <td>Data file's type</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>curated_cell_type</td>\n",
              "      <td>Types of cell present in the dataset</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>key</td>\n",
              "      <td>Key to the data file in source package</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>src_repo</td>\n",
              "      <td>Name of the repository this data file originat...</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>12</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>package</td>\n",
              "      <td>Name of the package this data file was part of</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>file_location</td>\n",
              "      <td>Unique S3 location for this data file</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>14</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>dataset_id</td>\n",
              "      <td>Unique ID assocaited with every dataset</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>15</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>curated_disease</td>\n",
              "      <td>Disease associated with the dataset</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>16</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>curated_drug</td>\n",
              "      <td>Drugs administered in the samples belonging to...</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>17</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>version</td>\n",
              "      <td>Version of data file</td>\n",
              "      <td>integer</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>18</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>bucket</td>\n",
              "      <td>S3 bucket in which the data file resides</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>19</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>curated_tissue</td>\n",
              "      <td>Tissue from which the samples were derivved</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>20</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>dataset_source</td>\n",
              "      <td>Source from where the data was fetched</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>21</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>data_type</td>\n",
              "      <td>The type of biomolecular data captured (eg - E...</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>22</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>is_current</td>\n",
              "      <td>Whether this is the current version of the dat...</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>23</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>region</td>\n",
              "      <td>Region (AWS) in which this data is located</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "   Source Datatype          Field Name  \\\n",
              "0     all      all    curated_organism   \n",
              "1     all      all             src_uri   \n",
              "2     all      all   total_num_samples   \n",
              "3     all      all         description   \n",
              "4     all      all   curated_cell_line   \n",
              "5     all      all     data_table_name   \n",
              "6     all      all  data_table_version   \n",
              "7     all      all          timestamp_   \n",
              "8     all      all           file_type   \n",
              "9     all      all   curated_cell_type   \n",
              "10    all      all                 key   \n",
              "11    all      all            src_repo   \n",
              "12    all      all             package   \n",
              "13    all      all       file_location   \n",
              "14    all      all          dataset_id   \n",
              "15    all      all     curated_disease   \n",
              "16    all      all        curated_drug   \n",
              "17    all      all             version   \n",
              "18    all      all              bucket   \n",
              "19    all      all      curated_tissue   \n",
              "20    all      all      dataset_source   \n",
              "21    all      all           data_type   \n",
              "22    all      all          is_current   \n",
              "23    all      all              region   \n",
              "\n",
              "                                    Field Description Field Type  Is Curated  \\\n",
              "0         Orgnism from which the samples were derived       text       False   \n",
              "1     Unique URI derived from data file's S3 location       text       False   \n",
              "2                Total number of samples in a dataset    integer       False   \n",
              "3                          Description of the dataset       text       False   \n",
              "4   Cell lines from which the samples were derived...       text       False   \n",
              "5    Name of the data table associated with data file       text       False   \n",
              "6   Current version of the data table associated w...    integer       False   \n",
              "7   Unix timestamp denoting time of creation for t...       text       False   \n",
              "8                                    Data file's type       text       False   \n",
              "9                Types of cell present in the dataset       text       False   \n",
              "10             Key to the data file in source package       text       False   \n",
              "11  Name of the repository this data file originat...       text       False   \n",
              "12     Name of the package this data file was part of       text       False   \n",
              "13              Unique S3 location for this data file       text       False   \n",
              "14            Unique ID assocaited with every dataset       text       False   \n",
              "15                Disease associated with the dataset       text       False   \n",
              "16  Drugs administered in the samples belonging to...       text       False   \n",
              "17                               Version of data file    integer       False   \n",
              "18           S3 bucket in which the data file resides       text       False   \n",
              "19        Tissue from which the samples were derivved       text       False   \n",
              "20             Source from where the data was fetched       text       False   \n",
              "21  The type of biomolecular data captured (eg - E...       text       False   \n",
              "22  Whether this is the current version of the dat...       text       False   \n",
              "23         Region (AWS) in which this data is located       text       False   \n",
              "\n",
              "    Is Array  \n",
              "0       True  \n",
              "1      False  \n",
              "2      False  \n",
              "3      False  \n",
              "4       True  \n",
              "5      False  \n",
              "6      False  \n",
              "7      False  \n",
              "8      False  \n",
              "9       True  \n",
              "10     False  \n",
              "11     False  \n",
              "12     False  \n",
              "13     False  \n",
              "14     False  \n",
              "15      True  \n",
              "16      True  \n",
              "17     False  \n",
              "18     False  \n",
              "19      True  \n",
              "20     False  \n",
              "21     False  \n",
              "22     False  \n",
              "23     False  "
            ]
          },
          "execution_count": 22,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "schema = omixatlas.get_schema(1657777478388)\n",
        "schema.dataset"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 23,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Source</th>\n",
              "      <th>Datatype</th>\n",
              "      <th>Field Name</th>\n",
              "      <th>Field Description</th>\n",
              "      <th>Field Type</th>\n",
              "      <th>Is Curated</th>\n",
              "      <th>Is Array</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>growth_protocol_ch1</td>\n",
              "      <td>NA</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>src_uri</td>\n",
              "      <td>Unique URI derived from source data file's S3 ...</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>sample_id</td>\n",
              "      <td>Unique ID associated with every sample</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>curated_gene_modified</td>\n",
              "      <td>Gene modified through genetic modification</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>dose_ch1</td>\n",
              "      <td>NA</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>curated_cohort_name</td>\n",
              "      <td>Name of the cohort to which the sample belongs</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>curated_control</td>\n",
              "      <td>Signifies whether the given sample is a contro...</td>\n",
              "      <td>integer</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>src_dataset_id</td>\n",
              "      <td>Dataset ID of the file this data entity origin...</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>extract_protocol_ch1</td>\n",
              "      <td>NA</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>characteristics_ch2</td>\n",
              "      <td>NA</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>perturbation_type</td>\n",
              "      <td>Whether a chemical or molecular perturbation w...</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>characteristics_ch1</td>\n",
              "      <td>NA</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>12</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>extract_protocol_ch2</td>\n",
              "      <td>NA</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>age_ch1</td>\n",
              "      <td>NA</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>14</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>curated_cell_type</td>\n",
              "      <td>Types of cell present in the sample</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>15</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>sex_ch1</td>\n",
              "      <td>NA</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>16</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>treatment_ch1</td>\n",
              "      <td>NA</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>17</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>dataset_id</td>\n",
              "      <td>Unique ID of the dataset to which the sample b...</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>18</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>taxid_ch1</td>\n",
              "      <td>NA</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>19</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>curated_gene</td>\n",
              "      <td>Gene of interest in the sample</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>20</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>curated_drug</td>\n",
              "      <td>Drug admistered in the sample</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>21</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>version</td>\n",
              "      <td>Version of data entity</td>\n",
              "      <td>integer</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>22</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>label_protocol_ch2</td>\n",
              "      <td>NA</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>23</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>label_protocol_ch1</td>\n",
              "      <td>NA</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>24</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>treatment_protocol_ch2</td>\n",
              "      <td>NA</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>25</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>characteristics_ch2_1</td>\n",
              "      <td>NA</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>26</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>treatment_protocol_ch1</td>\n",
              "      <td>NA</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>27</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>instrument_model</td>\n",
              "      <td>NA</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>28</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>name</td>\n",
              "      <td>Name of the data entity (local to its source f...</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>29</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>curated_genetic_modification_type</td>\n",
              "      <td>Signifies the kind of genetic modification per...</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>30</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>bmi_ch1</td>\n",
              "      <td>NA</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>31</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>data_processing</td>\n",
              "      <td>NA</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>32</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>dosage</td>\n",
              "      <td>Dosage of the treatment</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>33</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>curated_cell_line</td>\n",
              "      <td>Cell line from which the sample was derived</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>34</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>description</td>\n",
              "      <td>Description of the sample</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>35</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>title</td>\n",
              "      <td>NA</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>36</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>exposure_time</td>\n",
              "      <td>Exposure time of treatment</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>37</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>timestamp_</td>\n",
              "      <td>Unix timestamp denoting time of creation for t...</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>38</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>curated_cohort_id</td>\n",
              "      <td>Numeric ID of the cohort to which the sample b...</td>\n",
              "      <td>integer</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>39</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>data_id</td>\n",
              "      <td>Unique ID for this data entity on Polly</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>40</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>perturbation_modality</td>\n",
              "      <td>The modality of the perturbation performed on ...</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>41</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>src_repo</td>\n",
              "      <td>Name of the repository this data entity origin...</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>42</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>id_key</td>\n",
              "      <td>Name of the key that was used for creation of ...</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>43</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>gender_ch1</td>\n",
              "      <td>NA</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>44</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>smoking_status_ch1</td>\n",
              "      <td>NA</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>45</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>label_ch2</td>\n",
              "      <td>NA</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>46</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>curated_disease</td>\n",
              "      <td>Disease associated with the sample</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>47</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>label_ch1</td>\n",
              "      <td>NA</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>48</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>time_point_ch1</td>\n",
              "      <td>NA</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>49</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>pubchem_id</td>\n",
              "      <td>PubChem ID of the drug</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>50</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>characteristics_ch1_3</td>\n",
              "      <td>NA</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>51</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>characteristics_ch1_2</td>\n",
              "      <td>NA</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>52</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>curated_tissue</td>\n",
              "      <td>Tissue to which the sample belongs</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>53</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>curated_drug_smiles_code</td>\n",
              "      <td>SMILES code of the drug admistered in the sample</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>54</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>hyb_protocol</td>\n",
              "      <td>NA</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>55</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>platform_id</td>\n",
              "      <td>NA</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>56</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>is_current</td>\n",
              "      <td>Whether this is the current version of the dat...</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>57</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>characteristics_ch1_1</td>\n",
              "      <td>NA</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "   Source Datatype                         Field Name  \\\n",
              "0     all      all                growth_protocol_ch1   \n",
              "1     all      all                            src_uri   \n",
              "2     all      all                          sample_id   \n",
              "3     all      all              curated_gene_modified   \n",
              "4     all      all                           dose_ch1   \n",
              "5     all      all                curated_cohort_name   \n",
              "6     all      all                    curated_control   \n",
              "7     all      all                     src_dataset_id   \n",
              "8     all      all               extract_protocol_ch1   \n",
              "9     all      all                characteristics_ch2   \n",
              "10    all      all                  perturbation_type   \n",
              "11    all      all                characteristics_ch1   \n",
              "12    all      all               extract_protocol_ch2   \n",
              "13    all      all                            age_ch1   \n",
              "14    all      all                  curated_cell_type   \n",
              "15    all      all                            sex_ch1   \n",
              "16    all      all                      treatment_ch1   \n",
              "17    all      all                         dataset_id   \n",
              "18    all      all                          taxid_ch1   \n",
              "19    all      all                       curated_gene   \n",
              "20    all      all                       curated_drug   \n",
              "21    all      all                            version   \n",
              "22    all      all                 label_protocol_ch2   \n",
              "23    all      all                 label_protocol_ch1   \n",
              "24    all      all             treatment_protocol_ch2   \n",
              "25    all      all              characteristics_ch2_1   \n",
              "26    all      all             treatment_protocol_ch1   \n",
              "27    all      all                   instrument_model   \n",
              "28    all      all                               name   \n",
              "29    all      all  curated_genetic_modification_type   \n",
              "30    all      all                            bmi_ch1   \n",
              "31    all      all                    data_processing   \n",
              "32    all      all                             dosage   \n",
              "33    all      all                  curated_cell_line   \n",
              "34    all      all                        description   \n",
              "35    all      all                              title   \n",
              "36    all      all                      exposure_time   \n",
              "37    all      all                         timestamp_   \n",
              "38    all      all                  curated_cohort_id   \n",
              "39    all      all                            data_id   \n",
              "40    all      all              perturbation_modality   \n",
              "41    all      all                           src_repo   \n",
              "42    all      all                             id_key   \n",
              "43    all      all                         gender_ch1   \n",
              "44    all      all                 smoking_status_ch1   \n",
              "45    all      all                          label_ch2   \n",
              "46    all      all                    curated_disease   \n",
              "47    all      all                          label_ch1   \n",
              "48    all      all                     time_point_ch1   \n",
              "49    all      all                         pubchem_id   \n",
              "50    all      all              characteristics_ch1_3   \n",
              "51    all      all              characteristics_ch1_2   \n",
              "52    all      all                     curated_tissue   \n",
              "53    all      all           curated_drug_smiles_code   \n",
              "54    all      all                       hyb_protocol   \n",
              "55    all      all                        platform_id   \n",
              "56    all      all                         is_current   \n",
              "57    all      all              characteristics_ch1_1   \n",
              "\n",
              "                                    Field Description Field Type  Is Curated  \\\n",
              "0                                                  NA       text       False   \n",
              "1   Unique URI derived from source data file's S3 ...       text       False   \n",
              "2              Unique ID associated with every sample       text       False   \n",
              "3          Gene modified through genetic modification       text       False   \n",
              "4                                                  NA       text       False   \n",
              "5      Name of the cohort to which the sample belongs       text       False   \n",
              "6   Signifies whether the given sample is a contro...    integer       False   \n",
              "7   Dataset ID of the file this data entity origin...       text       False   \n",
              "8                                                  NA       text       False   \n",
              "9                                                  NA       text       False   \n",
              "10  Whether a chemical or molecular perturbation w...       text       False   \n",
              "11                                                 NA       text       False   \n",
              "12                                                 NA       text       False   \n",
              "13                                                 NA       text       False   \n",
              "14                Types of cell present in the sample       text       False   \n",
              "15                                                 NA       text       False   \n",
              "16                                                 NA       text       False   \n",
              "17  Unique ID of the dataset to which the sample b...       text       False   \n",
              "18                                                 NA       text       False   \n",
              "19                     Gene of interest in the sample       text       False   \n",
              "20                      Drug admistered in the sample       text       False   \n",
              "21                             Version of data entity    integer       False   \n",
              "22                                                 NA       text       False   \n",
              "23                                                 NA       text       False   \n",
              "24                                                 NA       text       False   \n",
              "25                                                 NA       text       False   \n",
              "26                                                 NA       text       False   \n",
              "27                                                 NA       text       False   \n",
              "28  Name of the data entity (local to its source f...       text       False   \n",
              "29  Signifies the kind of genetic modification per...       text       False   \n",
              "30                                                 NA       text       False   \n",
              "31                                                 NA       text       False   \n",
              "32                            Dosage of the treatment       text       False   \n",
              "33        Cell line from which the sample was derived       text       False   \n",
              "34                          Description of the sample       text       False   \n",
              "35                                                 NA       text       False   \n",
              "36                         Exposure time of treatment       text       False   \n",
              "37  Unix timestamp denoting time of creation for t...       text       False   \n",
              "38  Numeric ID of the cohort to which the sample b...    integer       False   \n",
              "39            Unique ID for this data entity on Polly       text       False   \n",
              "40  The modality of the perturbation performed on ...       text       False   \n",
              "41  Name of the repository this data entity origin...       text       False   \n",
              "42  Name of the key that was used for creation of ...       text       False   \n",
              "43                                                 NA       text       False   \n",
              "44                                                 NA       text       False   \n",
              "45                                                 NA       text       False   \n",
              "46                 Disease associated with the sample       text       False   \n",
              "47                                                 NA       text       False   \n",
              "48                                                 NA       text       False   \n",
              "49                             PubChem ID of the drug       text       False   \n",
              "50                                                 NA       text       False   \n",
              "51                                                 NA       text       False   \n",
              "52                 Tissue to which the sample belongs       text       False   \n",
              "53   SMILES code of the drug admistered in the sample       text       False   \n",
              "54                                                 NA       text       False   \n",
              "55                                                 NA       text       False   \n",
              "56  Whether this is the current version of the dat...       text       False   \n",
              "57                                                 NA       text       False   \n",
              "\n",
              "    Is Array  \n",
              "0      False  \n",
              "1      False  \n",
              "2      False  \n",
              "3       True  \n",
              "4      False  \n",
              "5      False  \n",
              "6      False  \n",
              "7      False  \n",
              "8      False  \n",
              "9      False  \n",
              "10      True  \n",
              "11     False  \n",
              "12     False  \n",
              "13     False  \n",
              "14      True  \n",
              "15     False  \n",
              "16     False  \n",
              "17      True  \n",
              "18     False  \n",
              "19      True  \n",
              "20      True  \n",
              "21     False  \n",
              "22     False  \n",
              "23     False  \n",
              "24     False  \n",
              "25     False  \n",
              "26     False  \n",
              "27     False  \n",
              "28     False  \n",
              "29      True  \n",
              "30     False  \n",
              "31     False  \n",
              "32     False  \n",
              "33     False  \n",
              "34     False  \n",
              "35     False  \n",
              "36     False  \n",
              "37     False  \n",
              "38     False  \n",
              "39     False  \n",
              "40      True  \n",
              "41     False  \n",
              "42     False  \n",
              "43     False  \n",
              "44     False  \n",
              "45     False  \n",
              "46      True  \n",
              "47     False  \n",
              "48     False  \n",
              "49     False  \n",
              "50     False  \n",
              "51     False  \n",
              "52     False  \n",
              "53      True  \n",
              "54     False  \n",
              "55     False  \n",
              "56     False  \n",
              "57     False  "
            ]
          },
          "execution_count": 23,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "schema.sample"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## Preparing dataset level metadata files (.json)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### fetching template for the same\n",
        "\n",
        "In order to prepare the dataset level metadata, the user should ensure the keys of json files they prepare adheres to the schema of dataset level. \n",
        "\n",
        "In order to facilitate this, users can use the function dataset_metadata_template to generate the template. These are mandatory fields to have in the dataset level metadata."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 24,
      "metadata": {},
      "outputs": [],
      "source": [
        "template = omixatlas.dataset_metadata_template(1657777478388)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 25,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/plain": [
              "{'curated_organism': [],\n",
              " 'kw_src_uri': 'text',\n",
              " 'total_num_samples': 'integer',\n",
              " 'description': 'text',\n",
              " 'curated_cell_line': [],\n",
              " 'data_table_name': 'text',\n",
              " 'data_table_version': 'integer',\n",
              " 'kw_timestamp_': 'text',\n",
              " 'kw_file_type': 'text',\n",
              " 'curated_cell_type': [],\n",
              " 'kw_key': 'text',\n",
              " 'kw_src_repo': 'text',\n",
              " 'kw_package': 'text',\n",
              " 'kw_file_location': 'text',\n",
              " 'dataset_id': 'text',\n",
              " 'curated_disease': [],\n",
              " 'curated_drug': [],\n",
              " 'version': 'integer',\n",
              " 'kw_bucket': 'text',\n",
              " 'curated_tissue': [],\n",
              " 'dataset_source': 'text',\n",
              " 'data_type': 'text',\n",
              " 'is_current': 'text',\n",
              " 'kw_region': 'text',\n",
              " '__index__': {'file_metadata': True,\n",
              "  'col_metadata': True,\n",
              "  'row_metadata': False,\n",
              "  'data_required': False}}"
            ]
          },
          "execution_count": 25,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "template"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### from source OmixAtlas, fetching the dataset level metadata for the datasets to be ingested in destination OmixAtlas\n",
        "\n",
        "Because some of the information present at dataset level metadata in source OA and destination OA will remain same, we can just query the source OA and use the information to prepare the dataset level metadata for destination OA."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 26,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Query execution succeeded (time taken: 1.83 seconds, data scanned: 0.002 MB)\n",
            "Fetched 2 rows\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>curated_organism</th>\n",
              "      <th>src_uri</th>\n",
              "      <th>total_num_samples</th>\n",
              "      <th>description</th>\n",
              "      <th>curated_cell_line</th>\n",
              "      <th>data_table_name</th>\n",
              "      <th>data_table_version</th>\n",
              "      <th>timestamp_</th>\n",
              "      <th>file_type</th>\n",
              "      <th>curated_cell_type</th>\n",
              "      <th>key</th>\n",
              "      <th>src_repo</th>\n",
              "      <th>package</th>\n",
              "      <th>file_location</th>\n",
              "      <th>dataset_id</th>\n",
              "      <th>curated_disease</th>\n",
              "      <th>curated_drug</th>\n",
              "      <th>version</th>\n",
              "      <th>bucket</th>\n",
              "      <th>curated_tissue</th>\n",
              "      <th>dataset_source</th>\n",
              "      <th>data_type</th>\n",
              "      <th>is_current</th>\n",
              "      <th>region</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>[Homo sapiens]</td>\n",
              "      <td>polly:data://valo_onco/data/220707-1146/PXD002...</td>\n",
              "      <td>7</td>\n",
              "      <td>Quantitative K-GG site analysis</td>\n",
              "      <td>[HCT 116]</td>\n",
              "      <td>valo_onco__pxd002408_biological_replicate_1</td>\n",
              "      <td>0</td>\n",
              "      <td>1657177092182</td>\n",
              "      <td>gct</td>\n",
              "      <td>[]</td>\n",
              "      <td>valo_onco/data/220707-1146/PXD002408_Biologica...</td>\n",
              "      <td>valo_onco</td>\n",
              "      <td>valo_onco/data</td>\n",
              "      <td>https://discover-test-datalake-v1.s3-ap-southe...</td>\n",
              "      <td>PXD002408_Biological_Replicate_1</td>\n",
              "      <td>[Colonic Neoplasms]</td>\n",
              "      <td>[dimethyl sulfoxide,   growth hormone]</td>\n",
              "      <td>0</td>\n",
              "      <td>discover-test-datalake-v1</td>\n",
              "      <td>[colorectum]</td>\n",
              "      <td>PRIDE</td>\n",
              "      <td>Proteomics</td>\n",
              "      <td>True</td>\n",
              "      <td>ap-southeast-1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>[Homo sapiens]</td>\n",
              "      <td>polly:data://valo_onco/data/220707-1146/PXD002...</td>\n",
              "      <td>3</td>\n",
              "      <td>Quantitative K-GG site analysis</td>\n",
              "      <td>[HCT 116]</td>\n",
              "      <td>valo_onco__pxd002408_biological_replicate_2</td>\n",
              "      <td>0</td>\n",
              "      <td>1657177198137</td>\n",
              "      <td>gct</td>\n",
              "      <td>[]</td>\n",
              "      <td>valo_onco/data/220707-1146/PXD002408_Biologica...</td>\n",
              "      <td>valo_onco</td>\n",
              "      <td>valo_onco/data</td>\n",
              "      <td>https://discover-test-datalake-v1.s3-ap-southe...</td>\n",
              "      <td>PXD002408_Biological_Replicate_2</td>\n",
              "      <td>[Colonic Neoplasms]</td>\n",
              "      <td>[dimethyl sulfoxide,   growth hormone]</td>\n",
              "      <td>0</td>\n",
              "      <td>discover-test-datalake-v1</td>\n",
              "      <td>[colorectum]</td>\n",
              "      <td>PRIDE</td>\n",
              "      <td>Proteomics</td>\n",
              "      <td>True</td>\n",
              "      <td>ap-southeast-1</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "  curated_organism                                            src_uri  \\\n",
              "0   [Homo sapiens]  polly:data://valo_onco/data/220707-1146/PXD002...   \n",
              "1   [Homo sapiens]  polly:data://valo_onco/data/220707-1146/PXD002...   \n",
              "\n",
              "   total_num_samples                      description curated_cell_line  \\\n",
              "0                  7  Quantitative K-GG site analysis         [HCT 116]   \n",
              "1                  3  Quantitative K-GG site analysis         [HCT 116]   \n",
              "\n",
              "                               data_table_name  data_table_version  \\\n",
              "0  valo_onco__pxd002408_biological_replicate_1                   0   \n",
              "1  valo_onco__pxd002408_biological_replicate_2                   0   \n",
              "\n",
              "      timestamp_ file_type curated_cell_type  \\\n",
              "0  1657177092182       gct                []   \n",
              "1  1657177198137       gct                []   \n",
              "\n",
              "                                                 key   src_repo  \\\n",
              "0  valo_onco/data/220707-1146/PXD002408_Biologica...  valo_onco   \n",
              "1  valo_onco/data/220707-1146/PXD002408_Biologica...  valo_onco   \n",
              "\n",
              "          package                                      file_location  \\\n",
              "0  valo_onco/data  https://discover-test-datalake-v1.s3-ap-southe...   \n",
              "1  valo_onco/data  https://discover-test-datalake-v1.s3-ap-southe...   \n",
              "\n",
              "                         dataset_id      curated_disease  \\\n",
              "0  PXD002408_Biological_Replicate_1  [Colonic Neoplasms]   \n",
              "1  PXD002408_Biological_Replicate_2  [Colonic Neoplasms]   \n",
              "\n",
              "                             curated_drug  version                     bucket  \\\n",
              "0  [dimethyl sulfoxide,   growth hormone]        0  discover-test-datalake-v1   \n",
              "1  [dimethyl sulfoxide,   growth hormone]        0  discover-test-datalake-v1   \n",
              "\n",
              "  curated_tissue dataset_source   data_type  is_current          region  \n",
              "0   [colorectum]          PRIDE  Proteomics        True  ap-southeast-1  \n",
              "1   [colorectum]          PRIDE  Proteomics        True  ap-southeast-1  "
            ]
          },
          "execution_count": 26,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "list = ['PXD002408_Biological_Replicate_1', 'PXD002408_Biological_Replicate_2']\n",
        "query = f\"SELECT * from 1657110718820.datasets WHERE dataset_id IN {tuple(list)}\"\n",
        "metadata = omixatlas.query_metadata(query)\n",
        "metadata"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "**Before running the next cell, user should create two folders = 'data_final' and 'metadata_final' in their working directory**"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### converting the dataset level metadata to .json file"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 27,
      "metadata": {},
      "outputs": [],
      "source": [
        "# Converting to json\n",
        "for i in metadata.index:\n",
        "    metadata.loc[i].to_json(f\"metadata_final/{metadata['dataset_id'][i]}.json\".format(i))\n",
        "filenames = os.listdir('metadata_final')"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### adding the _ _index_ _ OR any custom curated fields\n",
        "\n",
        "Once the dataset level metadata files are created then as per the template, _ _index_ _ needs to be appended to the .json file prior to ingestion.\n",
        "\n",
        "**If there are any custom curation fields, then those can also be added in a similar way**"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 28,
      "metadata": {},
      "outputs": [],
      "source": [
        "for i in filenames:\n",
        "    entry = { \"__index__\": {\n",
        "        'file_metadata': 'true',  \n",
        "        'col_metadata': 'true',  \n",
        "        'row_metadata': 'false',  \n",
        "        'data_required': 'false',\n",
        "    }}\n",
        "    file = open('metadata_final/'+i)\n",
        "    json_met = json.load(file)\n",
        "    entry.update(json_met)\n",
        "    with open('metadata_final/'+i, \"w\") as file:\n",
        "        json.dump(entry, file)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### Validating if the keys of json file matches the template\n",
        "\n",
        "Prior to ingestion, the user should ensure the keys in dataset level metadata matches with that of the template generated above."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 29,
      "metadata": {},
      "outputs": [],
      "source": [
        "template_keys = set()\n",
        "for key in template:\n",
        "        template_keys.add(key)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 30,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/plain": [
              "{'__index__',\n",
              " 'curated_cell_line',\n",
              " 'curated_cell_type',\n",
              " 'curated_disease',\n",
              " 'curated_drug',\n",
              " 'curated_organism',\n",
              " 'curated_tissue',\n",
              " 'data_table_name',\n",
              " 'data_table_version',\n",
              " 'data_type',\n",
              " 'dataset_id',\n",
              " 'dataset_source',\n",
              " 'description',\n",
              " 'is_current',\n",
              " 'kw_bucket',\n",
              " 'kw_file_location',\n",
              " 'kw_file_type',\n",
              " 'kw_key',\n",
              " 'kw_package',\n",
              " 'kw_region',\n",
              " 'kw_src_repo',\n",
              " 'kw_src_uri',\n",
              " 'kw_timestamp_',\n",
              " 'total_num_samples',\n",
              " 'version'}"
            ]
          },
          "execution_count": 30,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "template_keys"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 31,
      "metadata": {},
      "outputs": [],
      "source": [
        "import json\n",
        "keys_in_json = set()\n",
        "f = open('metadata_final/PXD002408_Biological_Replicate_1.json')\n",
        "data = json.load(f)\n",
        "\n",
        "for key in data:\n",
        "    keys_in_json.add(key)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 32,
      "metadata": {
        "scrolled": true
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "{'__index__',\n",
              " 'bucket',\n",
              " 'curated_cell_line',\n",
              " 'curated_cell_type',\n",
              " 'curated_disease',\n",
              " 'curated_drug',\n",
              " 'curated_organism',\n",
              " 'curated_tissue',\n",
              " 'data_table_name',\n",
              " 'data_table_version',\n",
              " 'data_type',\n",
              " 'dataset_id',\n",
              " 'dataset_source',\n",
              " 'description',\n",
              " 'file_location',\n",
              " 'file_type',\n",
              " 'is_current',\n",
              " 'key',\n",
              " 'package',\n",
              " 'region',\n",
              " 'src_repo',\n",
              " 'src_uri',\n",
              " 'timestamp_',\n",
              " 'total_num_samples',\n",
              " 'version'}"
            ]
          },
          "execution_count": 32,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "keys_in_json"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 33,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/plain": [
              "{'__index__',\n",
              " 'curated_cell_line',\n",
              " 'curated_cell_type',\n",
              " 'curated_disease',\n",
              " 'curated_drug',\n",
              " 'curated_organism',\n",
              " 'curated_tissue',\n",
              " 'data_table_name',\n",
              " 'data_table_version',\n",
              " 'data_type',\n",
              " 'dataset_id',\n",
              " 'dataset_source',\n",
              " 'description',\n",
              " 'is_current',\n",
              " 'total_num_samples',\n",
              " 'version'}"
            ]
          },
          "execution_count": 33,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "intersect = keys_in_json.intersection(template_keys)\n",
        "intersect"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### validating indexes are same\n",
        "\n",
        "In this case, given the intersection has only the fields which are automatically added to dataset level metadata file during ingestion, it will not make any impact on the ingestion job."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 34,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/plain": [
              "{'kw_bucket',\n",
              " 'kw_file_location',\n",
              " 'kw_file_type',\n",
              " 'kw_key',\n",
              " 'kw_package',\n",
              " 'kw_region',\n",
              " 'kw_src_repo',\n",
              " 'kw_src_uri',\n",
              " 'kw_timestamp_'}"
            ]
          },
          "execution_count": 34,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "template_keys.difference(intersect)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### Downloading corresponding gct files"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "As per the assumption, no custom curation is required hence the gct files don't need any modification. Hence, downloading the files from the source OA and uploading them as it is in the destination OA"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 35,
      "metadata": {
        "scrolled": true
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "PXD002408_Biological_Replicate_1\n"
          ]
        },
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "--2022-07-14 05:55:56--  https://discover-test-datalake-v1.s3.amazonaws.com/valo_onco/data/220713-13235/PXD002408_Biological_Replicate_1.gct?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=ASIARBT2ZRFJPGHAPSHS%2F20220714%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Date=20220714T055556Z&X-Amz-Expires=3600&X-Amz-SignedHeaders=host&X-Amz-Security-Token=IQoJb3JpZ2luX2VjEE4aDmFwLXNvdXRoZWFzdC0xIkcwRQIhAIykPPH7JG%2F89ouW88shMZbvGa8WB9kE1JGH%2BliqD%2BJ7AiBXl11EdL%2BAyMenUIfixOdgKHoHGYlvjOz5ppbWZ1Y5jCqqAgiH%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F8BEAIaDDA3MjE5ODIyODMwNiIMRtW5amK%2B7yo9VavJKv4BXIcqTtAYsBH3xqRYpq9aAFxAFbJIsKM5POxkP7xZ0lRsEVaK4OKptp13k%2ByVl3HHiQkClcqWMsa7aklShXm8sdIcQwVE8fpn%2FgDNr3hDp7ixvudbkpWW9hcJeNBoods7kQscO5AIVVx1SmtJO6NlVlQVrTQRXCSoWZoaw68ryO1x6xkdj2P4zA1CCql1VNp0vN%2Blm3hw026iaRO3Ci2iMHUHuOnu8d%2BONKUXPzOJiyyLgkJtasT4Kz4GbUeYlbdJvIbFLfQ6su4yTTH2Jh1UhzEnhqFGfsleEmmRGDORyAW2K%2Be8C%2Fwc8Ds%2F3CmF4vdE0MMIvkhn27x9ml4DRscw4tq%2BlgY6mgEBPdRpVjjlWsOsIGyYiYJJ2lIsiS8nxVqvOmReMl%2BYMTW2sh67FNXz3CMSvRtx74ehBBsikS%2BSO6QG0OmnvCj35CXeKCuPpi2HUbZJlJoUXtFlqsyO5qLcKuIWBhksQTmPKFByVTWVDhXCT7XVXCd7lPClLkCkiGlSY6qjChRpKVwWxog5FcfsTh5RJX2t5Cb6b6eeOl4AM5%2Bo&X-Amz-Signature=8dee9fefea9ebf5256c8b6e3a9fc9fc9f3945afeb82985be0d0f91b5a5d3a0e0\n",
            "Resolving discover-test-datalake-v1.s3.amazonaws.com (discover-test-datalake-v1.s3.amazonaws.com)... 52.219.128.240\n",
            "Connecting to discover-test-datalake-v1.s3.amazonaws.com (discover-test-datalake-v1.s3.amazonaws.com)|52.219.128.240|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 141053 (138K) [binary/octet-stream]\n",
            "Saving to: 'data_final/PXD002408_Biological_Replicate_1.gct'\n",
            "\n",
            "     0K .......... .......... .......... .......... .......... 36%  297K 0s\n",
            "    50K .......... .......... .......... .......... .......... 72%  149K 0s\n",
            "   100K .......... .......... .......... .......              100%  115M=0.5s\n",
            "\n",
            "2022-07-14 05:55:57 (273 KB/s) - 'data_final/PXD002408_Biological_Replicate_1.gct' saved [141053/141053]\n",
            "\n"
          ]
        },
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Downloaded data successfully\n",
            "PXD002408_Biological_Replicate_2\n"
          ]
        },
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "--2022-07-14 05:56:00--  https://discover-test-datalake-v1.s3.amazonaws.com/valo_onco/data/220713-13235/PXD002408_Biological_Replicate_2.gct?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=ASIARBT2ZRFJNVEYMIJP%2F20220714%2Fap-southeast-1%2Fs3%2Faws4_request&X-Amz-Date=20220714T055600Z&X-Amz-Expires=3600&X-Amz-SignedHeaders=host&X-Amz-Security-Token=IQoJb3JpZ2luX2VjEE4aDmFwLXNvdXRoZWFzdC0xIkYwRAIgbnnq5u7o9FYFhPSvYPSrNFfJnv0viog094wmY42HYKgCIA%2FDdH6DiaMSbluMlMydMfgEtOG%2B5U2OYUamUY0IOX0nKqoCCIf%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEQAhoMMDcyMTk4MjI4MzA2IgxcAuKF6rQFREZ5o8cq%2FgE2j9JuiCLgIaoJUaA1UU5oZkVVtcHi5phvHnkdBmfMpZ336%2BVoJIVv2KApYq9JrSoA1W8C%2FlwbF4KNICZcLoGQjB1CpxxuAn8qc9Y9g6q0SsJP2qNR3%2FezqhrlGuugEoBWof4UgaWR23BcD0EwyusRuw9JTiumzIByJhtLozPEsBrzOIeeAgnYXOIqD4BobQdJI1ljka%2Bwf0T7QihNhlvkqz3q5B5Ge1sf3yu6lF3rrW4Nitkmi0YUTe%2FR57OP8Qq4gSPvgQSbVKHCdZTT8%2Fixo2C6859fAr4%2Fsal%2FuM8Si8tnPaX%2Fkm6r%2Fd0YhRNdjm8JEze9HJwnIqLwgZGHCjDu376WBjqbAf%2FqZf4j2YHaw9hOUJYI4iZoXCBTzh9zLOxiY9jUI4tmvcDj1nSPEOEUlk%2FGRolFG3KTNVuVKqLroMtyRvI%2FUP5Z20JOEJ7Z6IHVO57G4Iu8xWePdyhZA1p2%2FFwUpyZFHO8m91V2qWzwqPM0S6pKI9r5p7eC7QvGoA6N3v64WSVjeJniMm5QWeCiWxctjjGRIjNdNgP5TCTGetaM&X-Amz-Signature=79d52b993719f5a17ab1102ea3a9f5aebb4afa2c80ed2b53bc8264d0e4cc653f\n",
            "Resolving discover-test-datalake-v1.s3.amazonaws.com (discover-test-datalake-v1.s3.amazonaws.com)... 52.219.32.193\n",
            "Connecting to discover-test-datalake-v1.s3.amazonaws.com (discover-test-datalake-v1.s3.amazonaws.com)|52.219.32.193|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 149121 (146K) [binary/octet-stream]\n",
            "Saving to: 'data_final/PXD002408_Biological_Replicate_2.gct'\n",
            "\n",
            "     0K .......... .......... .......... .......... .......... 34%  306K 0s"
          ]
        },
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Downloaded data successfully\n"
          ]
        },
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "\n",
            "    50K .......... .......... .......... .......... .......... 68%  306K 0s\n",
            "   100K .......... .......... .......... .......... .....     100%  280K=0.5s\n",
            "\n",
            "2022-07-14 05:56:02 (297 KB/s) - 'data_final/PXD002408_Biological_Replicate_2.gct' saved [149121/149121]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "#create a folder data_final in the working repository\n",
        "for i in metadata['dataset_id']:\n",
        "    print (str(i))\n",
        "    data = omixatlas.download_data(\"1657110718820\",str(i))\n",
        "    file_name = f\"{str(i)}.gct\"\n",
        "    url = data.get('data').get('attributes').get('download_url')\n",
        "    status = os.system(f\"wget -O 'data_final/{file_name}' '{url}'\")\n",
        "    if status == 0:\n",
        "        print(\"Downloaded data successfully\")\n",
        "    else:\n",
        "        raise Exception(\"Download not successful\")"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## Ingesting the data"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "In order to ingest the data, user can use the following function:-\n",
        "\n",
        "**add_datasets(repo_id (int/str), source_folder_path (dict), destination_folder (str) (optional), priority (str) (optiona))**\n",
        "\n",
        "**Input:** \n",
        "\n",
        "repo_id: This is the repository ID to which ingestion should be done\n",
        "\n",
        "source_folder_path: This is the dictionary with keys \"data\" and \"metadata\". The corresponding value pairs should be the folder containing the file (gct, h5ad, vcf, mmcif etc) for data and folder containing json of dataset level metadata for metadata.\n",
        "\n",
        "destination_folder (optional): This is the folder within S3 when data gets pushed\n",
        "\n",
        "priority (optional): This is the priority flag as per ingestion is being done. Default is 'medium'\n",
        "\n",
        "**Output:** \n",
        "\n",
        "Status of file upload for each dataset in a dataframe"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 37,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "{'data': '/import/data_final', 'metadata': '/import/metadata_final'}\n",
            "                              File Name        Message\n",
            "0                combined_metadata.json  File Uploaded\n",
            "1  PXD002408_Biological_Replicate_1.gct  File Uploaded\n",
            "2  PXD002408_Biological_Replicate_2.gct  File Uploaded\n"
          ]
        }
      ],
      "source": [
        "repo_id = \"1657777478388\"\n",
        "source_folder_path_data = \"/import/data_final\"\n",
        "source_folder_metadata = \"/import/metadata_final\"\n",
        "destination_folder = \"220714-1126\"\n",
        "priority = \"high\"\n",
        "source_folder_path = {\"data\":source_folder_path_data, \"metadata\":source_folder_metadata}\n",
        "print(source_folder_path)\n",
        "omixatlas.add_datasets(repo_id, source_folder_path, destination_folder, priority)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## Deletion of datset from the OmixAtlas"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "**delete_datasets(repo_id: int/str, dataset_ids: list)**\n",
        "\n",
        "This function is used to delete data from an omixatlas\n",
        "\n",
        "**Input:**\n",
        "\n",
        "repo_id: (int/str) This is the repository ID from which dataset should be deleted\n",
        "\n",
        "dataset_ids: (list) dataset_ids that users want to delete\n",
        "\n",
        "**Output:**\n",
        "\n",
        "Status of file deletion for each dataset in a dataframe"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 5,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "                         Dataset Id  \\\n",
            "0  PXD002408_Biological_Replicate_1   \n",
            "\n",
            "                                                                      Message  \n",
            "0  Request Accepted. Dataset Will be deleted in the next version of OmixAtlas  \n"
          ]
        }
      ],
      "source": [
        "repo_id = \"1657777478388\"\n",
        "dataset_ids = [\"PXD002408_Biological_Replicate_1\"]\n",
        "omixatlas.delete_datasets(repo_id, dataset_ids)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": []
    }
  ],
  "metadata": {
    "kernelspec": {
      "display_name": "Python 3 (ipykernel)",
      "language": "python",
      "name": "python3"
    },
    "language_info": {
      "codemirror_mode": {
        "name": "ipython",
        "version": 3
      },
      "file_extension": ".py",
      "mimetype": "text/x-python",
      "name": "python",
      "nbconvert_exporter": "python",
      "pygments_lexer": "ipython3",
      "version": "3.10.2"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 2
}