{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "colab_type": "text",
        "id": "view-in-github"
      },
      "source": [
        "<a href=\"https://polly.elucidata.io/manage/workspaces?action=open_polly_notebook&source=github&path=path_place_holder&kernel=elucidata/Python 3.10&machine=medium\" target=\"_parent\"><img src=\"https://elucidatainc.github.io/PublicAssets/open_polly.svg\" alt=\"Open in Polly\"/></a>\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "# Ingestion Case 1\n",
        "\n",
        "## Task: Upload selected datasets from an existing OmixAtlas to a new OmixAtlas\n",
        "\n",
        "Assumptions: \n",
        "1. No custom curation required\n",
        "2. Schema of new OA to be different from the existing OA\n",
        "3. Indexing to be done only for dataset and sample level metadata\n",
        "\n",
        "Steps:\n",
        "#### Create OA\n",
        "1. Create OA on Test Polly\n",
        "#### Schema management\n",
        "2. Create schema in csv file\n",
        "3. Prepare schema payload for both dataset and sample only.\n",
        "4. Insert schema for dataset and sample level metadata\n",
        "5. Verify if schema is properly inserted\n",
        "#### Preparing files\n",
        "6. Fetch the template which dataset level metadata should follow\n",
        "7. Prepare dataset level metadata\n",
        "8. Verify if dataset level metadata is as per the template \n",
        "9. Prepare gct to be uploaded\n",
        "#### Ingestion\n",
        "10. Ingestion\n",
        "11. Verify indexing"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "/bin/bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)\n",
            "Collecting polly-python==0.1.5\n",
            "  Downloading https://elucidatainc.github.io/PublicAssets/builds/polly-python/polly_python-0.1.5-py3-none-any.whl (52 kB)\n",
            "\u001b[K     |████████████████████████████████| 52 kB 1.5 MB/s eta 0:00:011\n",
            "\u001b[?25hCollecting python-magic==0.4.24\n",
            "  Downloading python_magic-0.4.24-py2.py3-none-any.whl (12 kB)\n",
            "Collecting rst2txt\n",
            "  Downloading rst2txt-1.1.0-py2.py3-none-any.whl (12 kB)\n",
            "Requirement already satisfied: tqdm in /usr/local/lib/python3.10/site-packages (from polly-python==0.1.5) (4.64.0)\n",
            "Collecting postpy2==0.0.6\n",
            "  Downloading postpy2-0.0.6-py3-none-any.whl (17 kB)\n",
            "Collecting Cerberus==1.3.2\n",
            "  Downloading Cerberus-1.3.2.tar.gz (52 kB)\n",
            "\u001b[K     |████████████████████████████████| 52 kB 2.2 MB/s eta 0:00:011\n",
            "\u001b[?25hRequirement already satisfied: urllib3==1.26.6 in /usr/local/lib/python3.10/site-packages (from polly-python==0.1.5) (1.26.6)\n",
            "Collecting pytest\n",
            "  Downloading pytest-7.1.2-py3-none-any.whl (297 kB)\n",
            "\u001b[K     |████████████████████████████████| 297 kB 6.4 MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting cmapPy\n",
            "  Downloading cmapPy-4.0.1-py2.py3-none-any.whl (150 kB)\n",
            "\u001b[K     |████████████████████████████████| 150 kB 131.8 MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting joblib\n",
            "  Downloading joblib-1.1.0-py2.py3-none-any.whl (306 kB)\n",
            "\u001b[K     |████████████████████████████████| 306 kB 127.5 MB/s eta 0:00:01\n",
            "\u001b[?25hRequirement already satisfied: idna==2.10 in /usr/local/lib/python3.10/site-packages (from polly-python==0.1.5) (2.10)\n",
            "Collecting sqlparse\n",
            "  Downloading sqlparse-0.4.2-py3-none-any.whl (42 kB)\n",
            "\u001b[K     |████████████████████████████████| 42 kB 2.9 MB/s s eta 0:00:01\n",
            "\u001b[?25hCollecting requests==2.25.1\n",
            "  Downloading requests-2.25.1-py2.py3-none-any.whl (61 kB)\n",
            "\u001b[K     |████████████████████████████████| 61 kB 21.2 MB/s  eta 0:00:01\n",
            "\u001b[?25hCollecting pytz==2021.1\n",
            "  Downloading pytz-2021.1-py2.py3-none-any.whl (510 kB)\n",
            "\u001b[K     |████████████████████████████████| 510 kB 139.0 MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting elucidatacloudpathlib==0.6.6\n",
            "  Downloading elucidatacloudpathlib-0.6.6-py3-none-any.whl (50 kB)\n",
            "\u001b[K     |████████████████████████████████| 50 kB 18.5 MB/s  eta 0:00:01\n",
            "\u001b[?25hCollecting certifi==2021.10.8\n",
            "  Downloading certifi-2021.10.8-py2.py3-none-any.whl (149 kB)\n",
            "\u001b[K     |████████████████████████████████| 149 kB 142.5 MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting retrying==1.3.3\n",
            "  Downloading retrying-1.3.3.tar.gz (10 kB)\n",
            "Requirement already satisfied: python-dateutil==2.8.1 in /usr/local/lib/python3.10/site-packages (from polly-python==0.1.5) (2.8.1)\n",
            "Requirement already satisfied: boto3>=1.17.73 in /usr/local/lib/python3.10/site-packages (from polly-python==0.1.5) (1.21.19)\n",
            "Requirement already satisfied: chardet==4.0.0 in /usr/local/lib/python3.10/site-packages (from polly-python==0.1.5) (4.0.0)\n",
            "Collecting Deprecated\n",
            "  Downloading Deprecated-1.2.13-py2.py3-none-any.whl (9.6 kB)\n",
            "Requirement already satisfied: botocore>=1.20.73 in /usr/local/lib/python3.10/site-packages (from polly-python==0.1.5) (1.24.19)\n",
            "Requirement already satisfied: six==1.16.0 in /usr/local/lib/python3.10/site-packages (from polly-python==0.1.5) (1.16.0)\n",
            "Collecting black\n",
            "  Downloading black-22.6.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.5 MB)\n",
            "\u001b[K     |████████████████████████████████| 1.5 MB 116.1 MB/s eta 0:00:01\n",
            "\u001b[?25hRequirement already satisfied: pandas in /usr/local/lib/python3.10/site-packages (from polly-python==0.1.5) (1.4.1)\n",
            "Requirement already satisfied: setuptools in /usr/local/lib/python3.10/site-packages (from Cerberus==1.3.2->polly-python==0.1.5) (58.1.0)\n",
            "Requirement already satisfied: s3transfer<0.6.0,>=0.5.0 in /usr/local/lib/python3.10/site-packages (from boto3>=1.17.73->polly-python==0.1.5) (0.5.2)\n",
            "Requirement already satisfied: jmespath<1.0.0,>=0.7.1 in /usr/local/lib/python3.10/site-packages (from boto3>=1.17.73->polly-python==0.1.5) (0.10.0)\n",
            "Collecting platformdirs>=2\n",
            "  Downloading platformdirs-2.5.2-py3-none-any.whl (14 kB)\n",
            "Collecting tomli>=1.1.0\n",
            "  Downloading tomli-2.0.1-py3-none-any.whl (12 kB)\n",
            "Collecting mypy-extensions>=0.4.3\n",
            "  Downloading mypy_extensions-0.4.3-py2.py3-none-any.whl (4.5 kB)\n",
            "Collecting click>=8.0.0\n",
            "  Downloading click-8.1.3-py3-none-any.whl (96 kB)\n",
            "\u001b[K     |████████████████████████████████| 96 kB 16.6 MB/s  eta 0:00:01\n",
            "\u001b[?25hCollecting pathspec>=0.9.0\n",
            "  Downloading pathspec-0.9.0-py2.py3-none-any.whl (31 kB)\n",
            "Requirement already satisfied: numpy>=1.11.2 in /usr/local/lib/python3.10/site-packages (from cmapPy->polly-python==0.1.5) (1.22.3)\n",
            "Collecting h5py>=2.6.0\n",
            "  Downloading h5py-3.7.0-cp310-cp310-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (4.5 MB)\n",
            "\u001b[K     |████████████████████████████████| 4.5 MB 29.0 MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting wrapt<2,>=1.10\n",
            "  Downloading wrapt-1.14.1-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (77 kB)\n",
            "\u001b[K     |████████████████████████████████| 77 kB 12.8 MB/s eta 0:00:01\n",
            "\u001b[?25hCollecting py>=1.8.2\n",
            "  Downloading py-1.11.0-py2.py3-none-any.whl (98 kB)\n",
            "\u001b[K     |████████████████████████████████| 98 kB 19.3 MB/s  eta 0:00:01\n",
            "\u001b[?25hCollecting pluggy<2.0,>=0.12\n",
            "  Downloading pluggy-1.0.0-py2.py3-none-any.whl (13 kB)\n",
            "Requirement already satisfied: attrs>=19.2.0 in /usr/local/lib/python3.10/site-packages (from pytest->polly-python==0.1.5) (21.2.0)\n",
            "Collecting iniconfig\n",
            "  Downloading iniconfig-1.1.1-py2.py3-none-any.whl (5.0 kB)\n",
            "Requirement already satisfied: packaging in /usr/local/lib/python3.10/site-packages (from pytest->polly-python==0.1.5) (21.3)\n",
            "Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in /usr/local/lib/python3.10/site-packages (from packaging->pytest->polly-python==0.1.5) (3.0.6)\n",
            "Requirement already satisfied: pygments in /usr/local/lib/python3.10/site-packages (from rst2txt->polly-python==0.1.5) (2.10.0)\n",
            "Requirement already satisfied: docutils>=0.11 in /usr/local/lib/python3.10/site-packages (from rst2txt->polly-python==0.1.5) (0.15.2)\n",
            "Using legacy 'setup.py install' for Cerberus, since package 'wheel' is not installed.\n",
            "Using legacy 'setup.py install' for retrying, since package 'wheel' is not installed.\n",
            "Installing collected packages: pytz, certifi, wrapt, tomli, requests, python-magic, py, pluggy, platformdirs, pathspec, mypy-extensions, iniconfig, h5py, click, sqlparse, rst2txt, retrying, pytest, postpy2, joblib, elucidatacloudpathlib, Deprecated, cmapPy, Cerberus, black, polly-python\n",
            "  Attempting uninstall: pytz\n",
            "    Found existing installation: pytz 2021.3\n",
            "    Uninstalling pytz-2021.3:\n",
            "      Successfully uninstalled pytz-2021.3\n",
            "  Attempting uninstall: certifi\n",
            "    Found existing installation: certifi 2021.5.30\n",
            "    Uninstalling certifi-2021.5.30:\n",
            "      Successfully uninstalled certifi-2021.5.30\n",
            "  Attempting uninstall: requests\n",
            "    Found existing installation: requests 2.27.1\n",
            "    Uninstalling requests-2.27.1:\n",
            "      Successfully uninstalled requests-2.27.1\n",
            "    Running setup.py install for retrying ... \u001b[?25ldone\n",
            "\u001b[?25h  Attempting uninstall: Cerberus\n",
            "    Found existing installation: Cerberus 1.3.4\n",
            "    Uninstalling Cerberus-1.3.4:\n",
            "      Successfully uninstalled Cerberus-1.3.4\n",
            "    Running setup.py install for Cerberus ... \u001b[?25ldone\n",
            "\u001b[?25hSuccessfully installed Cerberus-1.3.2 Deprecated-1.2.13 black-22.6.0 certifi-2021.10.8 click-8.1.3 cmapPy-4.0.1 elucidatacloudpathlib-0.6.6 h5py-3.7.0 iniconfig-1.1.1 joblib-1.1.0 mypy-extensions-0.4.3 pathspec-0.9.0 platformdirs-2.5.2 pluggy-1.0.0 polly-python-0.1.5 postpy2-0.0.6 py-1.11.0 pytest-7.1.2 python-magic-0.4.24 pytz-2021.1 requests-2.25.1 retrying-1.3.3 rst2txt-1.1.0 sqlparse-0.4.2 tomli-2.0.1 wrapt-1.14.1\n",
            "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\n",
            "\u001b[33mWARNING: You are using pip version 21.2.4; however, version 22.1.2 is available.\n",
            "You should consider upgrading via the '/usr/local/bin/python3.10 -m pip install --upgrade pip' command.\u001b[0m\n"
          ]
        }
      ],
      "source": [
        "!sudo pip3 install https://elucidatainc.github.io/PublicAssets/builds/polly-python/polly_python-0.1.5-py3-none-any.whl"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "/bin/bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)\n",
            "\u001b[32m\u001b[1mA new version of Polly CLI is available. To update, execute the command npm update -g @elucidatainc/pollycli\u001b[22m\u001b[39m\n",
            "\u001b[32m\u001b[1mRefreshing session...\u001b[22m\u001b[39m\n",
            "\u001b[32m\u001b[1mSession refreshed!\u001b[22m\u001b[39m\n",
            "polly://dataset_schema.csv\n",
            "./\n",
            "\u001b7\u001b[?7l\u001b[1Gprogress [░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░] 0% | NA | ETA: 0s | time elapsed: 0s\u001b[0K\u001b[1Gprogress [████████████████████████████████████████] 100% | 1.198 KB/1.198 KB | ETA: 0s | time elapsed: 0s\u001b[0K\u001b[?7h\u001b8\n",
            "\u001b[32m\u001b[1mSuccess: Download complete\u001b[22m\u001b[39m\n",
            "/bin/bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)\n",
            "\u001b[32m\u001b[1mA new version of Polly CLI is available. To update, execute the command npm update -g @elucidatainc/pollycli\u001b[22m\u001b[39m\n",
            "polly://sample_schema.csv\n",
            "./\n",
            "\u001b7\u001b[?7l\u001b[1Gprogress [░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░] 0% | NA | ETA: 0s | time elapsed: 0s\u001b[0K\u001b[1Gprogress [████████████████████████████████████████] 100% | 831 Bytes/831 Bytes | ETA: 0s | time elapsed: 0s\u001b[0K\u001b[?7h\u001b8\n",
            "\u001b[32m\u001b[1mSuccess: Download complete\u001b[22m\u001b[39m\n"
          ]
        }
      ],
      "source": [
        "! polly files copy -y -s polly://dataset_schema.csv -d ./ \n",
        "! polly files copy -y -s polly://sample_schema.csv -d ./ "
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### Importing dependecies and authenticating into Polly"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 31,
      "metadata": {},
      "outputs": [],
      "source": [
        "import os\n",
        "from polly.omixatlas import OmixAtlas\n",
        "from polly.workspaces import Workspaces\n",
        "\n",
        "from polly.auth import Polly\n",
        "#AUTH_KEY = 'KEY' # prod polly\n",
        "AUTH_KEY= \"KEY\" # test polly\n",
        "Polly.auth(AUTH_KEY, env = \"testpolly\")\n",
        "\n",
        "omixatlas = OmixAtlas()\n",
        "workspaces = Workspaces()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### generating schema payload"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "This block of code will be used to generate the payload of schema when the schema is prepared in the csv file. \n",
        "\n",
        "**Input:** csv file containing the schema \n",
        "\n",
        "**Output:** payload to be ingested to the OmixAtlas"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 4,
      "metadata": {},
      "outputs": [],
      "source": [
        "def get_fields(schema_file):\n",
        "    schema_df = pd.read_csv(schema_file)\n",
        "    col_types = {\n",
        "        \"field_name\": str,\n",
        "        \"original_name\": str,\n",
        "        \"type\": str,\n",
        "        \"is_keyword\": bool,\n",
        "        \"is_array\": bool,\n",
        "        \"is_filter\": bool,\n",
        "        \"is_column\": bool,\n",
        "        \"is_keyword\": bool,\n",
        "        \"filter_size\": int,\n",
        "        \"display_name\": str,\n",
        "        \"description\": str,\n",
        "    }\n",
        "    schema_df = schema_df.astype(col_types)\n",
        "    schema_df[\"field_name\"] = schema_df[\"field_name\"].apply(lambda x: x.lower().strip().replace(\" \", \"_\"))\n",
        "    schema_df[\"type\"] = schema_df[\"type\"].str.lower().str.strip()\n",
        "    schema_df[\"description\"] = schema_df[\"description\"].str.replace(\"nan\", \"NA\")\n",
        "    \n",
        "    object_rows = schema_df[schema_df[\"type\"] == \"object\"]\n",
        "    object_rows_array = object_rows[object_rows[\"is_array\"] == 1]\n",
        "    if len(object_rows_array.index) > 0:\n",
        "        raise ValueError(\"Fields with type 'object' cannot be arrays\")\n",
        "    object_rows_keyword = object_rows[object_rows[\"is_keyword\"] == 1]\n",
        "    if len(object_rows_keyword.index) > 0:\n",
        "        raise ValueError(\"Fields with type 'object' cannot be keywords\")\n",
        "    \n",
        "    field_list = schema_df.to_dict(orient=\"records\")\n",
        "    fields = {}\n",
        "    for field in field_list:\n",
        "        field_name = field.pop(\"field_name\")\n",
        "        fields[field_name] = field\n",
        "    \n",
        "    return fields\n",
        "\n",
        "def generate_schema_payload(schema_type, schema_file):\n",
        "    return {\n",
        "        \"data\": {\n",
        "            \"type\": \"schemas\",\n",
        "            \"id\": repo_id,\n",
        "            \"attributes\": {\n",
        "                \"repo_id\": repo_id,\n",
        "                \"schema_type\": schema_type,\n",
        "                \"schema\": {\n",
        "                    \"all\": {\n",
        "                        \"all\": get_fields(schema_file)\n",
        "                    }\n",
        "                }\n",
        "            }\n",
        "        }\n",
        "    }"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### generating schema for dataset level metadata (files)\n",
        "\n",
        "Calling the functions above to generate the schema payload. \n",
        "\n",
        "Here, the repo_id must be added as a string. The file is the input csv file which contains the schema in excel sheet."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 65,
      "metadata": {
        "scrolled": true
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Creating dataset-level metadata schema...\n"
          ]
        },
        {
          "data": {
            "text/plain": [
              "{'data': {'type': 'schemas',\n",
              "  'id': '1657110718820',\n",
              "  'attributes': {'repo_id': '1657110718820',\n",
              "   'schema_type': 'files',\n",
              "   'schema': {'all': {'all': {'dataset_source': {'original_name': 'dataset_source',\n",
              "       'type': 'text',\n",
              "       'is_keyword': True,\n",
              "       'is_array': False,\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Source',\n",
              "       'description': 'Source from where the data was fetched'},\n",
              "      'dataset_id': {'original_name': 'dataset_id',\n",
              "       'type': 'text',\n",
              "       'is_keyword': True,\n",
              "       'is_array': False,\n",
              "       'is_filter': False,\n",
              "       'is_column': True,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Dataset ID',\n",
              "       'description': 'Unique ID assocaited with every dataset'},\n",
              "      'description': {'original_name': 'description',\n",
              "       'type': 'text',\n",
              "       'is_keyword': False,\n",
              "       'is_array': False,\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Title',\n",
              "       'description': 'Description of the dataset'},\n",
              "      'curated_disease': {'original_name': 'curated_disease',\n",
              "       'type': 'text',\n",
              "       'is_keyword': True,\n",
              "       'is_array': True,\n",
              "       'is_filter': True,\n",
              "       'is_column': True,\n",
              "       'filter_size': 3000,\n",
              "       'display_name': 'Disease',\n",
              "       'description': 'Disease associated with the dataset'},\n",
              "      'curated_cell_line': {'original_name': 'curated_cell_line',\n",
              "       'type': 'text',\n",
              "       'is_keyword': True,\n",
              "       'is_array': True,\n",
              "       'is_filter': True,\n",
              "       'is_column': True,\n",
              "       'filter_size': 1000,\n",
              "       'display_name': 'Cell line',\n",
              "       'description': 'Cell lines from which the samples were derived for this dataset'},\n",
              "      'curated_cell_type': {'original_name': 'curated_cell_type',\n",
              "       'type': 'text',\n",
              "       'is_keyword': True,\n",
              "       'is_array': True,\n",
              "       'is_filter': False,\n",
              "       'is_column': True,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Cell type',\n",
              "       'description': 'Types of cell present in the dataset'},\n",
              "      'data_type': {'original_name': 'data_type',\n",
              "       'type': 'text',\n",
              "       'is_keyword': True,\n",
              "       'is_array': False,\n",
              "       'is_filter': True,\n",
              "       'is_column': True,\n",
              "       'filter_size': 10,\n",
              "       'display_name': 'Data type',\n",
              "       'description': 'The type of biomolecular data captured (eg - Exome Sequencing, Transcriptomics)'},\n",
              "      'curated_organism': {'original_name': 'curated_organism',\n",
              "       'type': 'text',\n",
              "       'is_keyword': True,\n",
              "       'is_array': True,\n",
              "       'is_filter': True,\n",
              "       'is_column': True,\n",
              "       'filter_size': 1000,\n",
              "       'display_name': 'Organism',\n",
              "       'description': 'Orgnism from which the samples were derived'},\n",
              "      'curated_tissue': {'original_name': 'curated_tissue',\n",
              "       'type': 'text',\n",
              "       'is_keyword': True,\n",
              "       'is_array': True,\n",
              "       'is_filter': True,\n",
              "       'is_column': True,\n",
              "       'filter_size': 1000,\n",
              "       'display_name': 'Tissue',\n",
              "       'description': 'Tissue from which the samples were derivved'},\n",
              "      'curated_drug': {'original_name': 'curated_drug',\n",
              "       'type': 'text',\n",
              "       'is_keyword': True,\n",
              "       'is_array': True,\n",
              "       'is_filter': True,\n",
              "       'is_column': True,\n",
              "       'filter_size': 3000,\n",
              "       'display_name': 'Drug',\n",
              "       'description': 'Drugs administered in the samples belonging to the dataset'},\n",
              "      'total_num_samples': {'original_name': 'total_num_samples',\n",
              "       'type': 'integer',\n",
              "       'is_keyword': True,\n",
              "       'is_array': False,\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Number of Samples',\n",
              "       'description': 'Total number of samples in a dataset'}}}}}}}"
            ]
          },
          "execution_count": 65,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "print(\"Creating dataset-level metadata schema...\")\n",
        "repo_id = \"1657110718820\"\n",
        "file = \"dataset_schema.csv\"\n",
        "payload_d = generate_schema_payload(\"files\", file)\n",
        "payload_d"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### adding dataset level schema to OA\n",
        "\n",
        "The payload above can be used with polly python function insert_schema to add the schema in a new OmixAtlas"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 67,
      "metadata": {
        "scrolled": true
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "'{\"data\": {\"type\": \"schemas\", \"id\": \"1657110718820\", \"attributes\": {\"repo_id\": \"1657110718820\", \"schema_type\": \"files\", \"schema\": {\"all\": {\"all\": {\"dataset_source\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"dataset_source\", \"display_name\": \"Source\", \"description\": \"Source from where the data was fetched\"}, \"dataset_id\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"dataset_id\", \"display_name\": \"Dataset ID\", \"description\": \"Unique ID assocaited with every dataset\"}, \"description\": {\"type\": \"text\", \"is_keyword\": false, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"description\", \"display_name\": \"Title\", \"description\": \"Description of the dataset\"}, \"curated_disease\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": true, \"is_filter\": true, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 3000, \"original_name\": \"curated_disease\", \"display_name\": \"Disease\", \"description\": \"Disease associated with the dataset\"}, \"curated_cell_line\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": true, \"is_filter\": true, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1000, \"original_name\": \"curated_cell_line\", \"display_name\": \"Cell line\", \"description\": \"Cell lines from which the samples were derived for this dataset\"}, \"curated_cell_type\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": true, \"is_filter\": false, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"curated_cell_type\", \"display_name\": \"Cell type\", \"description\": \"Types of cell present in the dataset\"}, \"data_type\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": true, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 10, \"original_name\": \"data_type\", \"display_name\": \"Data type\", \"description\": \"The type of biomolecular data captured (eg - Exome Sequencing, Transcriptomics)\"}, \"curated_organism\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": true, \"is_filter\": true, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1000, \"original_name\": \"curated_organism\", \"display_name\": \"Organism\", \"description\": \"Orgnism from which the samples were derived\"}, \"curated_tissue\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": true, \"is_filter\": true, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1000, \"original_name\": \"curated_tissue\", \"display_name\": \"Tissue\", \"description\": \"Tissue from which the samples were derivved\"}, \"curated_drug\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": true, \"is_filter\": true, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 3000, \"original_name\": \"curated_drug\", \"display_name\": \"Drug\", \"description\": \"Drugs administered in the samples belonging to the dataset\"}, \"total_num_samples\": {\"type\": \"integer\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"total_num_samples\", \"display_name\": \"Number of Samples\", \"description\": \"Total number of samples in a dataset\"}}}}, \"created_time\": 1657173537526, \"last_updated_time\": 1657183791788, \"enabled\": true}, \"links\": {\"self\": \"/repositories/1657110718820/schema/files\"}}}'"
            ]
          },
          "execution_count": 67,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "omixatlas.insert_schema(1657110718820, payload_d)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### generating schema for sample level metadata (gct_metadata)\n",
        "\n",
        "Just like the dataset level metadata, the sample level metadata should also be added to the OmixAtlas."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 19,
      "metadata": {
        "scrolled": true
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Creating dataset-level metadata schema...\n"
          ]
        },
        {
          "data": {
            "text/plain": [
              "{'data': {'type': 'schemas',\n",
              "  'id': '1657110718820',\n",
              "  'attributes': {'repo_id': '1657110718820',\n",
              "   'schema_type': 'gct_metadata',\n",
              "   'schema': {'all': {'all': {'sample_id': {'original_name': 'sample_id',\n",
              "       'type': 'text',\n",
              "       'is_keyword': True,\n",
              "       'is_array': False,\n",
              "       'is_filter': False,\n",
              "       'is_column': True,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Sample ID',\n",
              "       'description': 'Unique ID associated with every sample'},\n",
              "      'dataset_id': {'original_name': 'dataset_id',\n",
              "       'type': 'text',\n",
              "       'is_keyword': True,\n",
              "       'is_array': True,\n",
              "       'is_filter': False,\n",
              "       'is_column': True,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Dataset ID',\n",
              "       'description': 'Unique ID of the dataset to which the sample belongs'},\n",
              "      'timestamp_': {'original_name': 'timestamp_',\n",
              "       'type': 'text',\n",
              "       'is_keyword': False,\n",
              "       'is_array': False,\n",
              "       'is_filter': False,\n",
              "       'is_column': False,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'timestamp',\n",
              "       'description': 'Time stamp when the dataset was last updated'},\n",
              "      'curated_gene': {'original_name': 'curated_gene',\n",
              "       'type': 'text',\n",
              "       'is_keyword': True,\n",
              "       'is_array': True,\n",
              "       'is_filter': False,\n",
              "       'is_column': True,\n",
              "       'filter_size': 1,\n",
              "       'display_name': 'Gene',\n",
              "       'description': 'Gene of interest in the sample'},\n",
              "      'curated_drug': {'original_name': 'curated_drug',\n",
              "       'type': 'text',\n",
              "       'is_keyword': True,\n",
              "       'is_array': True,\n",
              "       'is_filter': True,\n",
              "       'is_column': True,\n",
              "       'filter_size': 3000,\n",
              "       'display_name': 'Drug',\n",
              "       'description': 'Drug admistered in the sample'},\n",
              "      'curated_cell_line': {'original_name': 'curated_cell_line',\n",
              "       'type': 'text',\n",
              "       'is_keyword': True,\n",
              "       'is_array': False,\n",
              "       'is_filter': True,\n",
              "       'is_column': True,\n",
              "       'filter_size': 1000,\n",
              "       'display_name': 'Cell line',\n",
              "       'description': 'Cell line from which the sample was derived'},\n",
              "      'curated_disease': {'original_name': 'curated_disease',\n",
              "       'type': 'text',\n",
              "       'is_keyword': True,\n",
              "       'is_array': True,\n",
              "       'is_filter': True,\n",
              "       'is_column': True,\n",
              "       'filter_size': 3000,\n",
              "       'display_name': 'Disease',\n",
              "       'description': 'Disease associated with the sample'},\n",
              "      'curated_tissue': {'original_name': 'curated_tissue',\n",
              "       'type': 'text',\n",
              "       'is_keyword': True,\n",
              "       'is_array': False,\n",
              "       'is_filter': True,\n",
              "       'is_column': True,\n",
              "       'filter_size': 1000,\n",
              "       'display_name': 'Tissue',\n",
              "       'description': 'Tissue to which the sample belongs'}}}}}}}"
            ]
          },
          "execution_count": 19,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "print(\"Creating dataset-level metadata schema...\")\n",
        "repo_id = \"1657110718820\" #to be given as string (repo_name is not acceptable here)\n",
        "file = \"sample_schema.csv\"\n",
        "payload_s = generate_schema_payload(\"gct_metadata\", file)\n",
        "payload_s"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### adding sample level schema to OA\n",
        "\n",
        "The payload above can be used with polly python function insert_schema to add the schema in a new OmixAtlas"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 20,
      "metadata": {
        "scrolled": true
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "'{\"data\": {\"type\": \"schemas\", \"id\": \"1657110718820\", \"attributes\": {\"repo_id\": \"1657110718820\", \"schema_type\": \"gct_metadata\", \"schema\": {\"all\": {\"all\": {\"sample_id\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"sample_id\", \"display_name\": \"Sample ID\", \"description\": \"Unique ID associated with every sample\"}, \"dataset_id\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": true, \"is_filter\": false, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"dataset_id\", \"display_name\": \"Dataset ID\", \"description\": \"Unique ID of the dataset to which the sample belongs\"}, \"timestamp_\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 2000, \"original_name\": \"kw_timestamp_\", \"display_name\": \"Timestamp\", \"description\": \"Unix timestamp denoting time of creation for this entry\"}, \"curated_gene\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": true, \"is_filter\": false, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1, \"original_name\": \"curated_gene\", \"display_name\": \"Gene\", \"description\": \"Gene of interest in the sample\"}, \"curated_drug\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": true, \"is_filter\": true, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 3000, \"original_name\": \"curated_drug\", \"display_name\": \"Drug\", \"description\": \"Drug admistered in the sample\"}, \"curated_cell_line\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": true, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1000, \"original_name\": \"curated_cell_line\", \"display_name\": \"Cell line\", \"description\": \"Cell line from which the sample was derived\"}, \"curated_disease\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": true, \"is_filter\": true, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 3000, \"original_name\": \"curated_disease\", \"display_name\": \"Disease\", \"description\": \"Disease associated with the sample\"}, \"curated_tissue\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": true, \"is_column\": true, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 1000, \"original_name\": \"curated_tissue\", \"display_name\": \"Tissue\", \"description\": \"Tissue to which the sample belongs\"}, \"version\": {\"type\": \"integer\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 2000, \"original_name\": \"version\", \"display_name\": \"Version\", \"description\": \"Version of data entity\"}, \"is_current\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 2000, \"original_name\": \"is_current\", \"display_name\": \"Is current\", \"description\": \"Whether this is the current version of the data entity or not\"}, \"id_key\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 2000, \"original_name\": \"kw_id_key\", \"display_name\": \"ID key\", \"description\": \"Name of the key that was used for creation of unique identity\"}, \"data_id\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 2000, \"original_name\": \"kw_data_id\", \"display_name\": \"Data ID\", \"description\": \"Unique ID for this data entity on Polly\"}, \"name\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 2000, \"original_name\": \"kw_name\", \"display_name\": \"Name\", \"description\": \"Name of the data entity (local to its source file)\"}, \"src_repo\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 2000, \"original_name\": \"kw_src_repo\", \"display_name\": \"Source repository\", \"description\": \"Name of the repository this data entity originated from\"}, \"src_dataset_id\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 2000, \"original_name\": \"kw_src_dataset_id\", \"display_name\": \"Source dataset ID\", \"description\": \"Dataset ID of the file this data entity originated from\"}, \"src_uri\": {\"type\": \"text\", \"is_keyword\": true, \"is_array\": false, \"is_filter\": false, \"is_column\": false, \"is_curated\": false, \"is_ontology\": false, \"filter_size\": 2000, \"original_name\": \"kw_src_uri\", \"display_name\": \"Source URI\", \"description\": \"Unique URI derived from source data file\\'s S3 location\"}}}}, \"created_time\": 1657173913956, \"last_updated_time\": 1657173913956, \"enabled\": true}, \"links\": {\"self\": \"/repositories/1657110718820/schema/gct_metadata\"}}}'"
            ]
          },
          "execution_count": 20,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "omixatlas.insert_schema(1657110718820, payload_s)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### validating if the schema is appropriately added"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 68,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Source</th>\n",
              "      <th>Datatype</th>\n",
              "      <th>Field Name</th>\n",
              "      <th>Field Description</th>\n",
              "      <th>Field Type</th>\n",
              "      <th>Is Curated</th>\n",
              "      <th>Is Array</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>curated_organism</td>\n",
              "      <td>Orgnism from which the samples were derived</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>curated_tissue</td>\n",
              "      <td>Tissue from which the samples were derivved</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>total_num_samples</td>\n",
              "      <td>Total number of samples in a dataset</td>\n",
              "      <td>integer</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>dataset_source</td>\n",
              "      <td>Source from where the data was fetched</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>dataset_id</td>\n",
              "      <td>Unique ID assocaited with every dataset</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>data_type</td>\n",
              "      <td>The type of biomolecular data captured (eg - E...</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>description</td>\n",
              "      <td>Description of the dataset</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>curated_cell_line</td>\n",
              "      <td>Cell lines from which the samples were derived...</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>curated_disease</td>\n",
              "      <td>Disease associated with the dataset</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>curated_drug</td>\n",
              "      <td>Drugs administered in the samples belonging to...</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>curated_cell_type</td>\n",
              "      <td>Types of cell present in the dataset</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "   Source Datatype         Field Name  \\\n",
              "0     all      all   curated_organism   \n",
              "1     all      all     curated_tissue   \n",
              "2     all      all  total_num_samples   \n",
              "3     all      all     dataset_source   \n",
              "4     all      all         dataset_id   \n",
              "5     all      all          data_type   \n",
              "6     all      all        description   \n",
              "7     all      all  curated_cell_line   \n",
              "8     all      all    curated_disease   \n",
              "9     all      all       curated_drug   \n",
              "10    all      all  curated_cell_type   \n",
              "\n",
              "                                    Field Description Field Type  Is Curated  \\\n",
              "0         Orgnism from which the samples were derived       text       False   \n",
              "1         Tissue from which the samples were derivved       text       False   \n",
              "2                Total number of samples in a dataset    integer       False   \n",
              "3              Source from where the data was fetched       text       False   \n",
              "4             Unique ID assocaited with every dataset       text       False   \n",
              "5   The type of biomolecular data captured (eg - E...       text       False   \n",
              "6                          Description of the dataset       text       False   \n",
              "7   Cell lines from which the samples were derived...       text       False   \n",
              "8                 Disease associated with the dataset       text       False   \n",
              "9   Drugs administered in the samples belonging to...       text       False   \n",
              "10               Types of cell present in the dataset       text       False   \n",
              "\n",
              "    Is Array  \n",
              "0       True  \n",
              "1       True  \n",
              "2      False  \n",
              "3      False  \n",
              "4      False  \n",
              "5      False  \n",
              "6      False  \n",
              "7       True  \n",
              "8       True  \n",
              "9       True  \n",
              "10      True  "
            ]
          },
          "execution_count": 68,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "schema = omixatlas.get_schema(1657110718820)\n",
        "schema.dataset"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 22,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Source</th>\n",
              "      <th>Datatype</th>\n",
              "      <th>Field Name</th>\n",
              "      <th>Field Description</th>\n",
              "      <th>Field Type</th>\n",
              "      <th>Is Curated</th>\n",
              "      <th>Is Array</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>src_repo</td>\n",
              "      <td>Name of the repository this data entity origin...</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>id_key</td>\n",
              "      <td>Name of the key that was used for creation of ...</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>src_uri</td>\n",
              "      <td>Unique URI derived from source data file's S3 ...</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>sample_id</td>\n",
              "      <td>Unique ID associated with every sample</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>dataset_id</td>\n",
              "      <td>Unique ID of the dataset to which the sample b...</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>curated_cell_line</td>\n",
              "      <td>Cell line from which the sample was derived</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>curated_gene</td>\n",
              "      <td>Gene of interest in the sample</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>curated_drug</td>\n",
              "      <td>Drug admistered in the sample</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>curated_disease</td>\n",
              "      <td>Disease associated with the sample</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>src_dataset_id</td>\n",
              "      <td>Dataset ID of the file this data entity origin...</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>version</td>\n",
              "      <td>Version of data entity</td>\n",
              "      <td>integer</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>timestamp_</td>\n",
              "      <td>Unix timestamp denoting time of creation for t...</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>12</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>curated_tissue</td>\n",
              "      <td>Tissue to which the sample belongs</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>data_id</td>\n",
              "      <td>Unique ID for this data entity on Polly</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>14</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>name</td>\n",
              "      <td>Name of the data entity (local to its source f...</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>15</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>is_current</td>\n",
              "      <td>Whether this is the current version of the dat...</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "   Source Datatype         Field Name  \\\n",
              "0     all      all           src_repo   \n",
              "1     all      all             id_key   \n",
              "2     all      all            src_uri   \n",
              "3     all      all          sample_id   \n",
              "4     all      all         dataset_id   \n",
              "5     all      all  curated_cell_line   \n",
              "6     all      all       curated_gene   \n",
              "7     all      all       curated_drug   \n",
              "8     all      all    curated_disease   \n",
              "9     all      all     src_dataset_id   \n",
              "10    all      all            version   \n",
              "11    all      all         timestamp_   \n",
              "12    all      all     curated_tissue   \n",
              "13    all      all            data_id   \n",
              "14    all      all               name   \n",
              "15    all      all         is_current   \n",
              "\n",
              "                                    Field Description Field Type  Is Curated  \\\n",
              "0   Name of the repository this data entity origin...       text       False   \n",
              "1   Name of the key that was used for creation of ...       text       False   \n",
              "2   Unique URI derived from source data file's S3 ...       text       False   \n",
              "3              Unique ID associated with every sample       text       False   \n",
              "4   Unique ID of the dataset to which the sample b...       text       False   \n",
              "5         Cell line from which the sample was derived       text       False   \n",
              "6                      Gene of interest in the sample       text       False   \n",
              "7                       Drug admistered in the sample       text       False   \n",
              "8                  Disease associated with the sample       text       False   \n",
              "9   Dataset ID of the file this data entity origin...       text       False   \n",
              "10                             Version of data entity    integer       False   \n",
              "11  Unix timestamp denoting time of creation for t...       text       False   \n",
              "12                 Tissue to which the sample belongs       text       False   \n",
              "13            Unique ID for this data entity on Polly       text       False   \n",
              "14  Name of the data entity (local to its source f...       text       False   \n",
              "15  Whether this is the current version of the dat...       text       False   \n",
              "\n",
              "    Is Array  \n",
              "0      False  \n",
              "1      False  \n",
              "2      False  \n",
              "3      False  \n",
              "4       True  \n",
              "5      False  \n",
              "6       True  \n",
              "7       True  \n",
              "8       True  \n",
              "9      False  \n",
              "10     False  \n",
              "11     False  \n",
              "12     False  \n",
              "13     False  \n",
              "14     False  \n",
              "15     False  "
            ]
          },
          "execution_count": 22,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "schema.sample"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## Preparing dataset level metadata files (.json)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### fetching template for the same\n",
        "\n",
        "In order to prepare the dataset level metadata, the user should ensure the keys of json files they prepare adheres to the schema of dataset level. \n",
        "\n",
        "In order to facilitate this, users can use the function dataset_metadata_template to generate the template. These are mandatory fields to have in the dataset level metadata."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 69,
      "metadata": {},
      "outputs": [],
      "source": [
        "template = omixatlas.dataset_metadata_template(1657110718820)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 70,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/plain": [
              "{'curated_organism': [],\n",
              " 'curated_tissue': [],\n",
              " 'total_num_samples': 'integer',\n",
              " 'dataset_source': 'text',\n",
              " 'dataset_id': 'text',\n",
              " 'data_type': 'text',\n",
              " 'description': 'text',\n",
              " 'curated_cell_line': [],\n",
              " 'curated_disease': [],\n",
              " 'curated_drug': [],\n",
              " 'curated_cell_type': [],\n",
              " '__index__': {'file_metadata': True,\n",
              "  'col_metadata': True,\n",
              "  'row_metadata': False,\n",
              "  'data_required': False}}"
            ]
          },
          "execution_count": 70,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "template"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### from source OmixAtlas, fetching the dataset level metadata for the datasets to be ingested in new OmixAtlas\n",
        "\n",
        "Because some of the information present at dataset level metadata in source OA and destination OA will remain same, we can just query the source OA and use the information to prepare the dataset level metadata for destination OA."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 26,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Query execution succeeded (time taken: 1.14 seconds, data scanned: 0.007 MB)\n",
            "Fetched 2 rows\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>curated_organism</th>\n",
              "      <th>dosage</th>\n",
              "      <th>src_uri</th>\n",
              "      <th>total_num_samples</th>\n",
              "      <th>year</th>\n",
              "      <th>ingestion_approved</th>\n",
              "      <th>description</th>\n",
              "      <th>curated_cell_line</th>\n",
              "      <th>data_table_name</th>\n",
              "      <th>data_table_version</th>\n",
              "      <th>platform</th>\n",
              "      <th>exposure_time</th>\n",
              "      <th>timestamp_</th>\n",
              "      <th>perturbation_type</th>\n",
              "      <th>file_type</th>\n",
              "      <th>publication</th>\n",
              "      <th>curated_cell_type</th>\n",
              "      <th>key</th>\n",
              "      <th>summary</th>\n",
              "      <th>perturbation_modality</th>\n",
              "      <th>src_repo</th>\n",
              "      <th>drug_smiles</th>\n",
              "      <th>package</th>\n",
              "      <th>file_location</th>\n",
              "      <th>author</th>\n",
              "      <th>dataset_id</th>\n",
              "      <th>curated_disease</th>\n",
              "      <th>curated_drug</th>\n",
              "      <th>curated_gene</th>\n",
              "      <th>abstract</th>\n",
              "      <th>version</th>\n",
              "      <th>curated_strain</th>\n",
              "      <th>pubchem_id</th>\n",
              "      <th>bucket</th>\n",
              "      <th>curated_tissue</th>\n",
              "      <th>dataset_source</th>\n",
              "      <th>data_type</th>\n",
              "      <th>overall_design</th>\n",
              "      <th>is_current</th>\n",
              "      <th>region</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>[Homo sapiens]</td>\n",
              "      <td>[30 μM,  50 μM]</td>\n",
              "      <td>polly:data://valo_onco_data_lake/data/PRIDE/Pr...</td>\n",
              "      <td>7</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Approved</td>\n",
              "      <td>Quantitative K-GG site analysis</td>\n",
              "      <td>[HCT 116]</td>\n",
              "      <td>valo_onco__pxd002408_biological_replicate_1</td>\n",
              "      <td>0</td>\n",
              "      <td>LTQ Orbitrap Velos</td>\n",
              "      <td>[2 hours,  6 hours]</td>\n",
              "      <td>1656064257394</td>\n",
              "      <td>[Chemical]</td>\n",
              "      <td>gct</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4...</td>\n",
              "      <td>[colorectal cancer cell line]</td>\n",
              "      <td>valo_onco_data_lake/data/PRIDE/Proteomics/PXD0...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>[Drug]</td>\n",
              "      <td>valo_onco</td>\n",
              "      <td>[]</td>\n",
              "      <td>valo_onco_data_lake/data</td>\n",
              "      <td>https://discover-prod-datalake-v1.s3-us-west-2...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>PXD002408_Biological_Replicate_1</td>\n",
              "      <td>[Colonic Neoplasms]</td>\n",
              "      <td>[dimethyl sulfoxide,  growth hormone]</td>\n",
              "      <td>[USP1,  USP7]</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0</td>\n",
              "      <td>[None]</td>\n",
              "      <td>[679,  46931953]</td>\n",
              "      <td>discover-prod-datalake-v1</td>\n",
              "      <td>[colorectum]</td>\n",
              "      <td>PRIDE</td>\n",
              "      <td>Proteomics</td>\n",
              "      <td>NaN</td>\n",
              "      <td>True</td>\n",
              "      <td>us-west-2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>[Homo sapiens]</td>\n",
              "      <td>[30 μM,  50 μM]</td>\n",
              "      <td>polly:data://valo_onco_data_lake/data/PRIDE/Pr...</td>\n",
              "      <td>3</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Approved</td>\n",
              "      <td>Quantitative K-GG site analysis</td>\n",
              "      <td>[HCT 116]</td>\n",
              "      <td>valo_onco__pxd002408_biological_replicate_2</td>\n",
              "      <td>0</td>\n",
              "      <td>LTQ Orbitrap Velos</td>\n",
              "      <td>[2 hours,  6 hours]</td>\n",
              "      <td>1656064386505</td>\n",
              "      <td>[Chemical]</td>\n",
              "      <td>gct</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4...</td>\n",
              "      <td>[colorectal cancer cell line]</td>\n",
              "      <td>valo_onco_data_lake/data/PRIDE/Proteomics/PXD0...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>[Drug]</td>\n",
              "      <td>valo_onco</td>\n",
              "      <td>[]</td>\n",
              "      <td>valo_onco_data_lake/data</td>\n",
              "      <td>https://discover-prod-datalake-v1.s3-us-west-2...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>PXD002408_Biological_Replicate_2</td>\n",
              "      <td>[Colonic Neoplasms]</td>\n",
              "      <td>[dimethyl sulfoxide,  growth hormone]</td>\n",
              "      <td>[USP1,  USP7]</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0</td>\n",
              "      <td>[None]</td>\n",
              "      <td>[679,  46931953]</td>\n",
              "      <td>discover-prod-datalake-v1</td>\n",
              "      <td>[colorectum]</td>\n",
              "      <td>PRIDE</td>\n",
              "      <td>Proteomics</td>\n",
              "      <td>NaN</td>\n",
              "      <td>True</td>\n",
              "      <td>us-west-2</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "  curated_organism           dosage  \\\n",
              "0   [Homo sapiens]  [30 μM,  50 μM]   \n",
              "1   [Homo sapiens]  [30 μM,  50 μM]   \n",
              "\n",
              "                                             src_uri  total_num_samples  year  \\\n",
              "0  polly:data://valo_onco_data_lake/data/PRIDE/Pr...                  7   NaN   \n",
              "1  polly:data://valo_onco_data_lake/data/PRIDE/Pr...                  3   NaN   \n",
              "\n",
              "  ingestion_approved                      description curated_cell_line  \\\n",
              "0           Approved  Quantitative K-GG site analysis         [HCT 116]   \n",
              "1           Approved  Quantitative K-GG site analysis         [HCT 116]   \n",
              "\n",
              "                               data_table_name  data_table_version  \\\n",
              "0  valo_onco__pxd002408_biological_replicate_1                   0   \n",
              "1  valo_onco__pxd002408_biological_replicate_2                   0   \n",
              "\n",
              "             platform        exposure_time     timestamp_ perturbation_type  \\\n",
              "0  LTQ Orbitrap Velos  [2 hours,  6 hours]  1656064257394        [Chemical]   \n",
              "1  LTQ Orbitrap Velos  [2 hours,  6 hours]  1656064386505        [Chemical]   \n",
              "\n",
              "  file_type                                        publication  \\\n",
              "0       gct  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4...   \n",
              "1       gct  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4...   \n",
              "\n",
              "               curated_cell_type  \\\n",
              "0  [colorectal cancer cell line]   \n",
              "1  [colorectal cancer cell line]   \n",
              "\n",
              "                                                 key  summary  \\\n",
              "0  valo_onco_data_lake/data/PRIDE/Proteomics/PXD0...      NaN   \n",
              "1  valo_onco_data_lake/data/PRIDE/Proteomics/PXD0...      NaN   \n",
              "\n",
              "  perturbation_modality   src_repo drug_smiles                   package  \\\n",
              "0                [Drug]  valo_onco          []  valo_onco_data_lake/data   \n",
              "1                [Drug]  valo_onco          []  valo_onco_data_lake/data   \n",
              "\n",
              "                                       file_location  author  \\\n",
              "0  https://discover-prod-datalake-v1.s3-us-west-2...     NaN   \n",
              "1  https://discover-prod-datalake-v1.s3-us-west-2...     NaN   \n",
              "\n",
              "                         dataset_id      curated_disease  \\\n",
              "0  PXD002408_Biological_Replicate_1  [Colonic Neoplasms]   \n",
              "1  PXD002408_Biological_Replicate_2  [Colonic Neoplasms]   \n",
              "\n",
              "                            curated_drug   curated_gene  abstract  version  \\\n",
              "0  [dimethyl sulfoxide,  growth hormone]  [USP1,  USP7]       NaN        0   \n",
              "1  [dimethyl sulfoxide,  growth hormone]  [USP1,  USP7]       NaN        0   \n",
              "\n",
              "  curated_strain        pubchem_id                     bucket curated_tissue  \\\n",
              "0         [None]  [679,  46931953]  discover-prod-datalake-v1   [colorectum]   \n",
              "1         [None]  [679,  46931953]  discover-prod-datalake-v1   [colorectum]   \n",
              "\n",
              "  dataset_source   data_type  overall_design  is_current     region  \n",
              "0          PRIDE  Proteomics             NaN        True  us-west-2  \n",
              "1          PRIDE  Proteomics             NaN        True  us-west-2  "
            ]
          },
          "execution_count": 26,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "list = ['PXD002408_Biological_Replicate_1', 'PXD002408_Biological_Replicate_2']\n",
        "query = f\"SELECT * from valo_onco.datasets WHERE dataset_id IN {tuple(list)}\"\n",
        "metadata = omixatlas.query_metadata(query)\n",
        "metadata"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "Before running the next cell, user should create two folders = 'data_final' and 'metadata_final' in their working directory"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### converting the dataset level metadata to .json file"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 28,
      "metadata": {},
      "outputs": [],
      "source": [
        "# Converting to json\n",
        "for i in metadata.index:\n",
        "    res.loc[i].to_json(f\"metadata_final/{metadata['dataset_id'][i]}.json\".format(i))\n",
        "filenames = os.listdir('metadata_final')"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### adding the _ _index_ _ OR any custom curated fields\n",
        "\n",
        "Once the dataset level metadata files are created then as per the template, _ _index_ _ needs to be appended to the .json file prior to ingestion.\n",
        "\n",
        "If there are any custom curation fields, then those can also be added in a similar way."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "for i in filenames:\n",
        "    entry = { \"__index__\": {\n",
        "        'file_metadata': 'true',  \n",
        "        'col_metadata': 'true',  \n",
        "        'row_metadata': 'false',  \n",
        "        'data_required': 'false',\n",
        "    }}\n",
        "    file = open('metadata_final/'+i)\n",
        "    json_met = json.load(file)\n",
        "    entry.update(json_met)\n",
        "    with open('metadata_final/'+i, \"w\") as file:\n",
        "        json.dump(entry, file)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### Validating if the keys of json file matches the template\n",
        "\n",
        "Prior to ingestion, the user should ensure the keys in dataset level metadata matches with that of the template generated above."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 78,
      "metadata": {},
      "outputs": [],
      "source": [
        "template_keys = set()\n",
        "for key in template:\n",
        "        template_keys.add(key)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 79,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/plain": [
              "{'__index__',\n",
              " 'curated_cell_line',\n",
              " 'curated_cell_type',\n",
              " 'curated_disease',\n",
              " 'curated_drug',\n",
              " 'curated_organism',\n",
              " 'curated_tissue',\n",
              " 'data_type',\n",
              " 'dataset_id',\n",
              " 'dataset_source',\n",
              " 'description',\n",
              " 'total_num_samples'}"
            ]
          },
          "execution_count": 79,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "template_keys"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 80,
      "metadata": {},
      "outputs": [],
      "source": [
        "import json\n",
        "keys_in_json = set()\n",
        "f = open('metadata_final/PXD002408_Biological_Replicate_1.json')\n",
        "data = json.load(f)\n",
        "\n",
        "for key in data:\n",
        "    keys_in_json.add(key)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 81,
      "metadata": {
        "scrolled": true
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "{'__index__',\n",
              " 'abstract',\n",
              " 'author',\n",
              " 'bucket',\n",
              " 'curated_cell_line',\n",
              " 'curated_cell_type',\n",
              " 'curated_disease',\n",
              " 'curated_drug',\n",
              " 'curated_gene',\n",
              " 'curated_organism',\n",
              " 'curated_strain',\n",
              " 'curated_tissue',\n",
              " 'data_table_name',\n",
              " 'data_table_version',\n",
              " 'data_type',\n",
              " 'dataset_id',\n",
              " 'dataset_source',\n",
              " 'description',\n",
              " 'dosage',\n",
              " 'drug_smiles',\n",
              " 'exposure_time',\n",
              " 'file_location',\n",
              " 'file_type',\n",
              " 'ingestion_approved',\n",
              " 'is_current',\n",
              " 'key',\n",
              " 'overall_design',\n",
              " 'package',\n",
              " 'perturbation_modality',\n",
              " 'perturbation_type',\n",
              " 'platform',\n",
              " 'pubchem_id',\n",
              " 'publication',\n",
              " 'region',\n",
              " 'src_repo',\n",
              " 'src_uri',\n",
              " 'summary',\n",
              " 'timestamp_',\n",
              " 'total_num_samples',\n",
              " 'version',\n",
              " 'year'}"
            ]
          },
          "execution_count": 81,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "keys_in_json"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 82,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/plain": [
              "{'__index__',\n",
              " 'curated_cell_line',\n",
              " 'curated_cell_type',\n",
              " 'curated_disease',\n",
              " 'curated_drug',\n",
              " 'curated_organism',\n",
              " 'curated_tissue',\n",
              " 'data_type',\n",
              " 'dataset_id',\n",
              " 'dataset_source',\n",
              " 'description',\n",
              " 'total_num_samples'}"
            ]
          },
          "execution_count": 82,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "intersect = keys_in_json.intersection(template_keys)\n",
        "intersect"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### validating indexes are same"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 83,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/plain": [
              "set()"
            ]
          },
          "execution_count": 83,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "template_keys.difference(intersect)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### Downloading corresponding gct files"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "As per the assumption, no custom curation is required hence the gct files don't need any modification. Hence, downloading the files from the source OA and uploading them as it is in the destination OA"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 29,
      "metadata": {
        "scrolled": true
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "PXD002408_Biological_Replicate_1\n"
          ]
        },
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "--2022-07-07 06:15:20--  https://discover-prod-datalake-v1.s3.amazonaws.com/valo_onco_data_lake/data/PRIDE/Proteomics/PXD002408_Biological_Replicate_1.gct?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=ASIAVRYB5UBIDT4GJDG3%2F20220707%2Fus-west-2%2Fs3%2Faws4_request&X-Amz-Date=20220707T061520Z&X-Amz-Expires=3600&X-Amz-SignedHeaders=host&X-Amz-Security-Token=IQoJb3JpZ2luX2VjEKb%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJHMEUCIGiIc25y6NdyTsFxBO6k35UH7tv5dN7QcGg2tcP7iRt0AiEAhvwTkZEpIMMqYbo6fCCGawCrPTNUfKAkiYsxQogguk4qqgIIz%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARADGgwzODE3MTkyNTcxNjgiDG%2B8EXHMxFxPQXdB4Sr%2BAQ7iJQ2aPKz%2F4V3bH76HODP0midBGdvkbkEnEeNVCUs6%2Bhm9Se%2BWttxQBwqImF%2F3FE1ubSTiWLz6u6nhLloZ%2F%2FC%2FeORRYJk4ObNFH0JH8A2NMxunnDaGvgn8FUbUiuVB%2BvRDTBvvFSq42qn7tX7k6l%2Bb5kM5pxq%2F%2F3hI%2FwzJsMdgpOB6TFm3VmEqMBYa3uErh4VbwDetXVL0mzB8dpagKBbtTZvGEZNs14lpn5q8nqAuyBClDdOFx3a8kRyTUbZVESHLK13g%2FQpiRTwFbMhGkrfTaLrnGZyiAsZeEW1UXH0m1BZaLTBhlsgqMgXHsqd%2FrTDWr3iYaBXwFERB17bKMNjzmZYGOpoBq5mm%2FnC1toRNsdByhYOzzHI1XNO96nQgItsQ%2FPsAuluzIeAM3Gvc9iHwAvrUWHT3LevLKq8FcQiH1LF1XcnLOk%2FulSmPEoXzs%2B57NWhVWcIKMd%2BmG0wxXCIyxAIiZMZuvK5EnNICdnCIr83vIQ5avBsJS7HSbN4acE4KCAkPL%2F1IFFTdrRY6P9jNzFSP5mF0WPnVI2kkrTjnKQ%3D%3D&X-Amz-Signature=e0b570a52f62184ea0f5502b47bd2ad85fe60b9dd19fb707af6f011f360bb9c5\n",
            "Resolving discover-prod-datalake-v1.s3.amazonaws.com (discover-prod-datalake-v1.s3.amazonaws.com)... 52.92.160.177\n",
            "Connecting to discover-prod-datalake-v1.s3.amazonaws.com (discover-prod-datalake-v1.s3.amazonaws.com)|52.92.160.177|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 141053 (138K) [binary/octet-stream]\n",
            "Saving to: 'data_final/PXD002408_Biological_Replicate_1.gct'\n",
            "\n",
            "     0K .......... .......... .......... .......... .......... 36% 23.8M 0s\n",
            "    50K .......... .......... .......... .......... .......... 72% 28.9M 0s\n",
            "   100K .......... .......... .......... .......              100% 33.5M=0.005s\n",
            "\n",
            "2022-07-07 06:15:21 (27.8 MB/s) - 'data_final/PXD002408_Biological_Replicate_1.gct' saved [141053/141053]\n",
            "\n"
          ]
        },
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Downloaded data successfully\n",
            "PXD002408_Biological_Replicate_2\n",
            "Downloaded data successfully\n"
          ]
        },
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "--2022-07-07 06:15:22--  https://discover-prod-datalake-v1.s3.amazonaws.com/valo_onco_data_lake/data/PRIDE/Proteomics/PXD002408_Biological_Replicate_2.gct?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=ASIAVRYB5UBIDT4GJDG3%2F20220707%2Fus-west-2%2Fs3%2Faws4_request&X-Amz-Date=20220707T061522Z&X-Amz-Expires=3600&X-Amz-SignedHeaders=host&X-Amz-Security-Token=IQoJb3JpZ2luX2VjEKb%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJHMEUCIGiIc25y6NdyTsFxBO6k35UH7tv5dN7QcGg2tcP7iRt0AiEAhvwTkZEpIMMqYbo6fCCGawCrPTNUfKAkiYsxQogguk4qqgIIz%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARADGgwzODE3MTkyNTcxNjgiDG%2B8EXHMxFxPQXdB4Sr%2BAQ7iJQ2aPKz%2F4V3bH76HODP0midBGdvkbkEnEeNVCUs6%2Bhm9Se%2BWttxQBwqImF%2F3FE1ubSTiWLz6u6nhLloZ%2F%2FC%2FeORRYJk4ObNFH0JH8A2NMxunnDaGvgn8FUbUiuVB%2BvRDTBvvFSq42qn7tX7k6l%2Bb5kM5pxq%2F%2F3hI%2FwzJsMdgpOB6TFm3VmEqMBYa3uErh4VbwDetXVL0mzB8dpagKBbtTZvGEZNs14lpn5q8nqAuyBClDdOFx3a8kRyTUbZVESHLK13g%2FQpiRTwFbMhGkrfTaLrnGZyiAsZeEW1UXH0m1BZaLTBhlsgqMgXHsqd%2FrTDWr3iYaBXwFERB17bKMNjzmZYGOpoBq5mm%2FnC1toRNsdByhYOzzHI1XNO96nQgItsQ%2FPsAuluzIeAM3Gvc9iHwAvrUWHT3LevLKq8FcQiH1LF1XcnLOk%2FulSmPEoXzs%2B57NWhVWcIKMd%2BmG0wxXCIyxAIiZMZuvK5EnNICdnCIr83vIQ5avBsJS7HSbN4acE4KCAkPL%2F1IFFTdrRY6P9jNzFSP5mF0WPnVI2kkrTjnKQ%3D%3D&X-Amz-Signature=4d8ab126a40c9275af1c28fd4ac4a67f17e2a21e114f82e08eaaf43069c36212\n",
            "Resolving discover-prod-datalake-v1.s3.amazonaws.com (discover-prod-datalake-v1.s3.amazonaws.com)... 52.218.250.227\n",
            "Connecting to discover-prod-datalake-v1.s3.amazonaws.com (discover-prod-datalake-v1.s3.amazonaws.com)|52.218.250.227|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 149121 (146K) [binary/octet-stream]\n",
            "Saving to: 'data_final/PXD002408_Biological_Replicate_2.gct'\n",
            "\n",
            "     0K .......... .......... .......... .......... .......... 34% 19.6M 0s\n",
            "    50K .......... .......... .......... .......... .......... 68% 41.9M 0s\n",
            "   100K .......... .......... .......... .......... .....     100% 40.9M=0.005s\n",
            "\n",
            "2022-07-07 06:15:22 (29.9 MB/s) - 'data_final/PXD002408_Biological_Replicate_2.gct' saved [149121/149121]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "#create a folder data_final in the working repository\n",
        "for i in metadata['dataset_id']:\n",
        "    print (str(i))\n",
        "    data = omixatlas.download_data(\"valo_onco\",str(i))\n",
        "    file_name = f\"{str(i)}.gct\"\n",
        "    url = data.get('data').get('attributes').get('download_url')\n",
        "    status = os.system(f\"wget -O 'data_final/{file_name}' '{url}'\")\n",
        "    if status == 0:\n",
        "        print(\"Downloaded data successfully\")\n",
        "    else:\n",
        "        raise Exception(\"Download not successful\")"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## Ingesting the data"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "In order to ingest the data, user can use the following function:-\n",
        "\n",
        "**add_datasets(repo_id (int/str), source_folder_path (dict), destination_folder (str) (optional), priority (str) (optiona))**\n",
        "\n",
        "**Input:** \n",
        "\n",
        "repo_id: This is the repository ID to which ingestion should be done\n",
        "\n",
        "source_folder_path: This is the dictionary with keys \"data\" and \"metadata\". The corresponding value pairs should be the folder containing the file (gct, h5ad, vcf, mmcif etc) for data and folder containing json of dataset level metadata for metadata.\n",
        "\n",
        "destination_folder (optional): This is the folder within S3 when data gets pushed\n",
        "\n",
        "priority (optional): This is the priority flag as per ingestion is being done. Default is 'medium'\n",
        "\n",
        "**Output:** \n",
        "\n",
        "Status of file upload for each dataset in a dataframe"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 84,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "{'data': '/import/data_final', 'metadata': '/import/metadata_final'}\n",
            "                              File Name        Message\n",
            "0                combined_metadata.json  File Uploaded\n",
            "1  PXD002408_Biological_Replicate_1.gct  File Uploaded\n",
            "2  PXD002408_Biological_Replicate_2.gct  File Uploaded\n"
          ]
        }
      ],
      "source": [
        "repo_id = \"1657110718820\"\n",
        "source_folder_path_data = \"/import/data_final\"\n",
        "source_folder_metadata = \"/import/metadata_final\"\n",
        "destination_folder = \"220707-1426\"\n",
        "priority = \"high\"\n",
        "source_folder_path = {\"data\":source_folder_path_data, \"metadata\":source_folder_metadata}\n",
        "print(source_folder_path)\n",
        "omixatlas.add_datasets(repo_id, source_folder_path, destination_folder, priority)"
      ]
    }
  ],
  "metadata": {
    "kernelspec": {
      "display_name": "Python 3 (ipykernel)",
      "language": "python",
      "name": "python3"
    },
    "language_info": {
      "codemirror_mode": {
        "name": "ipython",
        "version": 3
      },
      "file_extension": ".py",
      "mimetype": "text/x-python",
      "name": "python",
      "nbconvert_exporter": "python",
      "pygments_lexer": "ipython3",
      "version": "3.10.2"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 2
}