{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "colab_type": "text",
        "id": "view-in-github"
      },
      "source": [
        "<a href=\"https://polly.elucidata.io/manage/workspaces?action=open_polly_notebook&amp;source=github&amp;path=ElucidataInc%2Fpolly-python%2Fblob%2Fmain%2FDiscover%2Fmeta_analysis_transcriptomics%2FDifferentially+expressed+pathways+%28GO+and+HPO%29+in+Neuroblastoma.ipynb&amp;kernel=elucidata%2FPython+3.10&amp;machine=medium\" target=\"_parent\"><img alt=\"Open in Polly\" src=\"https://elucidatainc.github.io/PublicAssets/open_polly.svg\"/></a>\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "# Objective:-\n",
        "1. Identify datasets on GEO where Neuroblastoma samples are compared with Normal samples\n",
        "2. Determine which pathways are differentially expressed in the Disease conditions compared to Normal condition as per Gene Ontology and Human Phenotype Ontology"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "kernel": "SoS"
      },
      "source": [
        "## Install polly-python"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 2,
      "metadata": {
        "kernel": "Bash",
        "scrolled": true
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "/bin/bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)\n",
            "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
            "awscli 1.22.74 requires PyYAML<5.5,>=3.10, but you have pyyaml 6.0 which is incompatible.\u001b[0m\n",
            "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\n",
            "\u001b[33mWARNING: You are using pip version 21.2.4; however, version 22.2.2 is available.\n",
            "You should consider upgrading via the '/usr/local/bin/python3.10 -m pip install --upgrade pip' command.\u001b[0m\n"
          ]
        }
      ],
      "source": [
        "!sudo pip3 install polly-python --quiet"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "kernel": "Bash"
      },
      "source": [
        "## Import libraries and OA configuration"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 3,
      "metadata": {
        "kernel": "Python3"
      },
      "outputs": [],
      "source": [
        "import os\n",
        "from polly.auth import Polly\n",
        "from polly.omixatlas import OmixAtlas\n",
        "\n",
        "AUTH_TOKEN=(os.environ['POLLY_REFRESH_TOKEN']) # Obtain authentication tokens\n",
        "Polly.auth(AUTH_TOKEN)\n",
        "\n",
        "#Defining omixatlas object\n",
        "omixatlas = OmixAtlas()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "kernel": "Python3"
      },
      "source": [
        "### Params"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 4,
      "metadata": {
        "kernel": "Python3"
      },
      "outputs": [],
      "source": [
        "params = {\n",
        "    \"ATLAS_NAME\" : \"gsea_atlas\",\n",
        "    \"COMPARISION_NAME\" : \"Neuroblastoma\",  # Drug or Disease condition you are interested in\n",
        "    \"Pvalue_cutoff\" : 0.05,  # p-value cutoff for significance\n",
        "    }"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "kernel": "Python3"
      },
      "source": [
        "## Query GSEA atlas"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "kernel": "Python3"
      },
      "source": [
        "### 1. datasets table"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 5,
      "metadata": {
        "kernel": "Python3"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Query execution succeeded (time taken: 1.86 seconds, data scanned: 0.233 MB)\n",
            "Fetched 4995 rows\n",
            "(4995, 5)\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>src_dataset_id</th>\n",
              "      <th>src_description</th>\n",
              "      <th>condition_column</th>\n",
              "      <th>condition_control</th>\n",
              "      <th>condition_perturbation</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>GSE57275_GPL6887</td>\n",
              "      <td>Genome-wide analysis of gene expression in lun...</td>\n",
              "      <td>kw_curated_disease</td>\n",
              "      <td>[Normal]</td>\n",
              "      <td>[Tuberculosis]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>GSE57297_GPL17077</td>\n",
              "      <td>Identification of significant gene regulations...</td>\n",
              "      <td>kw_curated_disease</td>\n",
              "      <td>[Normal]</td>\n",
              "      <td>[Breast Neoplasms]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>GSE57314_GPL6246</td>\n",
              "      <td>Nfatc1 enhances stem cell contribution to squa...</td>\n",
              "      <td>kw_curated_disease</td>\n",
              "      <td>[Normal]</td>\n",
              "      <td>[Carcinoma,  Squamous Cell]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>GSE57380_GPL17400</td>\n",
              "      <td>Coexistent ARID1A-PIK3CA mutations promote ova...</td>\n",
              "      <td>kw_curated_disease</td>\n",
              "      <td>[Normal]</td>\n",
              "      <td>[Carcinoma,  Ovarian Epithelial]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>GSE57382_GPL6246</td>\n",
              "      <td>Calcitriol deregulates AR signaling to inhibit...</td>\n",
              "      <td>kw_curated_drug</td>\n",
              "      <td>[none]</td>\n",
              "      <td>[calcitriol]</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "      src_dataset_id                                    src_description  \\\n",
              "0   GSE57275_GPL6887  Genome-wide analysis of gene expression in lun...   \n",
              "1  GSE57297_GPL17077  Identification of significant gene regulations...   \n",
              "2   GSE57314_GPL6246  Nfatc1 enhances stem cell contribution to squa...   \n",
              "3  GSE57380_GPL17400  Coexistent ARID1A-PIK3CA mutations promote ova...   \n",
              "4   GSE57382_GPL6246  Calcitriol deregulates AR signaling to inhibit...   \n",
              "\n",
              "     condition_column condition_control            condition_perturbation  \n",
              "0  kw_curated_disease          [Normal]                    [Tuberculosis]  \n",
              "1  kw_curated_disease          [Normal]                [Breast Neoplasms]  \n",
              "2  kw_curated_disease          [Normal]       [Carcinoma,  Squamous Cell]  \n",
              "3  kw_curated_disease          [Normal]  [Carcinoma,  Ovarian Epithelial]  \n",
              "4     kw_curated_drug            [none]                      [calcitriol]  "
            ]
          },
          "execution_count": 5,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "q = \"\"\"SELECT src_dataset_id, \n",
        "              src_description,\n",
        "              condition_column,\n",
        "              condition_control,\n",
        "              condition_perturbation\n",
        "       FROM gsea_atlas.datasets\"\"\"\n",
        "df = omixatlas.query_metadata(q, query_api_version='v2')\n",
        "print(df.shape)\n",
        "df.head()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 18,
      "metadata": {
        "kernel": "Python3"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "4953"
            ]
          },
          "execution_count": 18,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "df['src_dataset_id'].nunique()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 19,
      "metadata": {
        "kernel": "Python3"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "kw_curated_disease    2686\n",
              "kw_curated_drug       2309\n",
              "Name: condition_column, dtype: int64"
            ]
          },
          "execution_count": 19,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "df['condition_column'].value_counts()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "kernel": "Python3"
      },
      "source": [
        "### 2. features table"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 6,
      "metadata": {
        "kernel": "Python3"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Query execution succeeded (time taken: 4.40 seconds, data scanned: 172.106 MB)\n",
            "Fetched 5 rows\n",
            "(5, 20)\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>src_repo</th>\n",
              "      <th>id_key</th>\n",
              "      <th>padj</th>\n",
              "      <th>src_uri</th>\n",
              "      <th>pval</th>\n",
              "      <th>pmid</th>\n",
              "      <th>src_dataset_id</th>\n",
              "      <th>version</th>\n",
              "      <th>es</th>\n",
              "      <th>timestamp_</th>\n",
              "      <th>size</th>\n",
              "      <th>systematicname</th>\n",
              "      <th>data_id</th>\n",
              "      <th>exactsource</th>\n",
              "      <th>name</th>\n",
              "      <th>pathway</th>\n",
              "      <th>nmoreextreme</th>\n",
              "      <th>id</th>\n",
              "      <th>is_current</th>\n",
              "      <th>nes</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>gsea_atlas</td>\n",
              "      <td>kw_row_id</td>\n",
              "      <td>0.985726</td>\n",
              "      <td>polly:data://gsea_atlas/data/GSE16176_GPL570-2...</td>\n",
              "      <td>0.948571</td>\n",
              "      <td>NaN</td>\n",
              "      <td>GSE16176_GPL570-2022-07-28-07-25-30</td>\n",
              "      <td>0</td>\n",
              "      <td>0.241467</td>\n",
              "      <td>1659016129013</td>\n",
              "      <td>8</td>\n",
              "      <td>M26585</td>\n",
              "      <td>_9001</td>\n",
              "      <td>GO:0034235</td>\n",
              "      <td>9001</td>\n",
              "      <td>GOMF_GPI_ANCHOR_BINDING</td>\n",
              "      <td>497</td>\n",
              "      <td>NaN</td>\n",
              "      <td>True</td>\n",
              "      <td>0.558746</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>gsea_atlas</td>\n",
              "      <td>kw_row_id</td>\n",
              "      <td>0.985796</td>\n",
              "      <td>polly:data://gsea_atlas/data/GSE16176_GPL570-2...</td>\n",
              "      <td>0.949275</td>\n",
              "      <td>NaN</td>\n",
              "      <td>GSE16176_GPL570-2022-07-28-07-25-30</td>\n",
              "      <td>0</td>\n",
              "      <td>0.274733</td>\n",
              "      <td>1659016129013</td>\n",
              "      <td>5</td>\n",
              "      <td>M34439</td>\n",
              "      <td>_9002</td>\n",
              "      <td>GO:0046978</td>\n",
              "      <td>9002</td>\n",
              "      <td>GOMF_TAP1_BINDING</td>\n",
              "      <td>523</td>\n",
              "      <td>NaN</td>\n",
              "      <td>True</td>\n",
              "      <td>0.558287</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>gsea_atlas</td>\n",
              "      <td>kw_row_id</td>\n",
              "      <td>0.991797</td>\n",
              "      <td>polly:data://gsea_atlas/data/GSE16176_GPL570-2...</td>\n",
              "      <td>0.965779</td>\n",
              "      <td>NaN</td>\n",
              "      <td>GSE16176_GPL570-2022-07-28-07-25-30</td>\n",
              "      <td>0</td>\n",
              "      <td>0.233708</td>\n",
              "      <td>1659016129013</td>\n",
              "      <td>9</td>\n",
              "      <td>M24446</td>\n",
              "      <td>_9003</td>\n",
              "      <td>GO:0071225</td>\n",
              "      <td>9003</td>\n",
              "      <td>GOBP_CELLULAR_RESPONSE_TO_MURAMYL_DIPEPTIDE</td>\n",
              "      <td>507</td>\n",
              "      <td>NaN</td>\n",
              "      <td>True</td>\n",
              "      <td>0.556963</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>gsea_atlas</td>\n",
              "      <td>kw_row_id</td>\n",
              "      <td>0.996980</td>\n",
              "      <td>polly:data://gsea_atlas/data/GSE16176_GPL570-2...</td>\n",
              "      <td>0.979206</td>\n",
              "      <td>NaN</td>\n",
              "      <td>GSE16176_GPL570-2022-07-28-07-25-30</td>\n",
              "      <td>0</td>\n",
              "      <td>0.195206</td>\n",
              "      <td>1659016129013</td>\n",
              "      <td>16</td>\n",
              "      <td>M18314</td>\n",
              "      <td>_9004</td>\n",
              "      <td>GO:0005092</td>\n",
              "      <td>9004</td>\n",
              "      <td>GOMF_GDP_DISSOCIATION_INHIBITOR_ACTIVITY</td>\n",
              "      <td>517</td>\n",
              "      <td>NaN</td>\n",
              "      <td>True</td>\n",
              "      <td>0.556912</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>gsea_atlas</td>\n",
              "      <td>kw_row_id</td>\n",
              "      <td>0.993506</td>\n",
              "      <td>polly:data://gsea_atlas/data/GSE16176_GPL570-2...</td>\n",
              "      <td>0.970205</td>\n",
              "      <td>NaN</td>\n",
              "      <td>GSE16176_GPL570-2022-07-28-07-25-30</td>\n",
              "      <td>0</td>\n",
              "      <td>0.200938</td>\n",
              "      <td>1659016129013</td>\n",
              "      <td>14</td>\n",
              "      <td>M35479</td>\n",
              "      <td>_9005</td>\n",
              "      <td>HP:0001841</td>\n",
              "      <td>9005</td>\n",
              "      <td>HP_PREAXIAL_FOOT_POLYDACTYLY</td>\n",
              "      <td>520</td>\n",
              "      <td>NaN</td>\n",
              "      <td>True</td>\n",
              "      <td>0.556112</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "     src_repo     id_key      padj  \\\n",
              "0  gsea_atlas  kw_row_id  0.985726   \n",
              "1  gsea_atlas  kw_row_id  0.985796   \n",
              "2  gsea_atlas  kw_row_id  0.991797   \n",
              "3  gsea_atlas  kw_row_id  0.996980   \n",
              "4  gsea_atlas  kw_row_id  0.993506   \n",
              "\n",
              "                                             src_uri      pval  pmid  \\\n",
              "0  polly:data://gsea_atlas/data/GSE16176_GPL570-2...  0.948571   NaN   \n",
              "1  polly:data://gsea_atlas/data/GSE16176_GPL570-2...  0.949275   NaN   \n",
              "2  polly:data://gsea_atlas/data/GSE16176_GPL570-2...  0.965779   NaN   \n",
              "3  polly:data://gsea_atlas/data/GSE16176_GPL570-2...  0.979206   NaN   \n",
              "4  polly:data://gsea_atlas/data/GSE16176_GPL570-2...  0.970205   NaN   \n",
              "\n",
              "                        src_dataset_id  version        es     timestamp_  \\\n",
              "0  GSE16176_GPL570-2022-07-28-07-25-30        0  0.241467  1659016129013   \n",
              "1  GSE16176_GPL570-2022-07-28-07-25-30        0  0.274733  1659016129013   \n",
              "2  GSE16176_GPL570-2022-07-28-07-25-30        0  0.233708  1659016129013   \n",
              "3  GSE16176_GPL570-2022-07-28-07-25-30        0  0.195206  1659016129013   \n",
              "4  GSE16176_GPL570-2022-07-28-07-25-30        0  0.200938  1659016129013   \n",
              "\n",
              "   size systematicname data_id exactsource  name  \\\n",
              "0     8         M26585   _9001  GO:0034235  9001   \n",
              "1     5         M34439   _9002  GO:0046978  9002   \n",
              "2     9         M24446   _9003  GO:0071225  9003   \n",
              "3    16         M18314   _9004  GO:0005092  9004   \n",
              "4    14         M35479   _9005  HP:0001841  9005   \n",
              "\n",
              "                                       pathway  nmoreextreme  id  is_current  \\\n",
              "0                      GOMF_GPI_ANCHOR_BINDING           497 NaN        True   \n",
              "1                            GOMF_TAP1_BINDING           523 NaN        True   \n",
              "2  GOBP_CELLULAR_RESPONSE_TO_MURAMYL_DIPEPTIDE           507 NaN        True   \n",
              "3     GOMF_GDP_DISSOCIATION_INHIBITOR_ACTIVITY           517 NaN        True   \n",
              "4                 HP_PREAXIAL_FOOT_POLYDACTYLY           520 NaN        True   \n",
              "\n",
              "        nes  \n",
              "0  0.558746  \n",
              "1  0.558287  \n",
              "2  0.556963  \n",
              "3  0.556912  \n",
              "4  0.556112  "
            ]
          },
          "execution_count": 6,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "q = \"\"\"SELECT *\n",
        "       FROM gsea_atlas.features LIMIT 5\"\"\"\n",
        "df = omixatlas.query_metadata(q, query_api_version='v2')\n",
        "print(df.shape)\n",
        "df.head()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "kernel": "Python3"
      },
      "source": [
        "## Get comparisons of interest"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 7,
      "metadata": {
        "kernel": "Python3"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Query execution succeeded (time taken: 1.93 seconds, data scanned: 0.238 MB)\n",
            "Fetched 13 rows\n",
            "(13, 6)\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>dataset_id</th>\n",
              "      <th>src_dataset_id</th>\n",
              "      <th>src_description</th>\n",
              "      <th>condition_column</th>\n",
              "      <th>condition_control</th>\n",
              "      <th>condition_perturbation</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>GSE87782_GPL10787-2022-07-28-07-43-39</td>\n",
              "      <td>GSE87782_GPL10787</td>\n",
              "      <td>MYCN-transformed neuroblasts from TH-MYCN mice...</td>\n",
              "      <td>kw_curated_disease</td>\n",
              "      <td>[Normal]</td>\n",
              "      <td>[Neuroblastoma]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>GSE87783_GPL10787-2022-07-28-07-43-40</td>\n",
              "      <td>GSE87783_GPL10787</td>\n",
              "      <td>Transcriptomic changes in MYC and PRC2 targets...</td>\n",
              "      <td>kw_curated_disease</td>\n",
              "      <td>[Normal]</td>\n",
              "      <td>[Neuroblastoma]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>GSE90711_GPL11154-2022-07-28-07-44-11</td>\n",
              "      <td>GSE90711_GPL11154</td>\n",
              "      <td>Proteomics and transcriptomics of peripheral n...</td>\n",
              "      <td>kw_curated_disease</td>\n",
              "      <td>[Normal]</td>\n",
              "      <td>[Neuroblastoma]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>GSE50959_GPL570-2022-07-28-07-59-34</td>\n",
              "      <td>GSE50959_GPL570</td>\n",
              "      <td>Differential gene expression in neuroblastoma ...</td>\n",
              "      <td>kw_curated_disease</td>\n",
              "      <td>[Normal]</td>\n",
              "      <td>[Neuroblastoma]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>GSE54721_GPL13534-2022-07-28-07-36-21</td>\n",
              "      <td>GSE54721_GPL13534</td>\n",
              "      <td>DNA methylation changes at CpG and non-CpG sit...</td>\n",
              "      <td>kw_curated_disease</td>\n",
              "      <td>[Normal]</td>\n",
              "      <td>[Neuroblastoma]</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "                              dataset_id     src_dataset_id  \\\n",
              "0  GSE87782_GPL10787-2022-07-28-07-43-39  GSE87782_GPL10787   \n",
              "1  GSE87783_GPL10787-2022-07-28-07-43-40  GSE87783_GPL10787   \n",
              "2  GSE90711_GPL11154-2022-07-28-07-44-11  GSE90711_GPL11154   \n",
              "3    GSE50959_GPL570-2022-07-28-07-59-34    GSE50959_GPL570   \n",
              "4  GSE54721_GPL13534-2022-07-28-07-36-21  GSE54721_GPL13534   \n",
              "\n",
              "                                     src_description    condition_column  \\\n",
              "0  MYCN-transformed neuroblasts from TH-MYCN mice...  kw_curated_disease   \n",
              "1  Transcriptomic changes in MYC and PRC2 targets...  kw_curated_disease   \n",
              "2  Proteomics and transcriptomics of peripheral n...  kw_curated_disease   \n",
              "3  Differential gene expression in neuroblastoma ...  kw_curated_disease   \n",
              "4  DNA methylation changes at CpG and non-CpG sit...  kw_curated_disease   \n",
              "\n",
              "  condition_control condition_perturbation  \n",
              "0          [Normal]        [Neuroblastoma]  \n",
              "1          [Normal]        [Neuroblastoma]  \n",
              "2          [Normal]        [Neuroblastoma]  \n",
              "3          [Normal]        [Neuroblastoma]  \n",
              "4          [Normal]        [Neuroblastoma]  "
            ]
          },
          "execution_count": 7,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "q = \"\"\"SELECT \n",
        "              dataset_id,\n",
        "              src_dataset_id, \n",
        "              src_description,\n",
        "              condition_column,\n",
        "              condition_control,\n",
        "              condition_perturbation\n",
        "       FROM \n",
        "              \"\"\" + params[\"ATLAS_NAME\"] + \"\"\".datasets\n",
        "       WHERE\n",
        "              CONTAINS(condition_perturbation, '\"\"\" + params[\"COMPARISION_NAME\"] + \"\"\"')\n",
        "       \"\"\"\n",
        "df = omixatlas.query_metadata(q, query_api_version='v2')\n",
        "print(df.shape)\n",
        "df.head()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "kernel": "Python3"
      },
      "source": [
        "## Query feature level info\n",
        "\n",
        "After having identified the differential comparisons of interest, we would like to understand what pathways are significantly enriched in a majority of those datasets. To do this, we count the number of enriched pathways found in, and along with that the average Enrichment score for that pathway."
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "kernel": "Python3"
      },
      "source": [
        "### 1. Enriched Pathways from Gene Ontology Biological Process (GOBP)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 8,
      "metadata": {
        "kernel": "Python3"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Query execution succeeded (time taken: 4.95 seconds, data scanned: 522.356 MB)\n",
            "Fetched 4459 rows\n",
            "(4459, 3)\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>pathway</th>\n",
              "      <th>dataset_count</th>\n",
              "      <th>NES_mean</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>GOBP_DEFENSE_RESPONSE_TO_OTHER_ORGANISM</td>\n",
              "      <td>7</td>\n",
              "      <td>-1.407988</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>GOBP_IMMUNE_RESPONSE</td>\n",
              "      <td>7</td>\n",
              "      <td>-1.030059</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>GOBP_PROTEIN_PROCESSING</td>\n",
              "      <td>7</td>\n",
              "      <td>-1.232059</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>GOBP_RESPONSE_TO_WOUNDING</td>\n",
              "      <td>7</td>\n",
              "      <td>-1.535205</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>GOBP_PROTEIN_MATURATION</td>\n",
              "      <td>7</td>\n",
              "      <td>-1.120485</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>GOBP_WOUND_HEALING</td>\n",
              "      <td>7</td>\n",
              "      <td>-1.623887</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>GOBP_REGULATION_OF_IMMUNE_SYSTEM_PROCESS</td>\n",
              "      <td>7</td>\n",
              "      <td>-0.986747</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>GOBP_PROTEOLYSIS</td>\n",
              "      <td>7</td>\n",
              "      <td>-1.084543</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8</th>\n",
              "      <td>GOBP_POSITIVE_REGULATION_OF_IMMUNE_SYSTEM_PROCESS</td>\n",
              "      <td>7</td>\n",
              "      <td>-0.998948</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>GOBP_PROTEIN_DNA_COMPLEX_SUBUNIT_ORGANIZATION</td>\n",
              "      <td>6</td>\n",
              "      <td>1.348849</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10</th>\n",
              "      <td>GOBP_RESPONSE_TO_STEROID_HORMONE</td>\n",
              "      <td>6</td>\n",
              "      <td>-1.176192</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11</th>\n",
              "      <td>GOBP_REGULATION_OF_CHROMOSOME_ORGANIZATION</td>\n",
              "      <td>6</td>\n",
              "      <td>1.401476</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>12</th>\n",
              "      <td>GOBP_LIPID_BIOSYNTHETIC_PROCESS</td>\n",
              "      <td>6</td>\n",
              "      <td>-0.973677</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13</th>\n",
              "      <td>GOBP_NEGATIVE_REGULATION_OF_CELL_POPULATION_PR...</td>\n",
              "      <td>6</td>\n",
              "      <td>-1.232155</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>14</th>\n",
              "      <td>GOBP_POSITIVE_REGULATION_OF_CELL_DIFFERENTIATION</td>\n",
              "      <td>6</td>\n",
              "      <td>-1.102075</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>15</th>\n",
              "      <td>GOBP_REGULATION_OF_INTRACELLULAR_SIGNAL_TRANSD...</td>\n",
              "      <td>6</td>\n",
              "      <td>-1.071215</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>16</th>\n",
              "      <td>GOBP_ANATOMICAL_STRUCTURE_FORMATION_INVOLVED_I...</td>\n",
              "      <td>6</td>\n",
              "      <td>-1.317864</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>17</th>\n",
              "      <td>GOBP_MUSCLE_CELL_DIFFERENTIATION</td>\n",
              "      <td>6</td>\n",
              "      <td>-1.119516</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>18</th>\n",
              "      <td>GOBP_CELLULAR_RESPONSE_TO_EXTERNAL_STIMULUS</td>\n",
              "      <td>6</td>\n",
              "      <td>-1.168684</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>19</th>\n",
              "      <td>GOBP_ENDOCYTOSIS</td>\n",
              "      <td>6</td>\n",
              "      <td>-0.558051</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "                                              pathway  dataset_count  NES_mean\n",
              "0             GOBP_DEFENSE_RESPONSE_TO_OTHER_ORGANISM              7 -1.407988\n",
              "1                                GOBP_IMMUNE_RESPONSE              7 -1.030059\n",
              "2                             GOBP_PROTEIN_PROCESSING              7 -1.232059\n",
              "3                           GOBP_RESPONSE_TO_WOUNDING              7 -1.535205\n",
              "4                             GOBP_PROTEIN_MATURATION              7 -1.120485\n",
              "5                                  GOBP_WOUND_HEALING              7 -1.623887\n",
              "6            GOBP_REGULATION_OF_IMMUNE_SYSTEM_PROCESS              7 -0.986747\n",
              "7                                    GOBP_PROTEOLYSIS              7 -1.084543\n",
              "8   GOBP_POSITIVE_REGULATION_OF_IMMUNE_SYSTEM_PROCESS              7 -0.998948\n",
              "9       GOBP_PROTEIN_DNA_COMPLEX_SUBUNIT_ORGANIZATION              6  1.348849\n",
              "10                   GOBP_RESPONSE_TO_STEROID_HORMONE              6 -1.176192\n",
              "11         GOBP_REGULATION_OF_CHROMOSOME_ORGANIZATION              6  1.401476\n",
              "12                    GOBP_LIPID_BIOSYNTHETIC_PROCESS              6 -0.973677\n",
              "13  GOBP_NEGATIVE_REGULATION_OF_CELL_POPULATION_PR...              6 -1.232155\n",
              "14   GOBP_POSITIVE_REGULATION_OF_CELL_DIFFERENTIATION              6 -1.102075\n",
              "15  GOBP_REGULATION_OF_INTRACELLULAR_SIGNAL_TRANSD...              6 -1.071215\n",
              "16  GOBP_ANATOMICAL_STRUCTURE_FORMATION_INVOLVED_I...              6 -1.317864\n",
              "17                   GOBP_MUSCLE_CELL_DIFFERENTIATION              6 -1.119516\n",
              "18        GOBP_CELLULAR_RESPONSE_TO_EXTERNAL_STIMULUS              6 -1.168684\n",
              "19                                   GOBP_ENDOCYTOSIS              6 -0.558051"
            ]
          },
          "execution_count": 8,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "pathway_db_name = \"GOBP\"\n",
        "q = \"\"\"SELECT pathway, COUNT(src_dataset_id) AS dataset_count, \n",
        "              SUM(NES)/COUNT(src_dataset_id) AS NES_mean \n",
        "       FROM \"\"\" + params[\"ATLAS_NAME\"] + \"\"\".features \n",
        "       WHERE src_dataset_id IN \n",
        "           (SELECT dataset_id FROM \"\"\" + params[\"ATLAS_NAME\"] + \"\"\".datasets WHERE CONTAINS(condition_perturbation, \n",
        "           '\"\"\" + params[\"COMPARISION_NAME\"] + \"\"\"')) \n",
        "       AND padj < \"\"\" + str(params[\"Pvalue_cutoff\"]) + \"\"\" AND pathway LIKE '\"\"\" + pathway_db_name + \"\"\"%'\n",
        "       GROUP BY pathway\n",
        "       ORDER BY dataset_count DESC\"\"\"\n",
        "\n",
        "df = omixatlas.query_metadata(q, query_api_version='v2')\n",
        "print(df.shape)\n",
        "df.head(20)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 9,
      "metadata": {
        "kernel": "Python3"
      },
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>pathway</th>\n",
              "      <th>dataset_count</th>\n",
              "      <th>NES_mean</th>\n",
              "      <th>metric</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>503</th>\n",
              "      <td>GOBP_MEIOSIS_I_CELL_CYCLE_PROCESS</td>\n",
              "      <td>5</td>\n",
              "      <td>2.094224</td>\n",
              "      <td>10.471122</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>275</th>\n",
              "      <td>GOBP_CENTROMERE_COMPLEX_ASSEMBLY</td>\n",
              "      <td>5</td>\n",
              "      <td>2.078301</td>\n",
              "      <td>10.391504</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>404</th>\n",
              "      <td>GOBP_POSITIVE_REGULATION_OF_CELL_CYCLE_G2_M_PH...</td>\n",
              "      <td>5</td>\n",
              "      <td>2.070064</td>\n",
              "      <td>10.350321</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>529</th>\n",
              "      <td>GOBP_MEIOTIC_CHROMOSOME_SEGREGATION</td>\n",
              "      <td>5</td>\n",
              "      <td>2.059274</td>\n",
              "      <td>10.296372</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>285</th>\n",
              "      <td>GOBP_HOMOLOGOUS_RECOMBINATION</td>\n",
              "      <td>5</td>\n",
              "      <td>1.956347</td>\n",
              "      <td>9.781735</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>394</th>\n",
              "      <td>GOBP_DENDRITE_EXTENSION</td>\n",
              "      <td>5</td>\n",
              "      <td>1.937409</td>\n",
              "      <td>9.687047</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>551</th>\n",
              "      <td>GOBP_NEUROTRANSMITTER_SECRETION</td>\n",
              "      <td>5</td>\n",
              "      <td>1.929397</td>\n",
              "      <td>9.646983</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>591</th>\n",
              "      <td>GOBP_FEMALE_MEIOTIC_NUCLEAR_DIVISION</td>\n",
              "      <td>5</td>\n",
              "      <td>1.916576</td>\n",
              "      <td>9.582880</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>311</th>\n",
              "      <td>GOBP_CHROMOSOME_ORGANIZATION_INVOLVED_IN_MEIOT...</td>\n",
              "      <td>5</td>\n",
              "      <td>1.888549</td>\n",
              "      <td>9.442746</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>372</th>\n",
              "      <td>GOBP_REGULATION_OF_DENDRITE_EXTENSION</td>\n",
              "      <td>5</td>\n",
              "      <td>1.881249</td>\n",
              "      <td>9.406247</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>595</th>\n",
              "      <td>GOBP_NEUROTRANSMITTER_TRANSPORT</td>\n",
              "      <td>5</td>\n",
              "      <td>1.878748</td>\n",
              "      <td>9.393740</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>144</th>\n",
              "      <td>GOBP_SISTER_CHROMATID_SEGREGATION</td>\n",
              "      <td>6</td>\n",
              "      <td>1.534345</td>\n",
              "      <td>9.206073</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>102</th>\n",
              "      <td>GOBP_MITOTIC_SISTER_CHROMATID_SEGREGATION</td>\n",
              "      <td>6</td>\n",
              "      <td>1.527134</td>\n",
              "      <td>9.162806</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>143</th>\n",
              "      <td>GOBP_DNA_DEPENDENT_DNA_REPLICATION</td>\n",
              "      <td>6</td>\n",
              "      <td>1.466809</td>\n",
              "      <td>8.800857</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>188</th>\n",
              "      <td>GOBP_CELL_CYCLE_G2_M_PHASE_TRANSITION</td>\n",
              "      <td>6</td>\n",
              "      <td>1.439124</td>\n",
              "      <td>8.634746</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>174</th>\n",
              "      <td>GOBP_METAPHASE_ANAPHASE_TRANSITION_OF_CELL_CYCLE</td>\n",
              "      <td>6</td>\n",
              "      <td>1.438325</td>\n",
              "      <td>8.629953</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>176</th>\n",
              "      <td>GOBP_MEIOTIC_CELL_CYCLE_PROCESS</td>\n",
              "      <td>6</td>\n",
              "      <td>1.423460</td>\n",
              "      <td>8.540758</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>79</th>\n",
              "      <td>GOBP_REGULATION_OF_CELL_CYCLE_G2_M_PHASE_TRANS...</td>\n",
              "      <td>6</td>\n",
              "      <td>1.423218</td>\n",
              "      <td>8.539310</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>266</th>\n",
              "      <td>GOBP_CHROMOSOME_SEPARATION</td>\n",
              "      <td>6</td>\n",
              "      <td>1.415429</td>\n",
              "      <td>8.492574</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>200</th>\n",
              "      <td>GOBP_NEGATIVE_REGULATION_OF_CHROMOSOME_ORGANIZ...</td>\n",
              "      <td>6</td>\n",
              "      <td>1.408457</td>\n",
              "      <td>8.450744</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "                                               pathway  dataset_count  \\\n",
              "503                  GOBP_MEIOSIS_I_CELL_CYCLE_PROCESS              5   \n",
              "275                   GOBP_CENTROMERE_COMPLEX_ASSEMBLY              5   \n",
              "404  GOBP_POSITIVE_REGULATION_OF_CELL_CYCLE_G2_M_PH...              5   \n",
              "529                GOBP_MEIOTIC_CHROMOSOME_SEGREGATION              5   \n",
              "285                      GOBP_HOMOLOGOUS_RECOMBINATION              5   \n",
              "394                            GOBP_DENDRITE_EXTENSION              5   \n",
              "551                    GOBP_NEUROTRANSMITTER_SECRETION              5   \n",
              "591               GOBP_FEMALE_MEIOTIC_NUCLEAR_DIVISION              5   \n",
              "311  GOBP_CHROMOSOME_ORGANIZATION_INVOLVED_IN_MEIOT...              5   \n",
              "372              GOBP_REGULATION_OF_DENDRITE_EXTENSION              5   \n",
              "595                    GOBP_NEUROTRANSMITTER_TRANSPORT              5   \n",
              "144                  GOBP_SISTER_CHROMATID_SEGREGATION              6   \n",
              "102          GOBP_MITOTIC_SISTER_CHROMATID_SEGREGATION              6   \n",
              "143                 GOBP_DNA_DEPENDENT_DNA_REPLICATION              6   \n",
              "188              GOBP_CELL_CYCLE_G2_M_PHASE_TRANSITION              6   \n",
              "174   GOBP_METAPHASE_ANAPHASE_TRANSITION_OF_CELL_CYCLE              6   \n",
              "176                    GOBP_MEIOTIC_CELL_CYCLE_PROCESS              6   \n",
              "79   GOBP_REGULATION_OF_CELL_CYCLE_G2_M_PHASE_TRANS...              6   \n",
              "266                         GOBP_CHROMOSOME_SEPARATION              6   \n",
              "200  GOBP_NEGATIVE_REGULATION_OF_CHROMOSOME_ORGANIZ...              6   \n",
              "\n",
              "     NES_mean     metric  \n",
              "503  2.094224  10.471122  \n",
              "275  2.078301  10.391504  \n",
              "404  2.070064  10.350321  \n",
              "529  2.059274  10.296372  \n",
              "285  1.956347   9.781735  \n",
              "394  1.937409   9.687047  \n",
              "551  1.929397   9.646983  \n",
              "591  1.916576   9.582880  \n",
              "311  1.888549   9.442746  \n",
              "372  1.881249   9.406247  \n",
              "595  1.878748   9.393740  \n",
              "144  1.534345   9.206073  \n",
              "102  1.527134   9.162806  \n",
              "143  1.466809   8.800857  \n",
              "188  1.439124   8.634746  \n",
              "174  1.438325   8.629953  \n",
              "176  1.423460   8.540758  \n",
              "79   1.423218   8.539310  \n",
              "266  1.415429   8.492574  \n",
              "200  1.408457   8.450744  "
            ]
          },
          "execution_count": 9,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "df['metric'] = df['dataset_count']*df['NES_mean']\n",
        "df = df.sort_values(by='metric', ascending=False)\n",
        "df.head(20)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "kernel": "Python3"
      },
      "source": [
        "### 2. Enriched Pathways from Gene Ontology Cellular Component (GOCC)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 10,
      "metadata": {
        "kernel": "Python3"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Query execution succeeded (time taken: 4.73 seconds, data scanned: 522.356 MB)\n",
            "Fetched 702 rows\n",
            "(702, 3)\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>pathway</th>\n",
              "      <th>dataset_count</th>\n",
              "      <th>NES_mean</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>GOCC_ENDOPLASMIC_RETICULUM_LUMEN</td>\n",
              "      <td>7</td>\n",
              "      <td>-1.569582</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>GOCC_ENDOPLASMIC_RETICULUM</td>\n",
              "      <td>7</td>\n",
              "      <td>-1.229097</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>GOCC_EARLY_ENDOSOME</td>\n",
              "      <td>6</td>\n",
              "      <td>-1.017206</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>GOCC_SPLICEOSOMAL_COMPLEX</td>\n",
              "      <td>6</td>\n",
              "      <td>1.297896</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>GOCC_SECRETORY_GRANULE</td>\n",
              "      <td>6</td>\n",
              "      <td>-1.307329</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "                            pathway  dataset_count  NES_mean\n",
              "0  GOCC_ENDOPLASMIC_RETICULUM_LUMEN              7 -1.569582\n",
              "1        GOCC_ENDOPLASMIC_RETICULUM              7 -1.229097\n",
              "2               GOCC_EARLY_ENDOSOME              6 -1.017206\n",
              "3         GOCC_SPLICEOSOMAL_COMPLEX              6  1.297896\n",
              "4            GOCC_SECRETORY_GRANULE              6 -1.307329"
            ]
          },
          "execution_count": 10,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "pathway_db_name = \"GOCC\"\n",
        "q = \"\"\"SELECT pathway, COUNT(src_dataset_id) AS dataset_count, \n",
        "              SUM(NES)/COUNT(src_dataset_id) AS NES_mean \n",
        "       FROM \"\"\" + params[\"ATLAS_NAME\"] + \"\"\".features \n",
        "       WHERE src_dataset_id IN \n",
        "           (SELECT dataset_id FROM \"\"\" + params[\"ATLAS_NAME\"] + \"\"\".datasets WHERE CONTAINS(condition_perturbation, \n",
        "           '\"\"\" + params[\"COMPARISION_NAME\"] + \"\"\"')) \n",
        "       AND padj < \"\"\" + str(params[\"Pvalue_cutoff\"]) + \"\"\" AND pathway LIKE '\"\"\" + pathway_db_name + \"\"\"%'\n",
        "       GROUP BY pathway\n",
        "       ORDER BY dataset_count DESC\"\"\"\n",
        "\n",
        "df = omixatlas.query_metadata(q, query_api_version='v2')\n",
        "print(df.shape)\n",
        "df.head()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 11,
      "metadata": {
        "kernel": "Python3"
      },
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>pathway</th>\n",
              "      <th>dataset_count</th>\n",
              "      <th>NES_mean</th>\n",
              "      <th>metric</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>77</th>\n",
              "      <td>GOCC_PRESYNAPTIC_MEMBRANE</td>\n",
              "      <td>5</td>\n",
              "      <td>1.952774</td>\n",
              "      <td>9.763871</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>94</th>\n",
              "      <td>GOCC_SYNAPTIC_MEMBRANE</td>\n",
              "      <td>5</td>\n",
              "      <td>1.866770</td>\n",
              "      <td>9.333848</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>62</th>\n",
              "      <td>GOCC_DISTAL_AXON</td>\n",
              "      <td>5</td>\n",
              "      <td>1.865197</td>\n",
              "      <td>9.325986</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>92</th>\n",
              "      <td>GOCC_PRESYNAPSE</td>\n",
              "      <td>5</td>\n",
              "      <td>1.860195</td>\n",
              "      <td>9.300976</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>31</th>\n",
              "      <td>GOCC_CHROMOSOME_CENTROMERIC_REGION</td>\n",
              "      <td>6</td>\n",
              "      <td>1.504451</td>\n",
              "      <td>9.026706</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>24</th>\n",
              "      <td>GOCC_CONDENSED_CHROMOSOME_CENTROMERIC_REGION</td>\n",
              "      <td>6</td>\n",
              "      <td>1.434843</td>\n",
              "      <td>8.609056</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>GOCC_NUCLEAR_CHROMOSOME</td>\n",
              "      <td>6</td>\n",
              "      <td>1.406465</td>\n",
              "      <td>8.438788</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>147</th>\n",
              "      <td>GOCC_KINESIN_COMPLEX</td>\n",
              "      <td>4</td>\n",
              "      <td>2.090061</td>\n",
              "      <td>8.360244</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>110</th>\n",
              "      <td>GOCC_NEURON_TO_NEURON_SYNAPSE</td>\n",
              "      <td>4</td>\n",
              "      <td>2.032852</td>\n",
              "      <td>8.131407</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>149</th>\n",
              "      <td>GOCC_SPLICEOSOMAL_SNRNP_COMPLEX</td>\n",
              "      <td>4</td>\n",
              "      <td>2.014187</td>\n",
              "      <td>8.056746</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>178</th>\n",
              "      <td>GOCC_INTRINSIC_COMPONENT_OF_SYNAPTIC_MEMBRANE</td>\n",
              "      <td>4</td>\n",
              "      <td>2.000797</td>\n",
              "      <td>8.003186</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>169</th>\n",
              "      <td>GOCC_INTRINSIC_COMPONENT_OF_POSTSYNAPTIC_DENSI...</td>\n",
              "      <td>4</td>\n",
              "      <td>1.997493</td>\n",
              "      <td>7.989972</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>37</th>\n",
              "      <td>GOCC_MITOTIC_SPINDLE</td>\n",
              "      <td>6</td>\n",
              "      <td>1.306004</td>\n",
              "      <td>7.836022</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>160</th>\n",
              "      <td>GOCC_TERMINAL_BOUTON</td>\n",
              "      <td>4</td>\n",
              "      <td>1.950498</td>\n",
              "      <td>7.801993</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>138</th>\n",
              "      <td>GOCC_INTRINSIC_COMPONENT_OF_POSTSYNAPTIC_SPECI...</td>\n",
              "      <td>4</td>\n",
              "      <td>1.949020</td>\n",
              "      <td>7.796080</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>GOCC_SPLICEOSOMAL_COMPLEX</td>\n",
              "      <td>6</td>\n",
              "      <td>1.297896</td>\n",
              "      <td>7.787375</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>117</th>\n",
              "      <td>GOCC_POSTSYNAPTIC_DENSITY_MEMBRANE</td>\n",
              "      <td>4</td>\n",
              "      <td>1.946416</td>\n",
              "      <td>7.785665</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>168</th>\n",
              "      <td>GOCC_AXON</td>\n",
              "      <td>4</td>\n",
              "      <td>1.942122</td>\n",
              "      <td>7.768489</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>139</th>\n",
              "      <td>GOCC_INTRINSIC_COMPONENT_OF_POSTSYNAPTIC_MEMBRANE</td>\n",
              "      <td>4</td>\n",
              "      <td>1.935876</td>\n",
              "      <td>7.743503</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>143</th>\n",
              "      <td>GOCC_POSTSYNAPSE</td>\n",
              "      <td>4</td>\n",
              "      <td>1.927253</td>\n",
              "      <td>7.709012</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "                                               pathway  dataset_count  \\\n",
              "77                           GOCC_PRESYNAPTIC_MEMBRANE              5   \n",
              "94                              GOCC_SYNAPTIC_MEMBRANE              5   \n",
              "62                                    GOCC_DISTAL_AXON              5   \n",
              "92                                     GOCC_PRESYNAPSE              5   \n",
              "31                  GOCC_CHROMOSOME_CENTROMERIC_REGION              6   \n",
              "24        GOCC_CONDENSED_CHROMOSOME_CENTROMERIC_REGION              6   \n",
              "5                              GOCC_NUCLEAR_CHROMOSOME              6   \n",
              "147                               GOCC_KINESIN_COMPLEX              4   \n",
              "110                      GOCC_NEURON_TO_NEURON_SYNAPSE              4   \n",
              "149                    GOCC_SPLICEOSOMAL_SNRNP_COMPLEX              4   \n",
              "178      GOCC_INTRINSIC_COMPONENT_OF_SYNAPTIC_MEMBRANE              4   \n",
              "169  GOCC_INTRINSIC_COMPONENT_OF_POSTSYNAPTIC_DENSI...              4   \n",
              "37                                GOCC_MITOTIC_SPINDLE              6   \n",
              "160                               GOCC_TERMINAL_BOUTON              4   \n",
              "138  GOCC_INTRINSIC_COMPONENT_OF_POSTSYNAPTIC_SPECI...              4   \n",
              "3                            GOCC_SPLICEOSOMAL_COMPLEX              6   \n",
              "117                 GOCC_POSTSYNAPTIC_DENSITY_MEMBRANE              4   \n",
              "168                                          GOCC_AXON              4   \n",
              "139  GOCC_INTRINSIC_COMPONENT_OF_POSTSYNAPTIC_MEMBRANE              4   \n",
              "143                                   GOCC_POSTSYNAPSE              4   \n",
              "\n",
              "     NES_mean    metric  \n",
              "77   1.952774  9.763871  \n",
              "94   1.866770  9.333848  \n",
              "62   1.865197  9.325986  \n",
              "92   1.860195  9.300976  \n",
              "31   1.504451  9.026706  \n",
              "24   1.434843  8.609056  \n",
              "5    1.406465  8.438788  \n",
              "147  2.090061  8.360244  \n",
              "110  2.032852  8.131407  \n",
              "149  2.014187  8.056746  \n",
              "178  2.000797  8.003186  \n",
              "169  1.997493  7.989972  \n",
              "37   1.306004  7.836022  \n",
              "160  1.950498  7.801993  \n",
              "138  1.949020  7.796080  \n",
              "3    1.297896  7.787375  \n",
              "117  1.946416  7.785665  \n",
              "168  1.942122  7.768489  \n",
              "139  1.935876  7.743503  \n",
              "143  1.927253  7.709012  "
            ]
          },
          "execution_count": 11,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "df['metric'] = df['dataset_count']*df['NES_mean']\n",
        "df = df.sort_values(by='metric', ascending=False)\n",
        "df.head(20)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "kernel": "Python3"
      },
      "source": [
        "### 3. Enriched Pathways from Gene Ontology Molecular Function (GOMF)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 12,
      "metadata": {
        "kernel": "Python3"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Query execution succeeded (time taken: 5.74 seconds, data scanned: 522.356 MB)\n",
            "Fetched 913 rows\n",
            "(913, 3)\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>pathway</th>\n",
              "      <th>dataset_count</th>\n",
              "      <th>NES_mean</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>GOMF_ENDOPEPTIDASE_ACTIVITY</td>\n",
              "      <td>7</td>\n",
              "      <td>-1.161230</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>GOMF_PEPTIDASE_ACTIVITY</td>\n",
              "      <td>7</td>\n",
              "      <td>-1.079669</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>GOMF_CELL_ADHESION_MOLECULE_BINDING</td>\n",
              "      <td>6</td>\n",
              "      <td>-1.235589</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>GOMF_ENZYME_REGULATOR_ACTIVITY</td>\n",
              "      <td>6</td>\n",
              "      <td>-0.846585</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>GOMF_PROTEASE_BINDING</td>\n",
              "      <td>6</td>\n",
              "      <td>-1.434183</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "                               pathway  dataset_count  NES_mean\n",
              "0          GOMF_ENDOPEPTIDASE_ACTIVITY              7 -1.161230\n",
              "1              GOMF_PEPTIDASE_ACTIVITY              7 -1.079669\n",
              "2  GOMF_CELL_ADHESION_MOLECULE_BINDING              6 -1.235589\n",
              "3       GOMF_ENZYME_REGULATOR_ACTIVITY              6 -0.846585\n",
              "4                GOMF_PROTEASE_BINDING              6 -1.434183"
            ]
          },
          "execution_count": 12,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "pathway_db_name = \"GOMF\"\n",
        "q = \"\"\"SELECT pathway, COUNT(src_dataset_id) AS dataset_count, \n",
        "              SUM(NES)/COUNT(src_dataset_id) AS NES_mean \n",
        "       FROM \"\"\" + params[\"ATLAS_NAME\"] + \"\"\".features \n",
        "       WHERE src_dataset_id IN \n",
        "           (SELECT dataset_id FROM \"\"\" + params[\"ATLAS_NAME\"] + \"\"\".datasets WHERE CONTAINS(condition_perturbation, \n",
        "           '\"\"\" + params[\"COMPARISION_NAME\"] + \"\"\"')) \n",
        "       AND padj < \"\"\" + str(params[\"Pvalue_cutoff\"]) + \"\"\" AND pathway LIKE '\"\"\" + pathway_db_name + \"\"\"%'\n",
        "       GROUP BY pathway\n",
        "       ORDER BY dataset_count DESC\"\"\"\n",
        "\n",
        "df = omixatlas.query_metadata(q, query_api_version='v2')\n",
        "print(df.shape)\n",
        "df.head()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 13,
      "metadata": {
        "kernel": "Python3"
      },
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>pathway</th>\n",
              "      <th>dataset_count</th>\n",
              "      <th>NES_mean</th>\n",
              "      <th>metric</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>29</th>\n",
              "      <td>GOMF_MICROTUBULE_MOTOR_ACTIVITY</td>\n",
              "      <td>5</td>\n",
              "      <td>2.009730</td>\n",
              "      <td>10.048648</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>25</th>\n",
              "      <td>GOMF_TUBULIN_BINDING</td>\n",
              "      <td>5</td>\n",
              "      <td>1.895192</td>\n",
              "      <td>9.475959</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>49</th>\n",
              "      <td>GOMF_MICROTUBULE_BINDING</td>\n",
              "      <td>5</td>\n",
              "      <td>1.886296</td>\n",
              "      <td>9.431480</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>21</th>\n",
              "      <td>GOMF_ATP_DEPENDENT_ACTIVITY_ACTING_ON_DNA</td>\n",
              "      <td>6</td>\n",
              "      <td>1.473506</td>\n",
              "      <td>8.841033</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>GOMF_DNA_HELICASE_ACTIVITY</td>\n",
              "      <td>6</td>\n",
              "      <td>1.443724</td>\n",
              "      <td>8.662342</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>23</th>\n",
              "      <td>GOMF_CATALYTIC_ACTIVITY_ACTING_ON_DNA</td>\n",
              "      <td>6</td>\n",
              "      <td>1.415170</td>\n",
              "      <td>8.491019</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>55</th>\n",
              "      <td>GOMF_ACETYLCHOLINE_RECEPTOR_ACTIVITY</td>\n",
              "      <td>4</td>\n",
              "      <td>2.002299</td>\n",
              "      <td>8.009197</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>69</th>\n",
              "      <td>GOMF_DNA_SECONDARY_STRUCTURE_BINDING</td>\n",
              "      <td>4</td>\n",
              "      <td>1.920315</td>\n",
              "      <td>7.681258</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>27</th>\n",
              "      <td>GOMF_SINGLE_STRANDED_DNA_HELICASE_ACTIVITY</td>\n",
              "      <td>5</td>\n",
              "      <td>1.283016</td>\n",
              "      <td>6.415080</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>46</th>\n",
              "      <td>GOMF_SINGLE_STRANDED_DNA_BINDING</td>\n",
              "      <td>5</td>\n",
              "      <td>1.274606</td>\n",
              "      <td>6.373028</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>36</th>\n",
              "      <td>GOMF_HISTONE_BINDING</td>\n",
              "      <td>5</td>\n",
              "      <td>1.267460</td>\n",
              "      <td>6.337301</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>196</th>\n",
              "      <td>GOMF_PLUS_END_DIRECTED_MICROTUBULE_MOTOR_ACTIVITY</td>\n",
              "      <td>3</td>\n",
              "      <td>2.003809</td>\n",
              "      <td>6.011426</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>73</th>\n",
              "      <td>GOMF_MRNA_BINDING</td>\n",
              "      <td>4</td>\n",
              "      <td>1.498413</td>\n",
              "      <td>5.993653</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>105</th>\n",
              "      <td>GOMF_FOUR_WAY_JUNCTION_DNA_BINDING</td>\n",
              "      <td>3</td>\n",
              "      <td>1.973218</td>\n",
              "      <td>5.919653</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>44</th>\n",
              "      <td>GOMF_CATALYTIC_ACTIVITY_ACTING_ON_A_TRNA</td>\n",
              "      <td>5</td>\n",
              "      <td>1.179818</td>\n",
              "      <td>5.899091</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>40</th>\n",
              "      <td>GOMF_RNA_METHYLTRANSFERASE_ACTIVITY</td>\n",
              "      <td>5</td>\n",
              "      <td>1.171681</td>\n",
              "      <td>5.858403</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>178</th>\n",
              "      <td>GOMF_METHYLATED_HISTONE_BINDING</td>\n",
              "      <td>3</td>\n",
              "      <td>1.950702</td>\n",
              "      <td>5.852105</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>182</th>\n",
              "      <td>GOMF_POSTSYNAPTIC_NEUROTRANSMITTER_RECEPTOR_AC...</td>\n",
              "      <td>3</td>\n",
              "      <td>1.937540</td>\n",
              "      <td>5.812620</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>142</th>\n",
              "      <td>GOMF_EXTRACELLULAR_LIGAND_GATED_ION_CHANNEL_AC...</td>\n",
              "      <td>3</td>\n",
              "      <td>1.925933</td>\n",
              "      <td>5.777800</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>127</th>\n",
              "      <td>GOMF_ACETYLCHOLINE_GATED_CATION_SELECTIVE_CHAN...</td>\n",
              "      <td>3</td>\n",
              "      <td>1.900293</td>\n",
              "      <td>5.700879</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "                                               pathway  dataset_count  \\\n",
              "29                     GOMF_MICROTUBULE_MOTOR_ACTIVITY              5   \n",
              "25                                GOMF_TUBULIN_BINDING              5   \n",
              "49                            GOMF_MICROTUBULE_BINDING              5   \n",
              "21           GOMF_ATP_DEPENDENT_ACTIVITY_ACTING_ON_DNA              6   \n",
              "7                           GOMF_DNA_HELICASE_ACTIVITY              6   \n",
              "23               GOMF_CATALYTIC_ACTIVITY_ACTING_ON_DNA              6   \n",
              "55                GOMF_ACETYLCHOLINE_RECEPTOR_ACTIVITY              4   \n",
              "69                GOMF_DNA_SECONDARY_STRUCTURE_BINDING              4   \n",
              "27          GOMF_SINGLE_STRANDED_DNA_HELICASE_ACTIVITY              5   \n",
              "46                    GOMF_SINGLE_STRANDED_DNA_BINDING              5   \n",
              "36                                GOMF_HISTONE_BINDING              5   \n",
              "196  GOMF_PLUS_END_DIRECTED_MICROTUBULE_MOTOR_ACTIVITY              3   \n",
              "73                                   GOMF_MRNA_BINDING              4   \n",
              "105                 GOMF_FOUR_WAY_JUNCTION_DNA_BINDING              3   \n",
              "44            GOMF_CATALYTIC_ACTIVITY_ACTING_ON_A_TRNA              5   \n",
              "40                 GOMF_RNA_METHYLTRANSFERASE_ACTIVITY              5   \n",
              "178                    GOMF_METHYLATED_HISTONE_BINDING              3   \n",
              "182  GOMF_POSTSYNAPTIC_NEUROTRANSMITTER_RECEPTOR_AC...              3   \n",
              "142  GOMF_EXTRACELLULAR_LIGAND_GATED_ION_CHANNEL_AC...              3   \n",
              "127  GOMF_ACETYLCHOLINE_GATED_CATION_SELECTIVE_CHAN...              3   \n",
              "\n",
              "     NES_mean     metric  \n",
              "29   2.009730  10.048648  \n",
              "25   1.895192   9.475959  \n",
              "49   1.886296   9.431480  \n",
              "21   1.473506   8.841033  \n",
              "7    1.443724   8.662342  \n",
              "23   1.415170   8.491019  \n",
              "55   2.002299   8.009197  \n",
              "69   1.920315   7.681258  \n",
              "27   1.283016   6.415080  \n",
              "46   1.274606   6.373028  \n",
              "36   1.267460   6.337301  \n",
              "196  2.003809   6.011426  \n",
              "73   1.498413   5.993653  \n",
              "105  1.973218   5.919653  \n",
              "44   1.179818   5.899091  \n",
              "40   1.171681   5.858403  \n",
              "178  1.950702   5.852105  \n",
              "182  1.937540   5.812620  \n",
              "142  1.925933   5.777800  \n",
              "127  1.900293   5.700879  "
            ]
          },
          "execution_count": 13,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "df['metric'] = df['dataset_count']*df['NES_mean']\n",
        "df = df.sort_values(by='metric', ascending=False)\n",
        "df.head(20)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "kernel": "Python3"
      },
      "source": [
        "### 4. Enriched HP"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 14,
      "metadata": {
        "kernel": "Python3"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Query execution succeeded (time taken: 5.34 seconds, data scanned: 522.356 MB)\n",
            "Fetched 3169 rows\n",
            "(3169, 3)\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>pathway</th>\n",
              "      <th>dataset_count</th>\n",
              "      <th>NES_mean</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>HP_ABNORMAL_BLEEDING</td>\n",
              "      <td>7</td>\n",
              "      <td>-1.337116</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>HP_VASCULAR_SKIN_ABNORMALITY</td>\n",
              "      <td>7</td>\n",
              "      <td>-1.318931</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>HP_GENERALIZED_ABNORMALITY_OF_SKIN</td>\n",
              "      <td>7</td>\n",
              "      <td>-1.204305</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>HP_SUBCUTANEOUS_HEMORRHAGE</td>\n",
              "      <td>7</td>\n",
              "      <td>-1.423481</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>HP_ABNORMAL_VASCULAR_PHYSIOLOGY</td>\n",
              "      <td>6</td>\n",
              "      <td>-1.374245</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "                              pathway  dataset_count  NES_mean\n",
              "0                HP_ABNORMAL_BLEEDING              7 -1.337116\n",
              "1        HP_VASCULAR_SKIN_ABNORMALITY              7 -1.318931\n",
              "2  HP_GENERALIZED_ABNORMALITY_OF_SKIN              7 -1.204305\n",
              "3          HP_SUBCUTANEOUS_HEMORRHAGE              7 -1.423481\n",
              "4     HP_ABNORMAL_VASCULAR_PHYSIOLOGY              6 -1.374245"
            ]
          },
          "execution_count": 14,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "pathway_db_name = \"HP\"\n",
        "q = \"\"\"SELECT pathway, COUNT(src_dataset_id) AS dataset_count, \n",
        "              SUM(NES)/COUNT(src_dataset_id) AS NES_mean \n",
        "       FROM \"\"\" + params[\"ATLAS_NAME\"] + \"\"\".features \n",
        "       WHERE src_dataset_id IN \n",
        "           (SELECT dataset_id FROM \"\"\" + params[\"ATLAS_NAME\"] + \"\"\".datasets WHERE CONTAINS(condition_perturbation, \n",
        "           '\"\"\" + params[\"COMPARISION_NAME\"] + \"\"\"')) \n",
        "       AND padj < \"\"\" + str(params[\"Pvalue_cutoff\"]) + \"\"\" AND pathway LIKE '\"\"\" + pathway_db_name + \"\"\"%'\n",
        "       GROUP BY pathway\n",
        "       ORDER BY dataset_count DESC\"\"\"\n",
        "\n",
        "df = omixatlas.query_metadata(q, query_api_version='v2')\n",
        "print(df.shape)\n",
        "df.head()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 15,
      "metadata": {
        "kernel": "Python3"
      },
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>pathway</th>\n",
              "      <th>dataset_count</th>\n",
              "      <th>NES_mean</th>\n",
              "      <th>metric</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>64</th>\n",
              "      <td>HP_ABNORMALITY_OF_URINE_CATECHOLAMINE_CONCENTR...</td>\n",
              "      <td>5</td>\n",
              "      <td>1.980936</td>\n",
              "      <td>9.904678</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>111</th>\n",
              "      <td>HP_PROMINENT_DIGIT_PAD</td>\n",
              "      <td>5</td>\n",
              "      <td>1.845298</td>\n",
              "      <td>9.226492</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>171</th>\n",
              "      <td>HP_OCULOGYRIC_CRISIS</td>\n",
              "      <td>5</td>\n",
              "      <td>1.838789</td>\n",
              "      <td>9.193945</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>24</th>\n",
              "      <td>HP_SIMPLIFIED_GYRAL_PATTERN</td>\n",
              "      <td>6</td>\n",
              "      <td>1.429893</td>\n",
              "      <td>8.579355</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>183</th>\n",
              "      <td>HP_BILATERAL_TONIC_CLONIC_SEIZURE</td>\n",
              "      <td>5</td>\n",
              "      <td>1.638029</td>\n",
              "      <td>8.190145</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>311</th>\n",
              "      <td>HP_EPILEPTIC_ENCEPHALOPATHY</td>\n",
              "      <td>4</td>\n",
              "      <td>1.970116</td>\n",
              "      <td>7.880465</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>HP_SLOPING_FOREHEAD</td>\n",
              "      <td>6</td>\n",
              "      <td>1.269897</td>\n",
              "      <td>7.619380</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>243</th>\n",
              "      <td>HP_TONIC_SEIZURE</td>\n",
              "      <td>4</td>\n",
              "      <td>1.863954</td>\n",
              "      <td>7.455817</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>321</th>\n",
              "      <td>HP_ABSENT_THUMB</td>\n",
              "      <td>4</td>\n",
              "      <td>1.858084</td>\n",
              "      <td>7.432337</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>227</th>\n",
              "      <td>HP_NON_MOTOR_SEIZURE</td>\n",
              "      <td>4</td>\n",
              "      <td>1.787961</td>\n",
              "      <td>7.151844</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>212</th>\n",
              "      <td>HP_HYPSARRHYTHMIA</td>\n",
              "      <td>4</td>\n",
              "      <td>1.761773</td>\n",
              "      <td>7.047091</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>323</th>\n",
              "      <td>HP_ABNORMAL_AGGRESSIVE_IMPULSIVE_OR_VIOLENT_BE...</td>\n",
              "      <td>4</td>\n",
              "      <td>1.737751</td>\n",
              "      <td>6.951005</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>352</th>\n",
              "      <td>HP_AGGRESSIVE_BEHAVIOR</td>\n",
              "      <td>4</td>\n",
              "      <td>1.735481</td>\n",
              "      <td>6.941923</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>353</th>\n",
              "      <td>HP_AUTISTIC_BEHAVIOR</td>\n",
              "      <td>4</td>\n",
              "      <td>1.729906</td>\n",
              "      <td>6.919623</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>356</th>\n",
              "      <td>HP_INTERICTAL_EEG_ABNORMALITY</td>\n",
              "      <td>4</td>\n",
              "      <td>1.729683</td>\n",
              "      <td>6.918730</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>211</th>\n",
              "      <td>HP_GENERALIZED_ONSET_MOTOR_SEIZURE</td>\n",
              "      <td>4</td>\n",
              "      <td>1.709128</td>\n",
              "      <td>6.836512</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>342</th>\n",
              "      <td>HP_DYSKINESIA</td>\n",
              "      <td>4</td>\n",
              "      <td>1.695275</td>\n",
              "      <td>6.781099</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>292</th>\n",
              "      <td>HP_TRIANGULAR_SHAPED_PHALANGES_OF_THE_HAND</td>\n",
              "      <td>4</td>\n",
              "      <td>1.695031</td>\n",
              "      <td>6.780125</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>357</th>\n",
              "      <td>HP_ABSENT_SPEECH</td>\n",
              "      <td>4</td>\n",
              "      <td>1.662556</td>\n",
              "      <td>6.650225</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>276</th>\n",
              "      <td>HP_EPILEPTIC_SPASM</td>\n",
              "      <td>4</td>\n",
              "      <td>1.649503</td>\n",
              "      <td>6.598013</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "                                               pathway  dataset_count  \\\n",
              "64   HP_ABNORMALITY_OF_URINE_CATECHOLAMINE_CONCENTR...              5   \n",
              "111                             HP_PROMINENT_DIGIT_PAD              5   \n",
              "171                               HP_OCULOGYRIC_CRISIS              5   \n",
              "24                         HP_SIMPLIFIED_GYRAL_PATTERN              6   \n",
              "183                  HP_BILATERAL_TONIC_CLONIC_SEIZURE              5   \n",
              "311                        HP_EPILEPTIC_ENCEPHALOPATHY              4   \n",
              "6                                  HP_SLOPING_FOREHEAD              6   \n",
              "243                                   HP_TONIC_SEIZURE              4   \n",
              "321                                    HP_ABSENT_THUMB              4   \n",
              "227                               HP_NON_MOTOR_SEIZURE              4   \n",
              "212                                  HP_HYPSARRHYTHMIA              4   \n",
              "323  HP_ABNORMAL_AGGRESSIVE_IMPULSIVE_OR_VIOLENT_BE...              4   \n",
              "352                             HP_AGGRESSIVE_BEHAVIOR              4   \n",
              "353                               HP_AUTISTIC_BEHAVIOR              4   \n",
              "356                      HP_INTERICTAL_EEG_ABNORMALITY              4   \n",
              "211                 HP_GENERALIZED_ONSET_MOTOR_SEIZURE              4   \n",
              "342                                      HP_DYSKINESIA              4   \n",
              "292         HP_TRIANGULAR_SHAPED_PHALANGES_OF_THE_HAND              4   \n",
              "357                                   HP_ABSENT_SPEECH              4   \n",
              "276                                 HP_EPILEPTIC_SPASM              4   \n",
              "\n",
              "     NES_mean    metric  \n",
              "64   1.980936  9.904678  \n",
              "111  1.845298  9.226492  \n",
              "171  1.838789  9.193945  \n",
              "24   1.429893  8.579355  \n",
              "183  1.638029  8.190145  \n",
              "311  1.970116  7.880465  \n",
              "6    1.269897  7.619380  \n",
              "243  1.863954  7.455817  \n",
              "321  1.858084  7.432337  \n",
              "227  1.787961  7.151844  \n",
              "212  1.761773  7.047091  \n",
              "323  1.737751  6.951005  \n",
              "352  1.735481  6.941923  \n",
              "353  1.729906  6.919623  \n",
              "356  1.729683  6.918730  \n",
              "211  1.709128  6.836512  \n",
              "342  1.695275  6.781099  \n",
              "292  1.695031  6.780125  \n",
              "357  1.662556  6.650225  \n",
              "276  1.649503  6.598013  "
            ]
          },
          "execution_count": 15,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "df['metric'] = df['dataset_count']*df['NES_mean']\n",
        "df = df.sort_values(by='metric', ascending=False)\n",
        "df.head(20)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "kernel": "Python3"
      },
      "outputs": [],
      "source": []
    }
  ],
  "metadata": {
    "kernelspec": {
      "display_name": "Python 3 (ipykernel)",
      "language": "python",
      "name": "python3"
    },
    "language_info": {
      "codemirror_mode": {
        "name": "ipython",
        "version": 3
      },
      "file_extension": ".py",
      "mimetype": "text/x-python",
      "name": "python",
      "nbconvert_exporter": "python",
      "pygments_lexer": "ipython3",
      "version": "3.10.2"
    },
    "sos": {
      "kernels": [
        [
          "Bash",
          "bash",
          "Bash",
          "#E6EEFF",
          "shell"
        ],
        [
          "Python3",
          "python3",
          "Python3",
          "#FFD91A",
          ""
        ]
      ],
      "panel": {
        "displayed": true,
        "height": 0
      },
      "version": "1.0.0"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 2
}