{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "colab_type": "text",
        "id": "view-in-github"
      },
      "source": [
        "<a href=\"https://polly.elucidata.io/manage/workspaces?action=open_polly_notebook&amp;source=github&amp;path=ElucidataInc%2Fpolly-python%2Fblob%2Fmain%2FDiscover%2Fmeta_analysis_transcriptomics%2FGene+bio-markers+for+Medulloblastoma.ipynb&amp;kernel=elucidata%2FPython+3.10&amp;machine=medium\" target=\"_parent\"><img alt=\"Open in Polly\" src=\"https://elucidatainc.github.io/PublicAssets/open_polly.svg\"/></a>\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## Objective:\n",
        "\n",
        "Identify GEO datasets where Medulloblastoma samples are compared with Normal samples. \n",
        "'mycn' gene is differentially expressed."
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "kernel": "SoS"
      },
      "source": [
        "## Install polly-python"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {
        "kernel": "Python3",
        "scrolled": true
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "/bin/bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)\n",
            "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
            "awscli 1.22.74 requires PyYAML<5.5,>=3.10, but you have pyyaml 6.0 which is incompatible.\u001b[0m\n",
            "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\n",
            "\u001b[33mWARNING: You are using pip version 21.2.4; however, version 22.2.2 is available.\n",
            "You should consider upgrading via the '/usr/local/bin/python3.10 -m pip install --upgrade pip' command.\u001b[0m\n"
          ]
        }
      ],
      "source": [
        "! sudo pip3 install polly-python --quiet"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "kernel": "Bash"
      },
      "source": [
        "## Import libraries and OA configuration"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 2,
      "metadata": {
        "kernel": "Python3"
      },
      "outputs": [],
      "source": [
        "import os\n",
        "from polly.auth import Polly\n",
        "from polly.omixatlas import OmixAtlas\n",
        "\n",
        "AUTH_TOKEN=(os.environ['POLLY_REFRESH_TOKEN']) # Obtain authentication tokens\n",
        "Polly.auth(AUTH_TOKEN)\n",
        "\n",
        "#Defining omixatlas object\n",
        "omixatlas = OmixAtlas()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "kernel": "Python3"
      },
      "source": [
        "## Tables in the GDX atlas"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 3,
      "metadata": {
        "kernel": "Python3"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Query execution succeeded (time taken: 1.01 seconds, data scanned: 0.000 MB)\n",
            "Fetched 2 rows\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>table_name</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>gdx_atlas.datasets</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>gdx_atlas.features</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "           table_name\n",
              "0  gdx_atlas.datasets\n",
              "1  gdx_atlas.features"
            ]
          },
          "execution_count": 3,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "q = \"SHOW TABLES IN gdx_atlas\"\n",
        "omixatlas.query_metadata(q)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "kernel": "Python3"
      },
      "source": [
        "## Get the schema of the GDX atlas"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 4,
      "metadata": {
        "kernel": "Python3"
      },
      "outputs": [],
      "source": [
        "gdx_schema = omixatlas.get_schema('gdx_atlas', schema_level=['dataset'])"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 5,
      "metadata": {
        "kernel": "Python3"
      },
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Source</th>\n",
              "      <th>Datatype</th>\n",
              "      <th>Field Name</th>\n",
              "      <th>Field Description</th>\n",
              "      <th>Field Type</th>\n",
              "      <th>Is Curated</th>\n",
              "      <th>Is Array</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>bucket</td>\n",
              "      <td>S3 bucket in which the data file resides</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>cohort_separability</td>\n",
              "      <td>Cohort separability in PCA</td>\n",
              "      <td>float</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>condition_column</td>\n",
              "      <td>Condition column from sample metadata</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>condition_control</td>\n",
              "      <td>condition_column value for control</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>condition_perturbation</td>\n",
              "      <td>condition_column value for perturbation</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>curated_cell_line</td>\n",
              "      <td>curated_cell_line</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>curated_cell_type</td>\n",
              "      <td>curated_cell_type</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>curated_disease</td>\n",
              "      <td>curated_disease</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>curated_drug</td>\n",
              "      <td>curated_drug</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>curated_gene</td>\n",
              "      <td>curated_gene</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>curated_tissue</td>\n",
              "      <td>curated_tissue</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>data_table_name</td>\n",
              "      <td>Name of the data table associated with data file</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>12</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>data_table_version</td>\n",
              "      <td>Current version of the data table associated w...</td>\n",
              "      <td>integer</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>data_type</td>\n",
              "      <td>data_type</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>14</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>dataset_id</td>\n",
              "      <td>dataset_id</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>15</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>dataset_source</td>\n",
              "      <td>Dataset source</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>16</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>file_location</td>\n",
              "      <td>Unique S3 location for this data file</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>17</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>file_type</td>\n",
              "      <td>Data file's type</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>18</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>is_current</td>\n",
              "      <td>Whether this is the current version of the dat...</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>19</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>key</td>\n",
              "      <td>Key to the data file in source package</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>20</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>package</td>\n",
              "      <td>Name of the package this data file was part of</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>21</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>region</td>\n",
              "      <td>Region (AWS) in which this data is located</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>22</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>samples_control</td>\n",
              "      <td>Sample ids for control cohort</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>23</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>samples_perturbation</td>\n",
              "      <td>Sample ids for perturbation cohort</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>24</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>src_dataset_id</td>\n",
              "      <td>src_dataset_id</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>25</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>src_description</td>\n",
              "      <td>src_description</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>26</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>src_overall_design</td>\n",
              "      <td>src_overall_design</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>27</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>src_repo</td>\n",
              "      <td>Name of the repository this data file originat...</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>28</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>src_summary</td>\n",
              "      <td>src_summary</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>29</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>src_uri</td>\n",
              "      <td>Unique URI derived from data file's S3 location</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>30</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>state_1</td>\n",
              "      <td>Name given to control condition</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>31</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>state_2</td>\n",
              "      <td>Name given to perturbation condition</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>32</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>status</td>\n",
              "      <td>Dataset status</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>33</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>timestamp_</td>\n",
              "      <td>Unix timestamp denoting time of creation for t...</td>\n",
              "      <td>text</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>34</th>\n",
              "      <td>all</td>\n",
              "      <td>all</td>\n",
              "      <td>version</td>\n",
              "      <td>Version of data file</td>\n",
              "      <td>integer</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "   Source Datatype              Field Name  \\\n",
              "0     all      all                  bucket   \n",
              "1     all      all     cohort_separability   \n",
              "2     all      all        condition_column   \n",
              "3     all      all       condition_control   \n",
              "4     all      all  condition_perturbation   \n",
              "5     all      all       curated_cell_line   \n",
              "6     all      all       curated_cell_type   \n",
              "7     all      all         curated_disease   \n",
              "8     all      all            curated_drug   \n",
              "9     all      all            curated_gene   \n",
              "10    all      all          curated_tissue   \n",
              "11    all      all         data_table_name   \n",
              "12    all      all      data_table_version   \n",
              "13    all      all               data_type   \n",
              "14    all      all              dataset_id   \n",
              "15    all      all          dataset_source   \n",
              "16    all      all           file_location   \n",
              "17    all      all               file_type   \n",
              "18    all      all              is_current   \n",
              "19    all      all                     key   \n",
              "20    all      all                 package   \n",
              "21    all      all                  region   \n",
              "22    all      all         samples_control   \n",
              "23    all      all    samples_perturbation   \n",
              "24    all      all          src_dataset_id   \n",
              "25    all      all         src_description   \n",
              "26    all      all      src_overall_design   \n",
              "27    all      all                src_repo   \n",
              "28    all      all             src_summary   \n",
              "29    all      all                 src_uri   \n",
              "30    all      all                 state_1   \n",
              "31    all      all                 state_2   \n",
              "32    all      all                  status   \n",
              "33    all      all              timestamp_   \n",
              "34    all      all                 version   \n",
              "\n",
              "                                    Field Description Field Type  Is Curated  \\\n",
              "0            S3 bucket in which the data file resides       text       False   \n",
              "1                          Cohort separability in PCA      float       False   \n",
              "2               Condition column from sample metadata       text       False   \n",
              "3                  condition_column value for control       text       False   \n",
              "4             condition_column value for perturbation       text       False   \n",
              "5                                   curated_cell_line       text       False   \n",
              "6                                   curated_cell_type       text       False   \n",
              "7                                     curated_disease       text       False   \n",
              "8                                        curated_drug       text       False   \n",
              "9                                        curated_gene       text       False   \n",
              "10                                     curated_tissue       text       False   \n",
              "11   Name of the data table associated with data file       text       False   \n",
              "12  Current version of the data table associated w...    integer       False   \n",
              "13                                          data_type       text       False   \n",
              "14                                         dataset_id       text       False   \n",
              "15                                     Dataset source       text       False   \n",
              "16              Unique S3 location for this data file       text       False   \n",
              "17                                   Data file's type       text       False   \n",
              "18  Whether this is the current version of the dat...       text       False   \n",
              "19             Key to the data file in source package       text       False   \n",
              "20     Name of the package this data file was part of       text       False   \n",
              "21         Region (AWS) in which this data is located       text       False   \n",
              "22                      Sample ids for control cohort       text       False   \n",
              "23                 Sample ids for perturbation cohort       text       False   \n",
              "24                                     src_dataset_id       text       False   \n",
              "25                                    src_description       text       False   \n",
              "26                                 src_overall_design       text       False   \n",
              "27  Name of the repository this data file originat...       text       False   \n",
              "28                                        src_summary       text       False   \n",
              "29    Unique URI derived from data file's S3 location       text       False   \n",
              "30                   Name given to control condition        text       False   \n",
              "31              Name given to perturbation condition        text       False   \n",
              "32                                     Dataset status       text       False   \n",
              "33  Unix timestamp denoting time of creation for t...       text       False   \n",
              "34                               Version of data file    integer       False   \n",
              "\n",
              "    Is Array  \n",
              "0      False  \n",
              "1      False  \n",
              "2      False  \n",
              "3       True  \n",
              "4       True  \n",
              "5       True  \n",
              "6       True  \n",
              "7       True  \n",
              "8       True  \n",
              "9       True  \n",
              "10      True  \n",
              "11     False  \n",
              "12     False  \n",
              "13     False  \n",
              "14     False  \n",
              "15     False  \n",
              "16     False  \n",
              "17     False  \n",
              "18     False  \n",
              "19     False  \n",
              "20     False  \n",
              "21     False  \n",
              "22      True  \n",
              "23      True  \n",
              "24     False  \n",
              "25     False  \n",
              "26     False  \n",
              "27     False  \n",
              "28     False  \n",
              "29     False  \n",
              "30      True  \n",
              "31      True  \n",
              "32     False  \n",
              "33     False  \n",
              "34     False  "
            ]
          },
          "execution_count": 5,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "gdx_schema.dataset"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "kernel": "Python3"
      },
      "source": [
        "## Query GDX atlas"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "kernel": "Python3"
      },
      "source": [
        "### 1. datasets table"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 6,
      "metadata": {
        "kernel": "Python3"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Query execution succeeded (time taken: 1.76 seconds, data scanned: 0.060 MB)\n",
            "Fetched 5 rows\n",
            "(5, 5)\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>src_dataset_id</th>\n",
              "      <th>src_description</th>\n",
              "      <th>condition_column</th>\n",
              "      <th>condition_control</th>\n",
              "      <th>condition_perturbation</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>GSE119157_GPL24247</td>\n",
              "      <td>Effect of norepinephrine and focal adhesion ki...</td>\n",
              "      <td>kw_curated_drug</td>\n",
              "      <td>[none]</td>\n",
              "      <td>[(R)-noradrenaline]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>GSE119167_GPL17543</td>\n",
              "      <td>Rbpj expression in regulatory T cells is criti...</td>\n",
              "      <td>kw_curated_disease</td>\n",
              "      <td>[Normal]</td>\n",
              "      <td>[Myeloid splenomegaly]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>GSE119169_GPL17543</td>\n",
              "      <td>Rbpj expression in regulatory T cells is criti...</td>\n",
              "      <td>kw_curated_disease</td>\n",
              "      <td>[Normal]</td>\n",
              "      <td>[Myeloid splenomegaly]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>GSE119207_GPL16686</td>\n",
              "      <td>Plucked hair follicles from patients with chro...</td>\n",
              "      <td>kw_curated_disease</td>\n",
              "      <td>[Normal]</td>\n",
              "      <td>[Lupus Erythematosus,  Discoid]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>GSE119265_GPL18573</td>\n",
              "      <td>Early onset preeclampsia (EOPE) in a model for...</td>\n",
              "      <td>kw_curated_disease</td>\n",
              "      <td>[Normal]</td>\n",
              "      <td>[Pre-Eclampsia]</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "       src_dataset_id                                    src_description  \\\n",
              "0  GSE119157_GPL24247  Effect of norepinephrine and focal adhesion ki...   \n",
              "1  GSE119167_GPL17543  Rbpj expression in regulatory T cells is criti...   \n",
              "2  GSE119169_GPL17543  Rbpj expression in regulatory T cells is criti...   \n",
              "3  GSE119207_GPL16686  Plucked hair follicles from patients with chro...   \n",
              "4  GSE119265_GPL18573  Early onset preeclampsia (EOPE) in a model for...   \n",
              "\n",
              "     condition_column condition_control           condition_perturbation  \n",
              "0     kw_curated_drug            [none]              [(R)-noradrenaline]  \n",
              "1  kw_curated_disease          [Normal]           [Myeloid splenomegaly]  \n",
              "2  kw_curated_disease          [Normal]           [Myeloid splenomegaly]  \n",
              "3  kw_curated_disease          [Normal]  [Lupus Erythematosus,  Discoid]  \n",
              "4  kw_curated_disease          [Normal]                  [Pre-Eclampsia]  "
            ]
          },
          "execution_count": 6,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "q = \"\"\"SELECT src_dataset_id, \n",
        "              src_description,\n",
        "              condition_column,\n",
        "              condition_control,\n",
        "              condition_perturbation\n",
        "       FROM gdx_atlas.datasets LIMIT 5\"\"\"\n",
        "df = omixatlas.query_metadata(q, query_api_version='v2')\n",
        "print(df.shape)\n",
        "df.head()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "kernel": "Python3"
      },
      "source": [
        "### 2. features table"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 7,
      "metadata": {
        "kernel": "Python3",
        "scrolled": false
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Query execution succeeded (time taken: 3.77 seconds, data scanned: 59.085 MB)\n",
            "Fetched 5 rows\n",
            "(5, 17)\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>avg_exp</th>\n",
              "      <th>p_value</th>\n",
              "      <th>src_repo</th>\n",
              "      <th>id_key</th>\n",
              "      <th>b</th>\n",
              "      <th>src_uri</th>\n",
              "      <th>gene</th>\n",
              "      <th>bh</th>\n",
              "      <th>src_dataset_id</th>\n",
              "      <th>version</th>\n",
              "      <th>log_fc</th>\n",
              "      <th>adj_p_val</th>\n",
              "      <th>timestamp_</th>\n",
              "      <th>t</th>\n",
              "      <th>data_id</th>\n",
              "      <th>name</th>\n",
              "      <th>is_current</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>6.979292</td>\n",
              "      <td>0.000374</td>\n",
              "      <td>gdx_atlas</td>\n",
              "      <td>kw_row_id</td>\n",
              "      <td>-3.967370</td>\n",
              "      <td>polly:data://gdx_atlas/data/GSE96689_GPL23185-...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0.999932</td>\n",
              "      <td>GSE96689_GPL23185-2022-07-08-06-15-01</td>\n",
              "      <td>0</td>\n",
              "      <td>0.886917</td>\n",
              "      <td>0.999932</td>\n",
              "      <td>1657277152614</td>\n",
              "      <td>5.003541</td>\n",
              "      <td>kiaa1161</td>\n",
              "      <td>KIAA1161</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>3.884633</td>\n",
              "      <td>0.000423</td>\n",
              "      <td>gdx_atlas</td>\n",
              "      <td>kw_row_id</td>\n",
              "      <td>-3.974604</td>\n",
              "      <td>polly:data://gdx_atlas/data/GSE96689_GPL23185-...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0.999932</td>\n",
              "      <td>GSE96689_GPL23185-2022-07-08-06-15-01</td>\n",
              "      <td>0</td>\n",
              "      <td>1.883600</td>\n",
              "      <td>0.999932</td>\n",
              "      <td>1657277152614</td>\n",
              "      <td>4.927506</td>\n",
              "      <td>cd14</td>\n",
              "      <td>CD14</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>2.566425</td>\n",
              "      <td>0.000457</td>\n",
              "      <td>gdx_atlas</td>\n",
              "      <td>kw_row_id</td>\n",
              "      <td>-3.979209</td>\n",
              "      <td>polly:data://gdx_atlas/data/GSE96689_GPL23185-...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0.999932</td>\n",
              "      <td>GSE96689_GPL23185-2022-07-08-06-15-01</td>\n",
              "      <td>0</td>\n",
              "      <td>0.566650</td>\n",
              "      <td>0.999932</td>\n",
              "      <td>1657277152614</td>\n",
              "      <td>4.880098</td>\n",
              "      <td>dach1</td>\n",
              "      <td>DACH1</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>9.535658</td>\n",
              "      <td>0.000931</td>\n",
              "      <td>gdx_atlas</td>\n",
              "      <td>kw_row_id</td>\n",
              "      <td>-4.024778</td>\n",
              "      <td>polly:data://gdx_atlas/data/GSE96689_GPL23185-...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0.999932</td>\n",
              "      <td>GSE96689_GPL23185-2022-07-08-06-15-01</td>\n",
              "      <td>0</td>\n",
              "      <td>-0.545550</td>\n",
              "      <td>0.999932</td>\n",
              "      <td>1657277152614</td>\n",
              "      <td>-4.447537</td>\n",
              "      <td>wars2</td>\n",
              "      <td>WARS2</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>5.316342</td>\n",
              "      <td>0.001668</td>\n",
              "      <td>gdx_atlas</td>\n",
              "      <td>kw_row_id</td>\n",
              "      <td>-4.065846</td>\n",
              "      <td>polly:data://gdx_atlas/data/GSE96689_GPL23185-...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0.999932</td>\n",
              "      <td>GSE96689_GPL23185-2022-07-08-06-15-01</td>\n",
              "      <td>0</td>\n",
              "      <td>1.298983</td>\n",
              "      <td>0.999932</td>\n",
              "      <td>1657277152614</td>\n",
              "      <td>4.104426</td>\n",
              "      <td>fam53a</td>\n",
              "      <td>FAM53A</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "    avg_exp   p_value   src_repo     id_key         b  \\\n",
              "0  6.979292  0.000374  gdx_atlas  kw_row_id -3.967370   \n",
              "1  3.884633  0.000423  gdx_atlas  kw_row_id -3.974604   \n",
              "2  2.566425  0.000457  gdx_atlas  kw_row_id -3.979209   \n",
              "3  9.535658  0.000931  gdx_atlas  kw_row_id -4.024778   \n",
              "4  5.316342  0.001668  gdx_atlas  kw_row_id -4.065846   \n",
              "\n",
              "                                             src_uri  gene        bh  \\\n",
              "0  polly:data://gdx_atlas/data/GSE96689_GPL23185-...   NaN  0.999932   \n",
              "1  polly:data://gdx_atlas/data/GSE96689_GPL23185-...   NaN  0.999932   \n",
              "2  polly:data://gdx_atlas/data/GSE96689_GPL23185-...   NaN  0.999932   \n",
              "3  polly:data://gdx_atlas/data/GSE96689_GPL23185-...   NaN  0.999932   \n",
              "4  polly:data://gdx_atlas/data/GSE96689_GPL23185-...   NaN  0.999932   \n",
              "\n",
              "                          src_dataset_id  version    log_fc  adj_p_val  \\\n",
              "0  GSE96689_GPL23185-2022-07-08-06-15-01        0  0.886917   0.999932   \n",
              "1  GSE96689_GPL23185-2022-07-08-06-15-01        0  1.883600   0.999932   \n",
              "2  GSE96689_GPL23185-2022-07-08-06-15-01        0  0.566650   0.999932   \n",
              "3  GSE96689_GPL23185-2022-07-08-06-15-01        0 -0.545550   0.999932   \n",
              "4  GSE96689_GPL23185-2022-07-08-06-15-01        0  1.298983   0.999932   \n",
              "\n",
              "      timestamp_         t   data_id      name  is_current  \n",
              "0  1657277152614  5.003541  kiaa1161  KIAA1161        True  \n",
              "1  1657277152614  4.927506      cd14      CD14        True  \n",
              "2  1657277152614  4.880098     dach1     DACH1        True  \n",
              "3  1657277152614 -4.447537     wars2     WARS2        True  \n",
              "4  1657277152614  4.104426    fam53a    FAM53A        True  "
            ]
          },
          "execution_count": 7,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "q = \"\"\"SELECT *\n",
        "       FROM gdx_atlas.features LIMIT 5\"\"\"\n",
        "df = omixatlas.query_metadata(q, query_api_version='v2')\n",
        "print(df.shape)\n",
        "df.head()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "kernel": "Python3"
      },
      "source": [
        "## PARAMS"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 8,
      "metadata": {
        "kernel": "Python3"
      },
      "outputs": [],
      "source": [
        "condition_perturbation = 'Medulloblastoma'\n",
        "condition_control = 'Normal'"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "kernel": "Python3"
      },
      "source": [
        "## Get comparisons of interest"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 9,
      "metadata": {
        "kernel": "Python3"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Query execution succeeded (time taken: 2.04 seconds, data scanned: 0.239 MB)\n",
            "Fetched 14 rows\n",
            "(14, 5)\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>src_dataset_id</th>\n",
              "      <th>src_description</th>\n",
              "      <th>condition_column</th>\n",
              "      <th>condition_control</th>\n",
              "      <th>condition_perturbation</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>GSE28192_GPL6102</td>\n",
              "      <td>MB_xenograft_models</td>\n",
              "      <td>kw_curated_disease</td>\n",
              "      <td>[Normal]</td>\n",
              "      <td>[Medulloblastoma]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>GSE33199_GPL1261</td>\n",
              "      <td>A mouse model of the most aggressive subgroup ...</td>\n",
              "      <td>kw_curated_disease</td>\n",
              "      <td>[Normal]</td>\n",
              "      <td>[Medulloblastoma]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>GSE39182_GPL6480</td>\n",
              "      <td>Epigenetic Silencing of DKK3 in Medulloblastoma.</td>\n",
              "      <td>kw_curated_disease</td>\n",
              "      <td>[Normal]</td>\n",
              "      <td>[Medulloblastoma]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>GSE46406_GPL11180</td>\n",
              "      <td>Pemetrexed and gemcitabine as combination ther...</td>\n",
              "      <td>kw_curated_disease</td>\n",
              "      <td>[Normal]</td>\n",
              "      <td>[Medulloblastoma]</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>GSE62600_GPL96</td>\n",
              "      <td>Gene expression analysis of human medulloblast...</td>\n",
              "      <td>kw_curated_disease</td>\n",
              "      <td>[Normal]</td>\n",
              "      <td>[Medulloblastoma]</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "      src_dataset_id                                    src_description  \\\n",
              "0   GSE28192_GPL6102                                MB_xenograft_models   \n",
              "1   GSE33199_GPL1261  A mouse model of the most aggressive subgroup ...   \n",
              "2   GSE39182_GPL6480   Epigenetic Silencing of DKK3 in Medulloblastoma.   \n",
              "3  GSE46406_GPL11180  Pemetrexed and gemcitabine as combination ther...   \n",
              "4     GSE62600_GPL96  Gene expression analysis of human medulloblast...   \n",
              "\n",
              "     condition_column condition_control condition_perturbation  \n",
              "0  kw_curated_disease          [Normal]      [Medulloblastoma]  \n",
              "1  kw_curated_disease          [Normal]      [Medulloblastoma]  \n",
              "2  kw_curated_disease          [Normal]      [Medulloblastoma]  \n",
              "3  kw_curated_disease          [Normal]      [Medulloblastoma]  \n",
              "4  kw_curated_disease          [Normal]      [Medulloblastoma]  "
            ]
          },
          "execution_count": 9,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "q = f\"\"\"SELECT \n",
        "              src_dataset_id, \n",
        "              src_description,\n",
        "              condition_column,\n",
        "              condition_control,\n",
        "              condition_perturbation\n",
        "       FROM \n",
        "              gdx_atlas.datasets\n",
        "       WHERE\n",
        "              CONTAINS(condition_perturbation,'{condition_perturbation}')\n",
        "       AND \n",
        "              CONTAINS(condition_control,'{condition_control}')\n",
        "       \"\"\"\n",
        "df = omixatlas.query_metadata(q, query_api_version='v2')\n",
        "print(df.shape)\n",
        "df.head()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "kernel": "Python3"
      },
      "source": [
        "## Query feature level info\n",
        "\n",
        "After having identified the differential comparisons of interest, we would like to understand what genes are significantly regulated in a majority of those datasets. To do this, we count the number of differential comparisons a gene is found in, and along with that the average logFoldChange for that gene."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 10,
      "metadata": {
        "kernel": "Python3"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Query execution succeeded (time taken: 4.95 seconds, data scanned: 165.078 MB)\n",
            "Fetched 37434 rows\n",
            "(37434, 3)\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>gene</th>\n",
              "      <th>dataset_count</th>\n",
              "      <th>log2FC_mean</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>ppip5k1</td>\n",
              "      <td>11</td>\n",
              "      <td>-0.342860</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>ncapd2</td>\n",
              "      <td>11</td>\n",
              "      <td>1.197678</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>arpp21</td>\n",
              "      <td>11</td>\n",
              "      <td>-0.996196</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>map4k4</td>\n",
              "      <td>10</td>\n",
              "      <td>1.214604</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>egfr</td>\n",
              "      <td>10</td>\n",
              "      <td>0.006275</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "      gene  dataset_count  log2FC_mean\n",
              "0  ppip5k1             11    -0.342860\n",
              "1   ncapd2             11     1.197678\n",
              "2   arpp21             11    -0.996196\n",
              "3   map4k4             10     1.214604\n",
              "4     egfr             10     0.006275"
            ]
          },
          "execution_count": 10,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "padj_cutoff = 0.05\n",
        "\n",
        "q = f\"\"\"SELECT data_id AS gene, COUNT(src_dataset_id) AS dataset_count, \n",
        "              SUM(log_fc)/COUNT(src_dataset_id) AS log2FC_mean \n",
        "       FROM gdx_atlas.features \n",
        "       WHERE src_dataset_id IN \n",
        "           (SELECT dataset_id \n",
        "           FROM gdx_atlas.datasets \n",
        "           WHERE CONTAINS(condition_perturbation, '{condition_perturbation}')\n",
        "           AND CONTAINS(condition_control, '{condition_control}'))\n",
        "           AND adj_p_val < {padj_cutoff}\n",
        "       GROUP BY data_id\n",
        "       ORDER BY dataset_count DESC\"\"\"\n",
        "\n",
        "df = omixatlas.query_metadata(q, query_api_version='v2')\n",
        "print(df.shape)\n",
        "df.head()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "kernel": "Python3"
      },
      "source": [
        "## Rank the genes by combining avg log2FC and count of the genes into a single metric"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 11,
      "metadata": {
        "kernel": "Python3"
      },
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>gene</th>\n",
              "      <th>dataset_count</th>\n",
              "      <th>log2FC_mean</th>\n",
              "      <th>importance</th>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>gene</th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>mycn</th>\n",
              "      <td>mycn</td>\n",
              "      <td>8</td>\n",
              "      <td>4.482495</td>\n",
              "      <td>35.859962</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>lhx2</th>\n",
              "      <td>lhx2</td>\n",
              "      <td>7</td>\n",
              "      <td>3.712369</td>\n",
              "      <td>25.986586</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>sox11</th>\n",
              "      <td>sox11</td>\n",
              "      <td>8</td>\n",
              "      <td>2.646387</td>\n",
              "      <td>21.171099</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>insm1</th>\n",
              "      <td>insm1</td>\n",
              "      <td>6</td>\n",
              "      <td>3.434770</td>\n",
              "      <td>20.608621</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>top2a</th>\n",
              "      <td>top2a</td>\n",
              "      <td>5</td>\n",
              "      <td>4.055910</td>\n",
              "      <td>20.279550</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>cenpf</th>\n",
              "      <td>cenpf</td>\n",
              "      <td>4</td>\n",
              "      <td>5.064936</td>\n",
              "      <td>20.259744</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>dcc</th>\n",
              "      <td>dcc</td>\n",
              "      <td>8</td>\n",
              "      <td>2.521027</td>\n",
              "      <td>20.168217</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>tbx1</th>\n",
              "      <td>tbx1</td>\n",
              "      <td>6</td>\n",
              "      <td>3.355053</td>\n",
              "      <td>20.130315</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ccnb2</th>\n",
              "      <td>ccnb2</td>\n",
              "      <td>5</td>\n",
              "      <td>3.818996</td>\n",
              "      <td>19.094980</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>rpgrip1</th>\n",
              "      <td>rpgrip1</td>\n",
              "      <td>5</td>\n",
              "      <td>3.785028</td>\n",
              "      <td>18.925138</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>aspm</th>\n",
              "      <td>aspm</td>\n",
              "      <td>6</td>\n",
              "      <td>3.137670</td>\n",
              "      <td>18.826018</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>cenpe</th>\n",
              "      <td>cenpe</td>\n",
              "      <td>6</td>\n",
              "      <td>3.128887</td>\n",
              "      <td>18.773320</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>bub1b</th>\n",
              "      <td>bub1b</td>\n",
              "      <td>7</td>\n",
              "      <td>2.602667</td>\n",
              "      <td>18.218667</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>cdhr1</th>\n",
              "      <td>cdhr1</td>\n",
              "      <td>8</td>\n",
              "      <td>2.255267</td>\n",
              "      <td>18.042132</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ube2c</th>\n",
              "      <td>ube2c</td>\n",
              "      <td>6</td>\n",
              "      <td>2.970964</td>\n",
              "      <td>17.825782</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>tshz3</th>\n",
              "      <td>tshz3</td>\n",
              "      <td>9</td>\n",
              "      <td>1.941722</td>\n",
              "      <td>17.475495</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>dtl</th>\n",
              "      <td>dtl</td>\n",
              "      <td>7</td>\n",
              "      <td>2.469564</td>\n",
              "      <td>17.286948</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ncapg</th>\n",
              "      <td>ncapg</td>\n",
              "      <td>5</td>\n",
              "      <td>3.435964</td>\n",
              "      <td>17.179819</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>cdca7</th>\n",
              "      <td>cdca7</td>\n",
              "      <td>6</td>\n",
              "      <td>2.834863</td>\n",
              "      <td>17.009175</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>cacna2d1</th>\n",
              "      <td>cacna2d1</td>\n",
              "      <td>9</td>\n",
              "      <td>1.828817</td>\n",
              "      <td>16.459351</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "              gene  dataset_count  log2FC_mean  importance\n",
              "gene                                                      \n",
              "mycn          mycn              8     4.482495   35.859962\n",
              "lhx2          lhx2              7     3.712369   25.986586\n",
              "sox11        sox11              8     2.646387   21.171099\n",
              "insm1        insm1              6     3.434770   20.608621\n",
              "top2a        top2a              5     4.055910   20.279550\n",
              "cenpf        cenpf              4     5.064936   20.259744\n",
              "dcc            dcc              8     2.521027   20.168217\n",
              "tbx1          tbx1              6     3.355053   20.130315\n",
              "ccnb2        ccnb2              5     3.818996   19.094980\n",
              "rpgrip1    rpgrip1              5     3.785028   18.925138\n",
              "aspm          aspm              6     3.137670   18.826018\n",
              "cenpe        cenpe              6     3.128887   18.773320\n",
              "bub1b        bub1b              7     2.602667   18.218667\n",
              "cdhr1        cdhr1              8     2.255267   18.042132\n",
              "ube2c        ube2c              6     2.970964   17.825782\n",
              "tshz3        tshz3              9     1.941722   17.475495\n",
              "dtl            dtl              7     2.469564   17.286948\n",
              "ncapg        ncapg              5     3.435964   17.179819\n",
              "cdca7        cdca7              6     2.834863   17.009175\n",
              "cacna2d1  cacna2d1              9     1.828817   16.459351"
            ]
          },
          "execution_count": 11,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "df['importance'] = df['dataset_count']*df['log2FC_mean']\n",
        "df = df.sort_values(by='importance', ascending=False)\n",
        "df.index = df['gene']\n",
        "df.head(20)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 12,
      "metadata": {
        "kernel": "Python3"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "Text(0.5, 1.0, 'Medulloblastoma markers')"
            ]
          },
          "execution_count": 12,
          "metadata": {},
          "output_type": "execute_result"
        },
        {
          "data": {
            "text/plain": [
              "<Figure size 1080x720 with 0 Axes>"
            ]
          },
          "metadata": {},
          "output_type": "display_data"
        },
        {
          "data": {
            "image/png": "iVBORw0KGgoAAAANSUhEUgAAAZYAAAEICAYAAAB8lNKlAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjUuMSwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/YYfK9AAAACXBIWXMAAAsTAAALEwEAmpwYAAAjDUlEQVR4nO3de7xd453H8c83FxJyEzImpETQpAiRSU4RKlIJOm21qGtbUaliaKcz1aYznamZ1gxlGq12KBUpIk1Raih1DRLV3OQqJERUyDCichGJXH7zx3pObMc5J/ska99yvu/X67yy9ro867fXYf/O8zxrr58iAjMzs7y0qXQAZma2fXFiMTOzXDmxmJlZrpxYzMwsV04sZmaWKycWMzPLlROL1TxJIWm/lu4raZykH27rOSSNlDS5+IhN0iRJoyodh5WGE4uVjaQlkt6TtFuD9c+kD+7eFQqt4iT1TtegXaVjMdtWTixWbi8BZ9S/kNQf2Kly4Vg5KZPL506ebVm+/EuxcrsF+HLB67OBmwt3kLSjpKsk/VnS65Kuk9SxYPslkpZJek3SVxoc+4EhlpYMU0n6qqQXJL0l6R5JezTY5VOSFkt6U9KVTX2oSfqJpFckrZQ0Q9JRBdvqJE1P216X9OO06Yn079uSVks6XFIbSd+T9LKkNyTdLKlraqe+h3NOOtdfJJ0vabCkOZLelvSzgvPuK+lRSctT/OMldWvmWoSkCyUtkrRK0g9SG0+l2H8jaYe07y6S7pX0fymOeyX1avA7uUzSFGAN0KfBuXqmmC9Jrw9L53lb0mxJQ5trK/2OF6c4X5J0VlPvy8okIvzjn7L8AEuAY4HngY8BbYGlwN5AAL3TfmOAe4DuQGfgf4D/TNuOB14HDgJ2Bm5Lx+6Xtk8CRhWccyQwueB14b7jgB+m5WHAm8BAYEfgGuCJBsc9lmLaC1hYf55GzvFFYFegHfCPwP8CHdK2PwJfSsudgMPScu90jnYF7XwFeIHsg7gT8Fvglgb7Xwd0AEYAa4G7gb8C9gTeAI5O++8HDE/vrQdZIru6md9VAL8DugAHAuuAR1IsXYFngbPTvrsCJ5P1PDsDtwN3F7Q1Cfhzaqcd0L7+9wTsk67leWnfPYHlwKfI/vAdnl73aKKtrsBKoG/a3hM4sNL/rbf2H/dYrBLqey3DgQXAq/UbJAk4D/hmRLwVEauA/wBOT7ucCtwUEfMi4h3g0pxiOgsYGxEzI2Id8F3g8AbzPlekmP4MXE3BkF6hiLg1IpZHxIaI+C+yD/O+afN6YD9Ju0XE6oh4egsx/TgiFkfE6hTT6Q3mYX4QEWsj4kHgHWBCRLwREa8CTwKHppheiIiHImJdRPwf8GPg6C1ckx9FxMqImA/MAx5MsawA7i9oe3lE3BkRa9Lv67JG2h4XEfPTNVmf1h1Alqy/HxHXp3VfBH4fEb+PiE0R8RAwnSzRfKgtYAOwCThIUseIWJbitQpyYrFKuAU4k+wv/ZsbbOtB9pfvjDQU8jbwQFoPsAfwSsH+L+cU0x6FbaUP8uVkf0HXa3jehkNlAEj6lqQFklak+LsC9TcsnAt8FHhO0jRJny42prTcDti9YN3rBcvvNvK6U4ppd0m/lvSqpJXArQUxNaXYtneS9Is0ZLeSrDfUTVLbgv0Lr129s8j+qLijYN3ewBfqf/fp+h1J1hP5UFvpj4vTgPOBZZLuk9RvC+/LSsyJxcouIl4mm8T/FNnwTqE3yT60DoyIbumna0R0StuXAR8p2H+vBse/wwdvBvjrIsN6jexDDQBJO5MN8bxasE/D877WsJE0n/Jtsp7VLhHRDVgBCCAiFkXEGWTDVVcAd6RzNfaY8Q/ElM65gQ9+wBfrP9I5+kdEF7Kegbaincb8I1mP7OOp7U+k9YXtN/b+LiX7fd9WkIReIRvu61bws3NEXN5UWxHxh4gYTpZ8ngNu2OZ3ZNvEicUq5VxgWPqLc7OI2ET2wTBG0l8BSNpT0nFpl98AIyUdIGkn4PsN2p0FnJT+it4vnacYE4BzJA2QtCPZB/GfImJJwT6XpInqjwDfACY20k5nsg///wPaSfpXsnkK0nv5oqQe6X2+nVZvSvtv4oMT2xOAb0raR1KnFNPENATUUp2B1cAKSXsCl2xFG821/S7ZjQfd+fDvpCnrgS+QzZXdrOxmiFuBz0g6TlJbSR0kDS28GaBQ6omdmJLzOrL3uGlb35BtGycWq4iIeDEipjex+Ttkk9ZPp6GVh0lzFBFxP9n8xqNpn0cbHDsGeI/sr/pfAeOLjOdh4F+AO8l6Rfvy/rxOvd8BM8iS133AjY009QeyobuFZENXa/ngMNDxwHxJq4GfAKdHxLsRsYZsbmJKGgI6DBhLNmz4BFkPby1wcTHvpxH/RnZjwooUe8Oe4ra4GuhI1vt4muz9FyUi3gNOIhveG0vWQzwR+CeyZPsKWRJs6rOqDfAPZL27t8jmdi7YivdgOVKEC32ZmVl+3GMxM7NcObGYmVmunFjMzCxXTixmZpYrP0kV2G233aJ3796VDsPMrGbMmDHjzYjo0dg2Jxagd+/eTJ/e1J2vZmbWkKQmn3rhoTAzM8uVE4uZmeXKicXMzHLlORZg7qsr6D36vkqHYWZWNksu/9uSte0ei5mZ5cqJxczMcuXEYmZmuapoYpHUW9JzksZJWihpvKRjJU2RtEhSXfq3R9q/jaQXJPVIdRjukjQ7/RyR2lsg6QZJ8yU9KKljJd+jmVlrUw09lv2A/wL6pZ8zyUqRfousJsOtZCVMAY4FZqea3T8FHo+IQ8jqTNTXud4f+HlEHEhWSOnkxk4q6TxJ0yVN37hmRSnel5lZq1QNieWliJibKurNBx6JrEjMXKA3WfGfL6d9vwLclJaHAdcCRMTGiFhR0N6stDwjtfEhEXF9RAyKiEFtd+qa7zsyM2vFqiGxrCtY3lTwehPQLiJeAV6XNAyoA+5vQXsb8S3VZmZlVQ2JpRi/JBsSuz0iNqZ1j5BKkKba2O52mJlVgVpJLPcAnXh/GAzgG8AxkuaSDXkdUInAzMzsg2qi5r2kQcCYiDiqFO0PGjQo/HRjM7PiSZoREYMa21b18w+SRpMNeZ21pX3NzKzyqn4oLCIuj4i9I2JypWMxM7Mtq/rEYmZmtcWJxczMcuXEYmZmuXJiMTOzXDmxmJlZrpxYzMwsV04sZmaWq6r/gmQ5uOa9WXUqZV12Kx33WMzMLFdVmVgkrU7/DpV0bwuO20nSfakq5XxJl5cuSjMza0xVJpZtdFVE9AMOBYZIOqHSAZmZtSa1kFg6Sboj9ULGK9NV0vOS+gJImiDpqxGxJiIeA4iI94CZQK9KBm9m1trUQmI5FPh7snorfYAhqQzxRcA4SacDu0TEDYUHSeoGfIasINiHuOa9mVlp1EJimRoRSyNiEzCLVMM+Ih4C5gI/B0YVHiCpHTAB+GlELG6sUde8NzMrjVpILI3WsJfUBvgYsAbYpcEx1wOLIuLqcgRoZmbvq4XE0pRvAguAM4GbJLUHkPRDoCvZ8JmZmZVZTSaWNGk/CvjHiHgSeAL4nqRewD+TzcfMlDRL0qhmmjIzs5zVRM37UnPNezOzlmmu5n1N9ljMzKx6ObGYmVmunFjMzCxXTixmZpYrJxYzM8uVE4uZmeXKicXMzHLlxGJmZrlyYjEzs1y55j2ueW+ti+vIW6m5x2JmZrmq2cQi6QFJb0u6t8H6iyS9ICkk7Vap+MzMWquaTSzAlcCXGlk/BTgWeLm84ZiZGVQosUjaWdJ9kmZLmifpNEmflPSMpLmSxkraUdJgSXMkdUjHzJd0EEBEPAKsath2RDwTEUvK/Z7MzCxTqcn744HXIuJvASR1BeYBn4yIhZJuBi6IiKsl3QP8EOgI3BoR8/IIQNJ5wHkAbbv0yKNJMzOjckNhc4Hhkq6QdBRZHfuXImJh2v4r4BNp+d+B4cAg4Ed5BeCa92ZmpVGRxJISyECyBPND4HPN7L4r0AnoDHQoeXBmZrZNKjXHsgewJiJuJZuEPxzoLWm/tMuXgMfT8i+AfwHGA1eUO1YzM2uZSs2x9AeulLQJWA9cAHQFbpfUDpgGXCfpy8D6iLhNUlvgKUnDIuJRSU8C/YBOkpYC50bEHyR9Hfg28NfAHEm/jwjXvTczKxPXvMc1783MWso1783MrGycWMzMLFdOLGZmlisnFjMzy5UTi5mZ5cqJxczMcuXEYmZmuXJiMTOzXDmxmJlZrlzzHte8t+2X69tbJZS9xyLpqRK3f5mkVyStLuV5zMyscWVPLBFxRIlP8T9AXYnPYWZmTahEj2V1+neopEmS7pD0nKTxkpS2XS7p2VSW+Kq0bpykayU9LWlxOn6spAWSxtW3HxFPR8Sycr8vMzPLVHqO5VDgQOA1YAowRNIC4PNAv4gISd0K9t+FrHbLZ4F7gCHAKGCapAERMauMsZuZWSMqfVfY1IhYGhGbgFlkJYpXAGuBGyWdBKwp2P9/InvO/1zg9YiYm46dn44tmqTzJE2XNH3jmhXb/k7MzAyofGJZV7C8EWgXERvI5kjuAD4NPNDI/psaHLuJFva+XPPezKw0Kj0U9iGSOgE7RcTvJU0BFlc6JjMzK16leyyN6QzcK2kOMBn4h5YcLOlHqVTxTpKWSrq0BDGamVkTXJoYlyY2M2splyY2M7OycWIxM7NcObGYmVmunFjMzCxXTixmZpYrJxYzM8uVE4uZmeXKicXMzHLlxGJmZrmqumeFVYJLE9v2xOWIrdLcYzEzs1xVJLFI6ibpwq08didJ96Wqk/MlXZ53fGZmtvUq1WPpBmxVYkmuioh+ZBUoh0g6IZeozMxsm1UqsVwO7CtplqQr0888SXMlnQaQato/kXonz0u6TlKbiFgTEY8BRMR7wEygVzrmM5L+JOkZSQ9L2r1C78/MrNWqVGIZDbwYEQOAp4EBwCHAscCVknqm/eqAi4EDgH2BkwobkdQN+AzwSFo1GTgsIg4Ffg18u6kAXJrYzKw0qmHy/khgQkRsjIjXgceBwWnb1IhYHBEbgQlpXwAktUvrfhoR9VUmewF/kDQXuAQ4sKmTujSxmVlpVENiaU7DKmSFr68HFkXE1QXrrgF+FhH9ga8BHUobnpmZNVSpxLKKrAQxwJPAaZLaSuoBfAKYmrbVSdpHUhvgNLKhLiT9EOgK/H2DdrsCr6bls0sXvpmZNaUiiSUilgNTJM0DDgfmALOBR4FvR8T/pl2nAT8DFgAvAXdJ6gX8M9m8y8x0A8CotP+lwO2SZgBvluv9mJnZ+6q25r2kocC3IuLTpT6Xa96bmbWMa96bmVnZVO2zwiJiEjCpwmGYmVkLucdiZma5cmIxM7NcFZ1YJB0p6Zy03EPSPqULy8zMalVRiUXS94HvAN9Nq9oDt5YqKDMzq13F9lg+D3wWeAcgIl7j/S84mpmZbVZsYnkvsi+8BICknUsXkpmZ1bJiE8tvJP0C6Cbpq8DDwA2lC8vMzGpV0d+8lzQcGAEI+ENEPFTKwMppx577R8+zr650GGa5cM17K4fmvnlf9BckUyLZbpKJmZmVRrF3hZ0kaZGkFZJWSlolaWWpg9sWko6SND89pLJjpeMxM2stip1j+RHw2YjoGhFdIqJzRHQpZWA5OAv4z4gYEBHvVjoYM7PWotjE8npELMj75JK+LGmOpNmSbklfvLxT0rT0MyTtd6mksZImSVos6etpfW9Jz0kaL2mBpDsk7ZQeo38q8ANJ4/OO28zMmlbsHMt0SROBu4F19Ssj4rdbe2JJBwLfA46IiDcldServTImIiZL2gv4A/CxdEg/4Biy7888L+natL4vcG5ETJE0FrgwIq6SdCRwb0Tc0cT5zwPOA2jbpcfWvg0zM2ug2MTSBVhDdldYvQC2OrEAw4DbI+JNgIh4S9KxwAGSNp9XUqe0fF9ErAPWSXoD2D2tfyUipqTlW4GvA1dt6eQRcT1ZeWN27Ll/dRalMTOrQUUllog4p9SBJG2AwyJibeHKlGjWFazayPuxN0wKThJmZhVU7F1hH5X0SColjKSDJX1vG8/9KPAFSbumNrsDDwIXF5x3QBHt7CXp8LR8JjB5G+MyM7NtUOzk/Q1kD6BcDxARc4DTt+XEETEfuAx4XNJs4Mdkw1iD0oT+s8D5RTT1PPB3khYAuwDXbmF/MzMroaK+eS9pWkQMlvRMRBya1s2KiAGlDnALcfUmm6A/aFvacc17M7OWyaPm/ZuS9uX9h1CeAizLKT4zM9uOFHtX2N+R3UHVT9KrwEtkX0CsqIhYAmxTb8XMzPJVbGL5HPB74DGyXs47wLGpKzSrNKGZmVktKnYobBDZRPouQDfga8DxwA2Svl2a0MzMrBYV22PpBQyMiNWwuVTxfcAngBlkzxIzMzMrusfyV3zwC4rrgd3Twx3XNX6ImZm1RsX2WMYDf5L0u/T6M8BtqUTxsyWJzMzMalKxj3T5gaT7gSFp1fkRUf/Fj4rfHWZmZtWjJRUkpwP+FqGZmTWr6MSyPZv76gp6j76v0mGYbTPXu7dqUOzkvZmZWVFqMrGkipLfqnQcZmb2YTWZWMzMrHrVTGKR9M+SFkqaTFaOGEn7SXpY0mxJM9ODMpH0HUlz0/rLKxq4mVkrUxOT95L+hqz+ywCymGeSfeN/PHB5RNwlqQPQRtIJwInAxyNiTSog1librnlvZlYCtdJjOQq4KyLWRMRK4B6gI7BnRNwFEBFrI2INcCxwU1omIt5qrMGIuD4iBkXEoLY7dS3PuzAzawVqJbGYmVmNqJXE8gTwOUkdJXUme6TMu8BSSZ8DkLSjpJ2Ah4Bz0jJNDYWZmVlp1ERiiYiZwERgNnA/MC1t+hLwdUlzgKeAv46IB8iGyqZLmgX4tmQzszIqqub99s41783MWiaPmvdmZmZFcWIxM7NcObGYmVmunFjMzCxXTixmZpYrJxYzM8uVE4uZmeXKicXMzHLlxGJmZrmqicfml5pr3tv2wjXvrRpUVY9FUjdJF6bloZLubeHxYyW9IWleaSI0M7MtqarEAnQDLtyG48cBx+cSiZmZbZVqSyyXA/umpxJfCXSRdJ+k5yVdJ6mNpL0lLZK0W3r9pKQRABHxBNBoYS8zMyuPaptjGQ0cFBEDJA0FHgAOAF5OyydFxB2SrgCuBaYCz0bEgxWK18zMGqi2HktDUyNicURsBCYARwJExC+BLsD5bGW9FUnnSZouafrGNStyC9jMrLWr9sTSsFhMAKTqkL3Suk5b1bBr3puZlUS1JZZVQOeC13WS9pHUBjgNmJzWXwGMB/4VuKG8IZqZWXOqKrFExHJgSrpd+EqyEsQ/AxYALwF3SToaGAxcERHjgfcknQMgaQLwR6CvpKWSzq3E+zAza82qbfKeiDhzC7s8DhxWsP9JBctnlCouMzMrTtUllkrov2dXpvsby2ZmuaiqoTAzM6t9TixmZpYrJxYzM8uVE4uZmeXKicXMzHLlxGJmZrlyYjEzs1w5sZiZWa78BUlcmtiqn0sOWy1xj8XMzHJVk4lF0iRJgxpZf5akOZLmSnpK0iGViM/MrDXb3obCXgKOjoi/SDoBuB74eIVjMjNrVSrSY5H05dSzmC3pFkm7S7orvZ4t6QhJvSUtkHSDpPmSHpTUsaCZL0maJWmepDqAiHgqIv6Stj/N+8XAzMysTMqeWCQdCHwPGBYRhwDfAH4KPJ5eDwTmp933B34eEQcCbwMnFzS1U0QMAC4ExjZyqnOB+5uJw6WJzcxKoBI9lmHA7RHxJkBEvJXWXZteb4yI+k/6lyJiVlqeAfQuaGdC2v8JoIukbvUbJB1Dlli+01QQLk1sZlYa1T7Hsq5geSNQOBQWDfYNAEkHA78ETkgVKc3MrIwq0WN5FPiCpF0BJHUHHgEuSK/bSiqmC3Fa2v9IYEVErJC0F/Bb4EsRsbAk0ZuZWbPK3mOJiPmSLgMel7QReIZsnuX6VKN+I1mSWbaFptZKegZoD3wlrftXYFfgvyUBbIiID92WbGZmpaOIhiNKrc+gQYNi+vTplQ7DzKxmSJrR1B/uNfkFSTMzq15OLGZmlisnFjMzy5UTi5mZ5cqJxczMcuXEYmZmuXJiMTOzXDmxmJlZrpxYzMwsV9X+EMqycM17q2aud2+1xj0WMzPLVc0kFkl7SLqjiP3GSnpD0rxyxGVmZh9UlsSizFafS1K7iHgtIk4pYvdxwPFbey4zM9s2JUssqWb985JuBuYBGyWNSfXrH5HUI+03WNKcVL/+yvqehqSRku6R9CjwSGqvcNvvJE2StEjS9+vPmypKvlWq92VmZs0rdY9lf+C/U816gOlp+XGgPhncBHwt1a/f2OD4gcApEXF0I23XAScDB5MVDmtR3RXXvDczK41SJ5aXI+LptLwJmJiWbwWOTHXqO0fEH9P62xoc/1BENNX7eCgilkfEu2RVI49sSWCueW9mVhqlTizvNLOtmApjLTneFcvMzKpAOe8KawPUT76fCUyOiLeBVZI+ntaf3oL2hkvqLqkj8DlgSl6BmpnZ1itnYnkHqEsT8MOAf0/rzwVukDQL2BkodsJjKnAnMAe4MyKmA0iaAPwR6CtpqaRz83sLZma2JWWreS9pdUR0amR9p4hYnZZHAz0j4htbaGskMCgiLsojNte8NzNrmeZq3lfDI13+VtJ3yWJ5GRhZ2XDMzGxblC2xNNZbSesn8v7dYsW2NY7si5Bm1gqtX7+epUuXsnbt2kqHst3r0KEDvXr1on379kUfUw09FjOzFlm6dCmdO3emd+/eSKp0ONutiGD58uUsXbqUffbZp+jjauZZYWZm9dauXcuuu+7qpFJikth1111b3DN0YjGzmuSkUh5bc52dWMzMLFeeYzGzmpd3ob4tFVc74ogjeOqpp3I9Z3OWLFnCU089xZlnnlm2c24L91jMzFqonEllw4YNLFmyhNtua/goxerlxGJm1kKdOmXfnpg0aRJHH300J554In369GH06NGMHz+euro6+vfvz4svvgjAyJEjOf/88xk0aBAf/ehHuffee4HsJoRzzjmH/v37c+ihh/LYY48BMG7cOD772c8ybNgwPvnJTzJ69GiefPJJBgwYwJgxY1iyZAlHHXUUAwcOZODAgZsT3aRJkxg6dCinnHIK/fr146yzzqL+S/DTpk3jiCOO4JBDDqGuro5Vq1axceNGLrnkEgYPHszBBx/ML37xi1yuj4fCcM17q16ud1/9Zs+ezYIFC+jevTt9+vRh1KhRTJ06lZ/85Cdcc801XH311UA2nDV16lRefPFFjjnmGF544QV+/vOfI4m5c+fy3HPPMWLECBYuXAjAzJkzmTNnDt27d2fSpElcddVVmxPSmjVreOihh+jQoQOLFi3ijDPOoP7pIc888wzz589njz32YMiQIUyZMoW6ujpOO+00Jk6cyODBg1m5ciUdO3bkxhtvpGvXrkybNo1169YxZMgQRowY0aJbixvjxGJmtg0GDx5Mz549Adh3330ZMWIEAP3799/cAwE49dRTadOmDfvvvz99+vThueeeY/LkyVx88cUA9OvXj7333ntzYhk+fDjdu3dv9Jzr16/noosuYtasWbRt23bzMQB1dXX06tULgAEDBrBkyRK6du1Kz549GTx4MABdunQB4MEHH2TOnDnccUdW9X3FihUsWrTIicXMrJJ23HHHzctt2rTZ/LpNmzZs2LBh87aGt+1u6TbenXfeucltY8aMYffdd2f27Nls2rSJDh06NBpP27ZtPxBDQxHBNddcw3HHHddsLC3lORYzszK4/fbb2bRpEy+++CKLFy+mb9++HHXUUYwfPx6AhQsX8uc//5m+fft+6NjOnTuzatWqza9XrFhBz549adOmDbfccgsbNzYsvvtBffv2ZdmyZUybNg2AVatWsWHDBo477jiuvfZa1q9fvzmGd95prgxWcSreY5F0N/ARoAPwE+DG9DOIrHjX2IgYI2kSMBs4mizur0TEVEmXAvsAfYC9gG8ChwEnAK8Cn4mI9WV8S2ZWZrUwF7XXXntRV1fHypUrue666+jQoQMXXnghF1xwAf3796ddu3aMGzfuAz2OegcffDBt27blkEMOYeTIkVx44YWcfPLJ3HzzzRx//PHN9m4AdthhByZOnMjFF1/Mu+++S8eOHXn44YcZNWoUS5YsYeDAgUQEPXr04O67797m91q2x+Y3GYDUPSLeSgW7pgFnA5dHxPC0vVtEvJ0Sy6KI+KqkTwD/HREHpcRyLHAMcABZLZaTI+J+SXcBv4qIuxs573nAeQBtu/T4m14X3FTy92rWUrXwgVkJCxYs4GMf+1ilwyjayJEj+fSnP80pp5yy5Z2rUGPXu7nH5lfDUNjXJc0GnibruewA9JF0jaTjgZUF+04AiIgngC6SuqX196deyVygLfBAWj8X6N3YSV3z3sysNCo6FCZpKFlv4/CIWJN6JTsChwDHAecDpwJfSYc0Ved+HUBEbJK0Pt7vhm2iCob7zKx1GzduXKVDKKtK91i6An9JSaUf2dzIbkCbiLgT+B4wsGD/0wAkHQmsiIhiyxib2Xam0sP4rcXWXOdK/zX/AHC+pAXA82TDYXsCkyTVJ73vFuy/VtIzQHve78WYWSvToUMHli9f7kfnl1h9PZbC25mLUfHJ+2KlYbJvRUTuxeld896striCZPk0VUGy2mvem5m1SPv27bf52+FWOjWTWCJiaKVjMDOzLav05L2ZmW1nnFjMzCxXNTN5X0qSVpHdlVYrdgPerHQQLVBr8ULtxVxr8ULtxVxr8UJpY947Ino0tqFm5lhK7Pmm7m6oRpKmO97SqrWYay1eqL2Yay1eqFzMHgozM7NcObGYmVmunFgy11c6gBZyvKVXazHXWrxQezHXWrxQoZg9eW9mZrlyj8XMzHLlxGJmZrlq1YlF0vGSnpf0gqTRlY6nGJKWSJoraZakqntypqSxkt6QNK9gXXdJD0lalP7dpZIxNtREzJdKejVd51mSPlXJGAtJ+oikxyQ9K2m+pG+k9VV5nZuJt5qvcQdJUyXNTjH/W1q/j6Q/pc+MiZJ2qHSs0Gy84yS9VHCNB5QlntY6xyKpLbAQGA4sJSuLfEZEPFvRwLZA0hJgUERU5Re1Utno1cDNEXFQWvcj4K2IuDwl8F0i4juVjLNQEzFfCqyOiKsqGVtjJPUEekbETEmdgRnA54CRVOF1bibeU6neayxg54hYLak9MBn4BvAPwG8j4teSrgNmR8S1lYwVmo33fODeiLijnPG05h5LHfBCRCyOiPeAXwMnVjimmpfKRr/VYPWJwK/S8q/IPlSqRhMxV62IWBYRM9PyKmABWR2jqrzOzcRbtSKzOr1sn34CGAbUf0hX0zVuKt6KaM2JZU/glYLXS6ny/9iTAB6UNEPSeZUOpki7R8SytPy/wO6VDKYFLpI0Jw2VVcWwUkOSegOHAn+iBq5zg3ihiq+xpLaSZgFvAA8BLwJvR8SGtEtVfWY0jDci6q/xZekaj5G0Yzliac2JpVYdGREDgROAv0vDODUjsrHXWhh/vRbYFxgALAP+q6LRNEJSJ+BO4O8jYmXhtmq8zo3EW9XXOCI2RsQAoBfZCEe/ykbUvIbxSjqIrAJvP2Aw0B0oy9Boa04srwIfKXjdK62rahHxavr3DeAusv/gq93raZy9frz9jQrHs0UR8Xr6H3UTcANVdp3TOPqdwPiI+G1aXbXXubF4q/0a14uIt4HHgMOBbpLqn7FYlZ8ZBfEen4YhIyLWATdRpmvcmhPLNGD/dJfHDsDpwD0VjqlZknZOk59I2hkYAcxr/qiqcA9wdlo+G/hdBWMpSv0HdPJ5qug6p4naG4EFEfHjgk1VeZ2birfKr3EPSd3Sckeym3wWkH1gn5J2q6Zr3Fi8zxX8oSGy+aCyXONWe1cYQLq98WqgLTA2Ii6rbETNk9SHrJcC2ZOpb6u2mCVNAIaSPa77deD7wN3Ab4C9gJeBUyOiaibLm4h5KNkQTQBLgK8VzF9UlKQjgSeBucCmtPqfyOYtqu46NxPvGVTvNT6YbHK+Ldkf4L+JiH9P/w/+mmxY6Rngi6k3UFHNxPso0AMQMAs4v2CSv3TxtObEYmZm+WvNQ2FmZlYCTixmZpYrJxYzM8uVE4uZmeXKicXMzHLlxGJmZrlyYjEzs1z9PzhtY3WhU+GsAAAAAElFTkSuQmCC\n",
            "text/plain": [
              "<Figure size 432x288 with 1 Axes>"
            ]
          },
          "metadata": {
            "needs_background": "light"
          },
          "output_type": "display_data"
        }
      ],
      "source": [
        "plt.figure(figsize=(15,10))\n",
        "df.iloc[range(10,-1,-1)].plot.barh(x='gene', y='importance')\n",
        "plt.title(f'{condition_perturbation} markers')"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "kernel": "Python3"
      },
      "outputs": [],
      "source": []
    }
  ],
  "metadata": {
    "kernelspec": {
      "display_name": "Python 3 (ipykernel)",
      "language": "python",
      "name": "python3"
    },
    "language_info": {
      "codemirror_mode": {
        "name": "ipython",
        "version": 3
      },
      "file_extension": ".py",
      "mimetype": "text/x-python",
      "name": "python",
      "nbconvert_exporter": "python",
      "pygments_lexer": "ipython3",
      "version": "3.10.2"
    },
    "sos": {
      "kernels": [
        [
          "Python3",
          "python3",
          "Python3",
          "#FFD91A",
          {
            "name": "ipython",
            "version": 3
          }
        ]
      ],
      "panel": {
        "displayed": true,
        "height": 0
      },
      "version": "1.0.0"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 2
}